{"metadata":{"kernelspec":{"language":"python","display_name":"Python 3","name":"python3"},"language_info":{"name":"python","version":"3.7.12","mimetype":"text/x-python","codemirror_mode":{"name":"ipython","version":3},"pygments_lexer":"ipython3","nbconvert_exporter":"python","file_extension":".py"}},"nbformat_minor":4,"nbformat":4,"cells":[{"cell_type":"markdown","source":"## Assignment 2\n### Zhengjie Deng a1865926","metadata":{}},{"cell_type":"markdown","source":"#### Dependency version:\n- Python: 3.8.16\n- pandas: 1.4.2\n- sklearn: 1.0.2\n- nltk: 3.7\n- tqdm: 4.64.1\n- matplotlib: 3.7.0\n- spacy: 3.5.0\n- numpy: 1.23.5\n- gensim: 3.8.3","metadata":{}},{"cell_type":"code","source":"import warnings\nimport gensim.downloader as api\nfrom collections import Counter\nimport spacy\nimport pandas as pd\nfrom tqdm import tqdm\nimport nltk\nfrom nltk.corpus import stopwords\n\n\nimport numpy as np\nimport json\nimport re\nimport os\n\n# Install faiss\n!pip install faiss\n!pip install pyserini\n\n# Download stopwords\nnltk.download('stopwords')\nnltk.download('wordnet2022')\n\n# Download en_core_web_lg model\nspacy.cli.download(\"en_core_web_sm\")\n\n\n# Download the pre-trained GloVe model\nglove_model = api.load('glove-wiki-gigaword-100')\n\n# ignore the warning\nwarnings.filterwarnings(\"ignore\")\n\n# Some Kaggle-wordnet patch\n! cp -rf /usr/share/nltk_data/corpora/wordnet2022 /usr/share/nltk_data/corpora/wordnet \n\n# Install java for pyserini\n\n!wget \"https://download.java.net/java/GA/jdk11/9/GPL/openjdk-11.0.2_linux-x64_bin.tar.gz\"\n!tar -xvf openjdk-11.0.2_linux-x64_bin.tar.gz\n\n!export JAVA_HOME='/kaggle/working/jdk-11.0.2/'\n!export PATH='/kaggle/working/jdk-11.0.2/bin':$PATH\n!mkdir -p /kaggle/working/jdk-11.0.2/jre/lib/amd64/server/\n\n!ln -s /kaggle/working/jdk-11.0.2/lib/server/libjvm.so /kaggle/working/jdk-11.0.2/jre/lib/amd64/server/libjvm.so\nos.environ[\"JAVA_HOME\"] = \"/kaggle/working/jdk-11.0.2/\"","metadata":{"execution":{"iopub.status.busy":"2023-04-05T06:02:22.508598Z","iopub.execute_input":"2023-04-05T06:02:22.509152Z","iopub.status.idle":"2023-04-05T06:03:39.250002Z","shell.execute_reply.started":"2023-04-05T06:02:22.509109Z","shell.execute_reply":"2023-04-05T06:03:39.248361Z"},"trusted":true},"execution_count":2,"outputs":[{"name":"stdout","text":"Requirement already satisfied: faiss in /opt/conda/lib/python3.7/site-packages (1.5.3)\nRequirement already satisfied: numpy in /opt/conda/lib/python3.7/site-packages (from faiss) (1.21.6)\n\u001b[33mWARNING: Error parsing requirements for pybind11: [Errno 2] No such file or directory: '/opt/conda/lib/python3.7/site-packages/pybind11-2.10.4.dist-info/METADATA'\u001b[0m\u001b[33m\n\u001b[0m\u001b[33mWARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv\u001b[0m\u001b[33m\n\u001b[0mRequirement already satisfied: pyserini in /opt/conda/lib/python3.7/site-packages (0.19.0)\nRequirement already satisfied: pandas>=1.1.5 in /opt/conda/lib/python3.7/site-packages (from pyserini) (1.3.5)\nRequirement already satisfied: numpy>=1.18.1 in /opt/conda/lib/python3.7/site-packages (from pyserini) (1.21.6)\nRequirement already satisfied: tqdm in /opt/conda/lib/python3.7/site-packages (from pyserini) (4.64.1)\nRequirement already satisfied: pyjnius>=1.4.0 in /opt/conda/lib/python3.7/site-packages (from pyserini) (1.4.2)\nRequirement already satisfied: sentencepiece>=0.1.95 in /opt/conda/lib/python3.7/site-packages (from pyserini) (0.1.97)\nRequirement already satisfied: scikit-learn>=0.22.1 in /opt/conda/lib/python3.7/site-packages (from pyserini) (1.0.2)\nRequirement already satisfied: lightgbm>=3.3.2 in /opt/conda/lib/python3.7/site-packages (from pyserini) (3.3.2)\nRequirement already satisfied: nmslib>=2.1.1 in /opt/conda/lib/python3.7/site-packages (from pyserini) (2.1.1)\nRequirement already satisfied: Cython>=0.29.21 in /opt/conda/lib/python3.7/site-packages (from pyserini) (0.29.33)\nRequirement already satisfied: transformers>=4.6.0 in /opt/conda/lib/python3.7/site-packages (from pyserini) (4.27.3)\nRequirement already satisfied: scipy>=1.4.1 in /opt/conda/lib/python3.7/site-packages (from pyserini) (1.7.3)\nRequirement already satisfied: onnxruntime>=1.8.1 in /opt/conda/lib/python3.7/site-packages (from pyserini) (1.14.1)\nRequirement already satisfied: spacy>=3.2.1 in /opt/conda/lib/python3.7/site-packages (from pyserini) (3.5.1)\nRequirement already satisfied: wheel in /opt/conda/lib/python3.7/site-packages (from lightgbm>=3.3.2->pyserini) (0.38.4)\nRequirement already satisfied: psutil in /opt/conda/lib/python3.7/site-packages (from nmslib>=2.1.1->pyserini) (5.9.3)\nCollecting pybind11<2.6.2\n  Using cached pybind11-2.6.1-py2.py3-none-any.whl (188 kB)\nRequirement already satisfied: packaging in /opt/conda/lib/python3.7/site-packages (from onnxruntime>=1.8.1->pyserini) (23.0)\nRequirement already satisfied: coloredlogs in /opt/conda/lib/python3.7/site-packages (from onnxruntime>=1.8.1->pyserini) (15.0.1)\nRequirement already satisfied: protobuf in /opt/conda/lib/python3.7/site-packages (from onnxruntime>=1.8.1->pyserini) (3.20.3)\nRequirement already satisfied: sympy in /opt/conda/lib/python3.7/site-packages (from onnxruntime>=1.8.1->pyserini) (1.10.1)\nRequirement already satisfied: flatbuffers in /opt/conda/lib/python3.7/site-packages (from onnxruntime>=1.8.1->pyserini) (23.1.21)\nRequirement already satisfied: python-dateutil>=2.7.3 in /opt/conda/lib/python3.7/site-packages (from pandas>=1.1.5->pyserini) (2.8.2)\nRequirement already satisfied: pytz>=2017.3 in /opt/conda/lib/python3.7/site-packages (from pandas>=1.1.5->pyserini) (2022.7.1)\nRequirement already satisfied: six>=1.7.0 in /opt/conda/lib/python3.7/site-packages (from pyjnius>=1.4.0->pyserini) (1.16.0)\nRequirement already satisfied: joblib>=0.11 in /opt/conda/lib/python3.7/site-packages (from scikit-learn>=0.22.1->pyserini) (1.2.0)\nRequirement already satisfied: threadpoolctl>=2.0.0 in /opt/conda/lib/python3.7/site-packages (from scikit-learn>=0.22.1->pyserini) (3.1.0)\nRequirement already satisfied: jinja2 in /opt/conda/lib/python3.7/site-packages (from spacy>=3.2.1->pyserini) (3.1.2)\nRequirement already satisfied: catalogue<2.1.0,>=2.0.6 in /opt/conda/lib/python3.7/site-packages (from spacy>=3.2.1->pyserini) (2.0.8)\nRequirement already satisfied: cymem<2.1.0,>=2.0.2 in /opt/conda/lib/python3.7/site-packages (from spacy>=3.2.1->pyserini) (2.0.7)\nRequirement already satisfied: spacy-legacy<3.1.0,>=3.0.11 in /opt/conda/lib/python3.7/site-packages (from spacy>=3.2.1->pyserini) (3.0.12)\nRequirement already satisfied: wasabi<1.2.0,>=0.9.1 in /opt/conda/lib/python3.7/site-packages (from spacy>=3.2.1->pyserini) (1.1.1)\nRequirement already satisfied: langcodes<4.0.0,>=3.2.0 in /opt/conda/lib/python3.7/site-packages (from spacy>=3.2.1->pyserini) (3.3.0)\nRequirement already satisfied: typer<0.8.0,>=0.3.0 in /opt/conda/lib/python3.7/site-packages (from spacy>=3.2.1->pyserini) (0.7.0)\nRequirement already satisfied: smart-open<7.0.0,>=5.2.1 in /opt/conda/lib/python3.7/site-packages (from spacy>=3.2.1->pyserini) (6.3.0)\nRequirement already satisfied: pathy>=0.10.0 in /opt/conda/lib/python3.7/site-packages (from spacy>=3.2.1->pyserini) (0.10.1)\nRequirement already satisfied: spacy-loggers<2.0.0,>=1.0.0 in /opt/conda/lib/python3.7/site-packages (from spacy>=3.2.1->pyserini) (1.0.4)\nRequirement already satisfied: typing-extensions<4.5.0,>=3.7.4.1 in /opt/conda/lib/python3.7/site-packages (from spacy>=3.2.1->pyserini) (4.4.0)\nRequirement already satisfied: murmurhash<1.1.0,>=0.28.0 in /opt/conda/lib/python3.7/site-packages (from spacy>=3.2.1->pyserini) (1.0.9)\nRequirement already satisfied: requests<3.0.0,>=2.13.0 in /opt/conda/lib/python3.7/site-packages (from spacy>=3.2.1->pyserini) (2.28.2)\nRequirement already satisfied: preshed<3.1.0,>=3.0.2 in /opt/conda/lib/python3.7/site-packages (from spacy>=3.2.1->pyserini) (3.0.8)\nRequirement already satisfied: setuptools in /opt/conda/lib/python3.7/site-packages (from spacy>=3.2.1->pyserini) (59.8.0)\nRequirement already satisfied: pydantic!=1.8,!=1.8.1,<1.11.0,>=1.7.4 in /opt/conda/lib/python3.7/site-packages (from spacy>=3.2.1->pyserini) (1.10.4)\nRequirement already satisfied: thinc<8.2.0,>=8.1.8 in /opt/conda/lib/python3.7/site-packages (from spacy>=3.2.1->pyserini) (8.1.9)\nRequirement already satisfied: srsly<3.0.0,>=2.4.3 in /opt/conda/lib/python3.7/site-packages (from spacy>=3.2.1->pyserini) (2.4.6)\nRequirement already satisfied: filelock in /opt/conda/lib/python3.7/site-packages (from transformers>=4.6.0->pyserini) (3.9.0)\nRequirement already satisfied: tokenizers!=0.11.3,<0.14,>=0.11.1 in /opt/conda/lib/python3.7/site-packages (from transformers>=4.6.0->pyserini) (0.13.2)\nRequirement already satisfied: huggingface-hub<1.0,>=0.11.0 in /opt/conda/lib/python3.7/site-packages (from transformers>=4.6.0->pyserini) (0.13.3)\nRequirement already satisfied: pyyaml>=5.1 in /opt/conda/lib/python3.7/site-packages (from transformers>=4.6.0->pyserini) (6.0)\nRequirement already satisfied: regex!=2019.12.17 in /opt/conda/lib/python3.7/site-packages (from transformers>=4.6.0->pyserini) (2021.11.10)\nRequirement already satisfied: importlib-metadata in /opt/conda/lib/python3.7/site-packages (from transformers>=4.6.0->pyserini) (4.11.4)\nRequirement already satisfied: zipp>=0.5 in /opt/conda/lib/python3.7/site-packages (from catalogue<2.1.0,>=2.0.6->spacy>=3.2.1->pyserini) (3.11.0)\nRequirement already satisfied: urllib3<1.27,>=1.21.1 in /opt/conda/lib/python3.7/site-packages (from requests<3.0.0,>=2.13.0->spacy>=3.2.1->pyserini) (1.26.14)\nRequirement already satisfied: certifi>=2017.4.17 in /opt/conda/lib/python3.7/site-packages (from requests<3.0.0,>=2.13.0->spacy>=3.2.1->pyserini) (2022.12.7)\nRequirement already satisfied: idna<4,>=2.5 in /opt/conda/lib/python3.7/site-packages (from requests<3.0.0,>=2.13.0->spacy>=3.2.1->pyserini) (3.4)\nRequirement already satisfied: charset-normalizer<4,>=2 in /opt/conda/lib/python3.7/site-packages (from requests<3.0.0,>=2.13.0->spacy>=3.2.1->pyserini) (2.1.1)\nRequirement already satisfied: confection<1.0.0,>=0.0.1 in /opt/conda/lib/python3.7/site-packages (from thinc<8.2.0,>=8.1.8->spacy>=3.2.1->pyserini) (0.0.4)\nRequirement already satisfied: blis<0.8.0,>=0.7.8 in /opt/conda/lib/python3.7/site-packages (from thinc<8.2.0,>=8.1.8->spacy>=3.2.1->pyserini) (0.7.9)\nRequirement already satisfied: click<9.0.0,>=7.1.1 in /opt/conda/lib/python3.7/site-packages (from typer<0.8.0,>=0.3.0->spacy>=3.2.1->pyserini) (8.1.3)\nRequirement already satisfied: humanfriendly>=9.1 in /opt/conda/lib/python3.7/site-packages (from coloredlogs->onnxruntime>=1.8.1->pyserini) (10.0)\nRequirement already satisfied: MarkupSafe>=2.0 in /opt/conda/lib/python3.7/site-packages (from jinja2->spacy>=3.2.1->pyserini) (2.1.1)\nRequirement already satisfied: mpmath>=0.19 in /opt/conda/lib/python3.7/site-packages (from sympy->onnxruntime>=1.8.1->pyserini) (1.3.0)\n\u001b[33mWARNING: Error parsing requirements for pybind11: [Errno 2] No such file or directory: '/opt/conda/lib/python3.7/site-packages/pybind11-2.10.4.dist-info/METADATA'\u001b[0m\u001b[33m\n\u001b[0mInstalling collected packages: pybind11\n  Attempting uninstall: pybind11\n\u001b[33m    WARNING: No metadata found in /opt/conda/lib/python3.7/site-packages\u001b[0m\u001b[33m\n\u001b[0m    Found existing installation: pybind11 2.10.4\n\u001b[31mERROR: Cannot uninstall pybind11 2.10.4, RECORD file not found. You might be able to recover from this via: 'pip install --force-reinstall --no-deps pybind11==2.10.4'.\u001b[0m\u001b[31m\n\u001b[0m[nltk_data] Downloading package stopwords to /usr/share/nltk_data...\n[nltk_data]   Package stopwords is already up-to-date!\n[nltk_data] Downloading package wordnet2022 to /usr/share/nltk_data...\n[nltk_data]   Package wordnet2022 is already up-to-date!\nCollecting en-core-web-sm==3.5.0\n  Downloading https://github.com/explosion/spacy-models/releases/download/en_core_web_sm-3.5.0/en_core_web_sm-3.5.0-py3-none-any.whl (12.8 MB)\n     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 12.8/12.8 MB 57.2 MB/s eta 0:00:00\nRequirement already satisfied: spacy<3.6.0,>=3.5.0 in /opt/conda/lib/python3.7/site-packages (from en-core-web-sm==3.5.0) (3.5.1)\nRequirement already satisfied: cymem<2.1.0,>=2.0.2 in /opt/conda/lib/python3.7/site-packages (from spacy<3.6.0,>=3.5.0->en-core-web-sm==3.5.0) (2.0.7)\nRequirement already satisfied: packaging>=20.0 in /opt/conda/lib/python3.7/site-packages (from spacy<3.6.0,>=3.5.0->en-core-web-sm==3.5.0) (23.0)\nRequirement already satisfied: spacy-legacy<3.1.0,>=3.0.11 in /opt/conda/lib/python3.7/site-packages (from spacy<3.6.0,>=3.5.0->en-core-web-sm==3.5.0) (3.0.12)\nRequirement already satisfied: preshed<3.1.0,>=3.0.2 in /opt/conda/lib/python3.7/site-packages (from spacy<3.6.0,>=3.5.0->en-core-web-sm==3.5.0) (3.0.8)\nRequirement already satisfied: murmurhash<1.1.0,>=0.28.0 in /opt/conda/lib/python3.7/site-packages (from spacy<3.6.0,>=3.5.0->en-core-web-sm==3.5.0) (1.0.9)\nRequirement already satisfied: tqdm<5.0.0,>=4.38.0 in /opt/conda/lib/python3.7/site-packages (from spacy<3.6.0,>=3.5.0->en-core-web-sm==3.5.0) (4.64.1)\nRequirement already satisfied: setuptools in /opt/conda/lib/python3.7/site-packages (from spacy<3.6.0,>=3.5.0->en-core-web-sm==3.5.0) (59.8.0)\nRequirement already satisfied: catalogue<2.1.0,>=2.0.6 in /opt/conda/lib/python3.7/site-packages (from spacy<3.6.0,>=3.5.0->en-core-web-sm==3.5.0) (2.0.8)\nRequirement already satisfied: jinja2 in /opt/conda/lib/python3.7/site-packages (from spacy<3.6.0,>=3.5.0->en-core-web-sm==3.5.0) (3.1.2)\nRequirement already satisfied: numpy>=1.15.0 in /opt/conda/lib/python3.7/site-packages (from spacy<3.6.0,>=3.5.0->en-core-web-sm==3.5.0) (1.21.6)\nRequirement already satisfied: pydantic!=1.8,!=1.8.1,<1.11.0,>=1.7.4 in /opt/conda/lib/python3.7/site-packages (from spacy<3.6.0,>=3.5.0->en-core-web-sm==3.5.0) (1.10.4)\nRequirement already satisfied: langcodes<4.0.0,>=3.2.0 in /opt/conda/lib/python3.7/site-packages (from spacy<3.6.0,>=3.5.0->en-core-web-sm==3.5.0) (3.3.0)\nRequirement already satisfied: spacy-loggers<2.0.0,>=1.0.0 in /opt/conda/lib/python3.7/site-packages (from spacy<3.6.0,>=3.5.0->en-core-web-sm==3.5.0) (1.0.4)\nRequirement already satisfied: wasabi<1.2.0,>=0.9.1 in /opt/conda/lib/python3.7/site-packages (from spacy<3.6.0,>=3.5.0->en-core-web-sm==3.5.0) (1.1.1)\nRequirement already satisfied: requests<3.0.0,>=2.13.0 in /opt/conda/lib/python3.7/site-packages (from spacy<3.6.0,>=3.5.0->en-core-web-sm==3.5.0) (2.28.2)\nRequirement already satisfied: srsly<3.0.0,>=2.4.3 in /opt/conda/lib/python3.7/site-packages (from spacy<3.6.0,>=3.5.0->en-core-web-sm==3.5.0) (2.4.6)\nRequirement already satisfied: typing-extensions<4.5.0,>=3.7.4.1 in /opt/conda/lib/python3.7/site-packages (from spacy<3.6.0,>=3.5.0->en-core-web-sm==3.5.0) (4.4.0)\nRequirement already satisfied: typer<0.8.0,>=0.3.0 in /opt/conda/lib/python3.7/site-packages (from spacy<3.6.0,>=3.5.0->en-core-web-sm==3.5.0) (0.7.0)\nRequirement already satisfied: pathy>=0.10.0 in /opt/conda/lib/python3.7/site-packages (from spacy<3.6.0,>=3.5.0->en-core-web-sm==3.5.0) (0.10.1)\nRequirement already satisfied: smart-open<7.0.0,>=5.2.1 in /opt/conda/lib/python3.7/site-packages (from spacy<3.6.0,>=3.5.0->en-core-web-sm==3.5.0) (6.3.0)\nRequirement already satisfied: thinc<8.2.0,>=8.1.8 in /opt/conda/lib/python3.7/site-packages (from spacy<3.6.0,>=3.5.0->en-core-web-sm==3.5.0) (8.1.9)\nRequirement already satisfied: zipp>=0.5 in /opt/conda/lib/python3.7/site-packages (from catalogue<2.1.0,>=2.0.6->spacy<3.6.0,>=3.5.0->en-core-web-sm==3.5.0) (3.11.0)\nRequirement already satisfied: certifi>=2017.4.17 in /opt/conda/lib/python3.7/site-packages (from requests<3.0.0,>=2.13.0->spacy<3.6.0,>=3.5.0->en-core-web-sm==3.5.0) (2022.12.7)\nRequirement already satisfied: urllib3<1.27,>=1.21.1 in /opt/conda/lib/python3.7/site-packages (from requests<3.0.0,>=2.13.0->spacy<3.6.0,>=3.5.0->en-core-web-sm==3.5.0) (1.26.14)\nRequirement already satisfied: charset-normalizer<4,>=2 in /opt/conda/lib/python3.7/site-packages (from requests<3.0.0,>=2.13.0->spacy<3.6.0,>=3.5.0->en-core-web-sm==3.5.0) (2.1.1)\nRequirement already satisfied: idna<4,>=2.5 in /opt/conda/lib/python3.7/site-packages (from requests<3.0.0,>=2.13.0->spacy<3.6.0,>=3.5.0->en-core-web-sm==3.5.0) (3.4)\nRequirement already satisfied: confection<1.0.0,>=0.0.1 in /opt/conda/lib/python3.7/site-packages (from thinc<8.2.0,>=8.1.8->spacy<3.6.0,>=3.5.0->en-core-web-sm==3.5.0) (0.0.4)\nRequirement already satisfied: blis<0.8.0,>=0.7.8 in /opt/conda/lib/python3.7/site-packages (from thinc<8.2.0,>=8.1.8->spacy<3.6.0,>=3.5.0->en-core-web-sm==3.5.0) (0.7.9)\nRequirement already satisfied: click<9.0.0,>=7.1.1 in /opt/conda/lib/python3.7/site-packages (from typer<0.8.0,>=0.3.0->spacy<3.6.0,>=3.5.0->en-core-web-sm==3.5.0) (8.1.3)\nRequirement already satisfied: MarkupSafe>=2.0 in /opt/conda/lib/python3.7/site-packages (from jinja2->spacy<3.6.0,>=3.5.0->en-core-web-sm==3.5.0) (2.1.1)\nRequirement already satisfied: importlib-metadata in /opt/conda/lib/python3.7/site-packages (from click<9.0.0,>=7.1.1->typer<0.8.0,>=0.3.0->spacy<3.6.0,>=3.5.0->en-core-web-sm==3.5.0) (4.11.4)\n","output_type":"stream"},{"name":"stderr","text":"WARNING: Error parsing requirements for pybind11: [Errno 2] No such file or directory: '/opt/conda/lib/python3.7/site-packages/pybind11-2.10.4.dist-info/METADATA'\nWARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv\n","output_type":"stream"},{"name":"stdout","text":"\u001b[38;5;2m✔ Download and installation successful\u001b[0m\nYou can now load the package via spacy.load('en_core_web_sm')\n--2023-04-05 06:03:29--  https://download.java.net/java/GA/jdk11/9/GPL/openjdk-11.0.2_linux-x64_bin.tar.gz\nResolving download.java.net (download.java.net)... 23.192.208.88\nConnecting to download.java.net (download.java.net)|23.192.208.88|:443... connected.\nHTTP request sent, awaiting response... 200 OK\nLength: 187513052 (179M) [application/x-gzip]\nSaving to: ‘openjdk-11.0.2_linux-x64_bin.tar.gz.1’\n\nopenjdk-11.0.2_linu 100%[===================>] 178.83M   243MB/s    in 0.7s    \n\n2023-04-05 06:03:30 (243 MB/s) - ‘openjdk-11.0.2_linux-x64_bin.tar.gz.1’ saved [187513052/187513052]\n\njdk-11.0.2/bin/jaotc\njdk-11.0.2/bin/jar\njdk-11.0.2/bin/jarsigner\njdk-11.0.2/bin/java\njdk-11.0.2/bin/javac\njdk-11.0.2/bin/javadoc\njdk-11.0.2/bin/javap\njdk-11.0.2/bin/jcmd\njdk-11.0.2/bin/jconsole\njdk-11.0.2/bin/jdb\njdk-11.0.2/bin/jdeprscan\njdk-11.0.2/bin/jdeps\njdk-11.0.2/bin/jhsdb\njdk-11.0.2/bin/jimage\njdk-11.0.2/bin/jinfo\njdk-11.0.2/bin/jjs\njdk-11.0.2/bin/jlink\njdk-11.0.2/bin/jmap\njdk-11.0.2/bin/jmod\njdk-11.0.2/bin/jps\njdk-11.0.2/bin/jrunscript\njdk-11.0.2/bin/jshell\njdk-11.0.2/bin/jstack\njdk-11.0.2/bin/jstat\njdk-11.0.2/bin/jstatd\njdk-11.0.2/bin/keytool\njdk-11.0.2/bin/pack200\njdk-11.0.2/bin/rmic\njdk-11.0.2/bin/rmid\njdk-11.0.2/bin/rmiregistry\njdk-11.0.2/bin/serialver\njdk-11.0.2/bin/unpack200\njdk-11.0.2/conf/logging.properties\njdk-11.0.2/conf/management/jmxremote.access\njdk-11.0.2/conf/management/jmxremote.password.template\njdk-11.0.2/conf/management/management.properties\njdk-11.0.2/conf/net.properties\njdk-11.0.2/conf/security/java.policy\njdk-11.0.2/conf/security/java.security\njdk-11.0.2/conf/security/policy/README.txt\njdk-11.0.2/conf/security/policy/limited/default_US_export.policy\njdk-11.0.2/conf/security/policy/limited/default_local.policy\njdk-11.0.2/conf/security/policy/limited/exempt_local.policy\njdk-11.0.2/conf/security/policy/unlimited/default_US_export.policy\njdk-11.0.2/conf/security/policy/unlimited/default_local.policy\njdk-11.0.2/conf/sound.properties\njdk-11.0.2/include/classfile_constants.h\njdk-11.0.2/include/jawt.h\njdk-11.0.2/include/jdwpTransport.h\njdk-11.0.2/include/jni.h\njdk-11.0.2/include/jvmti.h\njdk-11.0.2/include/jvmticmlr.h\njdk-11.0.2/include/linux/jawt_md.h\njdk-11.0.2/include/linux/jni_md.h\njdk-11.0.2/jmods/java.base.jmod\njdk-11.0.2/jmods/java.compiler.jmod\njdk-11.0.2/jmods/java.datatransfer.jmod\njdk-11.0.2/jmods/java.desktop.jmod\njdk-11.0.2/jmods/java.instrument.jmod\njdk-11.0.2/jmods/java.logging.jmod\njdk-11.0.2/jmods/java.management.jmod\njdk-11.0.2/jmods/java.management.rmi.jmod\njdk-11.0.2/jmods/java.naming.jmod\njdk-11.0.2/jmods/java.net.http.jmod\njdk-11.0.2/jmods/java.prefs.jmod\njdk-11.0.2/jmods/java.rmi.jmod\njdk-11.0.2/jmods/java.scripting.jmod\njdk-11.0.2/jmods/java.se.jmod\njdk-11.0.2/jmods/java.security.jgss.jmod\njdk-11.0.2/jmods/java.security.sasl.jmod\njdk-11.0.2/jmods/java.smartcardio.jmod\njdk-11.0.2/jmods/java.sql.jmod\njdk-11.0.2/jmods/java.sql.rowset.jmod\njdk-11.0.2/jmods/java.transaction.xa.jmod\njdk-11.0.2/jmods/java.xml.crypto.jmod\njdk-11.0.2/jmods/java.xml.jmod\njdk-11.0.2/jmods/jdk.accessibility.jmod\njdk-11.0.2/jmods/jdk.aot.jmod\njdk-11.0.2/jmods/jdk.attach.jmod\njdk-11.0.2/jmods/jdk.charsets.jmod\njdk-11.0.2/jmods/jdk.compiler.jmod\njdk-11.0.2/jmods/jdk.crypto.cryptoki.jmod\njdk-11.0.2/jmods/jdk.crypto.ec.jmod\njdk-11.0.2/jmods/jdk.dynalink.jmod\njdk-11.0.2/jmods/jdk.editpad.jmod\njdk-11.0.2/jmods/jdk.hotspot.agent.jmod\njdk-11.0.2/jmods/jdk.httpserver.jmod\njdk-11.0.2/jmods/jdk.internal.ed.jmod\njdk-11.0.2/jmods/jdk.internal.jvmstat.jmod\njdk-11.0.2/jmods/jdk.internal.le.jmod\njdk-11.0.2/jmods/jdk.internal.opt.jmod\njdk-11.0.2/jmods/jdk.internal.vm.ci.jmod\njdk-11.0.2/jmods/jdk.internal.vm.compiler.jmod\njdk-11.0.2/jmods/jdk.internal.vm.compiler.management.jmod\njdk-11.0.2/jmods/jdk.jartool.jmod\njdk-11.0.2/jmods/jdk.javadoc.jmod\njdk-11.0.2/jmods/jdk.jcmd.jmod\njdk-11.0.2/jmods/jdk.jconsole.jmod\njdk-11.0.2/jmods/jdk.jdeps.jmod\njdk-11.0.2/jmods/jdk.jdi.jmod\njdk-11.0.2/jmods/jdk.jdwp.agent.jmod\njdk-11.0.2/jmods/jdk.jfr.jmod\njdk-11.0.2/jmods/jdk.jlink.jmod\njdk-11.0.2/jmods/jdk.jshell.jmod\njdk-11.0.2/jmods/jdk.jsobject.jmod\njdk-11.0.2/jmods/jdk.jstatd.jmod\njdk-11.0.2/jmods/jdk.localedata.jmod\njdk-11.0.2/jmods/jdk.management.agent.jmod\njdk-11.0.2/jmods/jdk.management.jfr.jmod\njdk-11.0.2/jmods/jdk.management.jmod\njdk-11.0.2/jmods/jdk.naming.dns.jmod\njdk-11.0.2/jmods/jdk.naming.rmi.jmod\njdk-11.0.2/jmods/jdk.net.jmod\njdk-11.0.2/jmods/jdk.pack.jmod\njdk-11.0.2/jmods/jdk.rmic.jmod\njdk-11.0.2/jmods/jdk.scripting.nashorn.jmod\njdk-11.0.2/jmods/jdk.scripting.nashorn.shell.jmod\njdk-11.0.2/jmods/jdk.sctp.jmod\njdk-11.0.2/jmods/jdk.security.auth.jmod\njdk-11.0.2/jmods/jdk.security.jgss.jmod\njdk-11.0.2/jmods/jdk.unsupported.desktop.jmod\njdk-11.0.2/jmods/jdk.unsupported.jmod\njdk-11.0.2/jmods/jdk.xml.dom.jmod\njdk-11.0.2/jmods/jdk.zipfs.jmod\njdk-11.0.2/legal/java.base/ADDITIONAL_LICENSE_INFO\njdk-11.0.2/legal/java.base/ASSEMBLY_EXCEPTION\njdk-11.0.2/legal/java.base/LICENSE\njdk-11.0.2/legal/java.base/aes.md\njdk-11.0.2/legal/java.base/asm.md\njdk-11.0.2/legal/java.base/c-libutl.md\njdk-11.0.2/legal/java.base/cldr.md\njdk-11.0.2/legal/java.base/icu.md\njdk-11.0.2/legal/java.base/public_suffix.md\njdk-11.0.2/legal/java.base/unicode.md\njdk-11.0.2/legal/java.compiler/ADDITIONAL_LICENSE_INFO\njdk-11.0.2/legal/java.compiler/ASSEMBLY_EXCEPTION\njdk-11.0.2/legal/java.compiler/LICENSE\njdk-11.0.2/legal/java.datatransfer/ADDITIONAL_LICENSE_INFO\njdk-11.0.2/legal/java.datatransfer/ASSEMBLY_EXCEPTION\njdk-11.0.2/legal/java.datatransfer/LICENSE\njdk-11.0.2/legal/java.desktop/ADDITIONAL_LICENSE_INFO\njdk-11.0.2/legal/java.desktop/ASSEMBLY_EXCEPTION\njdk-11.0.2/legal/java.desktop/LICENSE\njdk-11.0.2/legal/java.desktop/colorimaging.md\njdk-11.0.2/legal/java.desktop/giflib.md\njdk-11.0.2/legal/java.desktop/harfbuzz.md\njdk-11.0.2/legal/java.desktop/jpeg.md\njdk-11.0.2/legal/java.desktop/lcms.md\njdk-11.0.2/legal/java.desktop/libpng.md\njdk-11.0.2/legal/java.desktop/mesa3d.md\njdk-11.0.2/legal/java.desktop/opengl.md\njdk-11.0.2/legal/java.desktop/xwindows.md\njdk-11.0.2/legal/java.instrument/ADDITIONAL_LICENSE_INFO\njdk-11.0.2/legal/java.instrument/ASSEMBLY_EXCEPTION\njdk-11.0.2/legal/java.instrument/LICENSE\njdk-11.0.2/legal/java.logging/ADDITIONAL_LICENSE_INFO\njdk-11.0.2/legal/java.logging/ASSEMBLY_EXCEPTION\njdk-11.0.2/legal/java.logging/LICENSE\njdk-11.0.2/legal/java.management.rmi/ADDITIONAL_LICENSE_INFO\njdk-11.0.2/legal/java.management.rmi/ASSEMBLY_EXCEPTION\njdk-11.0.2/legal/java.management.rmi/LICENSE\njdk-11.0.2/legal/java.management/ADDITIONAL_LICENSE_INFO\njdk-11.0.2/legal/java.management/ASSEMBLY_EXCEPTION\njdk-11.0.2/legal/java.management/LICENSE\njdk-11.0.2/legal/java.naming/ADDITIONAL_LICENSE_INFO\njdk-11.0.2/legal/java.naming/ASSEMBLY_EXCEPTION\njdk-11.0.2/legal/java.naming/LICENSE\njdk-11.0.2/legal/java.net.http/ADDITIONAL_LICENSE_INFO\njdk-11.0.2/legal/java.net.http/ASSEMBLY_EXCEPTION\njdk-11.0.2/legal/java.net.http/LICENSE\njdk-11.0.2/legal/java.prefs/ADDITIONAL_LICENSE_INFO\njdk-11.0.2/legal/java.prefs/ASSEMBLY_EXCEPTION\njdk-11.0.2/legal/java.prefs/LICENSE\njdk-11.0.2/legal/java.rmi/ADDITIONAL_LICENSE_INFO\njdk-11.0.2/legal/java.rmi/ASSEMBLY_EXCEPTION\njdk-11.0.2/legal/java.rmi/LICENSE\njdk-11.0.2/legal/java.scripting/ADDITIONAL_LICENSE_INFO\njdk-11.0.2/legal/java.scripting/ASSEMBLY_EXCEPTION\njdk-11.0.2/legal/java.scripting/LICENSE\njdk-11.0.2/legal/java.se/ADDITIONAL_LICENSE_INFO\njdk-11.0.2/legal/java.se/ASSEMBLY_EXCEPTION\njdk-11.0.2/legal/java.se/LICENSE\njdk-11.0.2/legal/java.security.jgss/ADDITIONAL_LICENSE_INFO\njdk-11.0.2/legal/java.security.jgss/ASSEMBLY_EXCEPTION\njdk-11.0.2/legal/java.security.jgss/LICENSE\njdk-11.0.2/legal/java.security.sasl/ADDITIONAL_LICENSE_INFO\njdk-11.0.2/legal/java.security.sasl/ASSEMBLY_EXCEPTION\njdk-11.0.2/legal/java.security.sasl/LICENSE\njdk-11.0.2/legal/java.smartcardio/ADDITIONAL_LICENSE_INFO\njdk-11.0.2/legal/java.smartcardio/ASSEMBLY_EXCEPTION\njdk-11.0.2/legal/java.smartcardio/LICENSE\njdk-11.0.2/legal/java.smartcardio/pcsclite.md\njdk-11.0.2/legal/java.sql.rowset/ADDITIONAL_LICENSE_INFO\njdk-11.0.2/legal/java.sql.rowset/ASSEMBLY_EXCEPTION\njdk-11.0.2/legal/java.sql.rowset/LICENSE\njdk-11.0.2/legal/java.sql/ADDITIONAL_LICENSE_INFO\njdk-11.0.2/legal/java.sql/ASSEMBLY_EXCEPTION\njdk-11.0.2/legal/java.sql/LICENSE\njdk-11.0.2/legal/java.transaction.xa/ADDITIONAL_LICENSE_INFO\njdk-11.0.2/legal/java.transaction.xa/ASSEMBLY_EXCEPTION\njdk-11.0.2/legal/java.transaction.xa/LICENSE\njdk-11.0.2/legal/java.xml.crypto/ADDITIONAL_LICENSE_INFO\njdk-11.0.2/legal/java.xml.crypto/ASSEMBLY_EXCEPTION\njdk-11.0.2/legal/java.xml.crypto/LICENSE\njdk-11.0.2/legal/java.xml.crypto/santuario.md\njdk-11.0.2/legal/java.xml/ADDITIONAL_LICENSE_INFO\njdk-11.0.2/legal/java.xml/ASSEMBLY_EXCEPTION\njdk-11.0.2/legal/java.xml/LICENSE\njdk-11.0.2/legal/java.xml/bcel.md\njdk-11.0.2/legal/java.xml/dom.md\njdk-11.0.2/legal/java.xml/jcup.md\njdk-11.0.2/legal/java.xml/xalan.md\njdk-11.0.2/legal/java.xml/xerces.md\njdk-11.0.2/legal/jdk.accessibility/ADDITIONAL_LICENSE_INFO\njdk-11.0.2/legal/jdk.accessibility/ASSEMBLY_EXCEPTION\njdk-11.0.2/legal/jdk.accessibility/LICENSE\njdk-11.0.2/legal/jdk.aot/ADDITIONAL_LICENSE_INFO\njdk-11.0.2/legal/jdk.aot/ASSEMBLY_EXCEPTION\njdk-11.0.2/legal/jdk.aot/LICENSE\njdk-11.0.2/legal/jdk.attach/ADDITIONAL_LICENSE_INFO\njdk-11.0.2/legal/jdk.attach/ASSEMBLY_EXCEPTION\njdk-11.0.2/legal/jdk.attach/LICENSE\njdk-11.0.2/legal/jdk.charsets/ADDITIONAL_LICENSE_INFO\njdk-11.0.2/legal/jdk.charsets/ASSEMBLY_EXCEPTION\njdk-11.0.2/legal/jdk.charsets/LICENSE\njdk-11.0.2/legal/jdk.compiler/ADDITIONAL_LICENSE_INFO\njdk-11.0.2/legal/jdk.compiler/ASSEMBLY_EXCEPTION\njdk-11.0.2/legal/jdk.compiler/LICENSE\njdk-11.0.2/legal/jdk.crypto.cryptoki/ADDITIONAL_LICENSE_INFO\njdk-11.0.2/legal/jdk.crypto.cryptoki/ASSEMBLY_EXCEPTION\njdk-11.0.2/legal/jdk.crypto.cryptoki/LICENSE\njdk-11.0.2/legal/jdk.crypto.cryptoki/pkcs11cryptotoken.md\njdk-11.0.2/legal/jdk.crypto.cryptoki/pkcs11wrapper.md\njdk-11.0.2/legal/jdk.crypto.ec/ADDITIONAL_LICENSE_INFO\njdk-11.0.2/legal/jdk.crypto.ec/ASSEMBLY_EXCEPTION\njdk-11.0.2/legal/jdk.crypto.ec/LICENSE\njdk-11.0.2/legal/jdk.crypto.ec/ecc.md\njdk-11.0.2/legal/jdk.dynalink/ADDITIONAL_LICENSE_INFO\njdk-11.0.2/legal/jdk.dynalink/ASSEMBLY_EXCEPTION\njdk-11.0.2/legal/jdk.dynalink/LICENSE\njdk-11.0.2/legal/jdk.dynalink/dynalink.md\njdk-11.0.2/legal/jdk.editpad/ADDITIONAL_LICENSE_INFO\njdk-11.0.2/legal/jdk.editpad/ASSEMBLY_EXCEPTION\njdk-11.0.2/legal/jdk.editpad/LICENSE\njdk-11.0.2/legal/jdk.hotspot.agent/ADDITIONAL_LICENSE_INFO\njdk-11.0.2/legal/jdk.hotspot.agent/ASSEMBLY_EXCEPTION\njdk-11.0.2/legal/jdk.hotspot.agent/LICENSE\njdk-11.0.2/legal/jdk.httpserver/ADDITIONAL_LICENSE_INFO\njdk-11.0.2/legal/jdk.httpserver/ASSEMBLY_EXCEPTION\njdk-11.0.2/legal/jdk.httpserver/LICENSE\njdk-11.0.2/legal/jdk.internal.ed/ADDITIONAL_LICENSE_INFO\njdk-11.0.2/legal/jdk.internal.ed/ASSEMBLY_EXCEPTION\njdk-11.0.2/legal/jdk.internal.ed/LICENSE\njdk-11.0.2/legal/jdk.internal.jvmstat/ADDITIONAL_LICENSE_INFO\njdk-11.0.2/legal/jdk.internal.jvmstat/ASSEMBLY_EXCEPTION\njdk-11.0.2/legal/jdk.internal.jvmstat/LICENSE\njdk-11.0.2/legal/jdk.internal.le/ADDITIONAL_LICENSE_INFO\njdk-11.0.2/legal/jdk.internal.le/ASSEMBLY_EXCEPTION\njdk-11.0.2/legal/jdk.internal.le/LICENSE\njdk-11.0.2/legal/jdk.internal.le/jline.md\njdk-11.0.2/legal/jdk.internal.opt/ADDITIONAL_LICENSE_INFO\njdk-11.0.2/legal/jdk.internal.opt/ASSEMBLY_EXCEPTION\njdk-11.0.2/legal/jdk.internal.opt/LICENSE\njdk-11.0.2/legal/jdk.internal.opt/jopt-simple.md\njdk-11.0.2/legal/jdk.internal.vm.ci/ADDITIONAL_LICENSE_INFO\njdk-11.0.2/legal/jdk.internal.vm.ci/ASSEMBLY_EXCEPTION\njdk-11.0.2/legal/jdk.internal.vm.ci/LICENSE\njdk-11.0.2/legal/jdk.internal.vm.compiler.management/ADDITIONAL_LICENSE_INFO\njdk-11.0.2/legal/jdk.internal.vm.compiler.management/ASSEMBLY_EXCEPTION\njdk-11.0.2/legal/jdk.internal.vm.compiler.management/LICENSE\njdk-11.0.2/legal/jdk.internal.vm.compiler/ADDITIONAL_LICENSE_INFO\njdk-11.0.2/legal/jdk.internal.vm.compiler/ASSEMBLY_EXCEPTION\njdk-11.0.2/legal/jdk.internal.vm.compiler/LICENSE\njdk-11.0.2/legal/jdk.jartool/ADDITIONAL_LICENSE_INFO\njdk-11.0.2/legal/jdk.jartool/ASSEMBLY_EXCEPTION\njdk-11.0.2/legal/jdk.jartool/LICENSE\njdk-11.0.2/legal/jdk.javadoc/ADDITIONAL_LICENSE_INFO\njdk-11.0.2/legal/jdk.javadoc/ASSEMBLY_EXCEPTION\njdk-11.0.2/legal/jdk.javadoc/LICENSE\njdk-11.0.2/legal/jdk.javadoc/jquery-migrate.md\njdk-11.0.2/legal/jdk.javadoc/jquery.md\njdk-11.0.2/legal/jdk.javadoc/jqueryUI.md\njdk-11.0.2/legal/jdk.javadoc/jszip.md\njdk-11.0.2/legal/jdk.javadoc/pako.md\njdk-11.0.2/legal/jdk.jcmd/ADDITIONAL_LICENSE_INFO\njdk-11.0.2/legal/jdk.jcmd/ASSEMBLY_EXCEPTION\njdk-11.0.2/legal/jdk.jcmd/LICENSE\njdk-11.0.2/legal/jdk.jconsole/ADDITIONAL_LICENSE_INFO\njdk-11.0.2/legal/jdk.jconsole/ASSEMBLY_EXCEPTION\njdk-11.0.2/legal/jdk.jconsole/LICENSE\njdk-11.0.2/legal/jdk.jdeps/ADDITIONAL_LICENSE_INFO\njdk-11.0.2/legal/jdk.jdeps/ASSEMBLY_EXCEPTION\njdk-11.0.2/legal/jdk.jdeps/LICENSE\njdk-11.0.2/legal/jdk.jdi/ADDITIONAL_LICENSE_INFO\njdk-11.0.2/legal/jdk.jdi/ASSEMBLY_EXCEPTION\njdk-11.0.2/legal/jdk.jdi/LICENSE\njdk-11.0.2/legal/jdk.jdwp.agent/ADDITIONAL_LICENSE_INFO\njdk-11.0.2/legal/jdk.jdwp.agent/ASSEMBLY_EXCEPTION\njdk-11.0.2/legal/jdk.jdwp.agent/LICENSE\njdk-11.0.2/legal/jdk.jfr/ADDITIONAL_LICENSE_INFO\njdk-11.0.2/legal/jdk.jfr/ASSEMBLY_EXCEPTION\njdk-11.0.2/legal/jdk.jfr/LICENSE\njdk-11.0.2/legal/jdk.jlink/ADDITIONAL_LICENSE_INFO\njdk-11.0.2/legal/jdk.jlink/ASSEMBLY_EXCEPTION\njdk-11.0.2/legal/jdk.jlink/LICENSE\njdk-11.0.2/legal/jdk.jshell/ADDITIONAL_LICENSE_INFO\njdk-11.0.2/legal/jdk.jshell/ASSEMBLY_EXCEPTION\njdk-11.0.2/legal/jdk.jshell/LICENSE\njdk-11.0.2/legal/jdk.jsobject/ADDITIONAL_LICENSE_INFO\njdk-11.0.2/legal/jdk.jsobject/ASSEMBLY_EXCEPTION\njdk-11.0.2/legal/jdk.jsobject/LICENSE\njdk-11.0.2/legal/jdk.jstatd/ADDITIONAL_LICENSE_INFO\njdk-11.0.2/legal/jdk.jstatd/ASSEMBLY_EXCEPTION\njdk-11.0.2/legal/jdk.jstatd/LICENSE\njdk-11.0.2/legal/jdk.localedata/ADDITIONAL_LICENSE_INFO\njdk-11.0.2/legal/jdk.localedata/ASSEMBLY_EXCEPTION\njdk-11.0.2/legal/jdk.localedata/LICENSE\njdk-11.0.2/legal/jdk.localedata/cldr.md\njdk-11.0.2/legal/jdk.localedata/thaidict.md\njdk-11.0.2/legal/jdk.management.agent/ADDITIONAL_LICENSE_INFO\njdk-11.0.2/legal/jdk.management.agent/ASSEMBLY_EXCEPTION\njdk-11.0.2/legal/jdk.management.agent/LICENSE\njdk-11.0.2/legal/jdk.management.jfr/ADDITIONAL_LICENSE_INFO\njdk-11.0.2/legal/jdk.management.jfr/ASSEMBLY_EXCEPTION\njdk-11.0.2/legal/jdk.management.jfr/LICENSE\njdk-11.0.2/legal/jdk.management/ADDITIONAL_LICENSE_INFO\njdk-11.0.2/legal/jdk.management/ASSEMBLY_EXCEPTION\njdk-11.0.2/legal/jdk.management/LICENSE\njdk-11.0.2/legal/jdk.naming.dns/ADDITIONAL_LICENSE_INFO\njdk-11.0.2/legal/jdk.naming.dns/ASSEMBLY_EXCEPTION\njdk-11.0.2/legal/jdk.naming.dns/LICENSE\njdk-11.0.2/legal/jdk.naming.rmi/ADDITIONAL_LICENSE_INFO\njdk-11.0.2/legal/jdk.naming.rmi/ASSEMBLY_EXCEPTION\njdk-11.0.2/legal/jdk.naming.rmi/LICENSE\njdk-11.0.2/legal/jdk.net/ADDITIONAL_LICENSE_INFO\njdk-11.0.2/legal/jdk.net/ASSEMBLY_EXCEPTION\njdk-11.0.2/legal/jdk.net/LICENSE\njdk-11.0.2/legal/jdk.pack/ADDITIONAL_LICENSE_INFO\njdk-11.0.2/legal/jdk.pack/ASSEMBLY_EXCEPTION\njdk-11.0.2/legal/jdk.pack/LICENSE\njdk-11.0.2/legal/jdk.rmic/ADDITIONAL_LICENSE_INFO\njdk-11.0.2/legal/jdk.rmic/ASSEMBLY_EXCEPTION\njdk-11.0.2/legal/jdk.rmic/LICENSE\njdk-11.0.2/legal/jdk.scripting.nashorn.shell/ADDITIONAL_LICENSE_INFO\njdk-11.0.2/legal/jdk.scripting.nashorn.shell/ASSEMBLY_EXCEPTION\njdk-11.0.2/legal/jdk.scripting.nashorn.shell/LICENSE\njdk-11.0.2/legal/jdk.scripting.nashorn/ADDITIONAL_LICENSE_INFO\njdk-11.0.2/legal/jdk.scripting.nashorn/ASSEMBLY_EXCEPTION\njdk-11.0.2/legal/jdk.scripting.nashorn/LICENSE\njdk-11.0.2/legal/jdk.scripting.nashorn/double-conversion.md\njdk-11.0.2/legal/jdk.scripting.nashorn/joni.md\njdk-11.0.2/legal/jdk.sctp/ADDITIONAL_LICENSE_INFO\njdk-11.0.2/legal/jdk.sctp/ASSEMBLY_EXCEPTION\njdk-11.0.2/legal/jdk.sctp/LICENSE\njdk-11.0.2/legal/jdk.security.auth/ADDITIONAL_LICENSE_INFO\njdk-11.0.2/legal/jdk.security.auth/ASSEMBLY_EXCEPTION\njdk-11.0.2/legal/jdk.security.auth/LICENSE\njdk-11.0.2/legal/jdk.security.jgss/ADDITIONAL_LICENSE_INFO\njdk-11.0.2/legal/jdk.security.jgss/ASSEMBLY_EXCEPTION\njdk-11.0.2/legal/jdk.security.jgss/LICENSE\njdk-11.0.2/legal/jdk.unsupported.desktop/ADDITIONAL_LICENSE_INFO\njdk-11.0.2/legal/jdk.unsupported.desktop/ASSEMBLY_EXCEPTION\njdk-11.0.2/legal/jdk.unsupported.desktop/LICENSE\njdk-11.0.2/legal/jdk.unsupported/ADDITIONAL_LICENSE_INFO\njdk-11.0.2/legal/jdk.unsupported/ASSEMBLY_EXCEPTION\njdk-11.0.2/legal/jdk.unsupported/LICENSE\njdk-11.0.2/legal/jdk.xml.dom/ADDITIONAL_LICENSE_INFO\njdk-11.0.2/legal/jdk.xml.dom/ASSEMBLY_EXCEPTION\njdk-11.0.2/legal/jdk.xml.dom/LICENSE\njdk-11.0.2/legal/jdk.zipfs/ADDITIONAL_LICENSE_INFO\njdk-11.0.2/legal/jdk.zipfs/ASSEMBLY_EXCEPTION\njdk-11.0.2/legal/jdk.zipfs/LICENSE\njdk-11.0.2/lib/classlist\njdk-11.0.2/lib/ct.sym\njdk-11.0.2/lib/jexec\njdk-11.0.2/lib/jfr/default.jfc\njdk-11.0.2/lib/jfr/profile.jfc\njdk-11.0.2/lib/jli/libjli.so\njdk-11.0.2/lib/jrt-fs.jar\njdk-11.0.2/lib/jvm.cfg\njdk-11.0.2/lib/libattach.so\njdk-11.0.2/lib/libawt.so\njdk-11.0.2/lib/libawt_headless.so\njdk-11.0.2/lib/libawt_xawt.so\njdk-11.0.2/lib/libdt_socket.so\njdk-11.0.2/lib/libextnet.so\njdk-11.0.2/lib/libfontmanager.so\njdk-11.0.2/lib/libinstrument.so\njdk-11.0.2/lib/libj2gss.so\njdk-11.0.2/lib/libj2pcsc.so\njdk-11.0.2/lib/libj2pkcs11.so\njdk-11.0.2/lib/libjaas.so\njdk-11.0.2/lib/libjava.so\njdk-11.0.2/lib/libjavajpeg.so\njdk-11.0.2/lib/libjawt.so\njdk-11.0.2/lib/libjdwp.so\njdk-11.0.2/lib/libjimage.so\njdk-11.0.2/lib/libjsig.so\njdk-11.0.2/lib/libjsound.so\njdk-11.0.2/lib/liblcms.so\njdk-11.0.2/lib/libmanagement.so\njdk-11.0.2/lib/libmanagement_agent.so\njdk-11.0.2/lib/libmanagement_ext.so\njdk-11.0.2/lib/libmlib_image.so\njdk-11.0.2/lib/libnet.so\njdk-11.0.2/lib/libnio.so\njdk-11.0.2/lib/libprefs.so\njdk-11.0.2/lib/librmi.so\njdk-11.0.2/lib/libsaproc.so\njdk-11.0.2/lib/libsctp.so\njdk-11.0.2/lib/libsplashscreen.so\njdk-11.0.2/lib/libsunec.so\njdk-11.0.2/lib/libunpack.so\njdk-11.0.2/lib/libverify.so\njdk-11.0.2/lib/libzip.so\njdk-11.0.2/lib/modules\njdk-11.0.2/lib/psfont.properties.ja\njdk-11.0.2/lib/psfontj2d.properties\njdk-11.0.2/lib/security/blacklisted.certs\njdk-11.0.2/lib/security/cacerts\njdk-11.0.2/lib/security/default.policy\njdk-11.0.2/lib/security/public_suffix_list.dat\njdk-11.0.2/lib/server/Xusage.txt\njdk-11.0.2/lib/server/libjsig.so\njdk-11.0.2/lib/server/libjvm.so\njdk-11.0.2/lib/src.zip\njdk-11.0.2/lib/tzdb.dat\njdk-11.0.2/release\nln: failed to create symbolic link '/kaggle/working/jdk-11.0.2/jre/lib/amd64/server/libjvm.so': File exists\n","output_type":"stream"}]},{"cell_type":"markdown","source":"### 1. Reading dataset and pre-processing","metadata":{}},{"cell_type":"markdown","source":"#### 1.1 Read dataset","metadata":{}},{"cell_type":"code","source":"# load the dataset\nf_metadata_path = \"../input/CORD-19-research-challenge/metadata.csv\"\n\n# construct the data frame of metadata\ndf = pd.read_csv(f_metadata_path)\ndf.head(5)","metadata":{"execution":{"iopub.status.busy":"2023-04-05T06:03:39.253652Z","iopub.execute_input":"2023-04-05T06:03:39.254230Z","iopub.status.idle":"2023-04-05T06:04:26.017602Z","shell.execute_reply.started":"2023-04-05T06:03:39.254172Z","shell.execute_reply":"2023-04-05T06:04:26.016127Z"},"trusted":true},"execution_count":3,"outputs":[{"execution_count":3,"output_type":"execute_result","data":{"text/plain":"   cord_uid                                       sha source_x  \\\n0  ug7v899j  d1aafb70c066a2068b02786f8929fd9c900897fb      PMC   \n1  02tnwd4m  6b0567729c2143a66d737eb0a2f63f2dce2e5a7d      PMC   \n2  ejv2xln0  06ced00a5fc04215949aa72528f2eeaae1d58927      PMC   \n3  2b73a28n  348055649b6b8cf2b9a376498df9bf41f7123605      PMC   \n4  9785vg6d  5f48792a5fa08bed9f56016f4981ae2ca6031b32      PMC   \n\n                                               title                    doi  \\\n0  Clinical features of culture-proven Mycoplasma...  10.1186/1471-2334-1-6   \n1  Nitric oxide: a pro-inflammatory mediator in l...           10.1186/rr14   \n2    Surfactant protein-D and pulmonary host defense           10.1186/rr19   \n3               Role of endothelin-1 in lung disease           10.1186/rr44   \n4  Gene expression in epithelial cells in respons...           10.1186/rr61   \n\n      pmcid pubmed_id license  \\\n0  PMC35282  11472636   no-cc   \n1  PMC59543  11667967   no-cc   \n2  PMC59549  11667972   no-cc   \n3  PMC59574  11686871   no-cc   \n4  PMC59580  11686888   no-cc   \n\n                                            abstract publish_time  \\\n0  OBJECTIVE: This retrospective chart review des...   2001-07-04   \n1  Inflammatory diseases of the respiratory tract...   2000-08-15   \n2  Surfactant protein-D (SP-D) participates in th...   2000-08-25   \n3  Endothelin-1 (ET-1) is a 21 amino acid peptide...   2001-02-22   \n4  Respiratory syncytial virus (RSV) and pneumoni...   2001-05-11   \n\n                                             authors         journal  mag_id  \\\n0                Madani, Tariq A; Al-Ghamdi, Aisha A  BMC Infect Dis     NaN   \n1  Vliet, Albert van der; Eiserich, Jason P; Cros...      Respir Res     NaN   \n2                                    Crouch, Erika C      Respir Res     NaN   \n3  Fagan, Karen A; McMurtry, Ivan F; Rodman, David M      Respir Res     NaN   \n4  Domachowske, Joseph B; Bonville, Cynthia A; Ro...      Respir Res     NaN   \n\n  who_covidence_id arxiv_id  \\\n0              NaN      NaN   \n1              NaN      NaN   \n2              NaN      NaN   \n3              NaN      NaN   \n4              NaN      NaN   \n\n                                      pdf_json_files  \\\n0  document_parses/pdf_json/d1aafb70c066a2068b027...   \n1  document_parses/pdf_json/6b0567729c2143a66d737...   \n2  document_parses/pdf_json/06ced00a5fc04215949aa...   \n3  document_parses/pdf_json/348055649b6b8cf2b9a37...   \n4  document_parses/pdf_json/5f48792a5fa08bed9f560...   \n\n                               pmc_json_files  \\\n0  document_parses/pmc_json/PMC35282.xml.json   \n1  document_parses/pmc_json/PMC59543.xml.json   \n2  document_parses/pmc_json/PMC59549.xml.json   \n3  document_parses/pmc_json/PMC59574.xml.json   \n4  document_parses/pmc_json/PMC59580.xml.json   \n\n                                                 url  s2_id  \n0  https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3...    NaN  \n1  https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5...    NaN  \n2  https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5...    NaN  \n3  https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5...    NaN  \n4  https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5...    NaN  ","text/html":"<div>\n<style scoped>\n    .dataframe tbody tr th:only-of-type {\n        vertical-align: middle;\n    }\n\n    .dataframe tbody tr th {\n        vertical-align: top;\n    }\n\n    .dataframe thead th {\n        text-align: right;\n    }\n</style>\n<table border=\"1\" class=\"dataframe\">\n  <thead>\n    <tr style=\"text-align: right;\">\n      <th></th>\n      <th>cord_uid</th>\n      <th>sha</th>\n      <th>source_x</th>\n      <th>title</th>\n      <th>doi</th>\n      <th>pmcid</th>\n      <th>pubmed_id</th>\n      <th>license</th>\n      <th>abstract</th>\n      <th>publish_time</th>\n      <th>authors</th>\n      <th>journal</th>\n      <th>mag_id</th>\n      <th>who_covidence_id</th>\n      <th>arxiv_id</th>\n      <th>pdf_json_files</th>\n      <th>pmc_json_files</th>\n      <th>url</th>\n      <th>s2_id</th>\n    </tr>\n  </thead>\n  <tbody>\n    <tr>\n      <th>0</th>\n      <td>ug7v899j</td>\n      <td>d1aafb70c066a2068b02786f8929fd9c900897fb</td>\n      <td>PMC</td>\n      <td>Clinical features of culture-proven Mycoplasma...</td>\n      <td>10.1186/1471-2334-1-6</td>\n      <td>PMC35282</td>\n      <td>11472636</td>\n      <td>no-cc</td>\n      <td>OBJECTIVE: This retrospective chart review des...</td>\n      <td>2001-07-04</td>\n      <td>Madani, Tariq A; Al-Ghamdi, Aisha A</td>\n      <td>BMC Infect Dis</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>document_parses/pdf_json/d1aafb70c066a2068b027...</td>\n      <td>document_parses/pmc_json/PMC35282.xml.json</td>\n      <td>https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3...</td>\n      <td>NaN</td>\n    </tr>\n    <tr>\n      <th>1</th>\n      <td>02tnwd4m</td>\n      <td>6b0567729c2143a66d737eb0a2f63f2dce2e5a7d</td>\n      <td>PMC</td>\n      <td>Nitric oxide: a pro-inflammatory mediator in l...</td>\n      <td>10.1186/rr14</td>\n      <td>PMC59543</td>\n      <td>11667967</td>\n      <td>no-cc</td>\n      <td>Inflammatory diseases of the respiratory tract...</td>\n      <td>2000-08-15</td>\n      <td>Vliet, Albert van der; Eiserich, Jason P; Cros...</td>\n      <td>Respir Res</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>document_parses/pdf_json/6b0567729c2143a66d737...</td>\n      <td>document_parses/pmc_json/PMC59543.xml.json</td>\n      <td>https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5...</td>\n      <td>NaN</td>\n    </tr>\n    <tr>\n      <th>2</th>\n      <td>ejv2xln0</td>\n      <td>06ced00a5fc04215949aa72528f2eeaae1d58927</td>\n      <td>PMC</td>\n      <td>Surfactant protein-D and pulmonary host defense</td>\n      <td>10.1186/rr19</td>\n      <td>PMC59549</td>\n      <td>11667972</td>\n      <td>no-cc</td>\n      <td>Surfactant protein-D (SP-D) participates in th...</td>\n      <td>2000-08-25</td>\n      <td>Crouch, Erika C</td>\n      <td>Respir Res</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>document_parses/pdf_json/06ced00a5fc04215949aa...</td>\n      <td>document_parses/pmc_json/PMC59549.xml.json</td>\n      <td>https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5...</td>\n      <td>NaN</td>\n    </tr>\n    <tr>\n      <th>3</th>\n      <td>2b73a28n</td>\n      <td>348055649b6b8cf2b9a376498df9bf41f7123605</td>\n      <td>PMC</td>\n      <td>Role of endothelin-1 in lung disease</td>\n      <td>10.1186/rr44</td>\n      <td>PMC59574</td>\n      <td>11686871</td>\n      <td>no-cc</td>\n      <td>Endothelin-1 (ET-1) is a 21 amino acid peptide...</td>\n      <td>2001-02-22</td>\n      <td>Fagan, Karen A; McMurtry, Ivan F; Rodman, David M</td>\n      <td>Respir Res</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>document_parses/pdf_json/348055649b6b8cf2b9a37...</td>\n      <td>document_parses/pmc_json/PMC59574.xml.json</td>\n      <td>https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5...</td>\n      <td>NaN</td>\n    </tr>\n    <tr>\n      <th>4</th>\n      <td>9785vg6d</td>\n      <td>5f48792a5fa08bed9f56016f4981ae2ca6031b32</td>\n      <td>PMC</td>\n      <td>Gene expression in epithelial cells in respons...</td>\n      <td>10.1186/rr61</td>\n      <td>PMC59580</td>\n      <td>11686888</td>\n      <td>no-cc</td>\n      <td>Respiratory syncytial virus (RSV) and pneumoni...</td>\n      <td>2001-05-11</td>\n      <td>Domachowske, Joseph B; Bonville, Cynthia A; Ro...</td>\n      <td>Respir Res</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>document_parses/pdf_json/5f48792a5fa08bed9f560...</td>\n      <td>document_parses/pmc_json/PMC59580.xml.json</td>\n      <td>https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5...</td>\n      <td>NaN</td>\n    </tr>\n  </tbody>\n</table>\n</div>"},"metadata":{}}]},{"cell_type":"markdown","source":"#### 1.2 Dropping useless columns","metadata":{}},{"cell_type":"code","source":"# drop useless columns for this assignment\ndf_trimed = df.drop([\"sha\", \"source_x\", \"doi\", \"license\", \"authors\", \"journal\", \"pmc_json_files\",\n                    \"pmcid\", \"pubmed_id\", \"mag_id\", \"who_covidence_id\", \"arxiv_id\", \"url\", \"s2_id\"], axis=1)\ndf_trimed.head(5)\n","metadata":{"execution":{"iopub.status.busy":"2023-04-05T06:04:26.018936Z","iopub.execute_input":"2023-04-05T06:04:26.020020Z","iopub.status.idle":"2023-04-05T06:04:26.110920Z","shell.execute_reply.started":"2023-04-05T06:04:26.019952Z","shell.execute_reply":"2023-04-05T06:04:26.109518Z"},"trusted":true},"execution_count":4,"outputs":[{"execution_count":4,"output_type":"execute_result","data":{"text/plain":"   cord_uid                                              title  \\\n0  ug7v899j  Clinical features of culture-proven Mycoplasma...   \n1  02tnwd4m  Nitric oxide: a pro-inflammatory mediator in l...   \n2  ejv2xln0    Surfactant protein-D and pulmonary host defense   \n3  2b73a28n               Role of endothelin-1 in lung disease   \n4  9785vg6d  Gene expression in epithelial cells in respons...   \n\n                                            abstract publish_time  \\\n0  OBJECTIVE: This retrospective chart review des...   2001-07-04   \n1  Inflammatory diseases of the respiratory tract...   2000-08-15   \n2  Surfactant protein-D (SP-D) participates in th...   2000-08-25   \n3  Endothelin-1 (ET-1) is a 21 amino acid peptide...   2001-02-22   \n4  Respiratory syncytial virus (RSV) and pneumoni...   2001-05-11   \n\n                                      pdf_json_files  \n0  document_parses/pdf_json/d1aafb70c066a2068b027...  \n1  document_parses/pdf_json/6b0567729c2143a66d737...  \n2  document_parses/pdf_json/06ced00a5fc04215949aa...  \n3  document_parses/pdf_json/348055649b6b8cf2b9a37...  \n4  document_parses/pdf_json/5f48792a5fa08bed9f560...  ","text/html":"<div>\n<style scoped>\n    .dataframe tbody tr th:only-of-type {\n        vertical-align: middle;\n    }\n\n    .dataframe tbody tr th {\n        vertical-align: top;\n    }\n\n    .dataframe thead th {\n        text-align: right;\n    }\n</style>\n<table border=\"1\" class=\"dataframe\">\n  <thead>\n    <tr style=\"text-align: right;\">\n      <th></th>\n      <th>cord_uid</th>\n      <th>title</th>\n      <th>abstract</th>\n      <th>publish_time</th>\n      <th>pdf_json_files</th>\n    </tr>\n  </thead>\n  <tbody>\n    <tr>\n      <th>0</th>\n      <td>ug7v899j</td>\n      <td>Clinical features of culture-proven Mycoplasma...</td>\n      <td>OBJECTIVE: This retrospective chart review des...</td>\n      <td>2001-07-04</td>\n      <td>document_parses/pdf_json/d1aafb70c066a2068b027...</td>\n    </tr>\n    <tr>\n      <th>1</th>\n      <td>02tnwd4m</td>\n      <td>Nitric oxide: a pro-inflammatory mediator in l...</td>\n      <td>Inflammatory diseases of the respiratory tract...</td>\n      <td>2000-08-15</td>\n      <td>document_parses/pdf_json/6b0567729c2143a66d737...</td>\n    </tr>\n    <tr>\n      <th>2</th>\n      <td>ejv2xln0</td>\n      <td>Surfactant protein-D and pulmonary host defense</td>\n      <td>Surfactant protein-D (SP-D) participates in th...</td>\n      <td>2000-08-25</td>\n      <td>document_parses/pdf_json/06ced00a5fc04215949aa...</td>\n    </tr>\n    <tr>\n      <th>3</th>\n      <td>2b73a28n</td>\n      <td>Role of endothelin-1 in lung disease</td>\n      <td>Endothelin-1 (ET-1) is a 21 amino acid peptide...</td>\n      <td>2001-02-22</td>\n      <td>document_parses/pdf_json/348055649b6b8cf2b9a37...</td>\n    </tr>\n    <tr>\n      <th>4</th>\n      <td>9785vg6d</td>\n      <td>Gene expression in epithelial cells in respons...</td>\n      <td>Respiratory syncytial virus (RSV) and pneumoni...</td>\n      <td>2001-05-11</td>\n      <td>document_parses/pdf_json/5f48792a5fa08bed9f560...</td>\n    </tr>\n  </tbody>\n</table>\n</div>"},"metadata":{}}]},{"cell_type":"markdown","source":"#### 1.3 Sampling","metadata":{}},{"cell_type":"markdown","source":"Given that COVID-19 was first reported in December 2019, we can limit our data selection to articles published after this date. Therefore, we will sample data from December 1st, 2019, onwards.","metadata":{}},{"cell_type":"code","source":"# only keep the documents published after 2019-11\ndf_trimed = df_trimed[df_trimed[\"publish_time\"] > '2019-12']\ndf_trimed.shape[0]\n","metadata":{"execution":{"iopub.status.busy":"2023-04-05T06:04:26.112594Z","iopub.execute_input":"2023-04-05T06:04:26.113047Z","iopub.status.idle":"2023-04-05T06:04:26.272063Z","shell.execute_reply.started":"2023-04-05T06:04:26.113011Z","shell.execute_reply":"2023-04-05T06:04:26.270714Z"},"trusted":true},"execution_count":5,"outputs":[{"execution_count":5,"output_type":"execute_result","data":{"text/plain":"958037"},"metadata":{}}]},{"cell_type":"markdown","source":"Check the missing values in the dataset.","metadata":{}},{"cell_type":"code","source":"# check the number of missing value in each column\nmissing_values_count = df_trimed.isnull().sum()\nmissing_values_count\n","metadata":{"execution":{"iopub.status.busy":"2023-04-05T06:04:26.275812Z","iopub.execute_input":"2023-04-05T06:04:26.276640Z","iopub.status.idle":"2023-04-05T06:04:26.611719Z","shell.execute_reply.started":"2023-04-05T06:04:26.276593Z","shell.execute_reply":"2023-04-05T06:04:26.610444Z"},"trusted":true},"execution_count":6,"outputs":[{"execution_count":6,"output_type":"execute_result","data":{"text/plain":"cord_uid               0\ntitle                488\nabstract          220150\npublish_time           0\npdf_json_files    628520\ndtype: int64"},"metadata":{}}]},{"cell_type":"markdown","source":"Many rows of data are missing titles, abstracts, and PDF files, as demonstrated above. We will drop these rows from our analysis.","metadata":{}},{"cell_type":"code","source":"df_trimed = df_trimed.dropna(subset=['pdf_json_files', 'title', 'abstract'])\ndf_trimed.shape[0]\n","metadata":{"execution":{"iopub.status.busy":"2023-04-05T06:04:26.613502Z","iopub.execute_input":"2023-04-05T06:04:26.614119Z","iopub.status.idle":"2023-04-05T06:04:27.012522Z","shell.execute_reply.started":"2023-04-05T06:04:26.614067Z","shell.execute_reply":"2023-04-05T06:04:27.011303Z"},"trusted":true},"execution_count":7,"outputs":[{"execution_count":7,"output_type":"execute_result","data":{"text/plain":"287065"},"metadata":{}}]},{"cell_type":"markdown","source":"Next, we will verify the validity of the PDF file URLs. Any rows with invalid URLs will be dropped from the dataset.","metadata":{}},{"cell_type":"code","source":"# remove all the rows that do not have the real file\nimport os\n# tqdm progress apply\ntqdm.pandas(desc=\"removing all the rows that do not have the real file...\")\nprint(df_trimed['pdf_json_files'])\n\n\ndf_trimed = df_trimed[df_trimed['pdf_json_files'].progress_apply(\n    lambda x: os.path.isfile(\"../input/CORD-19-research-challenge/\"+x))]\ndf_trimed.shape[0]\n","metadata":{"execution":{"iopub.status.busy":"2023-04-05T06:04:27.014657Z","iopub.execute_input":"2023-04-05T06:04:27.015763Z","iopub.status.idle":"2023-04-05T06:20:17.754788Z","shell.execute_reply.started":"2023-04-05T06:04:27.015702Z","shell.execute_reply":"2023-04-05T06:20:17.750734Z"},"trusted":true},"execution_count":8,"outputs":[{"name":"stdout","text":"4160       document_parses/pdf_json/d84f1c9036052ca0d9f82...\n4174       document_parses/pdf_json/aa3858e3cf21baa00c6f6...\n4177       document_parses/pdf_json/23d0641f1f78528bbf3fa...\n4179       document_parses/pdf_json/9be8c80283a7fed59057d...\n4180       document_parses/pdf_json/d5d29d1d48e1a4fe8f39d...\n                                 ...                        \n1056653    document_parses/pdf_json/96fd9bcc269abf4e0ef2c...\n1056655    document_parses/pdf_json/d86cfba423427426c10fa...\n1056657    document_parses/pdf_json/0d1698212a1fcbbed2fff...\n1056658    document_parses/pdf_json/9b2bbb5ceb21084931b5d...\n1056659    document_parses/pdf_json/c7ae1a19b70e6e2806b67...\nName: pdf_json_files, Length: 287065, dtype: object\n","output_type":"stream"},{"name":"stderr","text":"removing all the rows that do not have the real file...: 100%|██████████| 287065/287065 [15:50<00:00, 301.98it/s]\n","output_type":"stream"},{"traceback":["\u001b[0;31m---------------------------------------------------------------------------\u001b[0m","\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)","\u001b[0;32m/tmp/ipykernel_27/1283753991.py\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m      9\u001b[0m     lambda x: os.path.isfile(\"../input/CORD-19-research-challenge/\"+x))]\n\u001b[1;32m     10\u001b[0m \u001b[0mdf_trimed\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mshape\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 11\u001b[0;31m \u001b[0ms\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m","\u001b[0;31mNameError\u001b[0m: name 's' is not defined"],"ename":"NameError","evalue":"name 's' is not defined","output_type":"error"}]},{"cell_type":"markdown","source":"Based on the keywords of the test queries, we can sample the data further. We will only keep the rows whose title or abstract contain the keywords in the test queries. ","metadata":{}},{"cell_type":"code","source":"# function: select the rows whose title or abstract contain the given strings, ignoring the case\ndef select_rows_contain_string(df, strings):\n    tqdm.pandas(\n        desc=\"selecting the rows whose title or abstract contain the given strings...\")\n    return df[df.progress_apply(lambda row: any(string in row['title'].lower() or string in row['abstract'].lower() for string in strings), axis=1)]","metadata":{"execution":{"iopub.status.busy":"2023-04-05T06:24:29.595070Z","iopub.execute_input":"2023-04-05T06:24:29.595531Z","iopub.status.idle":"2023-04-05T06:24:29.603349Z","shell.execute_reply.started":"2023-04-05T06:24:29.595491Z","shell.execute_reply":"2023-04-05T06:24:29.601642Z"},"trusted":true},"execution_count":9,"outputs":[]},{"cell_type":"code","source":"# title or abstract contain \"COVID-19\", \"SARS-CoV-2\", \"coronavirus\", \"2019-nCoV\", \"covid\", \"covid-19\", \"Covid-19\"\nstrings = [\"COVID-19\", \"SARS-CoV-2\", \"coronavirus\",\n           \"2019-nCoV\", \"covid\", \"covid-19\", \"Covid-19\"]\ndf_trimed_covid = select_rows_contain_string(df_trimed, strings)\n\nkeyword_sampled_df_list = []\n\n# title or abstract contain \"origin\", \"Wuhan\" from the df_trimed_covid\nstrings = [\"origin\", \"Wuhan\"]\ndf_trimed_origin = select_rows_contain_string(df_trimed_covid, strings)\nkeyword_sampled_df_list.append(df_trimed_origin)\n\n# title or abstract contain \"rapid testing\"\nstrings = [\"rapid testing\"]\ndf_trimed_testing = select_rows_contain_string(df_trimed_covid, strings)\nkeyword_sampled_df_list.append(df_trimed_testing)\n\n# title or abstract contain \"social\", \"distancing\", \"lockdown\", \"quarantine\"\nstrings = [\"social distancing\", \"lockdown\", \"quarantine\"]\ndf_trimed_social = select_rows_contain_string(df_trimed_covid, strings)\nkeyword_sampled_df_list.append(df_trimed_social)\n\n# title or abstract contain \"transmission route\"\nstrings = [\"transmission route\"]\ndf_trimed_transmission = select_rows_contain_string(df_trimed_covid, strings)\nkeyword_sampled_df_list.append(df_trimed_transmission)\n\n# title or abstract contain \"best masks\", \"preventing infection\", \"prevent infection\", \"preventing transmission\", \"prevent transmission\", \"preventing spread\", \"prevent spread\"\nstrings = [\"best masks\", \"preventing infection\", \"prevent infection\",\n           \"preventing transmission\", \"prevent transmission\", \"preventing spread\", \"prevent spread\"]\ndf_trimed_masks = select_rows_contain_string(df_trimed_covid, strings)\nkeyword_sampled_df_list.append(df_trimed_masks)\n\n# title or abstract contain \"hand sanitizer\"\nstrings = [\"hand sanitizer\"]\ndf_trimed_sanitizer = select_rows_contain_string(df_trimed_covid, strings)\nkeyword_sampled_df_list.append(df_trimed_sanitizer)\n\n# title or abstract contain \"vaccine\", \"vaccination\", \"vaccines\", \"vaccinations\"\nstrings = [\"vaccine\", \"vaccination\", \"vaccines\", \"vaccinations\"]\ndf_trimed_vaccine = select_rows_contain_string(df_trimed_covid, strings)\nkeyword_sampled_df_list.append(df_trimed_vaccine)\n\n# title or abstract contain \"Vitamin\"\nstrings = [\"Vitamin\"]\ndf_trimed_vitamin = select_rows_contain_string(df_trimed_covid, strings)\nkeyword_sampled_df_list.append(df_trimed_vitamin)\n\n# title or abstract contain \"live outside the body\"\nstrings = [\"live outside the body\"]\ndf_trimed_outside = select_rows_contain_string(df_trimed_covid, strings)\nkeyword_sampled_df_list.append(df_trimed_outside)\n\n# title or abstract contain \"initial symptoms\"\nstrings = [\"initial symptoms\"]\ndf_trimed_symptoms = select_rows_contain_string(df_trimed_covid, strings)\nkeyword_sampled_df_list.append(df_trimed_symptoms)\n","metadata":{"execution":{"iopub.status.busy":"2023-04-05T06:24:33.360857Z","iopub.execute_input":"2023-04-05T06:24:33.361520Z","iopub.status.idle":"2023-04-05T06:26:00.404026Z","shell.execute_reply.started":"2023-04-05T06:24:33.361478Z","shell.execute_reply":"2023-04-05T06:26:00.402672Z"},"trusted":true},"execution_count":10,"outputs":[{"name":"stderr","text":"selecting the rows whose title or abstract contain the given strings...: 100%|██████████| 267675/267675 [00:21<00:00, 12240.41it/s]\nselecting the rows whose title or abstract contain the given strings...: 100%|██████████| 158540/158540 [00:06<00:00, 25778.83it/s]\nselecting the rows whose title or abstract contain the given strings...: 100%|██████████| 158540/158540 [00:04<00:00, 39509.09it/s]\nselecting the rows whose title or abstract contain the given strings...: 100%|██████████| 158540/158540 [00:08<00:00, 19152.38it/s]\nselecting the rows whose title or abstract contain the given strings...: 100%|██████████| 158540/158540 [00:03<00:00, 41083.76it/s]\nselecting the rows whose title or abstract contain the given strings...: 100%|██████████| 158540/158540 [00:17<00:00, 8973.72it/s]\nselecting the rows whose title or abstract contain the given strings...: 100%|██████████| 158540/158540 [00:03<00:00, 42333.42it/s]\nselecting the rows whose title or abstract contain the given strings...: 100%|██████████| 158540/158540 [00:10<00:00, 15774.05it/s]\nselecting the rows whose title or abstract contain the given strings...: 100%|██████████| 158540/158540 [00:03<00:00, 41875.72it/s]\nselecting the rows whose title or abstract contain the given strings...: 100%|██████████| 158540/158540 [00:03<00:00, 42381.28it/s]\nselecting the rows whose title or abstract contain the given strings...: 100%|██████████| 158540/158540 [00:03<00:00, 42063.73it/s]\n","output_type":"stream"}]},{"cell_type":"code","source":"# join all the dataframes above into one dataframe and remove the duplicates\ndf_trimed_covid = pd.concat(keyword_sampled_df_list).drop_duplicates()\ndf_trimed_covid.shape[0]","metadata":{"execution":{"iopub.status.busy":"2023-04-05T06:26:38.484161Z","iopub.execute_input":"2023-04-05T06:26:38.484903Z","iopub.status.idle":"2023-04-05T06:26:38.884316Z","shell.execute_reply.started":"2023-04-05T06:26:38.484856Z","shell.execute_reply":"2023-04-05T06:26:38.882894Z"},"trusted":true},"execution_count":11,"outputs":[{"execution_count":11,"output_type":"execute_result","data":{"text/plain":"43311"},"metadata":{}}]},{"cell_type":"markdown","source":"Finally, we randomly sample 10000 rows from the dataset.","metadata":{}},{"cell_type":"code","source":"# randomly pick 10000 documents from the dataset\n\ndf_sampled = df_trimed_covid.sample(n=10000, random_state=42)\ndf_sampled","metadata":{"execution":{"iopub.status.busy":"2023-04-05T06:26:41.408574Z","iopub.execute_input":"2023-04-05T06:26:41.408974Z","iopub.status.idle":"2023-04-05T06:26:41.436079Z","shell.execute_reply.started":"2023-04-05T06:26:41.408939Z","shell.execute_reply":"2023-04-05T06:26:41.434605Z"},"trusted":true},"execution_count":12,"outputs":[{"execution_count":12,"output_type":"execute_result","data":{"text/plain":"         cord_uid                                              title  \\\n753376   0ex7dy6g  In this together: Psychological wellbeing of f...   \n913530   rmvw3ko0  The Impact of COVID-19 on Self-Reported Substa...   \n712304   kf7lix7f  COVID-19 Pandemic and the Convalescent Plasma ...   \n736570   e0eki2rk  {fr}Who is hesitant about Covid-19 vaccines ? ...   \n816108   pyxmblll  Immediate reactions following the first dose o...   \n...           ...                                                ...   \n768690   3mwqk02l  Administrative Governance and Frontline Office...   \n737001   fniymnmf  Reverse genetic systems: Rational design of co...   \n1004462  ztrmpvbr  The cost-effectiveness of social distancing me...   \n79013    zslbp3f4  SAPV-Patienten in der COVID-19-Krise: Eine int...   \n25116    7cwve06g  COVID-19 Vaccination Did Not Change the Person...   \n\n                                                  abstract publish_time  \\\n753376   The COVID‐19 pandemic and the associated infec...   2021-06-14   \n913530   As the COVID-19 pandemic sweeps the globe, man...   2022-04-11   \n712304   PURPOSE OF REVIEW: COVID-19 pandemic has been ...   2021-07-03   \n736570   OBJECTIVES: : To report the characteristics of...   2022-03-28   \n816108   Addressing COVID19 vaccine hesitancy and minim...   2021-05-14   \n...                                                    ...          ...   \n768690   The spread of COVID-19 put prisons across the ...   2021-03-12   \n737001   Since the end of 2019, the global COVID-19 out...   2020-06-30   \n1004462  BACKGROUND: As one of the strategies to mitiga...   2021-12-23   \n79013    BACKGROUND: In the outpatient setting, COVID-1...   2022-05-25   \n25116    Personal protective behaviors of healthcare wo...   2021-12-21   \n\n                                            pdf_json_files  \n753376   document_parses/pdf_json/f6cc631cda193e16e84b3...  \n913530   document_parses/pdf_json/761ee085a77ce60863fe9...  \n712304   document_parses/pdf_json/60c6c463404554c0ad087...  \n736570   document_parses/pdf_json/8ee6271860745bf4d23d6...  \n816108   document_parses/pdf_json/2aa4576b561138b59ca67...  \n...                                                    ...  \n768690   document_parses/pdf_json/13c03c9ab9c2c4a5d8356...  \n737001   document_parses/pdf_json/8f2dd926eed4506073c2f...  \n1004462  document_parses/pdf_json/febd50367c452b4be9f5b...  \n79013    document_parses/pdf_json/8ef48ebbd9af1a8b0ffad...  \n25116    document_parses/pdf_json/8ad03bfb5811b2bb8e69b...  \n\n[10000 rows x 5 columns]","text/html":"<div>\n<style scoped>\n    .dataframe tbody tr th:only-of-type {\n        vertical-align: middle;\n    }\n\n    .dataframe tbody tr th {\n        vertical-align: top;\n    }\n\n    .dataframe thead th {\n        text-align: right;\n    }\n</style>\n<table border=\"1\" class=\"dataframe\">\n  <thead>\n    <tr style=\"text-align: right;\">\n      <th></th>\n      <th>cord_uid</th>\n      <th>title</th>\n      <th>abstract</th>\n      <th>publish_time</th>\n      <th>pdf_json_files</th>\n    </tr>\n  </thead>\n  <tbody>\n    <tr>\n      <th>753376</th>\n      <td>0ex7dy6g</td>\n      <td>In this together: Psychological wellbeing of f...</td>\n      <td>The COVID‐19 pandemic and the associated infec...</td>\n      <td>2021-06-14</td>\n      <td>document_parses/pdf_json/f6cc631cda193e16e84b3...</td>\n    </tr>\n    <tr>\n      <th>913530</th>\n      <td>rmvw3ko0</td>\n      <td>The Impact of COVID-19 on Self-Reported Substa...</td>\n      <td>As the COVID-19 pandemic sweeps the globe, man...</td>\n      <td>2022-04-11</td>\n      <td>document_parses/pdf_json/761ee085a77ce60863fe9...</td>\n    </tr>\n    <tr>\n      <th>712304</th>\n      <td>kf7lix7f</td>\n      <td>COVID-19 Pandemic and the Convalescent Plasma ...</td>\n      <td>PURPOSE OF REVIEW: COVID-19 pandemic has been ...</td>\n      <td>2021-07-03</td>\n      <td>document_parses/pdf_json/60c6c463404554c0ad087...</td>\n    </tr>\n    <tr>\n      <th>736570</th>\n      <td>e0eki2rk</td>\n      <td>{fr}Who is hesitant about Covid-19 vaccines ? ...</td>\n      <td>OBJECTIVES: : To report the characteristics of...</td>\n      <td>2022-03-28</td>\n      <td>document_parses/pdf_json/8ee6271860745bf4d23d6...</td>\n    </tr>\n    <tr>\n      <th>816108</th>\n      <td>pyxmblll</td>\n      <td>Immediate reactions following the first dose o...</td>\n      <td>Addressing COVID19 vaccine hesitancy and minim...</td>\n      <td>2021-05-14</td>\n      <td>document_parses/pdf_json/2aa4576b561138b59ca67...</td>\n    </tr>\n    <tr>\n      <th>...</th>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n    </tr>\n    <tr>\n      <th>768690</th>\n      <td>3mwqk02l</td>\n      <td>Administrative Governance and Frontline Office...</td>\n      <td>The spread of COVID-19 put prisons across the ...</td>\n      <td>2021-03-12</td>\n      <td>document_parses/pdf_json/13c03c9ab9c2c4a5d8356...</td>\n    </tr>\n    <tr>\n      <th>737001</th>\n      <td>fniymnmf</td>\n      <td>Reverse genetic systems: Rational design of co...</td>\n      <td>Since the end of 2019, the global COVID-19 out...</td>\n      <td>2020-06-30</td>\n      <td>document_parses/pdf_json/8f2dd926eed4506073c2f...</td>\n    </tr>\n    <tr>\n      <th>1004462</th>\n      <td>ztrmpvbr</td>\n      <td>The cost-effectiveness of social distancing me...</td>\n      <td>BACKGROUND: As one of the strategies to mitiga...</td>\n      <td>2021-12-23</td>\n      <td>document_parses/pdf_json/febd50367c452b4be9f5b...</td>\n    </tr>\n    <tr>\n      <th>79013</th>\n      <td>zslbp3f4</td>\n      <td>SAPV-Patienten in der COVID-19-Krise: Eine int...</td>\n      <td>BACKGROUND: In the outpatient setting, COVID-1...</td>\n      <td>2022-05-25</td>\n      <td>document_parses/pdf_json/8ef48ebbd9af1a8b0ffad...</td>\n    </tr>\n    <tr>\n      <th>25116</th>\n      <td>7cwve06g</td>\n      <td>COVID-19 Vaccination Did Not Change the Person...</td>\n      <td>Personal protective behaviors of healthcare wo...</td>\n      <td>2021-12-21</td>\n      <td>document_parses/pdf_json/8ad03bfb5811b2bb8e69b...</td>\n    </tr>\n  </tbody>\n</table>\n<p>10000 rows × 5 columns</p>\n</div>"},"metadata":{}}]},{"cell_type":"markdown","source":"#### 1.4 Pre-processing the text data ","metadata":{}},{"cell_type":"markdown","source":"After sampling the data, our next step is to pre-process the text data. This involves extracting the text from the PDF JSON files as our first task.","metadata":{}},{"cell_type":"code","source":"# function: get the pdf json text based on the URL\ndef extract_pdf(row, pdf_json):\n    # paragraphs other than abstract, introduction, conclusion\n    row[\"other_paragraph\"] = []\n    for body_paragraph in pdf_json['body_text']:\n        if body_paragraph[\"section\"] == \"Introduction\":\n            row[\"introduction\"] = body_paragraph[\"text\"]\n        elif body_paragraph[\"section\"] == \"Conclusion\":\n            row[\"conclusion\"] = body_paragraph[\"text\"]\n        else:\n            row[\"other_paragraph\"].append(body_paragraph[\"text\"])\n    return row\n\n# extract the pdf json text from the sampled dataset\ntqdm.pandas(desc=\"extracting the pdf json text from the sampled dataset...\")\ndf_sampled = df_sampled.progress_apply(lambda row: extract_pdf(\n    row, json.load(open(\"../input/CORD-19-research-challenge/\"+row['pdf_json_files']))), axis=1)\ndf_sampled.head()\n","metadata":{"execution":{"iopub.status.busy":"2023-04-05T06:26:44.304968Z","iopub.execute_input":"2023-04-05T06:26:44.305412Z","iopub.status.idle":"2023-04-05T06:28:41.784366Z","shell.execute_reply.started":"2023-04-05T06:26:44.305371Z","shell.execute_reply":"2023-04-05T06:28:41.783503Z"},"trusted":true},"execution_count":13,"outputs":[{"name":"stderr","text":"extracting the pdf json text from the sampled dataset...: 100%|██████████| 10000/10000 [01:57<00:00, 85.15it/s]\n","output_type":"stream"},{"execution_count":13,"output_type":"execute_result","data":{"text/plain":"                                                 abstract  \\\n753376  The COVID‐19 pandemic and the associated infec...   \n913530  As the COVID-19 pandemic sweeps the globe, man...   \n712304  PURPOSE OF REVIEW: COVID-19 pandemic has been ...   \n736570  OBJECTIVES: : To report the characteristics of...   \n816108  Addressing COVID19 vaccine hesitancy and minim...   \n\n                                               conclusion  cord_uid  \\\n753376                                                NaN  0ex7dy6g   \n913530                                                NaN  rmvw3ko0   \n712304  In such circumstance, the CP therapy is undoub...  kf7lix7f   \n736570                                                NaN  e0eki2rk   \n816108                                                NaN  pyxmblll   \n\n                                             introduction  \\\n753376                                                NaN   \n913530                                                NaN   \n712304  The immune plasma neutralizes the invading SAR...   \n736570  Comparison between hesitant and pro-vaccinatio...   \n816108                                                NaN   \n\n                                          other_paragraph  \\\n753376  [A severe acute respiratory syndrome, coronavi...   \n913530  [Since March 2020, the SARS-CoV-2 (COVID-19) p...   \n712304  [The concept of CP infusion evolved in 1880 wh...   \n736570  [The datasets generated and/or analyzed during...   \n816108  [M a n u s c r i p t Abbreviations: PEG -polye...   \n\n                                           pdf_json_files publish_time  \\\n753376  document_parses/pdf_json/f6cc631cda193e16e84b3...   2021-06-14   \n913530  document_parses/pdf_json/761ee085a77ce60863fe9...   2022-04-11   \n712304  document_parses/pdf_json/60c6c463404554c0ad087...   2021-07-03   \n736570  document_parses/pdf_json/8ee6271860745bf4d23d6...   2022-03-28   \n816108  document_parses/pdf_json/2aa4576b561138b59ca67...   2021-05-14   \n\n                                                    title  \n753376  In this together: Psychological wellbeing of f...  \n913530  The Impact of COVID-19 on Self-Reported Substa...  \n712304  COVID-19 Pandemic and the Convalescent Plasma ...  \n736570  {fr}Who is hesitant about Covid-19 vaccines ? ...  \n816108  Immediate reactions following the first dose o...  ","text/html":"<div>\n<style scoped>\n    .dataframe tbody tr th:only-of-type {\n        vertical-align: middle;\n    }\n\n    .dataframe tbody tr th {\n        vertical-align: top;\n    }\n\n    .dataframe thead th {\n        text-align: right;\n    }\n</style>\n<table border=\"1\" class=\"dataframe\">\n  <thead>\n    <tr style=\"text-align: right;\">\n      <th></th>\n      <th>abstract</th>\n      <th>conclusion</th>\n      <th>cord_uid</th>\n      <th>introduction</th>\n      <th>other_paragraph</th>\n      <th>pdf_json_files</th>\n      <th>publish_time</th>\n      <th>title</th>\n    </tr>\n  </thead>\n  <tbody>\n    <tr>\n      <th>753376</th>\n      <td>The COVID‐19 pandemic and the associated infec...</td>\n      <td>NaN</td>\n      <td>0ex7dy6g</td>\n      <td>NaN</td>\n      <td>[A severe acute respiratory syndrome, coronavi...</td>\n      <td>document_parses/pdf_json/f6cc631cda193e16e84b3...</td>\n      <td>2021-06-14</td>\n      <td>In this together: Psychological wellbeing of f...</td>\n    </tr>\n    <tr>\n      <th>913530</th>\n      <td>As the COVID-19 pandemic sweeps the globe, man...</td>\n      <td>NaN</td>\n      <td>rmvw3ko0</td>\n      <td>NaN</td>\n      <td>[Since March 2020, the SARS-CoV-2 (COVID-19) p...</td>\n      <td>document_parses/pdf_json/761ee085a77ce60863fe9...</td>\n      <td>2022-04-11</td>\n      <td>The Impact of COVID-19 on Self-Reported Substa...</td>\n    </tr>\n    <tr>\n      <th>712304</th>\n      <td>PURPOSE OF REVIEW: COVID-19 pandemic has been ...</td>\n      <td>In such circumstance, the CP therapy is undoub...</td>\n      <td>kf7lix7f</td>\n      <td>The immune plasma neutralizes the invading SAR...</td>\n      <td>[The concept of CP infusion evolved in 1880 wh...</td>\n      <td>document_parses/pdf_json/60c6c463404554c0ad087...</td>\n      <td>2021-07-03</td>\n      <td>COVID-19 Pandemic and the Convalescent Plasma ...</td>\n    </tr>\n    <tr>\n      <th>736570</th>\n      <td>OBJECTIVES: : To report the characteristics of...</td>\n      <td>NaN</td>\n      <td>e0eki2rk</td>\n      <td>Comparison between hesitant and pro-vaccinatio...</td>\n      <td>[The datasets generated and/or analyzed during...</td>\n      <td>document_parses/pdf_json/8ee6271860745bf4d23d6...</td>\n      <td>2022-03-28</td>\n      <td>{fr}Who is hesitant about Covid-19 vaccines ? ...</td>\n    </tr>\n    <tr>\n      <th>816108</th>\n      <td>Addressing COVID19 vaccine hesitancy and minim...</td>\n      <td>NaN</td>\n      <td>pyxmblll</td>\n      <td>NaN</td>\n      <td>[M a n u s c r i p t Abbreviations: PEG -polye...</td>\n      <td>document_parses/pdf_json/2aa4576b561138b59ca67...</td>\n      <td>2021-05-14</td>\n      <td>Immediate reactions following the first dose o...</td>\n    </tr>\n  </tbody>\n</table>\n</div>"},"metadata":{}}]},{"cell_type":"markdown","source":"Next, we can preprocess the text we got from the PDF JSON files. Preprocessing the text involves several tasks: first, we will convert all text to lowercase and remove stopwords. Next, we will perform lemmatization and store the preprocessed text in a separate column for further analysis.","metadata":{}},{"cell_type":"code","source":"stop_words = set(stopwords.words('english'))\n\n# function that preprocesses the input string\ndef preprocess_text(string):\n    string = string.lower()\n    # remove stopwords\n    string = \" \".join([word for word in string.split()\n                      if word not in stop_words])\n    # lemmatization\n    lemmatizer = nltk.stem.WordNetLemmatizer()\n    string = \" \".join([lemmatizer.lemmatize(word) for word in string.split()])\n    return string\n\n# function that preprocesses the text of a row in the sampled dataset\ndef preprocess_text_in_df(row):\n    title = preprocess_text(row['title'])\n    abstract = preprocess_text(row['abstract'])\n    introduction = \"\"\n    if pd.isnull(row['introduction']) == False:\n        introduction = preprocess_text(row['introduction'])\n    conclusion = \"\"\n    if pd.isnull(row['conclusion']) == False:\n        conclusion = preprocess_text(row['conclusion'])\n    other_paragraph = \"\"\n    for paragraph in row['other_paragraph']:\n        other_paragraph += preprocess_text(paragraph)\n    row[\"preprocessed_text\"] = title + \" \" + abstract + \" \" + \\\n        introduction + \" \" + conclusion + \" \" + other_paragraph\n    return row\n\n\ntqdm.pandas(desc=\"preprocessing the text in the sampled dataset...\")\ndf_sampled = df_sampled.progress_apply(\n    lambda row: preprocess_text_in_df(row), axis=1)\n","metadata":{"execution":{"iopub.status.busy":"2023-04-05T06:28:45.628434Z","iopub.execute_input":"2023-04-05T06:28:45.629458Z","iopub.status.idle":"2023-04-05T06:31:09.607308Z","shell.execute_reply.started":"2023-04-05T06:28:45.629405Z","shell.execute_reply":"2023-04-05T06:31:09.606155Z"},"trusted":true},"execution_count":14,"outputs":[{"name":"stderr","text":"preprocessing the text in the sampled dataset...: 100%|██████████| 10000/10000 [02:23<00:00, 69.47it/s]\n","output_type":"stream"}]},{"cell_type":"code","source":"df_sampled[\"preprocessed_text\"].head()","metadata":{"execution":{"iopub.status.busy":"2023-04-05T06:31:29.668910Z","iopub.execute_input":"2023-04-05T06:31:29.669838Z","iopub.status.idle":"2023-04-05T06:31:29.683079Z","shell.execute_reply.started":"2023-04-05T06:31:29.669798Z","shell.execute_reply":"2023-04-05T06:31:29.681857Z"},"trusted":true},"execution_count":15,"outputs":[{"execution_count":15,"output_type":"execute_result","data":{"text/plain":"753376    together: psychological wellbeing foreign work...\n913530    impact covid-19 self-reported substance use, w...\n712304    covid-19 pandemic convalescent plasma therapy:...\n736570    {fr}who hesitant covid-19 vaccine ? profiling ...\n816108    immediate reaction following first dose sars-c...\nName: preprocessed_text, dtype: object"},"metadata":{}}]},{"cell_type":"markdown","source":"### 2. Named Entity Recognition and Knowledge Base","metadata":{}},{"cell_type":"markdown","source":"#### 2.1 Entity extraction","metadata":{}},{"cell_type":"markdown","source":"To save time, we will only extract entities from the title, abstract, introduction, and conclusion sections of the text. These sections of the article are most likely to contain entities that are relevant to the topic of the article.","metadata":{}},{"cell_type":"code","source":"nlp = spacy.load(\"en_core_web_sm\")\n\n# function: get the name entities from the title, abstract, introduction, and conclusion of each data in the dataset\ndef get_name_entity(row):\n    name_entity = []\n    doc = nlp(row[\"title\"])\n    for ent in doc.ents:\n        # if ent is not number\n        if ent.label_ != \"CARDINAL\" and ent.label_ != \"PERCENT\" and ent.label_ != \"MONEY\":\n            name_entity.append(ent.text)\n    doc = nlp(row[\"abstract\"])\n    for ent in doc.ents:\n        if ent.label_ != \"CARDINAL\" and ent.label_ != \"PERCENT\" and ent.label_ != \"MONEY\":\n            name_entity.append(ent.text)\n    if pd.isna(row[\"introduction\"]) == False:\n        doc = nlp(row[\"introduction\"])\n        for ent in doc.ents:\n            if ent.label_ != \"CARDINAL\" and ent.label_ != \"PERCENT\" and ent.label_ != \"MONEY\":\n                name_entity.append(ent.text)\n    if pd.isna(row[\"conclusion\"]) == False:\n        doc = nlp(row[\"conclusion\"])\n        for ent in doc.ents:\n            if ent.label_ != \"CARDINAL\" and ent.label_ != \"PERCENT\" and ent.label_ != \"MONEY\":\n                name_entity.append(ent.text)\n    return name_entity\n","metadata":{"execution":{"iopub.status.busy":"2023-04-05T06:31:32.134190Z","iopub.execute_input":"2023-04-05T06:31:32.134900Z","iopub.status.idle":"2023-04-05T06:31:32.975628Z","shell.execute_reply.started":"2023-04-05T06:31:32.134860Z","shell.execute_reply":"2023-04-05T06:31:32.974461Z"},"trusted":true},"execution_count":16,"outputs":[]},{"cell_type":"code","source":"# get the name entity of the title, abstract, introduction, and conclusion of each data in the dataset\ntqdm.pandas(desc=\"Getting name entity...\")\ndf_sampled['name_entity'] = df_sampled.progress_apply(\n    lambda row: get_name_entity(row), axis=1)\n","metadata":{"execution":{"iopub.status.busy":"2023-04-05T06:31:34.827199Z","iopub.execute_input":"2023-04-05T06:31:34.828248Z","iopub.status.idle":"2023-04-05T06:43:59.816342Z","shell.execute_reply.started":"2023-04-05T06:31:34.828200Z","shell.execute_reply":"2023-04-05T06:43:59.814146Z"},"trusted":true},"execution_count":17,"outputs":[{"name":"stderr","text":"Getting name entity...: 100%|██████████| 10000/10000 [12:24<00:00, 13.42it/s]\n","output_type":"stream"}]},{"cell_type":"markdown","source":"Presenting the frequency of the extracted entities.","metadata":{}},{"cell_type":"code","source":"# merge the name entity into one list\nname_entity_list = []\nfor name_entity in df_sampled['name_entity']:\n    name_entity_list.extend(name_entity)\n\n# count the frequency of each name entity and sort it\nname_entity_count = Counter(name_entity_list)\nname_entity_count = sorted(name_entity_count.items(),\n                           key=lambda x: x[1], reverse=True)\n\n# show the top 100 name entity\nname_entity_count[:100]\n","metadata":{"execution":{"iopub.status.busy":"2023-04-05T06:44:46.454183Z","iopub.execute_input":"2023-04-05T06:44:46.454694Z","iopub.status.idle":"2023-04-05T06:44:46.543391Z","shell.execute_reply.started":"2023-04-05T06:44:46.454631Z","shell.execute_reply":"2023-04-05T06:44:46.542165Z"},"trusted":true},"execution_count":18,"outputs":[{"execution_count":18,"output_type":"execute_result","data":{"text/plain":"[('COVID-19', 34227),\n ('first', 3703),\n ('2019', 2925),\n ('China', 1873),\n ('2020', 1842),\n ('CI', 1834),\n ('second', 1709),\n ('COVID‐19', 1511),\n ('India', 1201),\n ('Wuhan', 919),\n ('daily', 875),\n ('Italy', 867),\n ('UK', 816),\n ('RNA', 794),\n ('IgG', 662),\n ('RBD', 654),\n ('al.', 644),\n ('Omicron', 637),\n ('the United States', 597),\n ('US', 591),\n ('2021', 578),\n ('third', 529),\n ('Delta', 527),\n ('Coronavirus', 423),\n ('ICU', 419),\n ('Italian', 406),\n ('Brazil', 398),\n ('Chinese', 394),\n ('Germany', 393),\n ('Spain', 377),\n ('WHO', 367),\n ('ACE2', 359),\n ('USA', 340),\n ('France', 339),\n ('U.S.', 336),\n ('COVID', 336),\n ('Pfizer', 333),\n ('December 2019', 331),\n ('Europe', 313),\n ('March 2020', 297),\n ('Moderna', 288),\n ('Vaccine', 287),\n ('PPE', 284),\n ('Japan', 284),\n ('European', 278),\n ('RT-PCR', 274),\n ('BCG', 272),\n ('England', 271),\n ('PCR', 270),\n ('Spike', 263),\n ('Australia', 261),\n ('S1', 256),\n ('Africa', 255),\n ('Saudi Arabia', 236),\n ('Israel', 232),\n ('Canada', 232),\n ('covid-19', 232),\n ('SUPPLEMENTARY', 231),\n ('South Africa', 227),\n ('Indian', 227),\n ('CDC', 224),\n ('First', 220),\n ('Bangladesh', 220),\n ('May 2020', 212),\n ('SARS-CoV-2', 207),\n ('weekly', 206),\n ('Pakistan', 206),\n ('VE', 206),\n ('CT', 204),\n ('p < 0.001', 204),\n ('FDA', 201),\n ('the World Health Organization', 199),\n ('CoronaVac', 192),\n ('French', 187),\n ('tertiary', 182),\n ('Twitter', 181),\n ('B.1.351', 179),\n ('PubMed', 177),\n ('South Korea', 174),\n ('English', 172),\n ('Spanish', 172),\n ('years', 169),\n ('Iran', 167),\n ('2018', 164),\n ('Nigeria', 163),\n ('Section 3', 160),\n ('PURPOSE', 160),\n ('March', 159),\n ('N', 159),\n ('April 2020', 159),\n ('AI', 156),\n ('NO(2', 156),\n ('Section 2', 154),\n ('the United Kingdom', 150),\n ('Alpha', 149),\n ('clinicians', 149),\n ('African', 149),\n ('mRNA-1273', 149),\n ('monthly', 147),\n ('Section 4', 145)]"},"metadata":{}}]},{"cell_type":"markdown","source":"#### 2.2 Knowledge base","metadata":{}},{"cell_type":"markdown","source":"We will manually build knowledge bases that display synonyms of the entities and their associated keywords. This will be based on the results of the Named Entity Recognition and the test query set.","metadata":{}},{"cell_type":"code","source":"# manually create the knowledge base\n\n# create the knowledge base\nknowledge_base_synonym = {\n    \"COVID-19\": [\"SARS-CoV-2\", \"coronavirus disease 2019\", \"coronavirus disease 19\", \"coronavirus 2019\", \"coronavirus 19\", \"2019 novel coronavirus\", \"2019-nCoV\", \"2019-novel coronavirus\", \"2019 novel coronavirus pneumonia\", \"2019-nCoV pneumonia\"],\n    \"rapid testing\": [\"RAT\", \"rapid test\", \"rapid antigen test\", \"rapid antigen tests\", \"rapid antigen testing\", \"rapid antigen tests\", \"rapid antigen test kit\"],\n    \"origin\": [\"origins\", \"source\", \"sources\"],\n    \"initial symtoms\": [\"early signs\"]\n}\n\nknowledge_base_association = {\n    \"mask\": [\"n95\", \"cloth mask\"],\n    \"vaccine\": [\"mrna\"],\n    \"origin\": [\"wuhan\", \"fish market\"],\n    \"symptoms\": [\"fever\", \"chill\", \"cough\", \"tired\", \"headache\", \"loss taste or small\", \"sore throat\", \"diarrhea\"],\n    \"sanitizer\": [\"alcohol\"],\n    \"social distancing\": [\"quarantine\", \"lockdown\"],\n    \"transmission route\": [\"airborne\", \"droplet\", \"contact\", \"fomite\"],\n    \"testing\": [\"PCR\"]\n}\n","metadata":{"execution":{"iopub.status.busy":"2023-04-05T06:44:53.594189Z","iopub.execute_input":"2023-04-05T06:44:53.595202Z","iopub.status.idle":"2023-04-05T06:44:53.602974Z","shell.execute_reply.started":"2023-04-05T06:44:53.595156Z","shell.execute_reply":"2023-04-05T06:44:53.601686Z"},"trusted":true},"execution_count":21,"outputs":[]},{"cell_type":"markdown","source":"### 3. Indexing method","metadata":{}},{"cell_type":"markdown","source":"To efficiently retrieve documents containing words from the query, we will utilize the inverted index method. This method creates a dictionary that maps words to the documents that contain them.","metadata":{}},{"cell_type":"code","source":"# the unique word set of the preprocessed_text of the dataset\nunique_word_set = set([])\n\ndef get_unique_words(row):\n    for word in row['preprocessed_text'].split():\n        # if not a single puctuation\n        if re.match(r'^[^\\w\\s]$', word) == None:\n            unique_word_set.add(word)\n\n\ntqdm.pandas(desc=\"Getting unique words...\")\nresult = df_sampled.progress_apply(lambda row: get_unique_words(row), axis=1)\n","metadata":{"execution":{"iopub.status.busy":"2023-04-05T06:44:56.491064Z","iopub.execute_input":"2023-04-05T06:44:56.491505Z","iopub.status.idle":"2023-04-05T06:45:32.731435Z","shell.execute_reply.started":"2023-04-05T06:44:56.491463Z","shell.execute_reply":"2023-04-05T06:45:32.729930Z"},"trusted":true},"execution_count":22,"outputs":[{"name":"stderr","text":"Getting unique words...: 100%|██████████| 10000/10000 [00:36<00:00, 276.04it/s]\n","output_type":"stream"}]},{"cell_type":"code","source":"# the inverted index of the preprocessed_text of the dataset\ninverted_index = inv_indx = {i: [] for i in unique_word_set}\n\n# function: get the inverted index of the preprocessed_text of the dataset\ndef get_inverted_index(row, inverted_index):\n    for word in row['preprocessed_text'].split():\n        if word in inverted_index:\n            inverted_index[word].append(row['cord_uid'])\n\n\ntqdm.pandas(desc=\"Getting inverted index...\")\nresult = df_sampled.progress_apply(\n    lambda row: get_inverted_index(row, inverted_index), axis=1)\n","metadata":{"execution":{"iopub.status.busy":"2023-04-05T06:45:39.965073Z","iopub.execute_input":"2023-04-05T06:45:39.965511Z","iopub.status.idle":"2023-04-05T06:47:48.793082Z","shell.execute_reply.started":"2023-04-05T06:45:39.965474Z","shell.execute_reply":"2023-04-05T06:47:48.791617Z"},"trusted":true},"execution_count":23,"outputs":[{"name":"stderr","text":"Getting inverted index...: 100%|██████████| 10000/10000 [02:06<00:00, 79.27it/s]\n","output_type":"stream"}]},{"cell_type":"code","source":"# get the documents containing the word \"mask\"\ninverted_index[\"mask\"]\n","metadata":{"execution":{"iopub.status.busy":"2023-04-05T06:48:04.246414Z","iopub.execute_input":"2023-04-05T06:48:04.246991Z","iopub.status.idle":"2023-04-05T06:48:04.272393Z","shell.execute_reply.started":"2023-04-05T06:48:04.246944Z","shell.execute_reply":"2023-04-05T06:48:04.271167Z"},"trusted":true},"execution_count":24,"outputs":[{"execution_count":24,"output_type":"execute_result","data":{"text/plain":"['jon0u3zi',\n 'jon0u3zi',\n 'jon0u3zi',\n '7f6xlts3',\n '7f6xlts3',\n 'x938j5rl',\n '8e8evwn4',\n '8e8evwn4',\n '1ykowy5k',\n '1ykowy5k',\n 'trufildm',\n 'trufildm',\n 'trufildm',\n 'trufildm',\n 'trufildm',\n 'trufildm',\n 'trufildm',\n 'trufildm',\n 'u1qb4t25',\n 'u1qb4t25',\n 'u1qb4t25',\n 'u1qb4t25',\n 'u1qb4t25',\n 'u1qb4t25',\n 'u1qb4t25',\n 'u1qb4t25',\n 'u1qb4t25',\n 'u1qb4t25',\n 'u1qb4t25',\n 'u1qb4t25',\n 'u1qb4t25',\n 'b9o6d24b',\n '75jxbqga',\n 'yfwpd3pe',\n 'mad91ezq',\n 'mad91ezq',\n 'mad91ezq',\n 'mad91ezq',\n 'igmhwm30',\n 'igmhwm30',\n 'gfqp659j',\n 'b55op5ep',\n 'b55op5ep',\n 'b55op5ep',\n 'eay6qfhz',\n 'eay6qfhz',\n 'eay6qfhz',\n 'eay6qfhz',\n 'eay6qfhz',\n 'eay6qfhz',\n 'eay6qfhz',\n 'eay6qfhz',\n 'eay6qfhz',\n 'eay6qfhz',\n 'eay6qfhz',\n 'eay6qfhz',\n 'eay6qfhz',\n 'eay6qfhz',\n 'eay6qfhz',\n 'eay6qfhz',\n 'eay6qfhz',\n 'eay6qfhz',\n 'eay6qfhz',\n 'eay6qfhz',\n 'eay6qfhz',\n 'eay6qfhz',\n 'eay6qfhz',\n 'eay6qfhz',\n 'eay6qfhz',\n 'eay6qfhz',\n 'eay6qfhz',\n 'eay6qfhz',\n 'eay6qfhz',\n 'eay6qfhz',\n 'eay6qfhz',\n 'eay6qfhz',\n 'eay6qfhz',\n 'eay6qfhz',\n 'eay6qfhz',\n 'eay6qfhz',\n 'eay6qfhz',\n 'eay6qfhz',\n 'eay6qfhz',\n 'eay6qfhz',\n 'eay6qfhz',\n 'eay6qfhz',\n 'eay6qfhz',\n 'eay6qfhz',\n 'eay6qfhz',\n 'eay6qfhz',\n 'eay6qfhz',\n 'eay6qfhz',\n 'eay6qfhz',\n 'eay6qfhz',\n 'eay6qfhz',\n 'eay6qfhz',\n 'eay6qfhz',\n 'eay6qfhz',\n 'eay6qfhz',\n 'eay6qfhz',\n 'eay6qfhz',\n 'eay6qfhz',\n 'eay6qfhz',\n 'eay6qfhz',\n 'eay6qfhz',\n 'eay6qfhz',\n 'eay6qfhz',\n 'eay6qfhz',\n 'eay6qfhz',\n 'eay6qfhz',\n 'eay6qfhz',\n 'eay6qfhz',\n 'eay6qfhz',\n 'eay6qfhz',\n 'eay6qfhz',\n 'eay6qfhz',\n 'eay6qfhz',\n 'eay6qfhz',\n 'tcgaacll',\n 'upu5kgm7',\n '2aib6jiw',\n 'uf1f9ypg',\n 'uf1f9ypg',\n 'uf1f9ypg',\n 'uf1f9ypg',\n 'udn90lso',\n 'udn90lso',\n 'udn90lso',\n 'udn90lso',\n 'udn90lso',\n 'udn90lso',\n 'udn90lso',\n 'udn90lso',\n 'udn90lso',\n '8qm6eqto',\n 'ydz7mpm0',\n 'ebwc3cc8',\n '0x4xqhoy',\n '0x4xqhoy',\n '0x4xqhoy',\n 'hb9l3gaq',\n 'chu0pxkf',\n 'qukldu77',\n 'vp279l3c',\n '4kpobeks',\n '4kpobeks',\n 'h1e5anwq',\n 'h1e5anwq',\n 'h1e5anwq',\n 'h1e5anwq',\n 'ntpt5is2',\n 'ntpt5is2',\n 'ppxdxdum',\n 'ppxdxdum',\n 'ppxdxdum',\n 'ppxdxdum',\n 'ppxdxdum',\n 'ppxdxdum',\n 'ppxdxdum',\n 'ppxdxdum',\n 'ppxdxdum',\n '0rxofyzq',\n 'cn75263g',\n 'n8ggv3ve',\n 'n8ggv3ve',\n '7937kyc2',\n '7937kyc2',\n '7937kyc2',\n '7937kyc2',\n '7937kyc2',\n 'hv82bivn',\n 'ismi7oxh',\n '2j8h8x3v',\n 'sbon3aes',\n '6zi88av8',\n 'gsxm3iiy',\n 'e2t6zq4x',\n 'w3zezn4d',\n '40aufihw',\n '40aufihw',\n '40aufihw',\n '40aufihw',\n '40aufihw',\n '40aufihw',\n '40aufihw',\n '40aufihw',\n '40aufihw',\n 'df65w0lx',\n 'df65w0lx',\n 'p6f7t9mp',\n 'p6f7t9mp',\n 'p6f7t9mp',\n 'p6f7t9mp',\n 'p6f7t9mp',\n 'p6f7t9mp',\n 'p6f7t9mp',\n 'p6f7t9mp',\n 'p6f7t9mp',\n 'p6f7t9mp',\n 'p6f7t9mp',\n 'p6f7t9mp',\n 'p6f7t9mp',\n 'p6f7t9mp',\n 'p6f7t9mp',\n 'p6f7t9mp',\n 'p6f7t9mp',\n 'p6f7t9mp',\n 'p6f7t9mp',\n 'p6f7t9mp',\n 'p6f7t9mp',\n 'p6f7t9mp',\n 'p6f7t9mp',\n 'p6f7t9mp',\n 'p6f7t9mp',\n 'p6f7t9mp',\n 'p6f7t9mp',\n 'p6f7t9mp',\n 'p6f7t9mp',\n 'p6f7t9mp',\n 'p6f7t9mp',\n 'p6f7t9mp',\n 'p6f7t9mp',\n 'p6f7t9mp',\n 'p6f7t9mp',\n 'p6f7t9mp',\n 'p6f7t9mp',\n 'p6f7t9mp',\n 'p6f7t9mp',\n 'p6f7t9mp',\n 'p6f7t9mp',\n 'p6f7t9mp',\n 'p6f7t9mp',\n 'p6f7t9mp',\n 'p6f7t9mp',\n 'p6f7t9mp',\n 'p6f7t9mp',\n 'p6f7t9mp',\n 'p6f7t9mp',\n 'oaj73szb',\n 'n7dhtm0t',\n 'yzrp054u',\n 'w1s0f1u0',\n 'w1s0f1u0',\n 'w1s0f1u0',\n 'squm46k9',\n 'squm46k9',\n 'squm46k9',\n 'squm46k9',\n 'squm46k9',\n 'squm46k9',\n 'squm46k9',\n 'kwqcoohp',\n 'kwqcoohp',\n 'kwqcoohp',\n 'kwqcoohp',\n 'kwqcoohp',\n 'kwqcoohp',\n 'kwqcoohp',\n 'kwqcoohp',\n 'kwqcoohp',\n 'kwqcoohp',\n 'kwqcoohp',\n 'kwqcoohp',\n 'kwqcoohp',\n 'kwqcoohp',\n 'kwqcoohp',\n 'kwqcoohp',\n 'kwqcoohp',\n 'ghmp34uk',\n '9kkhara2',\n '9kkhara2',\n '5hud7nf7',\n '5hud7nf7',\n '5hud7nf7',\n '5hud7nf7',\n '5hud7nf7',\n '5hud7nf7',\n '5hud7nf7',\n '5hud7nf7',\n '5hud7nf7',\n '5hud7nf7',\n '5hud7nf7',\n '5hud7nf7',\n '5hud7nf7',\n '5hud7nf7',\n '5hud7nf7',\n 'eg3vca2v',\n 'eg3vca2v',\n 'eg3vca2v',\n 'eg3vca2v',\n 'snsnbp06',\n 'mt2zaey2',\n 'mt2zaey2',\n 'mt2zaey2',\n 'mt2zaey2',\n 'mt2zaey2',\n 'mt2zaey2',\n 'mt2zaey2',\n 'mt2zaey2',\n 'mt2zaey2',\n 'mt2zaey2',\n 'mt2zaey2',\n 'mt2zaey2',\n 'mt2zaey2',\n 'mt2zaey2',\n 'mt2zaey2',\n 'mt2zaey2',\n 'mt2zaey2',\n 'mt2zaey2',\n 'mt2zaey2',\n 'mt2zaey2',\n 'mt2zaey2',\n 'mt2zaey2',\n 'mt2zaey2',\n 'mt2zaey2',\n 'mt2zaey2',\n 'mt2zaey2',\n 'mt2zaey2',\n 'mt2zaey2',\n 'mt2zaey2',\n 'mt2zaey2',\n 'mt2zaey2',\n 'mt2zaey2',\n 'mt2zaey2',\n 'mt2zaey2',\n 'mt2zaey2',\n 'mt2zaey2',\n 'mt2zaey2',\n 'mt2zaey2',\n 'mt2zaey2',\n 'mt2zaey2',\n 'mt2zaey2',\n 'mt2zaey2',\n 'mt2zaey2',\n 'mt2zaey2',\n 'mt2zaey2',\n 'mt2zaey2',\n 'mt2zaey2',\n 'mt2zaey2',\n 'mt2zaey2',\n 'mt2zaey2',\n 'mt2zaey2',\n 'mt2zaey2',\n 'mt2zaey2',\n 't9qpdoht',\n '5xsnfmxb',\n 'cxsdrrfn',\n 'cxsdrrfn',\n 'cxsdrrfn',\n 'cxsdrrfn',\n 'cxsdrrfn',\n 'cxsdrrfn',\n 'cxsdrrfn',\n '70qmpccs',\n '70qmpccs',\n '70qmpccs',\n '70qmpccs',\n '70qmpccs',\n 'ai18bt45',\n 'p5ifp1z3',\n '70c8rq5p',\n '30unk2c9',\n '30unk2c9',\n '7rbr53e5',\n '7rbr53e5',\n 'nx5s8xpj',\n 'nx5s8xpj',\n 'ndnztxov',\n 'o85vaz60',\n '8vinenwa',\n '8vinenwa',\n '8vinenwa',\n '8vinenwa',\n '8vinenwa',\n 'li851emv',\n 'rxd5w5kh',\n 'rxd5w5kh',\n 'rxd5w5kh',\n 'rxd5w5kh',\n '005mjewr',\n '005mjewr',\n '005mjewr',\n '005mjewr',\n 'qk0lgfiv',\n 'qk0lgfiv',\n 'rql80x4t',\n 'nzfsff7r',\n 'nzfsff7r',\n 'nzfsff7r',\n 'bdgjiqxr',\n 'bdgjiqxr',\n 'bdgjiqxr',\n 'gmhpekok',\n 'gmhpekok',\n 'm0spj5kj',\n 'c4amzro3',\n 'c4amzro3',\n 'fbw9x3ik',\n '7g7fkmuv',\n '7g7fkmuv',\n '7g7fkmuv',\n '7g7fkmuv',\n 'l4sv28hj',\n 'f3mvdpq1',\n 'uogabu1s',\n 'uogabu1s',\n 'uogabu1s',\n 'uogabu1s',\n 'uogabu1s',\n 'uogabu1s',\n 'uogabu1s',\n 'uogabu1s',\n 'uogabu1s',\n 'fka8c69p',\n 'hgj1tmc3',\n 'qk2t88qs',\n 'qk2t88qs',\n 'qk2t88qs',\n '5a9s1imr',\n '5a9s1imr',\n 's477tw3x',\n 's477tw3x',\n 's477tw3x',\n 's477tw3x',\n 's477tw3x',\n 's477tw3x',\n 's477tw3x',\n 's477tw3x',\n 's477tw3x',\n 's477tw3x',\n 's477tw3x',\n 's477tw3x',\n 's477tw3x',\n 's477tw3x',\n 's477tw3x',\n 's477tw3x',\n 's477tw3x',\n 's477tw3x',\n 's477tw3x',\n 's477tw3x',\n 's477tw3x',\n 's477tw3x',\n 's477tw3x',\n 's477tw3x',\n 's477tw3x',\n 's477tw3x',\n 's477tw3x',\n 's477tw3x',\n 's477tw3x',\n 's477tw3x',\n 's477tw3x',\n 's477tw3x',\n 's477tw3x',\n 's477tw3x',\n 's477tw3x',\n 's477tw3x',\n 's477tw3x',\n 's477tw3x',\n 's477tw3x',\n 's477tw3x',\n 's477tw3x',\n 's477tw3x',\n 's477tw3x',\n 's477tw3x',\n 's477tw3x',\n 's477tw3x',\n 's477tw3x',\n 's477tw3x',\n 's477tw3x',\n 's477tw3x',\n 'iwd82psd',\n 'x0m6lcn0',\n 'x0m6lcn0',\n 'x0m6lcn0',\n 'x0m6lcn0',\n '5s7p9fai',\n '4omocd8q',\n 'p3ek0qdi',\n 'me08w1ir',\n 'me08w1ir',\n 'me08w1ir',\n 'hhx0ozm3',\n 'hhx0ozm3',\n 'hhx0ozm3',\n 'hhx0ozm3',\n 'hhx0ozm3',\n 'hhx0ozm3',\n 'hhx0ozm3',\n 'hhx0ozm3',\n 'yptjzbzw',\n '8tqm4k6k',\n '8tqm4k6k',\n '8tqm4k6k',\n 'alop6lts',\n 'alop6lts',\n 'alop6lts',\n 'alop6lts',\n 'alop6lts',\n 'alop6lts',\n 'alop6lts',\n 'alop6lts',\n 'alop6lts',\n 'alop6lts',\n 'alop6lts',\n 'alop6lts',\n 'alop6lts',\n 'alop6lts',\n 'alop6lts',\n 'alop6lts',\n 'alop6lts',\n '3fbkv8r5',\n '3fbkv8r5',\n '3fbkv8r5',\n 'n2ye55zq',\n 'n2ye55zq',\n 'n2ye55zq',\n 'n2ye55zq',\n 'n2ye55zq',\n 'n2ye55zq',\n 'n2ye55zq',\n 'n2ye55zq',\n 'n2ye55zq',\n 'n2ye55zq',\n 'n2ye55zq',\n 'n2ye55zq',\n 'n2ye55zq',\n 'n2ye55zq',\n 'n2ye55zq',\n 'n2ye55zq',\n 'n2ye55zq',\n 'n2ye55zq',\n 'n2ye55zq',\n 'n2ye55zq',\n 'n2ye55zq',\n 'n2ye55zq',\n 'n2ye55zq',\n 'n2ye55zq',\n 'n2ye55zq',\n 'n2ye55zq',\n 'n2ye55zq',\n 'n2ye55zq',\n 'n2ye55zq',\n 'n2ye55zq',\n 'n2ye55zq',\n 'n2ye55zq',\n 'n2ye55zq',\n 'n2ye55zq',\n 'n2ye55zq',\n 'n2ye55zq',\n 'n2ye55zq',\n 'n2ye55zq',\n 'n2ye55zq',\n 'n2ye55zq',\n 'n2ye55zq',\n 'n2ye55zq',\n 'n2ye55zq',\n 'n2ye55zq',\n 'n2ye55zq',\n 'n2ye55zq',\n 'n2ye55zq',\n 'n2ye55zq',\n 'n2ye55zq',\n 'n2ye55zq',\n 'uky73x31',\n 'uky73x31',\n 'uky73x31',\n 'uky73x31',\n 'uky73x31',\n 'tpplbucu',\n 'tpplbucu',\n 'mt6rc3if',\n 'mt6rc3if',\n 'p7shv1iw',\n 'p7shv1iw',\n 'p7shv1iw',\n '65wtt18s',\n '65wtt18s',\n '65wtt18s',\n '65wtt18s',\n '65wtt18s',\n '65wtt18s',\n '65wtt18s',\n '65wtt18s',\n '65wtt18s',\n '65wtt18s',\n '65wtt18s',\n '65wtt18s',\n 'sp0577vw',\n 'sp0577vw',\n 'sp0577vw',\n 'sp0577vw',\n 'sp0577vw',\n 'sp0577vw',\n 'sp0577vw',\n 'sp0577vw',\n 'sp0577vw',\n 'sp0577vw',\n 'sp0577vw',\n 'sp0577vw',\n '2p2qat1f',\n '2p2qat1f',\n '2p2qat1f',\n '2p2qat1f',\n '2p2qat1f',\n '2p2qat1f',\n '2p2qat1f',\n '2p2qat1f',\n '2p2qat1f',\n '2p2qat1f',\n '2p2qat1f',\n '2p2qat1f',\n '2p2qat1f',\n '2vd658mj',\n 'n98w3vne',\n 'c9lt8kah',\n 'lr01djif',\n 'lr01djif',\n 'lr01djif',\n '9g275kmt',\n '9g275kmt',\n 'a8x4bqxb',\n 'a8x4bqxb',\n 'a8x4bqxb',\n 'a8x4bqxb',\n 'a8x4bqxb',\n 'a8x4bqxb',\n 'a8x4bqxb',\n 'a8x4bqxb',\n 'a8x4bqxb',\n 'a8x4bqxb',\n 'a8x4bqxb',\n 'a8x4bqxb',\n '1yf1ae1d',\n '1yf1ae1d',\n '1yf1ae1d',\n '1yf1ae1d',\n '1yf1ae1d',\n '1yf1ae1d',\n '1yf1ae1d',\n '1yf1ae1d',\n '1yf1ae1d',\n '1yf1ae1d',\n '1yf1ae1d',\n '1yf1ae1d',\n '1yf1ae1d',\n '1yf1ae1d',\n '1yf1ae1d',\n '1yf1ae1d',\n '1yf1ae1d',\n '1yf1ae1d',\n '1yf1ae1d',\n '1yf1ae1d',\n '1yf1ae1d',\n '1yf1ae1d',\n '1yf1ae1d',\n '1yf1ae1d',\n '1yf1ae1d',\n '1yf1ae1d',\n '1yf1ae1d',\n '1yf1ae1d',\n '1yf1ae1d',\n '1yf1ae1d',\n '1yf1ae1d',\n '1yf1ae1d',\n '1yf1ae1d',\n '1yf1ae1d',\n '1yf1ae1d',\n '1yf1ae1d',\n '1yf1ae1d',\n '1yf1ae1d',\n '1yf1ae1d',\n '1yf1ae1d',\n '1yf1ae1d',\n '1yf1ae1d',\n '1yf1ae1d',\n '1yf1ae1d',\n '1yf1ae1d',\n '1yf1ae1d',\n '1yf1ae1d',\n '1yf1ae1d',\n '1yf1ae1d',\n '1yf1ae1d',\n '1yf1ae1d',\n '1yf1ae1d',\n '1yf1ae1d',\n '1yf1ae1d',\n '1yf1ae1d',\n '1yf1ae1d',\n '1yf1ae1d',\n '1yf1ae1d',\n '1yf1ae1d',\n '1yf1ae1d',\n '1yf1ae1d',\n '1yf1ae1d',\n '1yf1ae1d',\n '2swl8zow',\n 'n61gz7pi',\n 'n61gz7pi',\n 'n61gz7pi',\n 'qyha7vdy',\n 'qyha7vdy',\n 'kd01q6wu',\n 'h0j8ldig',\n 'fettqyj6',\n '2vqawwwr',\n '2vqawwwr',\n '2vqawwwr',\n '2vqawwwr',\n '2vqawwwr',\n '2vqawwwr',\n '2vqawwwr',\n '2vqawwwr',\n '2vqawwwr',\n '2vqawwwr',\n 'f9nxajw2',\n 'f9nxajw2',\n 'm9icky9z',\n 'cigq1iin',\n 'cigq1iin',\n 'cigq1iin',\n 'cigq1iin',\n 'cigq1iin',\n 'cigq1iin',\n 'cigq1iin',\n 'cigq1iin',\n 'xjdiur42',\n 'jqcqkmav',\n 'pnxzkn7a',\n 'pnxzkn7a',\n 'e0gw285x',\n 'e0gw285x',\n 'e0gw285x',\n 'e0gw285x',\n 'e0gw285x',\n 'e0gw285x',\n 'jceqglyz',\n 'jceqglyz',\n 'jceqglyz',\n 'jceqglyz',\n 'jceqglyz',\n 'at3qo0ck',\n 'at3qo0ck',\n 'at3qo0ck',\n 'at3qo0ck',\n 'at3qo0ck',\n 'at3qo0ck',\n 'at3qo0ck',\n 'at3qo0ck',\n 'cs22l843',\n 'fmxaxfu3',\n 'ygehanhi',\n 'ygehanhi',\n '3e1ji2mw',\n 'xagjboqy',\n 'c8jy12jj',\n 'c8jy12jj',\n '9fzq45nm',\n '20vtnvmv',\n '20vtnvmv',\n '20vtnvmv',\n '0mwz8kwo',\n '0mwz8kwo',\n '0mwz8kwo',\n '0mwz8kwo',\n '0mwz8kwo',\n 'kuznnm0k',\n '5tzwoa4z',\n '5595sdzf',\n 'v45da786',\n 'v45da786',\n 'k00yauqz',\n 'k00yauqz',\n 'k00yauqz',\n 'k00yauqz',\n 'nhisb47i',\n '2m7z1pvp',\n '2m7z1pvp',\n '2m7z1pvp',\n '4i32jv9i',\n '4i32jv9i',\n '4i32jv9i',\n '4i32jv9i',\n '4i32jv9i',\n '4i32jv9i',\n '4i32jv9i',\n '4i32jv9i',\n '4i32jv9i',\n '4i32jv9i',\n '4i32jv9i',\n '4i32jv9i',\n '357tgplw',\n '357tgplw',\n '9epz06jk',\n '9epz06jk',\n 'hxrrvqtr',\n 'hxrrvqtr',\n 'hxrrvqtr',\n 'hxrrvqtr',\n 'hxrrvqtr',\n 'fdbmpar5',\n 'fdbmpar5',\n 'fdbmpar5',\n 'fdbmpar5',\n 'fdbmpar5',\n 'fdbmpar5',\n 'fdbmpar5',\n '4hlkxf7v',\n 'sqdsnub3',\n 'nb9osjcf',\n 'nb9osjcf',\n 'nb9osjcf',\n 'nb9osjcf',\n 'nb9osjcf',\n 'nb9osjcf',\n 'nb9osjcf',\n 'nb9osjcf',\n 'nb9osjcf',\n 'nb9osjcf',\n 'nb9osjcf',\n 'nb9osjcf',\n 'nb9osjcf',\n 'nb9osjcf',\n 'nb9osjcf',\n 'nb9osjcf',\n 'nb9osjcf',\n 'nb9osjcf',\n 'nb9osjcf',\n 'nb9osjcf',\n 'nb9osjcf',\n 'nb9osjcf',\n 'nb9osjcf',\n 'nb9osjcf',\n 'nb9osjcf',\n 'nb9osjcf',\n 'nb9osjcf',\n 'nb9osjcf',\n 'nb9osjcf',\n 'nb9osjcf',\n 'nb9osjcf',\n 'nb9osjcf',\n 'nb9osjcf',\n 'nb9osjcf',\n 'nb9osjcf',\n 'nb9osjcf',\n 'g6n8djem',\n 'g6n8djem',\n 'g6n8djem',\n 'g6n8djem',\n 'g6n8djem',\n 'xmck9pzs',\n 'xmck9pzs',\n 'xmck9pzs',\n 'xmck9pzs',\n 'xmck9pzs',\n 'xmck9pzs',\n 'xmck9pzs',\n 'xmck9pzs',\n 'xmck9pzs',\n 'xmck9pzs',\n '9mghf0q7',\n '7urdbldm',\n 'ac19ct3b',\n '5xktserz',\n '5xktserz',\n '5xktserz',\n '5xktserz',\n '5xktserz',\n 'v7s0v024',\n 'ouc1owpw',\n 'ouc1owpw',\n 'ouc1owpw',\n 'ouc1owpw',\n 'ouc1owpw',\n '2bm7mofj',\n '2ofdqmb2',\n 'vc4wg9hr',\n 'vc4wg9hr',\n 'vc4wg9hr',\n 'vc4wg9hr',\n 'vc4wg9hr',\n 'vc4wg9hr',\n 'vc4wg9hr',\n 'vc4wg9hr',\n 'vc4wg9hr',\n 'vc4wg9hr',\n 'vc4wg9hr',\n 'vc4wg9hr',\n 'vc4wg9hr',\n 'vc4wg9hr',\n 'vc4wg9hr',\n 'vc4wg9hr',\n 'vc4wg9hr',\n 'vc4wg9hr',\n 'vc4wg9hr',\n 'vc4wg9hr',\n 'x05mscyk',\n 'x05mscyk',\n 'x05mscyk',\n 'kx5iwu9r',\n 'kx5iwu9r',\n 'iwe0w2ke',\n '4ejcmts0',\n 'a2d0phjs',\n 'a2d0phjs',\n 'a2d0phjs',\n 'a2d0phjs',\n 'a2d0phjs',\n 'a2d0phjs',\n 'fux10x0w',\n '4zrvpbe9',\n 'e9bu6b8u',\n 'e9bu6b8u',\n 'e9bu6b8u',\n 'e9bu6b8u',\n 'e9bu6b8u',\n 'k06wt8yb',\n 'orsw8z2l',\n 'orsw8z2l',\n 'orsw8z2l',\n 'orsw8z2l',\n 'orsw8z2l',\n 'orsw8z2l',\n 'orsw8z2l',\n 'orsw8z2l',\n 'orsw8z2l',\n 'orsw8z2l',\n 'orsw8z2l',\n 'orsw8z2l',\n 'orsw8z2l',\n 'orsw8z2l',\n 'dz9b0o2j',\n 'dz9b0o2j',\n 'dz9b0o2j',\n 'dz9b0o2j',\n 'eymqlbix',\n 'q5x6a2b0',\n '1ueh4vu3',\n '1ueh4vu3',\n 'otlqhylf',\n 'otlqhylf',\n '1ycj3b66',\n '1ycj3b66',\n '1ycj3b66',\n '1ycj3b66',\n '1ycj3b66',\n '1ycj3b66',\n '1ycj3b66',\n '1ycj3b66',\n '1ycj3b66',\n '1ycj3b66',\n '1ycj3b66',\n '1ycj3b66',\n '1ycj3b66',\n '1ycj3b66',\n '1ycj3b66',\n '1ycj3b66',\n '1ycj3b66',\n '1ycj3b66',\n '1ycj3b66',\n '1ycj3b66',\n '1ycj3b66',\n '1ycj3b66',\n '1ycj3b66',\n '1ycj3b66',\n '1ycj3b66',\n '1ycj3b66',\n '1ycj3b66',\n '1ycj3b66',\n '1ycj3b66',\n '1ycj3b66',\n '1ycj3b66',\n '1ycj3b66',\n '1ycj3b66',\n '1ycj3b66',\n '1ycj3b66',\n '1ycj3b66',\n '1ycj3b66',\n '1ycj3b66',\n '1ycj3b66',\n '1ycj3b66',\n '1ycj3b66',\n '1ycj3b66',\n '1ycj3b66',\n '1ycj3b66',\n '1ycj3b66',\n '1ycj3b66',\n '1ycj3b66',\n '1ycj3b66',\n '1ycj3b66',\n '1ycj3b66',\n '1ycj3b66',\n '1ycj3b66',\n '1ycj3b66',\n '1ycj3b66',\n '1ycj3b66',\n '1ycj3b66',\n 'qo3cfytq',\n 'qo3cfytq',\n 'ctty9kl4',\n 'ctty9kl4',\n ...]"},"metadata":{}}]},{"cell_type":"markdown","source":"### 4. Text matching utility","metadata":{}},{"cell_type":"markdown","source":"To obtain the answer snippets to a query, we will first extend the query by adding related entities from the knowledge base. Next, we will use the inverted index method to locate documents containing the words from the extended query. We will then calculate cosine similarity between the query and the documents to rank them and select the top 5 as the target documents. Finally, we will again use cosine similarity to rank the paragraphs in the target documents and extract the top 3 paragraphs as answer snippets.","metadata":{}},{"cell_type":"markdown","source":"First, we define the function to preprocess the query.","metadata":{}},{"cell_type":"code","source":"# function: extend the query with the knowledge base\ndef extend_query(query, knowledge_base_synonym, knowledge_base_association):\n    # convert the query to lower case\n    query = query.lower()\n    # for each key in the knowledge base, if the key is in the query, then add the value to the query string\n    for key in knowledge_base_synonym:\n        # if the query string contains the key\n        if key.lower() in query:\n            # concatenate the query string with each value of the key\n            for value in knowledge_base_synonym[key]:\n                query += \" \" + value\n    for key in knowledge_base_association:\n        if key.lower() in query:\n            for value in knowledge_base_association[key]:\n                query += \" \" + value\n    return query\n\n# query preprocessing: extend the query with the knowledge base, remove stopwords, lemmatization, and remove question mark\ndef preprocess_query(query):\n    # extend the query with the knowledge base\n    query = extend_query(query, knowledge_base_synonym,\n                         knowledge_base_association)\n    query = query.lower()\n    # remove stopwords\n    query = \" \".join([word for word in query.split()\n                     if word not in stop_words])\n    # lemmatization\n    lemmatizer = nltk.stem.WordNetLemmatizer()\n    query = \" \".join([lemmatizer.lemmatize(word) for word in query.split()])\n    # remove question mark\n    query = query.replace(\"?\", \"\")\n    return query\n","metadata":{"execution":{"iopub.status.busy":"2023-04-05T06:48:12.862004Z","iopub.execute_input":"2023-04-05T06:48:12.862426Z","iopub.status.idle":"2023-04-05T06:48:12.873259Z","shell.execute_reply.started":"2023-04-05T06:48:12.862391Z","shell.execute_reply":"2023-04-05T06:48:12.871619Z"},"trusted":true},"execution_count":25,"outputs":[]},{"cell_type":"markdown","source":"Next, we need to define the function to embed the query and the documents.","metadata":{}},{"cell_type":"code","source":"# function: get the vector representation of the query by averaging the vector representation of each word in the query\ndef embedding_string(string):\n    vectors = []\n    for word in string.split():\n        if word in glove_model:\n            vectors.append(glove_model[word])\n    if vectors == []:\n        return np.zeros(100)\n    return np.mean(vectors, axis=0)\n\n# get the vector representation of each document in df_sampled\ndf_sampled['embedding'] = df_sampled.progress_apply(\n    lambda row: embedding_string(row['preprocessed_text']), axis=1)\n\ndf_sampled['embedding']","metadata":{"execution":{"iopub.status.busy":"2023-04-05T06:48:15.988071Z","iopub.execute_input":"2023-04-05T06:48:15.988499Z","iopub.status.idle":"2023-04-05T06:49:44.183093Z","shell.execute_reply.started":"2023-04-05T06:48:15.988461Z","shell.execute_reply":"2023-04-05T06:49:44.181812Z"},"trusted":true},"execution_count":26,"outputs":[{"name":"stderr","text":"Getting inverted index...: 100%|██████████| 10000/10000 [01:28<00:00, 113.42it/s]\n","output_type":"stream"},{"execution_count":26,"output_type":"execute_result","data":{"text/plain":"753376     [-0.062524386, 0.20725419, 0.11635797, 0.14362...\n913530     [-0.08254339, 0.176392, 0.11337998, 0.13646367...\n712304     [-0.0043793595, 0.18165466, -0.0017466267, 0.1...\n736570     [-0.045271114, 0.20421785, 0.08802674, 0.15633...\n816108     [-0.06078614, 0.21743849, 0.14221364, 0.168882...\n                                 ...                        \n768690     [-0.00047997944, 0.033717453, 0.10655, -0.0020...\n737001     [-0.03618455, 0.11873989, 0.016081613, 0.05789...\n1004462    [-0.06748906, 0.27562916, 0.14938413, 0.153400...\n79013      [-0.13417636, -0.071013786, -0.23611803, -0.13...\n25116      [-0.002016765, 0.18712643, 0.1122159, 0.080348...\nName: embedding, Length: 10000, dtype: object"},"metadata":{}}]},{"cell_type":"markdown","source":"#### 4.1 Cosine similarity","metadata":{}},{"cell_type":"code","source":"# function: calculate the cosine similarity between two vectors\ndef cosine_similarity(vector1, vector2):\n    return np.dot(vector1, vector2) / (np.linalg.norm(vector1) * np.linalg.norm(vector2))","metadata":{"execution":{"iopub.status.busy":"2023-04-05T06:49:46.447054Z","iopub.execute_input":"2023-04-05T06:49:46.447509Z","iopub.status.idle":"2023-04-05T06:49:46.454166Z","shell.execute_reply.started":"2023-04-05T06:49:46.447466Z","shell.execute_reply":"2023-04-05T06:49:46.453162Z"},"trusted":true},"execution_count":27,"outputs":[]},{"cell_type":"markdown","source":"#### 4.2 Get the top n documents or paragraphs","metadata":{}},{"cell_type":"code","source":"# function: get the top n documents that are most similar to the query from the indexed document set\ndef get_top_n_documents(query_vector, indexed_document_set, n):\n    # get the vector representation of each document in the indexed document set\n    df_indexed = df_sampled[df_sampled['cord_uid'].isin(indexed_document_set)]\n    # calculate the cosine similarity between the query and each document in the indexed document set\n    df_indexed['similarity'] = df_indexed.progress_apply(\n        lambda row: cosine_similarity(query_vector, row['embedding']), axis=1)\n    # sort the documents by the similarity score\n    df_indexed = df_indexed.sort_values(by=['similarity'], ascending=False)\n    # get the top n documents\n    df_top_n = df_indexed[:n]\n    return df_top_n\n","metadata":{"execution":{"iopub.status.busy":"2023-04-05T06:49:48.312398Z","iopub.execute_input":"2023-04-05T06:49:48.312837Z","iopub.status.idle":"2023-04-05T06:49:48.320921Z","shell.execute_reply.started":"2023-04-05T06:49:48.312800Z","shell.execute_reply":"2023-04-05T06:49:48.319577Z"},"trusted":true},"execution_count":28,"outputs":[]},{"cell_type":"code","source":"# function: get the top n paragraphs that are most similar to the query for each document in the top n documents\ndef get_top_n_paragraphs(query_vector, df_top_n_doc, n):\n    top_n_paragraphs_dict = {}\n    for index, row in df_top_n_doc.iterrows():\n        # get the paragraphs of the article\n        paragraph_list = {\"paragraph\": [], \"similarity\": []}\n        paragraph_list[\"paragraph\"].append(row[\"abstract\"])\n\n        # if the conclusion is not null, then add it to the paragraph list\n        if pd.isnull(row[\"conclusion\"]) == False:\n            paragraph_list[\"paragraph\"].append(row[\"conclusion\"])\n\n        if pd.isnull(row[\"introduction\"]) == False:\n            paragraph_list[\"paragraph\"].append(row[\"introduction\"])\n\n        paragraph_list[\"paragraph\"].extend(row[\"other_paragraph\"])\n\n        # calculate the cosine similarity between the query and each paragraph\n        for paragraph in paragraph_list[\"paragraph\"]:\n            paragraph_list[\"similarity\"].append(cosine_similarity(\n                query_vector, embedding_string(paragraph)))\n        \n        df_paragraph_list = pd.DataFrame(paragraph_list)\n        # sort the paragraphs by the similarity score\n        df_paragraph_list = df_paragraph_list.sort_values(\n            by=['similarity'], ascending=False)\n        top_n_paragraphs_dict[row[\"cord_uid\"]] = df_paragraph_list[:n]\n    return top_n_paragraphs_dict\n","metadata":{"execution":{"iopub.status.busy":"2023-04-05T06:49:50.557136Z","iopub.execute_input":"2023-04-05T06:49:50.558187Z","iopub.status.idle":"2023-04-05T06:49:50.568415Z","shell.execute_reply.started":"2023-04-05T06:49:50.558144Z","shell.execute_reply":"2023-04-05T06:49:50.566600Z"},"trusted":true},"execution_count":29,"outputs":[]},{"cell_type":"markdown","source":"#### 4.3 Get the final answer of a query","metadata":{}},{"cell_type":"code","source":"# function: integrate the above functions to get the answer of the query\ndef get_answer(query, n):\n    # preprocess the query\n    query = preprocess_query(query)\n    # get the indexed document set\n    indexed_document_set = set([])\n    for word in query.split():\n        if word in inverted_index:\n            for document in inverted_index[word]:\n                indexed_document_set.add(document)\n    # get the vector representation of the query\n    query_vector = embedding_string(query)\n    # get the top n documents that are most similar to the query\n    df_top_n = get_top_n_documents(query_vector, indexed_document_set, n)\n    # get the top n paragraphs that are most similar to the query for each document in the top n documents\n    top_n_paragraphs_dict = get_top_n_paragraphs(query_vector, df_top_n, 3)\n    return top_n_paragraphs_dict\n","metadata":{"execution":{"iopub.status.busy":"2023-04-05T06:49:54.349182Z","iopub.execute_input":"2023-04-05T06:49:54.349622Z","iopub.status.idle":"2023-04-05T06:49:54.358215Z","shell.execute_reply.started":"2023-04-05T06:49:54.349575Z","shell.execute_reply":"2023-04-05T06:49:54.356759Z"},"trusted":true},"execution_count":31,"outputs":[]},{"cell_type":"markdown","source":"### 5. Test utility and test results","metadata":{}},{"cell_type":"code","source":"# the test set with 10 queries\ntest_set = [\n    \"what is the origin of COVID-19?\",\n    \"what types of rapid testing for Covid-19 have been developed?\",\n    \"has social distancing had an impact on slowing the spread of COVID-19?\",\n    \"what are the transmission routes of coronavirus?\",\n    \"what are the best masks for preventing infection by Covid-19?\",\n    \"what type of hand sanitizer is needed to destroy Covid-19?\",\n    \"What vaccine candidates are being tested for Covid-19?\",\n    \"does Vitamin D impact COVID-19 prevention and treatment?\",\n    \"how long can the coronavirus live outside the body?\",\n    \"what are the initial symptoms of Covid-19?\",\n]\n","metadata":{"execution":{"iopub.status.busy":"2023-04-05T06:50:00.236850Z","iopub.execute_input":"2023-04-05T06:50:00.237273Z","iopub.status.idle":"2023-04-05T06:50:00.244774Z","shell.execute_reply.started":"2023-04-05T06:50:00.237238Z","shell.execute_reply":"2023-04-05T06:50:00.243141Z"},"trusted":true},"execution_count":33,"outputs":[]},{"cell_type":"code","source":"# present the top_n_paragraphs_dict in a more readable way\ndef present_answer(top_n_paragraphs_dict):\n    for key in top_n_paragraphs_dict:\n        # print the document title\n        print(\"Document: \" +\n              df_sampled[df_sampled[\"cord_uid\"] == key][\"title\"].values[0])\n        for index, row in top_n_paragraphs_dict[key].iterrows():\n            print(\"Paragraph: \" + row[\"paragraph\"])\n            print(\"Confidence: \" + str(row[\"similarity\"]))\n            print(\"\")\n        print(\"---------------------------------------------------------------------------------------------\")\n","metadata":{"execution":{"iopub.status.busy":"2023-04-05T06:50:02.125321Z","iopub.execute_input":"2023-04-05T06:50:02.125786Z","iopub.status.idle":"2023-04-05T06:50:02.133663Z","shell.execute_reply.started":"2023-04-05T06:50:02.125745Z","shell.execute_reply":"2023-04-05T06:50:02.132271Z"},"trusted":true},"execution_count":34,"outputs":[]},{"cell_type":"code","source":"for i in range(len(test_set)):\n    print(\"Question \" + str(i + 1) + \": \" + test_set[i])\n    present_answer(get_answer(test_set[i], 5))\n    print(\"\")\n    print(\"=================================================================================================\")\n    print(\"=================================================================================================\")\n    print(\"\")\n    print(\"\")\n","metadata":{"_kg_hide-output":false,"_kg_hide-input":false,"execution":{"iopub.status.busy":"2023-04-05T06:50:07.974003Z","iopub.execute_input":"2023-04-05T06:50:07.974563Z","iopub.status.idle":"2023-04-05T06:50:12.330121Z","shell.execute_reply.started":"2023-04-05T06:50:07.974500Z","shell.execute_reply":"2023-04-05T06:50:12.328793Z"},"trusted":true},"execution_count":35,"outputs":[{"name":"stdout","text":"Question 1: what is the origin of COVID-19?\n","output_type":"stream"},{"name":"stderr","text":"Getting inverted index...: 100%|██████████| 9990/9990 [00:00<00:00, 32431.44it/s]\n","output_type":"stream"},{"name":"stdout","text":"Document: Immune response in COVID-19: What do we currently know?\nParagraph: The acute respiratory syndrome is a disease caused by the SARS-CoV-2 virus (COVID-37 19), where symptoms include difficulty breathing, high fever, and cough (WHO, 2020). \nConfidence: 0.6983451843261719\n\nParagraph: In 2002/2003 there was a pandemic denominate SARS (severe acute respiratory syndrome), caused by the SARS-CoV virus that belongs to the genera Betacoranavirus and the family Coronaviridae, generally responsible for influenza infections. In mid of 2019, a new disease by the coronavirus named by COVID-19 (SARS-CoV-2) emerged, both infections have flu symptoms, however they are infections that variable intensity, being medium to severe. In medium infections individuals have the virus and exhibit symptoms, however hospitalization is not necessary, in severe infections, individuals are hospitalized, have high pathology and in some cases progress to death. The virus is formed by simple positive RNA, enveloped, non-segmented, and presenting the largest genome of viruses constituting 32 Kb, consisting of envelope proteins, membrane, nucleocapsid and spike protein, which is essential in the interaction with the host cells. As for the origin of this virus, research has been intensified to determine this paradox and although the similarity with SARS-CoV, this virus did not has necessarily the same place of origin. As for the immune system, it is currently unknown how this new virus interacts. In this brief review, we demonstrate important considerations about the responses to this infection.\nConfidence: 0.595037579536438\n\n---------------------------------------------------------------------------------------------\nDocument: Novel coronavirus disease 2019 (COVID‐19) and neurodegenerative disorders\nParagraph: During the last few months, the whole humanity is experiencing largest and most severe sudden influx of COVID‐19 outbreak caused by the novel coronavirus (CoV) originated from Wuhan, China. According to the WHO reports, total 3 862 676 positive cases and 265 961 deaths have been recorded worldwide due to COVID‐19 infection as of May 9, 2020. CoVs are a large family of viruses (enveloped, single‐stranded RNA viruses), which includes severe acute respiratory syndrome coronavirus (SARS‐CoV) and Middle East respiratory syndrome‐related coronavirus (MERS‐CoV). New SARS CoV2 is the members of Betacoronavirus genus. These viruses cause infections in bats, camels and humans, and a few other associated species. Despite many neurologic complications associated with SARS‐CoV‐2 infection, it is still unclear whether these symptoms results from direct neural injury or due to some other reason. Currently, it appears that most of the neurological symptoms of COVID‐19 are nonspecific and secondary to the systemic illness. A single case of acute hemorrhagic necrotizing encephalopathy has been reported. SARS‐CoV‐2 associated Guillain‐Barré syndrome is an atypical case. Till today, no convincing evidence is available to confirm that the SARS‐CoV‐2 virus directly affects nerves system in humans. However, postinfection surveillance will be necessary to identify the possible post‐COVID‐19 neurologic syndromes.\nConfidence: 0.6314988732337952\n\nParagraph: During the last few months, the whole humanity is experiencing the largest and most severe sudden influx of COVID-19 outbreak caused by the novel coronavirus (CoV) originated from Wuhan city.\nConfidence: 0.6112067103385925\n\nParagraph: Coronavirus was officially announced as the causative pathogen of COVID-19 by the Chinese Center for Disease Control and Prevention on January 8, 2020. The World Health Organization announced coronavirus outbreak as a public health emergency of international concern. It started to expect disturbing extents over the globe as forewarned by the World Health Organization (WHO) reports associated Guillain-Barré syndrome is an atypical case. Till today, no convincing evidence is available to confirm that the SARS-CoV-2 virus directly affects the CNS or PNS in humans. However, postinfection surveillance will be necessary to identify the possible post-COVID-19 neurologic syndromes.\nConfidence: 0.5646367073059082\n\n---------------------------------------------------------------------------------------------\nDocument: The First Disease X is Caused by a Highly Transmissible Acute Respiratory Syndrome Coronavirus\nParagraph: On February 9, 2018, the World Health Organization (WHO) announced the Blueprint list of priority diseases, including Middle East respiratory syndrome (MERS) and Severe Acute Respiratory Syndrome (SARS), as well as Disease X, for research and development in emergency contexts. Disease X would be a new disease with an epidemic or pandemic potential caused by an unknown pathogen (www.who.int/activities/prioritizing-diseasesfor-research-and-development-in-emergency-context). At that time, we believed that the first Disease X could be a transmissible infectious disease caused by a novel coronavirus originated from bats. This supposition was based on a live SARS-related coronavirus (SARSr-CoV), designated SARSr-CoV-WIV1, isolated from bat fecal samples in Vero E6 cells. It had 99.9% whole genome sequence identity to that of SARSr-CoV-Rs3367 identified from Chinese horseshoe bats and another novel strain, SARSr-CoV-RsSHC014, both of which could utilize human, civet and Chinese horseshoe bat angiotensin converting enzyme II (ACE2) as the host cell receptor for infecting human, civet and Chinese horseshoe bat target cells (Ge et al. 2013; Cui et al. 2019) . This evidence prompted us to accelerate our efforts to develop viral fusion inhibitors and neutralizing antibodies with broadspectrum inhibitory activity against divergent human coronaviruses and SARSr-CoVs (Zeng et al. 2017; Xia et al. 2019 ).\nConfidence: 0.6314553022384644\n\nParagraph: Based on the announcement of the World Health Organization (WHO) in 2018, the Wuhan pneumonia caused by an unknown etiology should be recognized as the first Disease X. Later, the pathogen was identified to be a novel coronavirus denoted 2019-nCoV, which has 79.5% and 96% whole genome sequence identify to SARS-CoV and bat SARS-related coronavirus (SARSr-CoV-RaTG13), respectively, suggesting its potential bat origin. With high human-to-human transmission rate (R0), 2019-nCoV has quickly spread in China and other countries, resulting in 34,953 confirmed cases and 725 deaths as of 8 February 2020, thus calling for urgent development of therapeutics and prophylactics. Here we suggest renaming 2019-nCoV as “transmissible acute respiratory syndrome coronavirus (TARS-CoV)” and briefly review the advancement of research and development of neutralizing antibodies and vaccines targeting the receptor-binding domain (RBD) and viral fusion inhibitors targeting the heptad repeat 1 (HR1) domain in spike protein of 2019-nCoV.\nConfidence: 0.6204904913902283\n\nParagraph: Soon after that, the pathogen causing Wuhan pneumonia was identified as a novel coronavirus, denoted 2019-nCoV by WHO (https://www.who.int/emergencies/diseases/ novel-coronavirus-2019), with 79.5% and 96% whole genome sequence identity to SARS-CoV and bat coronavirus SARSr-CoV-RaTG13, respectively, suggesting 2019-nCoV's potential bat origin (Zhou et al. 2020 ). However, no name for the pneumonia caused by 2019-nCoV infection has been provided by WHO or by the International Committee on Taxonomy of Viruses (ICTV) so far. Accordingly, some researchers decided to denote this novel coronavirus as ''novel coronavirus-infected pneumonia (NCIP)'' in their publication . However, in the generally accepted lexicon, one would then have to write ''novel coronavirus-infected pneumonia coronavirus (NCIP-CoV)'', which could be an awkward construct. Most reporters and media are still calling it ''Wuhan pneumonia'', but this terminology is detrimental to Wuhanese already facing lockdown and economic loss. The obvious dilemmas call for a renaming of the disease and virus causing the disease, one that is both lexicographically and taxonomically appropriate (news.ifeng.com/c/7tf2GFqwL4a).\nConfidence: 0.5875048637390137\n\n---------------------------------------------------------------------------------------------\nDocument: The First Two Imported Cases of SARS-CoV-2 Omicron Variant — Tianjin Municipality, China, December 13, 2021\nParagraph: On December 9, 2021, 2 international passengers arrived in Tianjin Binhai International Airport from Warsaw, Poland via airplane and tested positive for severe acute respiratory syndrome coronavirus 2 (SARS-CoV-2), which causes coronavirus disease 2019 , using nucleic acid tests. (3).\nConfidence: 0.7543003559112549\n\nParagraph: On December 9, 2021, 2 international passengers arrived in Tianjin Binhai International Airport from Warsaw, Poland via airplane and tested positive for severe acute respiratory syndrome coronavirus 2 (SARS-CoV-2), which causes coronavirus disease 2019 (COVID-19), using nucleic acid tests. The first case (Patient A) was a flight crew member, a 35-year-old male, who had returned to Warsaw with the same flight on December 10, 2021. The second case (Patient B) was a 17-year-old female student, who has been transferred to Haihe Hospital for isolation. Both cases are Polish nationality that had recently lived in Warsaw, Poland. According to the investigation, neither patient had disease symptoms nor medication history. Patient B received the Pfizer/BNT162b2 vaccine on September 27, 2021. Patient A was vaccinated on May 25, 2021, but no further vaccine information was collected as Patient A had left China. No other infections from this flight have been detected.\nConfidence: 0.6058023571968079\n\nParagraph: A total of 40 amino acid mutation sites (A67V, T95I, Y145D, L212I, G339D, S371L, S373P, S375F, K417N, N440K, G446S, S477N, T478K, E484A, Q493R, G496S, Q498R, N501Y, Y505H, T547K, D614G, H655Y, N679K, P681H, A701V, N764K, D796Y, N856K, Q954H, N969K, L981F, H69del, V70del, G142del, V143del, Y144del, N211del, and 214insEPE) and at least 26 amino acid mutation sites (T19R, G142D, C361R, K424I, N439I, N440K, G446S, S477N, T478K, E484A, Q493R, G496S, F497Y, Q498R, N501Y, Y505H, N679K, P681H, A701V, N764K, D796Y, N856K, Q954H, N969K, L981F, and P1069S) were detected in the spike protein of the strains from Patient A and B, respectively. Some of the mutations in the receptor binding domain and near the furin cleavage were concerning and may be associated with immune escape potency and higher transmissibility (4).\nConfidence: 0.5693592429161072\n\n---------------------------------------------------------------------------------------------\nDocument: COVID-19: the First Documented Coronavirus Pandemic in History\nParagraph: However, SARS-CoV, MERS-CoV, and SARS-CoV-2 can cause severe acute respiratory syndrome and result in life-threatening disease [17] [18] [19] [ Table 1 ].\nConfidence: 0.6812947988510132\n\nParagraph: Abstract The novel human coronavirus disease COVID-19 has become the fifth documented pandemic since the 1918 flu pandemic. COVID-19 was first reported in Wuhan, China, and subsequently spread worldwide. The coronavirus was officially named severe acute respiratory syndrome coronavirus 2 (SARS-CoV-2) by the International Committee on Taxonomy of Viruses based on phylogenetic analysis. SARS-CoV-2 is believed to be a spillover of an animal coronavirus and later adapted the ability of human-to-human transmission. Because the virus is highly contagious, it rapidly spreads and continuously evolves in the human population. In this review article, we discuss the basic properties, potential origin, and evolution of the novel human coronavirus. These factors may be critical for studies of pathogenicity, antiviral designs, and vaccine development against the virus.\nConfidence: 0.6485319137573242\n\nParagraph: agent is a novel coronavirus. Therefore, this virus is the seventh member of the coronavirus family to infect humans [3] . The World Health Organization (WHO) temporarily termed the new virus 2019 novel coronavirus (2019-nCoV) on 12 January 2020 and then officially named this infectious disease coronavirus disease 2019 (COVID-19) on 12 February 2020. Later, the International Committee on Taxonomy of Viruses (ICTV) officially designated the virus as SARS-CoV-2 based on phylogeny, taxonomy and established practice [4] . Subsequently, human-to-human transmission of COVID-19 occurring within Hong Kong has been shown in clinical data [5] . Since initially emerged in China, the virus has evolved for four months and rapidly spread to other countries worldwide as a global threat. On 11 March 2020, the WHO finally made the assessment that COVID-19 can be characterized as a pandemic, following 1918 Spanish flu (H1N1), 1957\nConfidence: 0.6306478977203369\n\n---------------------------------------------------------------------------------------------\n\n=================================================================================================\n=================================================================================================\n\n\nQuestion 2: what types of rapid testing for Covid-19 have been developed?\n","output_type":"stream"},{"name":"stderr","text":"Getting inverted index...: 100%|██████████| 9994/9994 [00:00<00:00, 34491.54it/s]\n","output_type":"stream"},{"name":"stdout","text":"Document: Reduced neutralization of SARS-CoV-2 B.1.617 variant by convalescent and vaccinated sera\nParagraph: Coronavirus disease 2019 (COVID-19) is caused by severe acute respiratory syndrome coronavirus 2 (SARS-CoV-2). The Spike protein that mediates coronavirus entry into host cells is a major target for COVID-19 vaccines and antibody therapeutics. However, multiple variants of SARS-CoV-2 have emerged, which may potentially compromise vaccine effectiveness. Using a pseudovirus-based assay, we evaluated SARS-CoV-2 cell entry mediated by the viral Spike B.1.617 and B.1.1.7 variants. We also compared the neutralization ability of monoclonal antibodies from convalescent sera and neutralizing antibodies (NAbs) elicited by CoronaVac (inactivated vaccine) and ZF2001 (RBD-subunit vaccine) against B.1.617 and B.1.1.7 variants. Our results showed that, compared to D614G and B.1.1.7 variants, B.1.617 shows enhanced viral entry and membrane fusion, as well as more resistant to antibody neutralization. These findings have important implications for understanding viral infectivity and for immunization policy against SARS-CoV-2 variants.\nConfidence: 0.7297025918960571\n\nParagraph: Coronavirus disease 2019 (COVID-19) is caused by severe acute respiratory Spike-specific antibodies elicited by natural infection or vaccination contribute 58 the majority of the neutralizing activity in human sera 3 . The receptor binding 59 domain (RBD) in the S1 subunit of Spike protein binds to its cellular receptor 60 angiotensin-converting enzyme 2 (ACE2) during viral entry, while the S2 61 subunit is required for the subsequent fusion of viral and cellular membranes 1 . 62 Therefore, RBD is believed to be a major target of neutralizing antibodies 63 (NAbs) and has been a focus of COVID-19 vaccine design 4,5 . Our previously 64 studies showed that mutations in SARS-CoV-2 Spike protein could affect viral 65 J o u r n a l P r e -p r o o f properties such as infectivity and neutralization resistance 6,7 . The newly 66 emerged SARS-CoV-2 variant, B.1.617, first reported from India, which 67 carries two mutations (L452R and E484Q) in its RBD is of particular concern. 68 The AstraZeneca ChAdOx1 nCoV-19 vaccine appeared less effective than the In this study, we used SARS-CoV-2 pseudovirus system to compare the viral 77 entry efficiency in vitro, as well as the neutralization activities of convalescent 78 sera, monoclonal antibodies (mAbs) and COVID-19 vaccine-elicited sera 79 against these newly emerging SARS-CoV-2 variants, including the highly The virus infectivity was determined by a Luc assay as previously described 16 .\nConfidence: 0.6783335208892822\n\nParagraph: As shown in Figure 1B (Fig. 2A) . Notably, the ID50 titer of 6 samples (30%) was 244 lower than the threshold against B.1.617 ( Fig. 2A) . Compared with D614G, CoV-2 Spike D614G, while those who received only two doses had lower but 282 detectable neutralization (Fig. 4A, B) . The GMTs of ZF2001-elicited serum concern will continue to arise, which may threaten vaccine efficacy. Therefore, binding domain) monoclonal antibody. To compare the S1 and S ratio, integrated density 495 of S1/(S+S1) was quantitatively analyzed using ImageJ software, n = 3. \nConfidence: 0.6231483817100525\n\n---------------------------------------------------------------------------------------------\nDocument: Influenza virus-like particle-based hybrid vaccine containing RBD induces immunity against influenza and SARS-CoV-2 viruses\nParagraph: Several approaches have produced an effective vaccine against severe acute respiratory syndrome coronavirus 2 (SARS-CoV-2). However, the influence of immune responses induced by other vaccinations on the durability and efficacy of the immune response to SARS-CoV-2 vaccine is still unknown. We have developed a hybrid vaccine for SARS-CoV-2 and influenza viruses using influenza virus-like particles (VLP) incorporated by protein transfer with glycosylphosphatidylinositol (GPI)-anchored SARS-CoV-2 S1 RBD fused to GM-CSF as an adjuvant. GPI-RBD-GM-CSF fusion protein was expressed in CHO-S cells, purified and incorporated onto influenza VLPs to develop the hybrid vaccine. Our results show that the hybrid vaccine induced a strong antibody response and protected mice from both influenza virus and mouse-adapted SARS-CoV-2 challenges, with vaccinated mice having significantly lower lung viral titers compared to naive mice. These results suggest that the hybrid vaccine strategy is a promising approach for developing multivalent vaccines to prevent influenza A and SARS-CoV-2 infections.\nConfidence: 0.7364246249198914\n\nParagraph: Abstract: 22 Several approaches have produced an effective vaccine against severe acute respiratory 23 syndrome coronavirus 2 (SARS-CoV-2). However, the influence of immune responses induced 24 by other vaccinations on the durability and efficacy of the immune response to SARS-CoV-2 25 vaccine is still unknown. We have developed a hybrid vaccine for SARS-CoV-2 and influenza 26 viruses using influenza virus-like particles (VLP) incorporated by protein transfer with 27 glycosylphosphatidylinositol (GPI)-anchored SARS-CoV-2 S1 RBD fused to GM-CSF as an 28 adjuvant. GPI-RBD-GM-CSF fusion protein was expressed in CHO-S cells, purified and 29 incorporated onto influenza VLPs to develop the hybrid vaccine. Our results show that the 30 hybrid vaccine induced a strong antibody response and protected mice from both influenza virus 31 and mouse-adapted SARS-CoV-2 challenges, with vaccinated mice having significantly lower 32 lung viral titers compared to naive mice. These results suggest that the hybrid vaccine strategy is 33 Introduction 37 Severe acute respiratory syndrome coronavirus 2 (SARS-CoV-2) first appeared in late 2019 38 in China before beginning its rapid spread across the globe (1). The disease, named COVID-19, 39 presents a severe respiratory disease course and high fatality rate in the elderly and 40 immunocompromised (2). The spike (S) protein of the virus binds to the human ACE2 protein 41 for entry into epithelial cells of the respiratory tract (1, 3). This S protein, and specifically the 42 conserved ACE2 receptor-binding domain (RBD), is a proven target for vaccine design. 43 Antibodies and vaccines directed to the S protein and the RBD are effective in preventing SARS- 44 CoV-2 infection (4). Virtually all recent vaccine strategies for SARS-CoV-2 target the full- 45 length S protein (Pfizer/BioNTech, Moderna, Johnson & Johnson, AstraZeneca/Oxford, 46 Novavax, Innovio, Curevac, and others) (5, 6). 47 uses an antigen fused to GM-CSF that delivers antigen effectively to the immune system (17, 60 18). Adjuvants that have a tolerable safety profile and generate a Th1 immune response are of 61 high importance. Moreover, proinflammatory cytokines produced by activated dendritic cells 62 (DCs) have been shown to play an important role in inducing a robust immune response (19, 20) . 63 IL-12 has been well documented to induce a Th1 response, with a promising clinical benefit in 64 cancer patients (8). IL-12, a heterodimeric cytokine (p35 and p40 subunits), activates DCs, T of adjuvant and antigen source leads to simultaneous adjuvant and antigen delivery to immune 83 cells, resulting in enhanced immune reactivity and increased vaccine efficacy when compared to 84 unconjugated antigen and adjuvant mixture (37, 38). In the present study, we demonstrate that 85 influenza VLP vaccine incorporated with GPI-RBD-GM-CSF fusion protein and GPI-IL-12 86 (hybrid vaccine) protected mice from influenza virus challenge and also induced a robust, 87 durable antibody response in BALB/c mice as well as decreased viral load and less weight loss 88 when challenged with mouse adapted SARS-CoV-2. 89 \nConfidence: 0.709616482257843\n\nParagraph: Characterization of GPI-RBD-GM-CSF fusion protein 92 We constructed a fusion protein gene by joining the DNA sequences specific to SARS CoV-2 S 93 protein RBD domain, mouse GM-CSF sequence and the GPI-anchor signal from human CD59 Figure 1B ). To verify whether the GPI-RBD-GM-CSF fusion protein contains the GPI tail, 97 CHO-S cells expressing the fusion protein were incubated with phosphatidylinositol-specific 98 phospholipase C (PIPLC), which is known to cleave the GPI moiety from the proteins and 99 release them from the cell membranes, and then analyzed by flow cytometry. The results show 100 that nearly 76% of the fusion protein was released from the cell surface, suggesting that the human ACE2 confirmed that the RBD in purified GPI-RBD-GM-CSF fusion protein retains its 129 receptor binding activity ( Figure 2C ). 130 To make the VLP hybrid vaccine, we incorporated the influenza VLPs with GPI-IL-12 and Biotech) or PBS. Booster dose was given 2-4 weeks after the first dose. The route of 149 administration was either subcutaneous (s.c.) or intramuscular (i.m.) (Fig. S2) . 150 Blood was collected every 2 to 4 weeks for antibody titer, ACE2 binding inhibition, and virus 151 neutralization assays. VLP vaccine incorporated with GPI-RBD-GM-CSF and GPI-IL-12 (hybrid 152 vaccine) induced robust antibody response after the booster dose (Fig. S3 ). The antibody 153 response against GPI-RBD-GM-CSF fusion protein was comparable in both i.m. and s.c. routes 154 of vaccine administration (Fig. S4) . The antibodies bind to CHO-S cells expressing the GPI- Hybrid vaccine induces IgG2a antibody response 177 We observed that purified GPI-RBD-GM-CSF fusion protein by itself or incorporated onto VLPs 178 induced a strong antibody response ( Figure 3A ). However, recombinant RBD-His tag failed to 179 induce an antibody response suggesting that the GM-CSF in our fusion protein is acting as an Figure 4C ). This may be because GPI-GM-CSF is more active than GPI-RBD-203 GM-CSF or may be due to the difference in the level of incorporation onto VLP or both, which 204 need further investigation. 243 We have demonstrated that a hybrid vaccine based on influenza VLPs induces effective 244 immunity against SARS-CoV-2 and influenza viruses. Our vaccine platform is based on cytokine 245 adjuvants linked to the VLPs that carry the antigens (7). This approach delivers both antigens 246 and biological adjuvants to the immune system simultaneously in a particulate form. We have 247 generated CHO-S cells expressing GPI-anchored spike RBD-GM-CSF fusion protein and GPI- In some experiments, Spike S1 protein from commercial sources (RayBiotech or Sino 416 Biologicals) was used as indicated in the figure legends. Plates were washed using a plate washer 417 and washing buffer (PBS with 0.05% Tween20). Plates were blocked with assay diluent (PBS 418 with 3% BSA) for 2 hours at room temperature on a rocker. Plates were washed again as above 419 and diluted serum samples were added and incubated for another 2 hours at room temperature. \nConfidence: 0.6617717742919922\n\n---------------------------------------------------------------------------------------------\nDocument: Immunogenicity and efficacy of the COVID-19 candidate vector vaccine MVA SARS 2 S in preclinical vaccination\nParagraph: The severe acute respiratory syndrome (SARS) coronavirus 2 (SARS-CoV-2) has emerged as the infectious agent causing the pandemic coronavirus disease 2019 (COVID-19) with dramatic consequences for global human health and economics. Previously, we reached clinical evaluation with our vector vaccine based on vaccinia virus MVA against the Middle East respiratory syndrome coronavirus (MERS-CoV), which causes an infection in humans similar to SARS and COVID-19. Here, we describe the construction and preclinical characterization of a recombinant MVA expressing full-length SARS-CoV-2 spike (S) protein (MVA-SARS-2-S). Genetic stability and growth characteristics of MVA-SARS-2-S, plus its robust synthesis of S antigen, make it a suitable candidate vaccine for industrial scale production. Vaccinated mice produced S antigen-specific CD8+ T cells and serum antibodies binding to S glycoprotein that neutralized SARS-CoV-2. Prime-boost vaccination with MVA-SARS-2-S protected mice sensitized with a human ACE2-expressing adenovirus from SARS-CoV-2 infection. MVA-SARS-2-S is currently being investigated in a phase I clinical trial as aspirant for developing a safe and efficacious vaccine against COVID-19. Significance Statement The highly attenuated vaccinia virus MVA is licensed as smallpox vaccine, and as vector it is a component of the approved Adenovirus-MVA-based prime-boost vaccine against Ebola virus disease. Here we provide results from testing the COVID-19 candidate vaccine MVA-SARS-2-S, a poxvirus-based vector vaccine that proceeded to clinical evaluation. When administered by intramuscular inoculation, MVA-SARS-2-S expresses and safely delivers the full-length SARS-CoV-2 spike (S) protein, inducing balanced SARS-CoV-2-specific cellular and humoral immunity, and protective efficacy in vaccinated mice. Substantial clinical experience has already been gained with MVA vectors using homologous and heterologous prime-boost applications, including the immunization of children and immunocompromised individuals. Thus, MVA-SARS-2-S represents an important resource for developing further optimized COVID-19 vaccines.\nConfidence: 0.7167670726776123\n\nParagraph: Western blot analysis using serum antibodies from a COVID-19 patient hospitalized with 170 pneumonia also revealed protein bands corresponding to the molecular masses of full-length S 171 protein or the S2 polypeptide (Fig. 2D ). intramuscular (i.m.) administration and prime-boost immunization schedules with a 3-week interval 176 (Fig. 3, SI Appendix, Fig. S2 ). At day 18 after the prime inoculation, we detected serum IgG 177 antibodies binding to whole recombinant SARS-CoV-2 S protein in the sera from 3/8 LD-vaccinated 178 and 4/6 HD-vaccinated animals by ELISA (Fig. 3A) . Following the booster immunization on day 21, 179 all vaccinated animals mounted high levels of S-binding serum IgG antibodies with mean titers of 180 1:900 for the LD vaccination group and 1:1257 for the HD group (Fig. 3A ). More importantly, sera 181 from vaccinated mice also contained antibodies binding to the S protein receptor-binding domain 182 (RBD). Already at day 18 post priming, the RBD-binding antibodies were detected in 33% of the 183 mice in the LD dose group (2/6 mice, mean OD value 0.35) and 50% of the mice receiving the HD 184 immunization (3/6, mean OD 0.63). The boost vaccinations increased the levels of RBD specific 185 antibodies with 87.5% seropositive mice in the 10 7 dose group (7/8, mean OD 1.81) and 100% of 186 the animals vaccinated with 10 8 PFU MVA-SARS2-S (8/8, mean OD 2.92) ( Figure 3B ). Since live 187 virus neutralization is the gold standard for coronavirus serologic analysis, we next assessed the 188 mouse sera in two different assays for SARS-CoV-2 neutralization, a plaque reduction 189 neutralization test 50 (PRNT50) (24) and a complete virus neutralization test (VNT100) (9) ( \nConfidence: 0.6869877576828003\n\nParagraph: Characterization of SARS-CoV-2 S expressed by recombinant MVA. To determine the 140 expression pattern of the recombinant SARS-CoV-2 S protein, we stained MVA-SARS-2-S infected To examine the MVA-produced recombinant S glycoprotein in more detail, we prepared total 150 lysates from MVA-SARS-2-S infected CEF or Vero cells for separation by SDS-PAGE and 151 subsequent immunoblot analysis (Fig. 2) . The mouse monoclonal antibody directed against the 152 HAtag at the C-terminus of the recombinant SARS-2-S protein revealed two prominent protein 153 bands that migrated with molecular masses of approximately 190 kDa and 90-100 kDa (Fig 2B) .\nConfidence: 0.6597922444343567\n\n---------------------------------------------------------------------------------------------\nDocument: Self-replicating vehicles based on negative strand RNA viruses\nParagraph: Oncolytic MV vectors have been utilized for the expression of granulocyte macrophage-colony stimulating factor (GM-CSF) in a mouse adenocarcinoma MC38cea model [116] . The tumor progression was delayed after intratumoral MV-GM-CSF administration, which also prolonged the survival time of immunized animals. Tumor remission was complete in one-third of mice and tumor re-engraftment was further rejected. In another approach, oncolytic MV vectors expressing IL-12 and IL-15 were evaluated in MC38cea and B16hCD46 tumor models [117] . The MV-IL-15 vector enhanced T-and NK-cell filtration in both tumor models, but the MV-IL-12 vector generated more robust viral gene expression and immune activation, which resulted in superior anticancer activity. In another study, MV-based expression of IL-12 was evaluated in colon cancer cells and in rats [118] . MV-IL-12 administration in rats enhanced anti-tumor activity and immunity. In the context of an oncolytic VSV vector, expression of enhanced secreted IL-15 resulted in strong local expression in CT-26 colon tumors [119] . Immunization of mice enhanced both anti-tumoral T-cell responses and prolonged the survival time. In another study, an LNCaP, PC-3 Sc-FV-PSMA MV-sc-Fv-PSMA Tumor regression in mice [160] PC-3 oMv, oMuV MV + MuV Immune responses, protection in mice [161] DU145, PC-3 GFP VSV-ΔM51-GFP Apoptosis in tumor cells, prolonged survival [162] DU-145, 22Rv1 LCMV GP VSV Long-term remission in mice [163] PC-3 VSV + curcumin VSV Promotion of oncolysis in vitro & in vivo [164] Ad adenovirus, ALL acute lymphocytic leukemia, AML acute myeloid leukemia, ATL adult T-cell leukemia, CEA carcinoembryonic antigen, CRC colorectal cancer, cHL classical Hodgkin lymphoma, CLL chronic lympocytic leukemia, CRPV cottontail rabbit papilloma virus, CSC glioma stem cell, CTCL cutaneous T-cell lymphoma, CTLA-4 CTL antigen-4, DC dendritic cell, GFP green fluorescent protein, GITR glucocorticoid-induced tumor necrosis factor receptor, GM-CSF granulocyte macrophage-colony stimulating factor, hDCT human dopachrome tautomerase, IL interleukin, LCMV lymphocytic choriomeningitis virus, Lewis lung carcinoma, MCL mantle cell lymphoma, MOSEC murine ovarian surface epithelial carcinoma, NIS sodium iodine symporter, PDAC pancreatic ductal adenocarcinoma, RROC refractory recurrent ovary cancer, TNBC triple-negative breast cancer, RROC recurrent refractory ovarian cancer, RSV respiratory syncytial virus, TRAMP transgenic adenocarcinoma mouse prostate, VSV vesicular stomatitis virus.\nConfidence: 0.7279717326164246\n\nParagraph: In the context of vaccines developed against hepatotropic viruses such as hepatitis B virus (HBV) negative-strand selfreplicating RNA viruses have been applied. For instance, expression of the HBV surface antigen (HBsAg) from an MV vector elicited strong immune responses and resulted in protection in 50% of immunized rhesus macaques [54] . Moreover, a VSV vector expressing the HBV middle envelope surface protein (MHBs) induced strong HBV-specific antibody responses in mice [55] . A single immunization induced CD8 + T-cell activation and protected mice against HBV challenge with hydrodynamic DNA transfection technique.\nConfidence: 0.7269976735115051\n\nParagraph: The annual global seasonal influenza virus threats have put pressure on vaccine development against influenza A virus and avian influenza virus strains. For instance, the HA protein of the highly pathogenic avian influenza virus (HPAIV) has been expressed from an MV vaccine strain (rMV-Ed-H5HA) or a wildtype MV-derived mutant (rMV-HL-Vko-H5HA) [62] . Immunization of cynomolgus monkeys elicited both MV-and HPAIV-specific antibodies. Furthermore, vaccinated monkeys showed less severe pneumonia, faster recovery from influenza symptoms, and suppression of viral shedding. Moreover, strong immune responses were observed in cotton rats immunized with an MV vector expressing the HA protein from the influenza A/Sapporo/ 107/2013 (H1N1pdm) strain [63] . Additionally, cotton rats immunized with MV AIK-C were protected against challenges with influenza A virus. In the case of VSV-based vaccine development, the VSV-G gene was replaced by the HPAIV HA gene, HPAIV A/ FPV/Rostock/34 (H7N1), generating the replication-deficient VSV*DeltaG(HA) vector [64] . Intramuscular immunization of chickens resulted in high antibody titers, which neutralized avian influenza virus subtypes H7N1 and H7N7 but not H5N2. Moreover, immunized chickens were protected against challenges with lethal doses of the HPAIV A/chicken/Italy/445/99 (H7N1) strain. In another study, VSV vectors expressing the influenza HA and neuraminidase (NA) provided protection against lethal challenges in mice immunized at least 24 h after the virus challenge [65] . It was determined that HA-specific immune responses were essential for achieving full protection. Moreover, the engineering of a VSV vector expressing the H5 HA of the H5N1 HPAIV A/ Vietnam/1203/04 (VN1203), and the NA of the mouse-adapted H1N1 influenza virus A/Puerto Rico/8/34 (PR8) generated a vaccine candidate, which was non-pathogenic in mice [66] . A single intramuscular or intranasal injection protected mice from lethal challenges with H5N1 influenza virus. A fast-acting VSV-based pan-H5 influenza virus vaccine was targeted by engineering different antigen designs [67] . Vaccination of mice confirmed that a single dose of VSV expressing the full-length HA gene (HAfl) provided 100% protection. The response was rapid as uniform protection was achieved when the vaccine candidate was administered only 3 days prior to the challenge. A single vaccination elicited cross-protective H5-specific antibodies and resulted in protection against challenges with different H5 clade 2 viruses, which is an indication of a pan-H5 influenza virus vaccine.\nConfidence: 0.7109046578407288\n\n---------------------------------------------------------------------------------------------\nDocument: Evaluation of the Xpert Xpress SARS-CoV-2/Flu/RSV Assay for Simultaneous Detection of SARS-CoV-2, Influenza A and B Viruses, and Respiratory Syncytial Virus in Nasopharyngeal Specimens\nParagraph: Patients infected with severe acute respiratory syndrome coronavirus 2 (SARS-CoV-2), influenza A (flu A), influenza B (flu B), and respiratory syncytial virus (RSV) have overlapping clinical presentations, but the approaches to treatment and management of infections caused by these viruses are different. Therefore, rapid diagnosis in conjunction with infection prevention measures is important to prevent transmission of the diseases. Recently, a new Xpert Xpress SARS-CoV-2/Flu/RSV (Xpert 4-in-1) assay enables the detection and differentiation of SARS-CoV-2, flu A, flu B, and RSV in upper respiratory tract specimens. In this study, we evaluated the performance of the Xpert 4-in-1 assay by comparing it with that of the Xpert Xpress SARS-CoV-2 and Xpert Xpress Flu/RSV assays for the detection of the four viruses in nasopharyngeal (NP) specimens. A total of 279 NP specimens, including 66, 56, 64, and 53 specimens positive for SARS-CoV-2, flu A, flu B, and RSV, respectively, were included. The Xpert 4-in-1 assay demonstrated high concordance with the comparator assays, with overall agreement for SARS-CoV-2, flu A, flu B, and RSV at 99.64%, 100%, 99.64%, and 100%, respectively, and a high Cohen’s kappa (κ) value ranging from 0.99 to 1.00, indicating an almost perfect correlation between assays. The cycle threshold value association between positive samples also showed a good correlation between assays. In conclusion, the overall performance of the Xpert 4-in-1 assay was highly comparable to that of the Xpert SARS-CoV-2 and Xpert Flu/RSV assays for the detection and differentiation of SARS CoV-2, flu A, flu B, and RSV in NP specimens.\nConfidence: 0.685740053653717\n\nParagraph: To assist global efforts in the fight against the spread of COVID-19 and influenza during the upcoming influenza season, a number of manufacturers are modifying existing assays to allow simultaneous multiplex testing for SARS-CoV-2 and flu A, flu B, and RSV. Recently, a new Xpert Xpress SARS-CoV-2/Flu/RSV (Xpert 4-in-1) assay (Cepheid, Sunnyvale, CA) for the qualitative detection and differentiation of SARS-CoV-2, flu A, flu B, and RSV in upper respiratory tract specimens in one cartridge for a single patient sample with results available in 36 min has been released with in vitro diagnostic emergency use authorization (EUA) by the U.S. Food and Drug Administration (FDA) in September 2020. The Xpert 4-in-1 assay aids in the combination of two individual rapid and random-access molecular assays, the Xpert Xpress SARS-CoV-2 (Xpert SARS-CoV-2) assay (Cepheid, Sunnyvale, CA) and the Xpert Xpress Flu/RSV (Xpert Flu/RSV) assay (Cepheid, Sunnyvale, CA). The Xpert Flu/RSV assay has been widely used for the detection and differentiation of flu A, flu B, and RSV from upper respiratory tract specimens worldwide, especially during the influenza seasons (8, 9) . To effectively manage the spread of COVID-19, the Xpert SARS-CoV-2 assay received FDA EUA in March 2020 for the rapid detection of SARS-CoV-2. Currently, we are using these two Xpert assays in our laboratory for the rapid diagnosis of SARS-CoV-2, flu A, flu B, and RSV. We have previously reported the high sensitivity and specificity of the Xpert SARS-CoV-2 assay for the detection of SARS-CoV-2 and those of the Xpert Flu/RSV assays for the detection of flu A, flu B, and RSV (8, 10) . In this study, we evaluated the performance of the Xpert 4-in-1 assay by comparing it with those of the Xpert SARS-CoV-2 and Xpert Flu/ RSV assays for the detection of SARS-CoV-2, flu A, flu B, and RSV in nasopharyngeal (NP) specimens.\nConfidence: 0.6671342849731445\n\nParagraph: Xpert 4-in-1 assay. The Xpert 4-in-1 assay is a multiplex real-time PCR assay for the simultaneous detection and differentiation of SARS-CoV-2, flu A, flu B, and RSV. The assay targets (i) two unique regions of the nucleocapsid gene (N2 gene) and the envelope protein gene (E gene) of SARS-CoV-2, (ii) genes encoding matrix protein, PB2, and PA of influenza A, (iii) genes encoding the matrix protein and nonstructural protein of influenza B, and (iv) genes encoding the nucleocapsids of RSV A and RSV B. This assay was performed on the GeneXpert instrument systems according to the manufacturer's instructions. Briefly, after opening the cartridge lid, 300 ml of the sample in the VTM was transferred with a disposable pipette supplied to the sample chamber. The cartridge lid was closed, and the cartridge was loaded into the instrument following the instructions displayed in the GeneXpert system window.\nConfidence: 0.6649575233459473\n\n---------------------------------------------------------------------------------------------\n\n=================================================================================================\n=================================================================================================\n\n\nQuestion 3: has social distancing had an impact on slowing the spread of COVID-19?\n","output_type":"stream"},{"name":"stderr","text":"Getting inverted index...: 100%|██████████| 9994/9994 [00:00<00:00, 34177.16it/s]\n","output_type":"stream"},{"name":"stdout","text":"Document: Novel coronavirus disease 2019 (COVID‐19) and neurodegenerative disorders\nParagraph: During the last few months, the whole humanity is experiencing largest and most severe sudden influx of COVID‐19 outbreak caused by the novel coronavirus (CoV) originated from Wuhan, China. According to the WHO reports, total 3 862 676 positive cases and 265 961 deaths have been recorded worldwide due to COVID‐19 infection as of May 9, 2020. CoVs are a large family of viruses (enveloped, single‐stranded RNA viruses), which includes severe acute respiratory syndrome coronavirus (SARS‐CoV) and Middle East respiratory syndrome‐related coronavirus (MERS‐CoV). New SARS CoV2 is the members of Betacoronavirus genus. These viruses cause infections in bats, camels and humans, and a few other associated species. Despite many neurologic complications associated with SARS‐CoV‐2 infection, it is still unclear whether these symptoms results from direct neural injury or due to some other reason. Currently, it appears that most of the neurological symptoms of COVID‐19 are nonspecific and secondary to the systemic illness. A single case of acute hemorrhagic necrotizing encephalopathy has been reported. SARS‐CoV‐2 associated Guillain‐Barré syndrome is an atypical case. Till today, no convincing evidence is available to confirm that the SARS‐CoV‐2 virus directly affects nerves system in humans. However, postinfection surveillance will be necessary to identify the possible post‐COVID‐19 neurologic syndromes.\nConfidence: 0.5912237763404846\n\nParagraph: During the last few months, the whole humanity is experiencing the largest and most severe sudden influx of COVID-19 outbreak caused by the novel coronavirus (CoV) originated from Wuhan city.\nConfidence: 0.5699441432952881\n\nParagraph: Coronavirus was officially announced as the causative pathogen of COVID-19 by the Chinese Center for Disease Control and Prevention on January 8, 2020. The World Health Organization announced coronavirus outbreak as a public health emergency of international concern. It started to expect disturbing extents over the globe as forewarned by the World Health Organization (WHO) reports associated Guillain-Barré syndrome is an atypical case. Till today, no convincing evidence is available to confirm that the SARS-CoV-2 virus directly affects the CNS or PNS in humans. However, postinfection surveillance will be necessary to identify the possible post-COVID-19 neurologic syndromes.\nConfidence: 0.5248440504074097\n\n---------------------------------------------------------------------------------------------\nDocument: Immune response in COVID-19: What do we currently know?\nParagraph: The acute respiratory syndrome is a disease caused by the SARS-CoV-2 virus (COVID-37 19), where symptoms include difficulty breathing, high fever, and cough (WHO, 2020). \nConfidence: 0.6834912300109863\n\nParagraph: In 2002/2003 there was a pandemic denominate SARS (severe acute respiratory syndrome), caused by the SARS-CoV virus that belongs to the genera Betacoranavirus and the family Coronaviridae, generally responsible for influenza infections. In mid of 2019, a new disease by the coronavirus named by COVID-19 (SARS-CoV-2) emerged, both infections have flu symptoms, however they are infections that variable intensity, being medium to severe. In medium infections individuals have the virus and exhibit symptoms, however hospitalization is not necessary, in severe infections, individuals are hospitalized, have high pathology and in some cases progress to death. The virus is formed by simple positive RNA, enveloped, non-segmented, and presenting the largest genome of viruses constituting 32 Kb, consisting of envelope proteins, membrane, nucleocapsid and spike protein, which is essential in the interaction with the host cells. As for the origin of this virus, research has been intensified to determine this paradox and although the similarity with SARS-CoV, this virus did not has necessarily the same place of origin. As for the immune system, it is currently unknown how this new virus interacts. In this brief review, we demonstrate important considerations about the responses to this infection.\nConfidence: 0.547529399394989\n\n---------------------------------------------------------------------------------------------\nDocument: Coronavirus Infections in Companion Animals: Virology, Epidemiology, Clinical and Pathologic Features\nParagraph: A genetically distinct coronavirus called ferret systemic coronavirus infection (FRSCV) was subsequently identified, which causes a systemic, progressive, and fatal pyogranulomatous inflammatory disease resembling the dry form of feline infectious peritonitis (FIP) in cats. The average age at the time of FRSCV diagnosis has been reported to be 11 months, and clinical signs include chronic weight loss, anorexia, diarrhea, palpable abdominal masses, and neurologic disease [97] .\nConfidence: 0.6613699793815613\n\nParagraph: Coronaviruses are enveloped RNA viruses capable of causing respiratory, enteric, or systemic diseases in a variety of mammalian hosts that vary in clinical severity from subclinical to fatal. The host range and tissue tropism are largely determined by the coronaviral spike protein, which initiates cellular infection by promoting fusion of the viral and host cell membranes. Companion animal coronaviruses responsible for causing enteric infection include feline enteric coronavirus, ferret enteric coronavirus, canine enteric coronavirus, equine coronavirus, and alpaca enteric coronavirus, while canine respiratory coronavirus and alpaca respiratory coronavirus result in respiratory infection. Ferret systemic coronavirus and feline infectious peritonitis virus, a mutated feline enteric coronavirus, can lead to lethal immuno-inflammatory systemic disease. Recent human viral pandemics, including severe acute respiratory syndrome (SARS), Middle East respiratory syndrome (MERS), and most recently, COVID-19, all thought to originate from bat coronaviruses, demonstrate the zoonotic potential of coronaviruses and their potential to have devastating impacts. A better understanding of the coronaviruses of companion animals, their capacity for cross-species transmission, and the sharing of genetic information may facilitate improved prevention and control strategies for future emerging zoonotic coronaviruses. This article reviews the clinical, epidemiologic, virologic, and pathologic characteristics of nine important coronaviruses of companion animals.\nConfidence: 0.650027871131897\n\nParagraph: A novel alpaca coronavirus belonging to the alphacoronavirus genus was isolated in 2007 and associated with acute respiratory disease rather than enteric disease [144] . Complete genome sequencing revealed less than 50% nucleotide identity with the previously reported enteric alpaca coronavirus but a much higher 92.2% nucleotide identity with the human coronavirus (HCoV) 229E, with striking similarity between the HCoV 229E and alpaca respiratory coronavirus spike proteins. Comparison of spike gene sequences revealed that alpaca respiratory coronavirus is most similar to HCoV 229E strains isolated between the 1960s and 1980s, suggesting the possibility that a transmission event may have occurred between alpacas and humans [151] .\nConfidence: 0.6382077932357788\n\n---------------------------------------------------------------------------------------------\nDocument: The peripheral T cell population is associated with pneumonia severity in cynomolgus monkeys experimentally infected with severe acute respiratory syndrome coronavirus 2\nParagraph: The coronavirus disease 2019 (COVID-19), caused by severe acute respiratory syndrome coronavirus 2 (SARS-CoV-2), is a global pandemic that began in December 2019. Lymphopenia is a common feature in severe cases of COVID-19; however, the role of T cell responses during infection is unclear. Here, we inoculated six cynomolgus monkeys, divided into two groups according to the CD3+ T cell population in peripheral blood, with two clinical isolates of SARS-CoV-2: one of East Asian lineage and one of European lineage. After initial infection with the isolate of East Asian lineage, all three monkeys in the CD3+ low group showed clinical symptoms, including loss of appetite, lethargy, and transient severe anemia with/without short-term fever, within 14 days post-infection (p.i.). By contrast, all three monkeys in the CD3+ high group showed mild clinical symptoms such as mild fever and loss of appetite within 4 days p.i. and then recovered. After a second inoculation with the isolate of European lineage, three of four animals in both groups showed mild clinical symptoms but recovered quickly. Hematological, immunological, and serological tests suggested that the CD3+ high and low groups mounted different immune responses during the initial and second infection stages. In both groups, anti-viral and innate immune responses were activated during the early phase of infection and re-infection. However, in the CD3+ low group, inflammatory responses, such as increased production of monocytes and neutrophils, were stronger than those in the CD3+ high group, leading to more severe immunopathology and failure to eliminate the virus. Taken together, the data suggest that the peripheral T lymphocyte population is associated with pneumonia severity in cynomolgus monkeys experimentally infected with SARS-CoV-2. Author summary SARS-CoV-2 infection causes an illness with clinical manifestations that vary from asymptomatic or mild to severe; examples include severe pneumonia and acute respiratory distress syndrome. Lymphopenia, which is common in severe COVID-19 cases, is characterized by markedly reduced numbers of CD4+ T cells, CD8+ T cells, B cells, and natural killer cells. Here, we showed that cynomolgus monkeys selected according to the T cell populations in peripheral blood have different outcomes after experimental infection with SARS-CoV-2. These findings will increase our understanding of disease pathogenesis and may facilitate the development of animal models for vaccine evaluation.\nConfidence: 0.5690917372703552\n\nParagraph: 303 These results suggest that the initially inoculated virus (WK-521) was maintained in the lungs 304 and/or accessory lymph nodes, and that the second inoculated virus (QH-329-037) was 305 eliminated from the lungs of these monkeys soon after the second inoculation. \nConfidence: 0.563288152217865\n\nParagraph: -16 -283 Table 2 The results revealed that the viral genome obtained from the tonsil of monkey #5404 287 after initial inoculation did not harbor any mutations (threshold = 50%). However, the viral 288 genome isolated from the lung of monkey #5412 harbored nine single nucleotide 289 polymorphisms (SNPs), including seven silent point mutations, one deletion resulting in a 290 frameshift mutation in the ORF1ab region, and a nonsynonymous mutation in ORF1b. The most 291 common base change was C > T. In addition, the genome isolated from mesenteric lymph nodes 292 from monkey #5412 harbored three SNPs, including two nonsynonymous mutations in the 293 ORF1a region and a synonymous mutation in the S region. The major sequence in these two 294 isolates was derived from WK-521, suggesting that the original inoculum replicated and resided 295 in the respiratory tract and intestine of monkey #5412, even after the second inoculation with 296 the heterologous strain. The viral genome obtained from the tonsil of monkey #5403 after the 297 second inoculation harbored a D614G mutation in the S region, suggesting the presence of 298 QH-329-037 in the tonsil after the 2nd inoculum (S3 Table) . The viral genome also obtained 299 from the tonsil of #5405 after the second inoculation harbored a D614G mutation in the S 300 region, suggesting the presence of QH-329-037 in the tonsil. Interestingly, the genome obtained 301 from the subcarinal lymph node of monkey #5405 did not harbor the D614G mutation, 302 suggesting that the original inoculum was maintained in the accessory lymph node of the lungs.\nConfidence: 0.5425497889518738\n\n---------------------------------------------------------------------------------------------\nDocument: The First Disease X is Caused by a Highly Transmissible Acute Respiratory Syndrome Coronavirus\nParagraph: On February 9, 2018, the World Health Organization (WHO) announced the Blueprint list of priority diseases, including Middle East respiratory syndrome (MERS) and Severe Acute Respiratory Syndrome (SARS), as well as Disease X, for research and development in emergency contexts. Disease X would be a new disease with an epidemic or pandemic potential caused by an unknown pathogen (www.who.int/activities/prioritizing-diseasesfor-research-and-development-in-emergency-context). At that time, we believed that the first Disease X could be a transmissible infectious disease caused by a novel coronavirus originated from bats. This supposition was based on a live SARS-related coronavirus (SARSr-CoV), designated SARSr-CoV-WIV1, isolated from bat fecal samples in Vero E6 cells. It had 99.9% whole genome sequence identity to that of SARSr-CoV-Rs3367 identified from Chinese horseshoe bats and another novel strain, SARSr-CoV-RsSHC014, both of which could utilize human, civet and Chinese horseshoe bat angiotensin converting enzyme II (ACE2) as the host cell receptor for infecting human, civet and Chinese horseshoe bat target cells (Ge et al. 2013; Cui et al. 2019) . This evidence prompted us to accelerate our efforts to develop viral fusion inhibitors and neutralizing antibodies with broadspectrum inhibitory activity against divergent human coronaviruses and SARSr-CoVs (Zeng et al. 2017; Xia et al. 2019 ).\nConfidence: 0.5812097191810608\n\nParagraph: Based on the announcement of the World Health Organization (WHO) in 2018, the Wuhan pneumonia caused by an unknown etiology should be recognized as the first Disease X. Later, the pathogen was identified to be a novel coronavirus denoted 2019-nCoV, which has 79.5% and 96% whole genome sequence identify to SARS-CoV and bat SARS-related coronavirus (SARSr-CoV-RaTG13), respectively, suggesting its potential bat origin. With high human-to-human transmission rate (R0), 2019-nCoV has quickly spread in China and other countries, resulting in 34,953 confirmed cases and 725 deaths as of 8 February 2020, thus calling for urgent development of therapeutics and prophylactics. Here we suggest renaming 2019-nCoV as “transmissible acute respiratory syndrome coronavirus (TARS-CoV)” and briefly review the advancement of research and development of neutralizing antibodies and vaccines targeting the receptor-binding domain (RBD) and viral fusion inhibitors targeting the heptad repeat 1 (HR1) domain in spike protein of 2019-nCoV.\nConfidence: 0.5753904581069946\n\nParagraph: Soon after that, the pathogen causing Wuhan pneumonia was identified as a novel coronavirus, denoted 2019-nCoV by WHO (https://www.who.int/emergencies/diseases/ novel-coronavirus-2019), with 79.5% and 96% whole genome sequence identity to SARS-CoV and bat coronavirus SARSr-CoV-RaTG13, respectively, suggesting 2019-nCoV's potential bat origin (Zhou et al. 2020 ). However, no name for the pneumonia caused by 2019-nCoV infection has been provided by WHO or by the International Committee on Taxonomy of Viruses (ICTV) so far. Accordingly, some researchers decided to denote this novel coronavirus as ''novel coronavirus-infected pneumonia (NCIP)'' in their publication . However, in the generally accepted lexicon, one would then have to write ''novel coronavirus-infected pneumonia coronavirus (NCIP-CoV)'', which could be an awkward construct. Most reporters and media are still calling it ''Wuhan pneumonia'', but this terminology is detrimental to Wuhanese already facing lockdown and economic loss. The obvious dilemmas call for a renaming of the disease and virus causing the disease, one that is both lexicographically and taxonomically appropriate (news.ifeng.com/c/7tf2GFqwL4a).\nConfidence: 0.5427755117416382\n\n---------------------------------------------------------------------------------------------\n\n=================================================================================================\n=================================================================================================\n\n\nQuestion 4: what are the transmission routes of coronavirus?\n","output_type":"stream"},{"name":"stderr","text":"Getting inverted index...: 100%|██████████| 8004/8004 [00:00<00:00, 34358.57it/s]\n","output_type":"stream"},{"name":"stdout","text":"Document: Coughs and Sneezes: Their Role in Transmission of Respiratory Viral Infections, Including SARS-CoV-2\nParagraph: To reduce transmission of respiratory tract viruses:\nConfidence: 0.6974228620529175\n\nParagraph: A susceptible individual is exposed to infectious viruses by direct physical contact with the infected patient.\nConfidence: 0.6546083688735962\n\nParagraph: The infected patient releases infectious agents via droplets to susceptible individuals in close proximity and via droplet nuclei suspended in air to other individuals who are farther away.\nConfidence: 0.6423292756080627\n\n---------------------------------------------------------------------------------------------\nDocument: Why Does the SARS-CoV-2 Delta VOC Spread So Rapidly? Universal Conditions for the Rapid Spread of Respiratory Viruses, Minimum Viral Loads for Viral Aerosol Generation, Effects of Vaccination on Viral Aerosol Generation, and Viral Aerosol Clouds\nParagraph: The aerosol-droplet cutoff particle diameter for respiratory particles depends on the relative humidity and air flow conditions. This analytic study demonstrates that higher viral loads in respiratory fluids decrease the minimum size of virus-laden respiratory particles, therefore generation of viral aerosols is induced by the higher viral loads [4] . As shown in Figure 1 , generation of viral aerosols is determined by the minimum size of respiratory particles carrying viruses (d v , which is determined by the viral load) and the aerosol-droplet cutoff particle diameter (d c , which depends on the relative humidity and air flow conditions) ( Figure 1 ) [4] . The minimum size of respiratory particles carrying viruses (d v ) decreases (d v moves to the left, in Figure 1 ) under high viral load conditions [4] ; consequently, more viral aerosols can be generated under high viral load conditions. Generation of viral aerosols is determined by the minimum size of respiratory particles carrying viruses (dv, decided by a viral load) and the aerosol-droplet cutoff particle diameter (dc, which depends on the relative humidity and air flow conditions). \nConfidence: 0.6372393369674683\n\nParagraph: The viral load principle for viral aerosol transmission is an important key in supporting that viral aerosols in the environment can be dominant factors in the COVID-19 issues. The viral loads and viral aerosols can explain the conundrum of viral spreading, including super-spreading events, during the pandemic [21] [22] [23] [24] [25] .\nConfidence: 0.6214439272880554\n\nParagraph: There is an original principle for aerosol generation of respiratory viruses. Smaller respiratory particles among various sized respiratory particles can be airborne for longer periods and water on their surfaces has a shorter evaporation time [1, 2] .Therefore, if sufficiently small virus-carrying respiratory particles are released from hosts into ambient air, viruses inside the particles transform into aerosol nuclei (viral aerosols) after water evaporation. The aforementioned original principle is the basis for the minimum viral loads required to generate sufficiently small virus-carrying respiratory particles (for viral aerosols). If the aerosol-droplet cutoff particle diameter distinguishing potential aerosols from earthbound respiratory particles (droplets = earthbound respiratory particles) is 100 µm [3] , the minimum viral load required in respiratory fluids for generating the sufficiently small virus-carrying respiratory particles would be~10 6 copies mL −1 (virus size, 0.09 µm). This theoretical analysis on minimum viral loads for viral aerosol generation is based on the assumptions of a homogeneous distribution of viruses in respiratory fluids, considering one gene copy as a single virion, and a spherical volume ratio model for both respiratory particles and viruses (Supplementary Material Section) [4] . These assumptions can be considered as a limitation of this theoretical analysis. If the aerosol-droplet cutoff values distinguishing potential aerosols from earthbound particles are 50 µm, 10 µm, and 5 µm, the minimum viral loads required for generating viral aerosols are~10 7 copies mL −1 , 10 9 copies mL −1 , and~10 10 copies mL −1 , respectively (Table 1) . 100 µm~1.9 × 10 6 copies mL −1 50 µm~1.5 × 10 7 copies mL −1 10 µm~1.9 × 10 9 copies mL −1 5 µm~1.5 × 10 10 copies mL −1\nConfidence: 0.6105262637138367\n\n---------------------------------------------------------------------------------------------\nDocument: Transmissible Gastroenteritis Virus of Pigs and Porcine Epidemic Diarrhea Virus\nParagraph: Because clinical signs cannot distinguish between TGEV and PEDV infections, additional assays are required for diagnosis. Common diagnostic assays include viral antigen detection by histoimmunochemistry and enzyme-linked immuosorbent assay (ELISA), viral RNA detection and genotyping by PCR and sequencing, virus isolation by cell culture, as well as antibody detection by serology.\nConfidence: 0.6211646795272827\n\nParagraph: Abstract TGEV and PEDV are porcine coronaviruses with typical coronavirus virion structures and replication cycles. Both viruses cause diarrhea in pigs with similar clinical symptoms. Disease severity is inversely related to the age of animals. While TGEV infections are currently under control, recent emergence of virulent PEDV strains results in significant mortality. No effective treatments are available once animals are infected. Detection of viral antigen and/or genomic RNA is required for diagnosis. Prevention measures include enhanced biosecurity and vaccination with the latter being the most effective approach. Commercial vaccines for both viruses are available and numerous experimental vaccines are in development.\nConfidence: 0.5941750407218933\n\nParagraph: Villous enterocytes in small and large intestine are the major target cells of TGEV and PEDV infections. Viral infection causes cell death that results in villous atrophy followed by malabsorption, diarrhea, dehydration, anorexia, and eventually animal death. The molecular mechanisms for pathogenesis are not very well understood. There is evidence to suggest that viral proteins, such as spike and ORF3 proteins may affect viral virulence. At the molecular level, TGEV and PEDV have been shown to modulate multiple cellular processes/pathways including endoplasmic reticulum stress, cell cycle, and mitogen-activated protein kinase signaling.\nConfidence: 0.589299738407135\n\n---------------------------------------------------------------------------------------------\nDocument: Susceptibility of white-tailed deer (Odocoileus virginianus) to SARS-CoV-2\nParagraph: Additionally, transmission of SARS-CoV-2 to indirect contact animals was also investigated.\nConfidence: 0.6465855240821838\n\nParagraph: The origin of severe acute respiratory syndrome coronavirus 2 (SARS-CoV-2), the virus causing the global coronavirus disease 19 (COVID-19) pandemic, remains a mystery. Current evidence suggests a likely spillover into humans from an animal reservoir. Understanding the host range and identifying animal species that are susceptible to SARS-CoV-2 infection may help to elucidate the origin of the virus and the mechanisms underlying cross-species transmission to humans. Here we demonstrated that white-tailed deer (Odocoileus virginianus), an animal species in which the angiotensin converting enzyme 2 (ACE2) – the SARS-CoV-2 receptor – shares a high degree of similarity to humans, are highly susceptible to infection. Intranasal inoculation of deer fawns with SARS-CoV-2 resulted in established subclinical viral infection and shedding of infectious virus in nasal secretions. Notably, infected animals transmitted the virus to non-inoculated contact deer. Viral RNA was detected in multiple tissues 21 days post-inoculation (pi). All inoculated and indirect contact animals seroconverted and developed neutralizing antibodies as early as day 7 pi. The work provides important insights into the animal host range of SARS-CoV-2 and identifies white-tailed deer as a susceptible wild animal species to the virus. IMPORTANCE Given the presumed zoonotic origin of SARS-CoV-2, the human-animal-environment interface of COVID-19 pandemic is an area of great scientific and public- and animal-health interest. Identification of animal species that are susceptible to infection by SARS-CoV-2 may help to elucidate the potential origin of the virus, identify potential reservoirs or intermediate hosts, and define the mechanisms underlying cross-species transmission to humans. Additionally, it may also provide information and help to prevent potential reverse zoonosis that could lead to the establishment of a new wildlife hosts. Our data show that upon intranasal inoculation, white-tailed deer became subclinically infected and shed infectious SARS-CoV-2 in nasal secretions and feces. Importantly, indirect contact animals were infected and shed infectious virus, indicating efficient SARS-CoV-2 transmission from inoculated animals. These findings support the inclusion of wild cervid species in investigations conducted to assess potential reservoirs or sources of SARS-CoV-2 of infection.\nConfidence: 0.6140402555465698\n\nParagraph: Given that white-tailed-deer have been recently identified as one of the species with an ACE2 protein with high binding probability to the SARS-CoV-2 S protein (18), we investigated the susceptibility of white-taileddeer fawns to SARS-CoV-2 infection. Six-week-old fawns (n = 4) were inoculated intranasally with 5 ml (2.5 ml per nostril) of a virus suspension containing 10 6.3 TCID 50 .ml -1 of SARS-CoV-2 TGR/NY/20, an animal SARS-CoV-2 strain that is identical to human viral strains (22) . To assess the potential transmission of SARS-CoV-2 between white-tailed-deer, two fawns (n = 2) were maintained as non-inoculated contacts in the same biosafety level 3 (BSL-3Ag) room. The inoculated and indirect contact animals were kept in separate pens divided by a plexiglass barrier, to prevent direct nose-to-nose contact between inoculated and contact animals ( Fig. 2A) . pi and decreasing thereafter through day 21 pi (Fig. 3A) . Notably, high levels of viral RNA were also detected in nasal secretions from indirect contact animals throughout the experimental period (Fig. 3B ). Viral RNA from the feces was detected in all inoculated and indirect contact animals ( Fig. 3A and B) , however, intermittent and short duration fecal shedding was observed, with most animals (4/5) only transiently shedding detectable SARS-CoV-2 in feces through days 6-7 pi ( Fig. 3A and B) . Tissues were also analyzed using the anti-genomic negative sense RNA probe (V-nCoV2019-Ssense probe) to determine virus replication. While intense labeling was noted with the genomic probe ( Fig 6E) , no labeling was observed with the anti-sense probe in any of the tissues (Fig.   6F ), suggesting lack of virus replication. These findings are consistent with lack of virus isolation from RT-rPCR positive tissues. The ability of the anti-sense probe to bind actively replicating virus, was confirmed in Vero cells infected with SARS-CoV-2 isolate TGR/NY/20 (Suppl Fig. 1 ). One of the most remarkable characteristics of SARS-CoV-2 is its highly efficient transmissibility (35, 36) . Consistent with this, rapid SARS-CoV-2 transmission was observed in several experimentally infected animal species (27, 31, (37) (38) (39) (40) (41) (42) (43) . Similarly, results here show efficient SARS-CoV-2 transmission between white-tailed-deer. Indirect contact fawns became infected and supported productive SARS-CoV-2 replication as evidenced by virus shedding in nasal secretions and seroconversion. Evidence to date indicates that transmission of SARS-CoV-2 occurs mainly through direct, indirect, or close contact with infected individuals through infected secretions such as respiratory-, salivary-, and/or fecal droplets (44) . Under experimental conditions, transmission via direct contact has been demonstrated in ferrets, minks, raccoon dogs, hamsters, cats, and deer mice (28, 31, 37, 39, 40, 43, 45) , while indirect/aerosol transmission was observed in ferrets and hamsters (37, 39) . The experimental setup in the present study prevented direct contact between animals suggesting that transmission of SARS-CoV-2 from inoculated to room indirect contact animals most likely occurred via aerosols or droplets generated by inoculated animals.\nConfidence: 0.6106818914413452\n\n---------------------------------------------------------------------------------------------\nDocument: Transmission routes for SARS-COV-2 infection: Review of Evidence\nParagraph: The SARS-CoV-2 RNA has been detected from the feces of infected patients and hospital toilet bowls. Although viral RNA detection is not equivalent to a viable virus, it is a trace of the virus, so fecal-oral transmission may be a possible route of transmission (5, 8) . ACE2 is present in the gastrointestinal tract, so SARS-CoV-2 may enter the gastrointestinal tract and replicate. Due to the digestive symptoms in some patients and the prolonged viral shedding in the feces, the virus appears to have active proliferation in the gastrointestinal tract (9) .Therefore, there is a possibility of infection transmission in toilets through creating aerosols by flushing toilets and surfaces contact. The potential for fecal-oral transmission also emphasize hand hygiene and washing.\nConfidence: 0.6290968656539917\n\nParagraph: In summary, considering high transmission capacity of SARS-COV-2, prolonged viral shedding, lack of vaccine and systematic medication at this time, the most important action to limit infection is to cut off the transmission chain.\nConfidence: 0.5927790999412537\n\nParagraph: The possible main route of transmission is thought to be the close contact and respiratory droplets secreted by the patient during coughing, sneezing, breathing and even normal speeching. Therefore, it is necessary to maintain physical distance and using the face mask. Another route of transmission is through contaminated objects or fomites. Touching the T-zone of the face, after contact with these objects, is also a means of transmission, and it emphasizes hand hygiene and hand washing.\nConfidence: 0.5921418070793152\n\n---------------------------------------------------------------------------------------------\n\n=================================================================================================\n=================================================================================================\n\n\nQuestion 5: what are the best masks for preventing infection by Covid-19?\n","output_type":"stream"},{"name":"stderr","text":"Getting inverted index...: 100%|██████████| 9994/9994 [00:00<00:00, 34409.79it/s]\n","output_type":"stream"},{"name":"stdout","text":"Document: Immune response in COVID-19: What do we currently know?\nParagraph: The acute respiratory syndrome is a disease caused by the SARS-CoV-2 virus (COVID-37 19), where symptoms include difficulty breathing, high fever, and cough (WHO, 2020). \nConfidence: 0.666410505771637\n\nParagraph: In 2002/2003 there was a pandemic denominate SARS (severe acute respiratory syndrome), caused by the SARS-CoV virus that belongs to the genera Betacoranavirus and the family Coronaviridae, generally responsible for influenza infections. In mid of 2019, a new disease by the coronavirus named by COVID-19 (SARS-CoV-2) emerged, both infections have flu symptoms, however they are infections that variable intensity, being medium to severe. In medium infections individuals have the virus and exhibit symptoms, however hospitalization is not necessary, in severe infections, individuals are hospitalized, have high pathology and in some cases progress to death. The virus is formed by simple positive RNA, enveloped, non-segmented, and presenting the largest genome of viruses constituting 32 Kb, consisting of envelope proteins, membrane, nucleocapsid and spike protein, which is essential in the interaction with the host cells. As for the origin of this virus, research has been intensified to determine this paradox and although the similarity with SARS-CoV, this virus did not has necessarily the same place of origin. As for the immune system, it is currently unknown how this new virus interacts. In this brief review, we demonstrate important considerations about the responses to this infection.\nConfidence: 0.5169873833656311\n\n---------------------------------------------------------------------------------------------\nDocument: Novel coronavirus disease 2019 (COVID‐19) and neurodegenerative disorders\nParagraph: During the last few months, the whole humanity is experiencing largest and most severe sudden influx of COVID‐19 outbreak caused by the novel coronavirus (CoV) originated from Wuhan, China. According to the WHO reports, total 3 862 676 positive cases and 265 961 deaths have been recorded worldwide due to COVID‐19 infection as of May 9, 2020. CoVs are a large family of viruses (enveloped, single‐stranded RNA viruses), which includes severe acute respiratory syndrome coronavirus (SARS‐CoV) and Middle East respiratory syndrome‐related coronavirus (MERS‐CoV). New SARS CoV2 is the members of Betacoronavirus genus. These viruses cause infections in bats, camels and humans, and a few other associated species. Despite many neurologic complications associated with SARS‐CoV‐2 infection, it is still unclear whether these symptoms results from direct neural injury or due to some other reason. Currently, it appears that most of the neurological symptoms of COVID‐19 are nonspecific and secondary to the systemic illness. A single case of acute hemorrhagic necrotizing encephalopathy has been reported. SARS‐CoV‐2 associated Guillain‐Barré syndrome is an atypical case. Till today, no convincing evidence is available to confirm that the SARS‐CoV‐2 virus directly affects nerves system in humans. However, postinfection surveillance will be necessary to identify the possible post‐COVID‐19 neurologic syndromes.\nConfidence: 0.5604574680328369\n\nParagraph: During the last few months, the whole humanity is experiencing the largest and most severe sudden influx of COVID-19 outbreak caused by the novel coronavirus (CoV) originated from Wuhan city.\nConfidence: 0.5296371579170227\n\nParagraph: Coronavirus was officially announced as the causative pathogen of COVID-19 by the Chinese Center for Disease Control and Prevention on January 8, 2020. The World Health Organization announced coronavirus outbreak as a public health emergency of international concern. It started to expect disturbing extents over the globe as forewarned by the World Health Organization (WHO) reports associated Guillain-Barré syndrome is an atypical case. Till today, no convincing evidence is available to confirm that the SARS-CoV-2 virus directly affects the CNS or PNS in humans. However, postinfection surveillance will be necessary to identify the possible post-COVID-19 neurologic syndromes.\nConfidence: 0.48591047525405884\n\n---------------------------------------------------------------------------------------------\nDocument: Coronavirus Infections in Companion Animals: Virology, Epidemiology, Clinical and Pathologic Features\nParagraph: A genetically distinct coronavirus called ferret systemic coronavirus infection (FRSCV) was subsequently identified, which causes a systemic, progressive, and fatal pyogranulomatous inflammatory disease resembling the dry form of feline infectious peritonitis (FIP) in cats. The average age at the time of FRSCV diagnosis has been reported to be 11 months, and clinical signs include chronic weight loss, anorexia, diarrhea, palpable abdominal masses, and neurologic disease [97] .\nConfidence: 0.6384496688842773\n\nParagraph: Coronaviruses are enveloped RNA viruses capable of causing respiratory, enteric, or systemic diseases in a variety of mammalian hosts that vary in clinical severity from subclinical to fatal. The host range and tissue tropism are largely determined by the coronaviral spike protein, which initiates cellular infection by promoting fusion of the viral and host cell membranes. Companion animal coronaviruses responsible for causing enteric infection include feline enteric coronavirus, ferret enteric coronavirus, canine enteric coronavirus, equine coronavirus, and alpaca enteric coronavirus, while canine respiratory coronavirus and alpaca respiratory coronavirus result in respiratory infection. Ferret systemic coronavirus and feline infectious peritonitis virus, a mutated feline enteric coronavirus, can lead to lethal immuno-inflammatory systemic disease. Recent human viral pandemics, including severe acute respiratory syndrome (SARS), Middle East respiratory syndrome (MERS), and most recently, COVID-19, all thought to originate from bat coronaviruses, demonstrate the zoonotic potential of coronaviruses and their potential to have devastating impacts. A better understanding of the coronaviruses of companion animals, their capacity for cross-species transmission, and the sharing of genetic information may facilitate improved prevention and control strategies for future emerging zoonotic coronaviruses. This article reviews the clinical, epidemiologic, virologic, and pathologic characteristics of nine important coronaviruses of companion animals.\nConfidence: 0.6236833930015564\n\nParagraph: A novel alpaca coronavirus belonging to the alphacoronavirus genus was isolated in 2007 and associated with acute respiratory disease rather than enteric disease [144] . Complete genome sequencing revealed less than 50% nucleotide identity with the previously reported enteric alpaca coronavirus but a much higher 92.2% nucleotide identity with the human coronavirus (HCoV) 229E, with striking similarity between the HCoV 229E and alpaca respiratory coronavirus spike proteins. Comparison of spike gene sequences revealed that alpaca respiratory coronavirus is most similar to HCoV 229E strains isolated between the 1960s and 1980s, suggesting the possibility that a transmission event may have occurred between alpacas and humans [151] .\nConfidence: 0.6154036521911621\n\n---------------------------------------------------------------------------------------------\nDocument: The peripheral T cell population is associated with pneumonia severity in cynomolgus monkeys experimentally infected with severe acute respiratory syndrome coronavirus 2\nParagraph: 303 These results suggest that the initially inoculated virus (WK-521) was maintained in the lungs 304 and/or accessory lymph nodes, and that the second inoculated virus (QH-329-037) was 305 eliminated from the lungs of these monkeys soon after the second inoculation. \nConfidence: 0.5457214713096619\n\nParagraph: The coronavirus disease 2019 (COVID-19), caused by severe acute respiratory syndrome coronavirus 2 (SARS-CoV-2), is a global pandemic that began in December 2019. Lymphopenia is a common feature in severe cases of COVID-19; however, the role of T cell responses during infection is unclear. Here, we inoculated six cynomolgus monkeys, divided into two groups according to the CD3+ T cell population in peripheral blood, with two clinical isolates of SARS-CoV-2: one of East Asian lineage and one of European lineage. After initial infection with the isolate of East Asian lineage, all three monkeys in the CD3+ low group showed clinical symptoms, including loss of appetite, lethargy, and transient severe anemia with/without short-term fever, within 14 days post-infection (p.i.). By contrast, all three monkeys in the CD3+ high group showed mild clinical symptoms such as mild fever and loss of appetite within 4 days p.i. and then recovered. After a second inoculation with the isolate of European lineage, three of four animals in both groups showed mild clinical symptoms but recovered quickly. Hematological, immunological, and serological tests suggested that the CD3+ high and low groups mounted different immune responses during the initial and second infection stages. In both groups, anti-viral and innate immune responses were activated during the early phase of infection and re-infection. However, in the CD3+ low group, inflammatory responses, such as increased production of monocytes and neutrophils, were stronger than those in the CD3+ high group, leading to more severe immunopathology and failure to eliminate the virus. Taken together, the data suggest that the peripheral T lymphocyte population is associated with pneumonia severity in cynomolgus monkeys experimentally infected with SARS-CoV-2. Author summary SARS-CoV-2 infection causes an illness with clinical manifestations that vary from asymptomatic or mild to severe; examples include severe pneumonia and acute respiratory distress syndrome. Lymphopenia, which is common in severe COVID-19 cases, is characterized by markedly reduced numbers of CD4+ T cells, CD8+ T cells, B cells, and natural killer cells. Here, we showed that cynomolgus monkeys selected according to the T cell populations in peripheral blood have different outcomes after experimental infection with SARS-CoV-2. These findings will increase our understanding of disease pathogenesis and may facilitate the development of animal models for vaccine evaluation.\nConfidence: 0.5356321930885315\n\nParagraph: -16 -283 Table 2 The results revealed that the viral genome obtained from the tonsil of monkey #5404 287 after initial inoculation did not harbor any mutations (threshold = 50%). However, the viral 288 genome isolated from the lung of monkey #5412 harbored nine single nucleotide 289 polymorphisms (SNPs), including seven silent point mutations, one deletion resulting in a 290 frameshift mutation in the ORF1ab region, and a nonsynonymous mutation in ORF1b. The most 291 common base change was C > T. In addition, the genome isolated from mesenteric lymph nodes 292 from monkey #5412 harbored three SNPs, including two nonsynonymous mutations in the 293 ORF1a region and a synonymous mutation in the S region. The major sequence in these two 294 isolates was derived from WK-521, suggesting that the original inoculum replicated and resided 295 in the respiratory tract and intestine of monkey #5412, even after the second inoculation with 296 the heterologous strain. The viral genome obtained from the tonsil of monkey #5403 after the 297 second inoculation harbored a D614G mutation in the S region, suggesting the presence of 298 QH-329-037 in the tonsil after the 2nd inoculum (S3 Table) . The viral genome also obtained 299 from the tonsil of #5405 after the second inoculation harbored a D614G mutation in the S 300 region, suggesting the presence of QH-329-037 in the tonsil. Interestingly, the genome obtained 301 from the subcarinal lymph node of monkey #5405 did not harbor the D614G mutation, 302 suggesting that the original inoculum was maintained in the accessory lymph node of the lungs.\nConfidence: 0.5191962122917175\n\n---------------------------------------------------------------------------------------------\nDocument: COVID‐19 vaccine (mRNA BNT162b2) and COVID‐19 infection‐induced thrombotic thrombocytopenic purpura in adolescents\nParagraph: The mRNA COVID‐19 vaccine and COVID‐19 infection caused by the SARS‐CoV‐2 virus may be immunologic triggers for the development of thrombotic thrombocytopenic purpura (TTP). There is not yet literature that discusses TTP induced by COVID‐19 vaccination or infection in pediatric or adolescent patients. We describe three adolescents presenting with TTP (both de novo and relapsed disease) following administration of the Pfizer COVID‐19 vaccine or after COVID‐19 infection. Our observations demonstrate that the Pfizer‐BioNTech mRNA vaccine and COVID‐19 infection can act as triggers for the development/relapse of both congenital and acquired TTP.\nConfidence: 0.5481597781181335\n\nParagraph: The mRNA COVID-19 vaccine and COVID-19 infection caused by the SARS-CoV-2 virus may be immunologic triggers for the development of both acquired and congenital thrombotic thrombocytopenic purpura (TTP). We present a case series of adolescents presenting with TTP (including both de novo and relapsed disease) following administration of the Pfizer-BioNTech mRNA BNT162b2 anti-COVID-19 vaccine or after COVID-19 infection.\nConfidence: 0.532699704170227\n\nParagraph: Our In the case of inherited TTP, Galbusera and colleagues address a \"two hit model,\" highlighting cases in which patients with ADAMTS13 mutations manifest TTP after an infection or pregnancy. 16, 17 Mouse models of ADAMTS13 deficiency suggest an environmental trigger may be needed in addition to a gene mutation to fully manifest TTP, with ADAMTS13-deficient mice developing TTP after the introduction of Shiga toxin. 18 These studies imply that an immunologic event, such as an infection or vaccination, triggers inherited TTP. reports of TTP, and five of those were in patients <21 years old. 23 Of note, the VAERS data may not be entirely accurate, as some cases may not have been reported and the search criteria may not have included all TTP cases. As we show, COVID-19 infection itself can also trigger TTP, among many other severe and devastating consequences, and therefore, we continue to stress that the benefits of the vaccine outweigh the risks, especially with the development of new, more transmissible, and potentially virulent SARS-CoV-2 variants.\nConfidence: 0.4891185462474823\n\n---------------------------------------------------------------------------------------------\n\n=================================================================================================\n=================================================================================================\n\n\nQuestion 6: what type of hand sanitizer is needed to destroy Covid-19?\n","output_type":"stream"},{"name":"stderr","text":"Getting inverted index...: 100%|██████████| 9992/9992 [00:00<00:00, 34348.01it/s]\n","output_type":"stream"},{"name":"stdout","text":"Document: Immune response in COVID-19: What do we currently know?\nParagraph: The acute respiratory syndrome is a disease caused by the SARS-CoV-2 virus (COVID-37 19), where symptoms include difficulty breathing, high fever, and cough (WHO, 2020). \nConfidence: 0.6953975558280945\n\nParagraph: In 2002/2003 there was a pandemic denominate SARS (severe acute respiratory syndrome), caused by the SARS-CoV virus that belongs to the genera Betacoranavirus and the family Coronaviridae, generally responsible for influenza infections. In mid of 2019, a new disease by the coronavirus named by COVID-19 (SARS-CoV-2) emerged, both infections have flu symptoms, however they are infections that variable intensity, being medium to severe. In medium infections individuals have the virus and exhibit symptoms, however hospitalization is not necessary, in severe infections, individuals are hospitalized, have high pathology and in some cases progress to death. The virus is formed by simple positive RNA, enveloped, non-segmented, and presenting the largest genome of viruses constituting 32 Kb, consisting of envelope proteins, membrane, nucleocapsid and spike protein, which is essential in the interaction with the host cells. As for the origin of this virus, research has been intensified to determine this paradox and although the similarity with SARS-CoV, this virus did not has necessarily the same place of origin. As for the immune system, it is currently unknown how this new virus interacts. In this brief review, we demonstrate important considerations about the responses to this infection.\nConfidence: 0.5663904547691345\n\n---------------------------------------------------------------------------------------------\nDocument: Novel coronavirus disease 2019 (COVID‐19) and neurodegenerative disorders\nParagraph: During the last few months, the whole humanity is experiencing largest and most severe sudden influx of COVID‐19 outbreak caused by the novel coronavirus (CoV) originated from Wuhan, China. According to the WHO reports, total 3 862 676 positive cases and 265 961 deaths have been recorded worldwide due to COVID‐19 infection as of May 9, 2020. CoVs are a large family of viruses (enveloped, single‐stranded RNA viruses), which includes severe acute respiratory syndrome coronavirus (SARS‐CoV) and Middle East respiratory syndrome‐related coronavirus (MERS‐CoV). New SARS CoV2 is the members of Betacoronavirus genus. These viruses cause infections in bats, camels and humans, and a few other associated species. Despite many neurologic complications associated with SARS‐CoV‐2 infection, it is still unclear whether these symptoms results from direct neural injury or due to some other reason. Currently, it appears that most of the neurological symptoms of COVID‐19 are nonspecific and secondary to the systemic illness. A single case of acute hemorrhagic necrotizing encephalopathy has been reported. SARS‐CoV‐2 associated Guillain‐Barré syndrome is an atypical case. Till today, no convincing evidence is available to confirm that the SARS‐CoV‐2 virus directly affects nerves system in humans. However, postinfection surveillance will be necessary to identify the possible post‐COVID‐19 neurologic syndromes.\nConfidence: 0.6086735129356384\n\nParagraph: During the last few months, the whole humanity is experiencing the largest and most severe sudden influx of COVID-19 outbreak caused by the novel coronavirus (CoV) originated from Wuhan city.\nConfidence: 0.5706837773323059\n\nParagraph: Coronavirus was officially announced as the causative pathogen of COVID-19 by the Chinese Center for Disease Control and Prevention on January 8, 2020. The World Health Organization announced coronavirus outbreak as a public health emergency of international concern. It started to expect disturbing extents over the globe as forewarned by the World Health Organization (WHO) reports associated Guillain-Barré syndrome is an atypical case. Till today, no convincing evidence is available to confirm that the SARS-CoV-2 virus directly affects the CNS or PNS in humans. However, postinfection surveillance will be necessary to identify the possible post-COVID-19 neurologic syndromes.\nConfidence: 0.5360773801803589\n\n---------------------------------------------------------------------------------------------\nDocument: Coronavirus Infections in Companion Animals: Virology, Epidemiology, Clinical and Pathologic Features\nParagraph: A genetically distinct coronavirus called ferret systemic coronavirus infection (FRSCV) was subsequently identified, which causes a systemic, progressive, and fatal pyogranulomatous inflammatory disease resembling the dry form of feline infectious peritonitis (FIP) in cats. The average age at the time of FRSCV diagnosis has been reported to be 11 months, and clinical signs include chronic weight loss, anorexia, diarrhea, palpable abdominal masses, and neurologic disease [97] .\nConfidence: 0.6806122064590454\n\nParagraph: Coronaviruses are enveloped RNA viruses capable of causing respiratory, enteric, or systemic diseases in a variety of mammalian hosts that vary in clinical severity from subclinical to fatal. The host range and tissue tropism are largely determined by the coronaviral spike protein, which initiates cellular infection by promoting fusion of the viral and host cell membranes. Companion animal coronaviruses responsible for causing enteric infection include feline enteric coronavirus, ferret enteric coronavirus, canine enteric coronavirus, equine coronavirus, and alpaca enteric coronavirus, while canine respiratory coronavirus and alpaca respiratory coronavirus result in respiratory infection. Ferret systemic coronavirus and feline infectious peritonitis virus, a mutated feline enteric coronavirus, can lead to lethal immuno-inflammatory systemic disease. Recent human viral pandemics, including severe acute respiratory syndrome (SARS), Middle East respiratory syndrome (MERS), and most recently, COVID-19, all thought to originate from bat coronaviruses, demonstrate the zoonotic potential of coronaviruses and their potential to have devastating impacts. A better understanding of the coronaviruses of companion animals, their capacity for cross-species transmission, and the sharing of genetic information may facilitate improved prevention and control strategies for future emerging zoonotic coronaviruses. This article reviews the clinical, epidemiologic, virologic, and pathologic characteristics of nine important coronaviruses of companion animals.\nConfidence: 0.6658029556274414\n\nParagraph: A novel alpaca coronavirus belonging to the alphacoronavirus genus was isolated in 2007 and associated with acute respiratory disease rather than enteric disease [144] . Complete genome sequencing revealed less than 50% nucleotide identity with the previously reported enteric alpaca coronavirus but a much higher 92.2% nucleotide identity with the human coronavirus (HCoV) 229E, with striking similarity between the HCoV 229E and alpaca respiratory coronavirus spike proteins. Comparison of spike gene sequences revealed that alpaca respiratory coronavirus is most similar to HCoV 229E strains isolated between the 1960s and 1980s, suggesting the possibility that a transmission event may have occurred between alpacas and humans [151] .\nConfidence: 0.6559971570968628\n\n---------------------------------------------------------------------------------------------\nDocument: The peripheral T cell population is associated with pneumonia severity in cynomolgus monkeys experimentally infected with severe acute respiratory syndrome coronavirus 2\nParagraph: 303 These results suggest that the initially inoculated virus (WK-521) was maintained in the lungs 304 and/or accessory lymph nodes, and that the second inoculated virus (QH-329-037) was 305 eliminated from the lungs of these monkeys soon after the second inoculation. \nConfidence: 0.588115930557251\n\nParagraph: The coronavirus disease 2019 (COVID-19), caused by severe acute respiratory syndrome coronavirus 2 (SARS-CoV-2), is a global pandemic that began in December 2019. Lymphopenia is a common feature in severe cases of COVID-19; however, the role of T cell responses during infection is unclear. Here, we inoculated six cynomolgus monkeys, divided into two groups according to the CD3+ T cell population in peripheral blood, with two clinical isolates of SARS-CoV-2: one of East Asian lineage and one of European lineage. After initial infection with the isolate of East Asian lineage, all three monkeys in the CD3+ low group showed clinical symptoms, including loss of appetite, lethargy, and transient severe anemia with/without short-term fever, within 14 days post-infection (p.i.). By contrast, all three monkeys in the CD3+ high group showed mild clinical symptoms such as mild fever and loss of appetite within 4 days p.i. and then recovered. After a second inoculation with the isolate of European lineage, three of four animals in both groups showed mild clinical symptoms but recovered quickly. Hematological, immunological, and serological tests suggested that the CD3+ high and low groups mounted different immune responses during the initial and second infection stages. In both groups, anti-viral and innate immune responses were activated during the early phase of infection and re-infection. However, in the CD3+ low group, inflammatory responses, such as increased production of monocytes and neutrophils, were stronger than those in the CD3+ high group, leading to more severe immunopathology and failure to eliminate the virus. Taken together, the data suggest that the peripheral T lymphocyte population is associated with pneumonia severity in cynomolgus monkeys experimentally infected with SARS-CoV-2. Author summary SARS-CoV-2 infection causes an illness with clinical manifestations that vary from asymptomatic or mild to severe; examples include severe pneumonia and acute respiratory distress syndrome. Lymphopenia, which is common in severe COVID-19 cases, is characterized by markedly reduced numbers of CD4+ T cells, CD8+ T cells, B cells, and natural killer cells. Here, we showed that cynomolgus monkeys selected according to the T cell populations in peripheral blood have different outcomes after experimental infection with SARS-CoV-2. These findings will increase our understanding of disease pathogenesis and may facilitate the development of animal models for vaccine evaluation.\nConfidence: 0.5837485790252686\n\nParagraph: -16 -283 Table 2 The results revealed that the viral genome obtained from the tonsil of monkey #5404 287 after initial inoculation did not harbor any mutations (threshold = 50%). However, the viral 288 genome isolated from the lung of monkey #5412 harbored nine single nucleotide 289 polymorphisms (SNPs), including seven silent point mutations, one deletion resulting in a 290 frameshift mutation in the ORF1ab region, and a nonsynonymous mutation in ORF1b. The most 291 common base change was C > T. In addition, the genome isolated from mesenteric lymph nodes 292 from monkey #5412 harbored three SNPs, including two nonsynonymous mutations in the 293 ORF1a region and a synonymous mutation in the S region. The major sequence in these two 294 isolates was derived from WK-521, suggesting that the original inoculum replicated and resided 295 in the respiratory tract and intestine of monkey #5412, even after the second inoculation with 296 the heterologous strain. The viral genome obtained from the tonsil of monkey #5403 after the 297 second inoculation harbored a D614G mutation in the S region, suggesting the presence of 298 QH-329-037 in the tonsil after the 2nd inoculum (S3 Table) . The viral genome also obtained 299 from the tonsil of #5405 after the second inoculation harbored a D614G mutation in the S 300 region, suggesting the presence of QH-329-037 in the tonsil. Interestingly, the genome obtained 301 from the subcarinal lymph node of monkey #5405 did not harbor the D614G mutation, 302 suggesting that the original inoculum was maintained in the accessory lymph node of the lungs.\nConfidence: 0.5655393600463867\n\n---------------------------------------------------------------------------------------------\nDocument: The First Two Imported Cases of SARS-CoV-2 Omicron Variant — Tianjin Municipality, China, December 13, 2021\nParagraph: On December 9, 2021, 2 international passengers arrived in Tianjin Binhai International Airport from Warsaw, Poland via airplane and tested positive for severe acute respiratory syndrome coronavirus 2 (SARS-CoV-2), which causes coronavirus disease 2019 , using nucleic acid tests. (3).\nConfidence: 0.7459763288497925\n\nParagraph: On December 9, 2021, 2 international passengers arrived in Tianjin Binhai International Airport from Warsaw, Poland via airplane and tested positive for severe acute respiratory syndrome coronavirus 2 (SARS-CoV-2), which causes coronavirus disease 2019 (COVID-19), using nucleic acid tests. The first case (Patient A) was a flight crew member, a 35-year-old male, who had returned to Warsaw with the same flight on December 10, 2021. The second case (Patient B) was a 17-year-old female student, who has been transferred to Haihe Hospital for isolation. Both cases are Polish nationality that had recently lived in Warsaw, Poland. According to the investigation, neither patient had disease symptoms nor medication history. Patient B received the Pfizer/BNT162b2 vaccine on September 27, 2021. Patient A was vaccinated on May 25, 2021, but no further vaccine information was collected as Patient A had left China. No other infections from this flight have been detected.\nConfidence: 0.5929173827171326\n\nParagraph: A total of 40 amino acid mutation sites (A67V, T95I, Y145D, L212I, G339D, S371L, S373P, S375F, K417N, N440K, G446S, S477N, T478K, E484A, Q493R, G496S, Q498R, N501Y, Y505H, T547K, D614G, H655Y, N679K, P681H, A701V, N764K, D796Y, N856K, Q954H, N969K, L981F, H69del, V70del, G142del, V143del, Y144del, N211del, and 214insEPE) and at least 26 amino acid mutation sites (T19R, G142D, C361R, K424I, N439I, N440K, G446S, S477N, T478K, E484A, Q493R, G496S, F497Y, Q498R, N501Y, Y505H, N679K, P681H, A701V, N764K, D796Y, N856K, Q954H, N969K, L981F, and P1069S) were detected in the spike protein of the strains from Patient A and B, respectively. Some of the mutations in the receptor binding domain and near the furin cleavage were concerning and may be associated with immune escape potency and higher transmissibility (4).\nConfidence: 0.5371033549308777\n\n---------------------------------------------------------------------------------------------\n\n=================================================================================================\n=================================================================================================\n\n\nQuestion 7: What vaccine candidates are being tested for Covid-19?\n","output_type":"stream"},{"name":"stderr","text":"Getting inverted index...: 100%|██████████| 9995/9995 [00:00<00:00, 33949.48it/s]\n","output_type":"stream"},{"name":"stdout","text":"Document: Immune response in COVID-19: What do we currently know?\nParagraph: The acute respiratory syndrome is a disease caused by the SARS-CoV-2 virus (COVID-37 19), where symptoms include difficulty breathing, high fever, and cough (WHO, 2020). \nConfidence: 0.6082919836044312\n\nParagraph: In 2002/2003 there was a pandemic denominate SARS (severe acute respiratory syndrome), caused by the SARS-CoV virus that belongs to the genera Betacoranavirus and the family Coronaviridae, generally responsible for influenza infections. In mid of 2019, a new disease by the coronavirus named by COVID-19 (SARS-CoV-2) emerged, both infections have flu symptoms, however they are infections that variable intensity, being medium to severe. In medium infections individuals have the virus and exhibit symptoms, however hospitalization is not necessary, in severe infections, individuals are hospitalized, have high pathology and in some cases progress to death. The virus is formed by simple positive RNA, enveloped, non-segmented, and presenting the largest genome of viruses constituting 32 Kb, consisting of envelope proteins, membrane, nucleocapsid and spike protein, which is essential in the interaction with the host cells. As for the origin of this virus, research has been intensified to determine this paradox and although the similarity with SARS-CoV, this virus did not has necessarily the same place of origin. As for the immune system, it is currently unknown how this new virus interacts. In this brief review, we demonstrate important considerations about the responses to this infection.\nConfidence: 0.4754858911037445\n\n---------------------------------------------------------------------------------------------\nDocument: Novel coronavirus disease 2019 (COVID‐19) and neurodegenerative disorders\nParagraph: During the last few months, the whole humanity is experiencing largest and most severe sudden influx of COVID‐19 outbreak caused by the novel coronavirus (CoV) originated from Wuhan, China. According to the WHO reports, total 3 862 676 positive cases and 265 961 deaths have been recorded worldwide due to COVID‐19 infection as of May 9, 2020. CoVs are a large family of viruses (enveloped, single‐stranded RNA viruses), which includes severe acute respiratory syndrome coronavirus (SARS‐CoV) and Middle East respiratory syndrome‐related coronavirus (MERS‐CoV). New SARS CoV2 is the members of Betacoronavirus genus. These viruses cause infections in bats, camels and humans, and a few other associated species. Despite many neurologic complications associated with SARS‐CoV‐2 infection, it is still unclear whether these symptoms results from direct neural injury or due to some other reason. Currently, it appears that most of the neurological symptoms of COVID‐19 are nonspecific and secondary to the systemic illness. A single case of acute hemorrhagic necrotizing encephalopathy has been reported. SARS‐CoV‐2 associated Guillain‐Barré syndrome is an atypical case. Till today, no convincing evidence is available to confirm that the SARS‐CoV‐2 virus directly affects nerves system in humans. However, postinfection surveillance will be necessary to identify the possible post‐COVID‐19 neurologic syndromes.\nConfidence: 0.5119127035140991\n\nParagraph: During the last few months, the whole humanity is experiencing the largest and most severe sudden influx of COVID-19 outbreak caused by the novel coronavirus (CoV) originated from Wuhan city.\nConfidence: 0.47777944803237915\n\nParagraph: Coronavirus was officially announced as the causative pathogen of COVID-19 by the Chinese Center for Disease Control and Prevention on January 8, 2020. The World Health Organization announced coronavirus outbreak as a public health emergency of international concern. It started to expect disturbing extents over the globe as forewarned by the World Health Organization (WHO) reports associated Guillain-Barré syndrome is an atypical case. Till today, no convincing evidence is available to confirm that the SARS-CoV-2 virus directly affects the CNS or PNS in humans. However, postinfection surveillance will be necessary to identify the possible post-COVID-19 neurologic syndromes.\nConfidence: 0.44425609707832336\n\n---------------------------------------------------------------------------------------------\nDocument: The peripheral T cell population is associated with pneumonia severity in cynomolgus monkeys experimentally infected with severe acute respiratory syndrome coronavirus 2\nParagraph: 303 These results suggest that the initially inoculated virus (WK-521) was maintained in the lungs 304 and/or accessory lymph nodes, and that the second inoculated virus (QH-329-037) was 305 eliminated from the lungs of these monkeys soon after the second inoculation. \nConfidence: 0.5062987208366394\n\nParagraph: The coronavirus disease 2019 (COVID-19), caused by severe acute respiratory syndrome coronavirus 2 (SARS-CoV-2), is a global pandemic that began in December 2019. Lymphopenia is a common feature in severe cases of COVID-19; however, the role of T cell responses during infection is unclear. Here, we inoculated six cynomolgus monkeys, divided into two groups according to the CD3+ T cell population in peripheral blood, with two clinical isolates of SARS-CoV-2: one of East Asian lineage and one of European lineage. After initial infection with the isolate of East Asian lineage, all three monkeys in the CD3+ low group showed clinical symptoms, including loss of appetite, lethargy, and transient severe anemia with/without short-term fever, within 14 days post-infection (p.i.). By contrast, all three monkeys in the CD3+ high group showed mild clinical symptoms such as mild fever and loss of appetite within 4 days p.i. and then recovered. After a second inoculation with the isolate of European lineage, three of four animals in both groups showed mild clinical symptoms but recovered quickly. Hematological, immunological, and serological tests suggested that the CD3+ high and low groups mounted different immune responses during the initial and second infection stages. In both groups, anti-viral and innate immune responses were activated during the early phase of infection and re-infection. However, in the CD3+ low group, inflammatory responses, such as increased production of monocytes and neutrophils, were stronger than those in the CD3+ high group, leading to more severe immunopathology and failure to eliminate the virus. Taken together, the data suggest that the peripheral T lymphocyte population is associated with pneumonia severity in cynomolgus monkeys experimentally infected with SARS-CoV-2. Author summary SARS-CoV-2 infection causes an illness with clinical manifestations that vary from asymptomatic or mild to severe; examples include severe pneumonia and acute respiratory distress syndrome. Lymphopenia, which is common in severe COVID-19 cases, is characterized by markedly reduced numbers of CD4+ T cells, CD8+ T cells, B cells, and natural killer cells. Here, we showed that cynomolgus monkeys selected according to the T cell populations in peripheral blood have different outcomes after experimental infection with SARS-CoV-2. These findings will increase our understanding of disease pathogenesis and may facilitate the development of animal models for vaccine evaluation.\nConfidence: 0.4908253848552704\n\nParagraph: -16 -283 Table 2 The results revealed that the viral genome obtained from the tonsil of monkey #5404 287 after initial inoculation did not harbor any mutations (threshold = 50%). However, the viral 288 genome isolated from the lung of monkey #5412 harbored nine single nucleotide 289 polymorphisms (SNPs), including seven silent point mutations, one deletion resulting in a 290 frameshift mutation in the ORF1ab region, and a nonsynonymous mutation in ORF1b. The most 291 common base change was C > T. In addition, the genome isolated from mesenteric lymph nodes 292 from monkey #5412 harbored three SNPs, including two nonsynonymous mutations in the 293 ORF1a region and a synonymous mutation in the S region. The major sequence in these two 294 isolates was derived from WK-521, suggesting that the original inoculum replicated and resided 295 in the respiratory tract and intestine of monkey #5412, even after the second inoculation with 296 the heterologous strain. The viral genome obtained from the tonsil of monkey #5403 after the 297 second inoculation harbored a D614G mutation in the S region, suggesting the presence of 298 QH-329-037 in the tonsil after the 2nd inoculum (S3 Table) . The viral genome also obtained 299 from the tonsil of #5405 after the second inoculation harbored a D614G mutation in the S 300 region, suggesting the presence of QH-329-037 in the tonsil. Interestingly, the genome obtained 301 from the subcarinal lymph node of monkey #5405 did not harbor the D614G mutation, 302 suggesting that the original inoculum was maintained in the accessory lymph node of the lungs.\nConfidence: 0.4848773181438446\n\n---------------------------------------------------------------------------------------------\nDocument: Local skin reaction to the AZD1222 vaccine in a patient who survived COVID‐19\nParagraph: Real allergic reactions to vaccines are rare events, and they may be linked to a direct response or cross-reaction to preservatives and excipients. 3 The AZD1222 vaccine developed by AstraZeneca/Oxford University contains the wild-type SARS-CoV-2 spike protein which is vectored by the attenuated adenovirus ChAdOx1 (a simian virus). 1 Notably, this vaccine against the novel coronavirus has been administered to the population since January 2021. 1 The potential excipient of this vaccine that may trigger allergic reactions is polysorbate 80. \nConfidence: 0.5361706018447876\n\nParagraph: To the Editor, Since the beginning of the COVID-19 pandemic, strenuous efforts to create a herd immunity against this deadly disease have been made. As a matter of fact, multiple vaccines harnessing different strategies have been developed: virus-like vaccines, protein subunit vaccines, RNA-based vaccines, inactivated vaccines, protein subunit vaccines, and vector vaccines have been created by the international scientific community. 1 The vaccines against the severe acute respiratory syndrome coronavirus 2 (SARS-CoV-2) that have been approved in Italy are the mRNA vaccines and the viral vector vaccines Ad26.COV2.S and AZD1222. 2 We report the cutaneous reaction after the AZD122 vaccine of a patient with no history of allergy.\nConfidence: 0.5273125171661377\n\nParagraph: Since the beginning of the COVID-19 pandemic strenuous efforts to create a herd immunity against this deadly disease have been made. As a matter of fact, multiple vaccines harnessing different strategies have been developed: virus-like vaccines, protein subunit vaccines, RNA-based vaccines, inactivated vaccines, protein subunit vaccines and vector vaccines have been created by the international scientific community.\nConfidence: 0.5238377451896667\n\n---------------------------------------------------------------------------------------------\nDocument: COVID‐19 vaccine (mRNA BNT162b2) and COVID‐19 infection‐induced thrombotic thrombocytopenic purpura in adolescents\nParagraph: The mRNA COVID‐19 vaccine and COVID‐19 infection caused by the SARS‐CoV‐2 virus may be immunologic triggers for the development of thrombotic thrombocytopenic purpura (TTP). There is not yet literature that discusses TTP induced by COVID‐19 vaccination or infection in pediatric or adolescent patients. We describe three adolescents presenting with TTP (both de novo and relapsed disease) following administration of the Pfizer COVID‐19 vaccine or after COVID‐19 infection. Our observations demonstrate that the Pfizer‐BioNTech mRNA vaccine and COVID‐19 infection can act as triggers for the development/relapse of both congenital and acquired TTP.\nConfidence: 0.504795253276825\n\nParagraph: The mRNA COVID-19 vaccine and COVID-19 infection caused by the SARS-CoV-2 virus may be immunologic triggers for the development of both acquired and congenital thrombotic thrombocytopenic purpura (TTP). We present a case series of adolescents presenting with TTP (including both de novo and relapsed disease) following administration of the Pfizer-BioNTech mRNA BNT162b2 anti-COVID-19 vaccine or after COVID-19 infection.\nConfidence: 0.4909549057483673\n\nParagraph: Our In the case of inherited TTP, Galbusera and colleagues address a \"two hit model,\" highlighting cases in which patients with ADAMTS13 mutations manifest TTP after an infection or pregnancy. 16, 17 Mouse models of ADAMTS13 deficiency suggest an environmental trigger may be needed in addition to a gene mutation to fully manifest TTP, with ADAMTS13-deficient mice developing TTP after the introduction of Shiga toxin. 18 These studies imply that an immunologic event, such as an infection or vaccination, triggers inherited TTP. reports of TTP, and five of those were in patients <21 years old. 23 Of note, the VAERS data may not be entirely accurate, as some cases may not have been reported and the search criteria may not have included all TTP cases. As we show, COVID-19 infection itself can also trigger TTP, among many other severe and devastating consequences, and therefore, we continue to stress that the benefits of the vaccine outweigh the risks, especially with the development of new, more transmissible, and potentially virulent SARS-CoV-2 variants.\nConfidence: 0.4446291923522949\n\n---------------------------------------------------------------------------------------------\n\n=================================================================================================\n=================================================================================================\n\n\nQuestion 8: does Vitamin D impact COVID-19 prevention and treatment?\n","output_type":"stream"},{"name":"stderr","text":"Getting inverted index...: 100%|██████████| 9991/9991 [00:00<00:00, 34954.31it/s]\n","output_type":"stream"},{"name":"stdout","text":"Document: Immune response in COVID-19: What do we currently know?\nParagraph: The acute respiratory syndrome is a disease caused by the SARS-CoV-2 virus (COVID-37 19), where symptoms include difficulty breathing, high fever, and cough (WHO, 2020). \nConfidence: 0.6615413427352905\n\nParagraph: In 2002/2003 there was a pandemic denominate SARS (severe acute respiratory syndrome), caused by the SARS-CoV virus that belongs to the genera Betacoranavirus and the family Coronaviridae, generally responsible for influenza infections. In mid of 2019, a new disease by the coronavirus named by COVID-19 (SARS-CoV-2) emerged, both infections have flu symptoms, however they are infections that variable intensity, being medium to severe. In medium infections individuals have the virus and exhibit symptoms, however hospitalization is not necessary, in severe infections, individuals are hospitalized, have high pathology and in some cases progress to death. The virus is formed by simple positive RNA, enveloped, non-segmented, and presenting the largest genome of viruses constituting 32 Kb, consisting of envelope proteins, membrane, nucleocapsid and spike protein, which is essential in the interaction with the host cells. As for the origin of this virus, research has been intensified to determine this paradox and although the similarity with SARS-CoV, this virus did not has necessarily the same place of origin. As for the immune system, it is currently unknown how this new virus interacts. In this brief review, we demonstrate important considerations about the responses to this infection.\nConfidence: 0.5082867741584778\n\n---------------------------------------------------------------------------------------------\nDocument: Novel coronavirus disease 2019 (COVID‐19) and neurodegenerative disorders\nParagraph: During the last few months, the whole humanity is experiencing largest and most severe sudden influx of COVID‐19 outbreak caused by the novel coronavirus (CoV) originated from Wuhan, China. According to the WHO reports, total 3 862 676 positive cases and 265 961 deaths have been recorded worldwide due to COVID‐19 infection as of May 9, 2020. CoVs are a large family of viruses (enveloped, single‐stranded RNA viruses), which includes severe acute respiratory syndrome coronavirus (SARS‐CoV) and Middle East respiratory syndrome‐related coronavirus (MERS‐CoV). New SARS CoV2 is the members of Betacoronavirus genus. These viruses cause infections in bats, camels and humans, and a few other associated species. Despite many neurologic complications associated with SARS‐CoV‐2 infection, it is still unclear whether these symptoms results from direct neural injury or due to some other reason. Currently, it appears that most of the neurological symptoms of COVID‐19 are nonspecific and secondary to the systemic illness. A single case of acute hemorrhagic necrotizing encephalopathy has been reported. SARS‐CoV‐2 associated Guillain‐Barré syndrome is an atypical case. Till today, no convincing evidence is available to confirm that the SARS‐CoV‐2 virus directly affects nerves system in humans. However, postinfection surveillance will be necessary to identify the possible post‐COVID‐19 neurologic syndromes.\nConfidence: 0.5549229383468628\n\nParagraph: During the last few months, the whole humanity is experiencing the largest and most severe sudden influx of COVID-19 outbreak caused by the novel coronavirus (CoV) originated from Wuhan city.\nConfidence: 0.5225597023963928\n\nParagraph: Coronavirus was officially announced as the causative pathogen of COVID-19 by the Chinese Center for Disease Control and Prevention on January 8, 2020. The World Health Organization announced coronavirus outbreak as a public health emergency of international concern. It started to expect disturbing extents over the globe as forewarned by the World Health Organization (WHO) reports associated Guillain-Barré syndrome is an atypical case. Till today, no convincing evidence is available to confirm that the SARS-CoV-2 virus directly affects the CNS or PNS in humans. However, postinfection surveillance will be necessary to identify the possible post-COVID-19 neurologic syndromes.\nConfidence: 0.4788472354412079\n\n---------------------------------------------------------------------------------------------\nDocument: Coronavirus Infections in Companion Animals: Virology, Epidemiology, Clinical and Pathologic Features\nParagraph: A genetically distinct coronavirus called ferret systemic coronavirus infection (FRSCV) was subsequently identified, which causes a systemic, progressive, and fatal pyogranulomatous inflammatory disease resembling the dry form of feline infectious peritonitis (FIP) in cats. The average age at the time of FRSCV diagnosis has been reported to be 11 months, and clinical signs include chronic weight loss, anorexia, diarrhea, palpable abdominal masses, and neurologic disease [97] .\nConfidence: 0.6344701051712036\n\nParagraph: Coronaviruses are enveloped RNA viruses capable of causing respiratory, enteric, or systemic diseases in a variety of mammalian hosts that vary in clinical severity from subclinical to fatal. The host range and tissue tropism are largely determined by the coronaviral spike protein, which initiates cellular infection by promoting fusion of the viral and host cell membranes. Companion animal coronaviruses responsible for causing enteric infection include feline enteric coronavirus, ferret enteric coronavirus, canine enteric coronavirus, equine coronavirus, and alpaca enteric coronavirus, while canine respiratory coronavirus and alpaca respiratory coronavirus result in respiratory infection. Ferret systemic coronavirus and feline infectious peritonitis virus, a mutated feline enteric coronavirus, can lead to lethal immuno-inflammatory systemic disease. Recent human viral pandemics, including severe acute respiratory syndrome (SARS), Middle East respiratory syndrome (MERS), and most recently, COVID-19, all thought to originate from bat coronaviruses, demonstrate the zoonotic potential of coronaviruses and their potential to have devastating impacts. A better understanding of the coronaviruses of companion animals, their capacity for cross-species transmission, and the sharing of genetic information may facilitate improved prevention and control strategies for future emerging zoonotic coronaviruses. This article reviews the clinical, epidemiologic, virologic, and pathologic characteristics of nine important coronaviruses of companion animals.\nConfidence: 0.6208584308624268\n\nParagraph: A novel alpaca coronavirus belonging to the alphacoronavirus genus was isolated in 2007 and associated with acute respiratory disease rather than enteric disease [144] . Complete genome sequencing revealed less than 50% nucleotide identity with the previously reported enteric alpaca coronavirus but a much higher 92.2% nucleotide identity with the human coronavirus (HCoV) 229E, with striking similarity between the HCoV 229E and alpaca respiratory coronavirus spike proteins. Comparison of spike gene sequences revealed that alpaca respiratory coronavirus is most similar to HCoV 229E strains isolated between the 1960s and 1980s, suggesting the possibility that a transmission event may have occurred between alpacas and humans [151] .\nConfidence: 0.600669264793396\n\n---------------------------------------------------------------------------------------------\nDocument: The peripheral T cell population is associated with pneumonia severity in cynomolgus monkeys experimentally infected with severe acute respiratory syndrome coronavirus 2\nParagraph: The coronavirus disease 2019 (COVID-19), caused by severe acute respiratory syndrome coronavirus 2 (SARS-CoV-2), is a global pandemic that began in December 2019. Lymphopenia is a common feature in severe cases of COVID-19; however, the role of T cell responses during infection is unclear. Here, we inoculated six cynomolgus monkeys, divided into two groups according to the CD3+ T cell population in peripheral blood, with two clinical isolates of SARS-CoV-2: one of East Asian lineage and one of European lineage. After initial infection with the isolate of East Asian lineage, all three monkeys in the CD3+ low group showed clinical symptoms, including loss of appetite, lethargy, and transient severe anemia with/without short-term fever, within 14 days post-infection (p.i.). By contrast, all three monkeys in the CD3+ high group showed mild clinical symptoms such as mild fever and loss of appetite within 4 days p.i. and then recovered. After a second inoculation with the isolate of European lineage, three of four animals in both groups showed mild clinical symptoms but recovered quickly. Hematological, immunological, and serological tests suggested that the CD3+ high and low groups mounted different immune responses during the initial and second infection stages. In both groups, anti-viral and innate immune responses were activated during the early phase of infection and re-infection. However, in the CD3+ low group, inflammatory responses, such as increased production of monocytes and neutrophils, were stronger than those in the CD3+ high group, leading to more severe immunopathology and failure to eliminate the virus. Taken together, the data suggest that the peripheral T lymphocyte population is associated with pneumonia severity in cynomolgus monkeys experimentally infected with SARS-CoV-2. Author summary SARS-CoV-2 infection causes an illness with clinical manifestations that vary from asymptomatic or mild to severe; examples include severe pneumonia and acute respiratory distress syndrome. Lymphopenia, which is common in severe COVID-19 cases, is characterized by markedly reduced numbers of CD4+ T cells, CD8+ T cells, B cells, and natural killer cells. Here, we showed that cynomolgus monkeys selected according to the T cell populations in peripheral blood have different outcomes after experimental infection with SARS-CoV-2. These findings will increase our understanding of disease pathogenesis and may facilitate the development of animal models for vaccine evaluation.\nConfidence: 0.5338848233222961\n\nParagraph: 303 These results suggest that the initially inoculated virus (WK-521) was maintained in the lungs 304 and/or accessory lymph nodes, and that the second inoculated virus (QH-329-037) was 305 eliminated from the lungs of these monkeys soon after the second inoculation. \nConfidence: 0.528554379940033\n\nParagraph: -16 -283 Table 2 The results revealed that the viral genome obtained from the tonsil of monkey #5404 287 after initial inoculation did not harbor any mutations (threshold = 50%). However, the viral 288 genome isolated from the lung of monkey #5412 harbored nine single nucleotide 289 polymorphisms (SNPs), including seven silent point mutations, one deletion resulting in a 290 frameshift mutation in the ORF1ab region, and a nonsynonymous mutation in ORF1b. The most 291 common base change was C > T. In addition, the genome isolated from mesenteric lymph nodes 292 from monkey #5412 harbored three SNPs, including two nonsynonymous mutations in the 293 ORF1a region and a synonymous mutation in the S region. The major sequence in these two 294 isolates was derived from WK-521, suggesting that the original inoculum replicated and resided 295 in the respiratory tract and intestine of monkey #5412, even after the second inoculation with 296 the heterologous strain. The viral genome obtained from the tonsil of monkey #5403 after the 297 second inoculation harbored a D614G mutation in the S region, suggesting the presence of 298 QH-329-037 in the tonsil after the 2nd inoculum (S3 Table) . The viral genome also obtained 299 from the tonsil of #5405 after the second inoculation harbored a D614G mutation in the S 300 region, suggesting the presence of QH-329-037 in the tonsil. Interestingly, the genome obtained 301 from the subcarinal lymph node of monkey #5405 did not harbor the D614G mutation, 302 suggesting that the original inoculum was maintained in the accessory lymph node of the lungs.\nConfidence: 0.5061053037643433\n\n---------------------------------------------------------------------------------------------\nDocument: COVID‐19 vaccine (mRNA BNT162b2) and COVID‐19 infection‐induced thrombotic thrombocytopenic purpura in adolescents\nParagraph: The mRNA COVID‐19 vaccine and COVID‐19 infection caused by the SARS‐CoV‐2 virus may be immunologic triggers for the development of thrombotic thrombocytopenic purpura (TTP). There is not yet literature that discusses TTP induced by COVID‐19 vaccination or infection in pediatric or adolescent patients. We describe three adolescents presenting with TTP (both de novo and relapsed disease) following administration of the Pfizer COVID‐19 vaccine or after COVID‐19 infection. Our observations demonstrate that the Pfizer‐BioNTech mRNA vaccine and COVID‐19 infection can act as triggers for the development/relapse of both congenital and acquired TTP.\nConfidence: 0.5427879095077515\n\nParagraph: The mRNA COVID-19 vaccine and COVID-19 infection caused by the SARS-CoV-2 virus may be immunologic triggers for the development of both acquired and congenital thrombotic thrombocytopenic purpura (TTP). We present a case series of adolescents presenting with TTP (including both de novo and relapsed disease) following administration of the Pfizer-BioNTech mRNA BNT162b2 anti-COVID-19 vaccine or after COVID-19 infection.\nConfidence: 0.5260177254676819\n\nParagraph: Our In the case of inherited TTP, Galbusera and colleagues address a \"two hit model,\" highlighting cases in which patients with ADAMTS13 mutations manifest TTP after an infection or pregnancy. 16, 17 Mouse models of ADAMTS13 deficiency suggest an environmental trigger may be needed in addition to a gene mutation to fully manifest TTP, with ADAMTS13-deficient mice developing TTP after the introduction of Shiga toxin. 18 These studies imply that an immunologic event, such as an infection or vaccination, triggers inherited TTP. reports of TTP, and five of those were in patients <21 years old. 23 Of note, the VAERS data may not be entirely accurate, as some cases may not have been reported and the search criteria may not have included all TTP cases. As we show, COVID-19 infection itself can also trigger TTP, among many other severe and devastating consequences, and therefore, we continue to stress that the benefits of the vaccine outweigh the risks, especially with the development of new, more transmissible, and potentially virulent SARS-CoV-2 variants.\nConfidence: 0.4839853048324585\n\n---------------------------------------------------------------------------------------------\n\n=================================================================================================\n=================================================================================================\n\n\nQuestion 9: how long can the coronavirus live outside the body?\n","output_type":"stream"},{"name":"stderr","text":"Getting inverted index...: 100%|██████████| 8227/8227 [00:00<00:00, 31932.97it/s]\n","output_type":"stream"},{"name":"stdout","text":"Document: Stitches in the Heart: A Small History of Scars\nParagraph: The human heart consists of knotted tubes through which the living flows are propelled onwards. It beats silently in the dark chamber of the chest. Oxygenised blood, coming from the lungs, is transported through arteries and veins mostly unperceived. A mode of disappearance of our inner rhythmic alternation is at work, a central absence of the visceral processes. Inspired by philosopher Drew Leder's study The Absent Body (1990) , 'I receive the surrounding world through my eyes, my ears, my hands. The structure of my perceptual organs shapes that which I apprehend' (1). I will introduce a concept in which the heart as carnal material of modification and as delicate fabric from which love is made are inextricably intertwined with each other.\nConfidence: 0.8525002002716064\n\nParagraph: Nevertheless, I took my 16 mm vintage camera for a walk a few days later. I found Vienna's heart at the first district deeply wounded. Bullet holes in windows and walls, traces of blood on the ground, lit candles and flowers in memory of those killed.\nConfidence: 0.8477011322975159\n\nParagraph: Hygiene precautions which are usually applied in the surgical theatre are currently intertwined with the modes of how we personally engage with each other. Touch and droplet transferrals of pathogens easily occur through direct contact with contagious people. As a consequence, social life needs to be brought to a standstill in order to avoid the spread of the coronavirus. Breathing changes in such a hostile situation. All living beings are impacted, though the animal and vegetal world much less so. We stay at home-in isolation-and the earth exhales.\nConfidence: 0.8469974994659424\n\n---------------------------------------------------------------------------------------------\nDocument: More-than-human kinship against proximal loneliness: practising emergent multispecies care with a dog in a pandemic and beyond\nParagraph: Bella and I walked in the same direction, but metres apart. We had to speak louder than usual to make sure our voices could carry through the distance between us. As we strolled around the field, the dogs were busy unsuccessfully hunting a vole or a mouse with their snouts in olfactory hyperdrive under the long, knotted grass. A month must be a very long time in dog time, and they were determined to make up for lost time.\nConfidence: 0.843758761882782\n\nParagraph: I realised that neither Frank nor I have ever experienced this specific spatial arrangement beforewalking down the middle of the road without a single car in sightand my mind took me to a fictional zombie-apocalyptic place: the scene from the poster of I Am Legend, where Will Smith walks down the street with his loyal companion, Samantha the German shepherd. In this moment, Frank and I were Samantha and Will Smith, alone together in this viral world that pried meand by extension, Frankaway from other humans.\nConfidence: 0.8413867950439453\n\nParagraph: For example, in our COVID-infused environment and its sensibilities, the act of walking outdoors created different possibilities that Frank and I could actualise (de Certeau, 1984: 254) . So, in the face of these new possibilities, we co-developed new techniques of the body in order to actualise the possibility of walking on the road for social distancing, which evoked proximal loneliness in both of us, though manifesting in different ways. While I've come to feel physically and emotionally distant from, as well as anxious with, those around me, Frank's proximal loneliness has manifested as nerves around strangers. This has affected our care practices: while I felt cared for by Frank, who kept me company whenever we braved stepping out into the contagious airand potentially violently racistoutdoors, I cared for him by engaging in training activities designed to help him understand that there was no need to bark at everything and everyone passing us. Connected by the leash and the lure of Frank's favourite treats, our bodies thus ongoingly actualise and adapt to the different possibilities in this viral world, fulfilling our responsibilities to one another. This is just one example from myriad other emergent practices that we have come to embody as techniques of our bodies since the start of the lockdown. Indeed, some of our indoor habits have changed as well. With more time than ever to practise due to the sheer number of hours I spend at home now, Frank has learned how to adjust his body around mine in such a way that we can now both enjoy a daily evening cuddle without him poking me with his pointy elbows. He does this by carefully surveying the arrangement of my body on the couch, and placing his body in the nooks, usually behind my slightly bent knees stacked together to the right side of my body on the couch. He awkwardly circles around a few times before he settles down, lying on my leg in such a way that I can feel every little thump of his heartbeat on my skin. Then, I make the final adjustments in my limbs as necessary in order to provide the most comfortable experience for both of us, feeling the warmth on my legs where his body meets mine, as well as in my heart. The fact that we each put thought and care into being in comfortable proximity means something to me, and it evidently means enough for him to continue cultivating this habit day after day.\nConfidence: 0.8398276567459106\n\n---------------------------------------------------------------------------------------------\nDocument: Simplest Way to Establish COVID-19 Quarantine Observation Wards Within 24 Hours\nParagraph: Reducing nosocomial transmission within health care facilities is important, but the number of negative-pressure airborne infection isolation rooms for SARS-CoV-2 (severe acute respiratory syndrome coronavirus 2) is limited. It is a daunting challenge to cope with a surge of suspected infectious patients in hospitals. We installed air exhaust fans on the windows to change the pressure direction within the wards rapidly. The best location for the fans was 90 cm from the floor and 90 cm from the edge of bed whether the indoor air conditioners were on or off. The noise level should be <60 dB(A) as per government regulations. General wards can be transformed into makeshift negative-pressure rooms easily and effectively within 24 hours, which is really the simple, fast, and effective way for the transformation being applied.\nConfidence: 0.8541801571846008\n\nParagraph: With an average of 9,838 people per square kilometer at Kaohsiung city, the population density of Taiwan ranks second highest worldwide, with 80% of its population concentrated in six municipalities. Moreover, Taiwan is one of only five countries and regions in the world that had not yet announced full suspension of schools and classes or required citizens to stay at home by March 31, 2020. Consequently, when it comes to the reception and treatment of suspected cases with SARS-COV-2 (severe acute respiratory syndrome coronavirus 2; COVID-19), it is not feasible to build mobile cabin hospitals with makeshift isolation wards or walkthrough negative-pressure booths by expropriating hotels, schools, parks, or spacious buildings within cities, walkthrough, 4 or drive-through experience. 5, 6 How to designate a specific hospital to diagnose and preempt quarantine patients with suspected COVID-19 and acute respiratory symptoms as a COVID-19 hospital is an essential challenge. This COVID-19 safety hospital will also operate a screening center during the COVID-19 epidemic. At the screening centers, prompt screening and specimen collection are important, but the safety of medical staffs and testees is the top priority. 6 Cross-contamination should also It is suggested that we either make use of existing wards in hospitals, as their utilization rate has fallen due to the COVID-19 outbreak, or build outdoor parking lots or transform them into makeshift quarantine observation wards. This would be the only way to respond to the influx of suspected cases into hospitals and to contain further infection as patients have to stay in negative-pressure booths while waiting for test results.\nConfidence: 0.8463939428329468\n\nParagraph: Installed at a height of over 90 cm above the ground, the exhaust fan is unable to effectively expel the mist over a standard hospital bed and the mist scatters in a nondirectional way, which indicates that the Installation height of exhaust fans has no effect on the airflow from corridor into the ward. As the mist from the nursing station to corridor and into the ward keeps flowing in the correct direction, it means contaminated air in the ward is under effective control without outflow. 1.3.\nConfidence: 0.8397077322006226\n\n---------------------------------------------------------------------------------------------\nDocument: COVID-19 and veterinarians for one health, zoonotic- and reverse-zoonotic transmissions\nParagraph: A novel coronavirus emerged in human populations and spread rapidly to cause the global coronavirus disease 2019 pandemic. Although the origin of the associated virus (severe acute respiratory syndrome coronavirus 2 [SARS-CoV-2]) remains unclear, genetic evidence suggests that bats are a reservoir host of the virus, and pangolins are a probable intermediate. SARS-CoV-2 has crossed the species barrier to infect humans and other animal species, and infected humans can facilitate reverse-zoonotic transmission to animals. Considering the rapidly changing interconnections among people, animals, and ecosystems, traditional roles of veterinarians should evolve to include transdisciplinary roles.\nConfidence: 0.8595113754272461\n\nParagraph: Cats and dogs are often in close contact with humans, and thus, it is important to determine their susceptibility to SARS-CoV-2. COVID-19 has been reported in two dogs in Hong Kong. Dogs that live with COVID-19 patients have become infected and tested positive. One of the dogs developed specific antibodies against SARS-CoV-2 and seroconverted, indicating an active infection. Canine cases of COVID-19 were also reported in the Netherlands and the US (https://6abc. com/coronavirus-dogs-dog-update/6134707/). A family in North Carolina experienced mild COVID-19 symptoms, and their pug also showed mild symptoms with inappetence. All three family members tested SARS-CoV-2 positive, and the virus was detected in the dog. The family owned two dogs and a cat, but only one dog tested positive. In the Netherlands, four house pets tested COVID-19 positive. A COVID-19 patient owned a dog and three cats, and the dog was suffering severe breathing problems. This bulldog tested SARS-CoV-2 positive and was euthanized due to the illness. The three cats also developed specific antibodies for SARS-CoV-2. All four animals appeared to have contracted the virus from their COVID-19 owner.\nConfidence: 0.8543074727058411\n\nParagraph: In contrast to dogs, cats appear to be highly susceptible to the virus. In Belgium, a cat living with its COVID-19 owner became clinically ill, exhibiting respiratory problems accompanied by diarrhea and vomiting. The specific viral sequence of SARS-CoV-2 was detected in the feces and gastric vomitus of the cat, and that sequence was identical to that of the cat owner, indicating the occurrence of reverse-zoonotic transmission of SARS-CoV-2 from human to animal. In Wuhan, 102 cats were tested, and 15% of them were seroconverted, indicating SARS-CoV-2 exposure of cats from either people or other cats [3] . An additional case of COVID-19 in cats was reported in Hong Kong. This cat did not show any clinical signs, but oral, nasal, and fecal samples tested positive. In the US, SARS-CoV-2 positive cats have been reported in two separate areas in the state of New York [4] . A veterinarian tested one cat after it exhibited mild respiratory symptoms. The cat was SARS-CoV-2 positive, but no one in the household was positive. This cat may have contracted the coronavirus outside of the household from a person with COVID-19. In another case, a cat with respiratory symptoms and living with a COVID-19 patient tested positive. These data clearly show that cats are susceptible to SARS-CoV-2 and may contract COVID-19 (Fig. 1) . A controlled experiment was conducted to support the observations showing that cats are susceptible to COVID-19 [5] . The results of that experiment showed that cats are highly susceptible to SARS-CoV, with the viral sequence detected in (on) the nasal turbinates, soft palate, tonsils, and small intestine. The infections led to massive lesions in their nasal passages, trachea, and lungs, indicating efficient replication of the virus in cats. Interestingly, kittens and adolescent cats were especially vulnerable to COVID-19. Notably, the coronavirus spread from infected cats to uninfected cats via respiratory droplets. Feline cases of COVID-19 have additionally been reported in Spain, France, and Germany (https://www.oie.int/scientific-expertise/specificinformation-and-recommendations/questions-and-answers-on-2019novel-coronavirus/).\nConfidence: 0.8523463606834412\n\n---------------------------------------------------------------------------------------------\nDocument: Animals, Vaccines, and COVID-19\nParagraph: While there are currently over 200 vaccine candidates, they fall into four main categories (Rhodes, 2021) . The first two, a dead virus and an attenuated live virus, follow the models of influenza and polio. However, the most successful COVID-19 vaccines thus far have followed two newer technologies: messenger RNA (or mRNA) and viral vectors. While mRNA vaccines are new, they are based on research over the past twenty years. 1 With an mRNA vaccine, messenger RNA is injected into a muscle, where it instructs the muscle cells to make a piece of spike protein that resembles the spikes on the SARS-CoV-2 but is harmless. The cells then break down the messenger RNA. The harmless spike protein in turn provokes the immune system to make antibodies just as an actual viral protein would. This vaccine uses a different virus, modified to be non-infectious (often an adenovirus, the virus behind the common cold) as a vector to deliver a gene that instructs cells to make antigens that in turn trigger production of antibodies. The Johnson and Johnson and AstraZeneca vaccines employ viral vectors.\nConfidence: 0.8503252267837524\n\nParagraph: Animals, especially mammals, have played a critical role in the COVID-19 pandemic. The COVID-19 virus originated in animals, and the virus can jump back and forth between humans and animals. Moreover, animals have been central to the development of the various vaccines against the virus now employed around the world, continuing a long history. The interrelationships between animals and humans in both disease transmission and its prevention call for an interdisciplinary approach to medicine.\nConfidence: 0.8483691215515137\n\nParagraph: COVID-19 is part of a family of viruses known as coronaviruses from their spiky, crown-like outer layer, and seven coronaviruses have made the leap from animals to humans in the past two decades. The original pathway of communication of COVID-19 to humans is not yet fully understood, nor is the animal or animals that serve as a reservoir for it. Bats were early suspects, since they appear to have been the reservoir for other viruses, and recent studies appear to confirm that bats do constitute a reservoir for SARS-COV-2, the official term for the COVID-19 virus (Liu et al., 2021) . A recent report by the World Health Organization suggested that animals sold for food at the market in Wuhan, China may have contracted the disease from bats and that this intermediate animal, which is still unknown, transmitted the disease to humans (Maxmen, 2021) . Pangolins, rare scaly anteaters, were an early suspect for an intermediary. (In China, their scales are used in traditional medicine and their meat is eaten, even though they are protected by international law.) But this theory has since been discarded (Cyranoski, 2020) .\nConfidence: 0.8463747501373291\n\n---------------------------------------------------------------------------------------------\n\n=================================================================================================\n=================================================================================================\n\n\nQuestion 10: what are the initial symptoms of Covid-19?\n","output_type":"stream"},{"name":"stderr","text":"Getting inverted index...: 100%|██████████| 9995/9995 [00:00<00:00, 34573.73it/s]\n","output_type":"stream"},{"name":"stdout","text":"Document: Immune response in COVID-19: What do we currently know?\nParagraph: The acute respiratory syndrome is a disease caused by the SARS-CoV-2 virus (COVID-37 19), where symptoms include difficulty breathing, high fever, and cough (WHO, 2020). \nConfidence: 0.7719785571098328\n\nParagraph: In 2002/2003 there was a pandemic denominate SARS (severe acute respiratory syndrome), caused by the SARS-CoV virus that belongs to the genera Betacoranavirus and the family Coronaviridae, generally responsible for influenza infections. In mid of 2019, a new disease by the coronavirus named by COVID-19 (SARS-CoV-2) emerged, both infections have flu symptoms, however they are infections that variable intensity, being medium to severe. In medium infections individuals have the virus and exhibit symptoms, however hospitalization is not necessary, in severe infections, individuals are hospitalized, have high pathology and in some cases progress to death. The virus is formed by simple positive RNA, enveloped, non-segmented, and presenting the largest genome of viruses constituting 32 Kb, consisting of envelope proteins, membrane, nucleocapsid and spike protein, which is essential in the interaction with the host cells. As for the origin of this virus, research has been intensified to determine this paradox and although the similarity with SARS-CoV, this virus did not has necessarily the same place of origin. As for the immune system, it is currently unknown how this new virus interacts. In this brief review, we demonstrate important considerations about the responses to this infection.\nConfidence: 0.580736517906189\n\n---------------------------------------------------------------------------------------------\nDocument: COVID-19 Presenting as Acute Necrotizing Pancreatitis\nParagraph: The ongoing pandemic of the novel coronavirus disease 2019 (COVID-19) originated in Wuhan, China, in December 2019. Various manifestations of coronavirus (SARS-CoV-2 [severe acute respiratory syndrome coronavirus 2]) have been reported since the pandemic began. Some articles have reported acute pancreatitis in several patients due to COVID-19 infection. In this case report, we introduce a patient in whom SARS-CoV-2 caused necrotizing pancreatitis. Acute abdomen is one of the manifestations of COVID-19. Various causes such as acute pancreatitis and necrotizing pancreatitis can be in the list of differential diagnoses.\nConfidence: 0.7069493532180786\n\nParagraph: He was admitted to the intensive care unit with diagnosed severe acute necrotizing pancreatitis with acute respiratory distress syndrome as per the modified Atlanta criteria. The patient received supportive care (the nasogastric tube was fixed, resuscitation with intravenous crystalloid fluid, empiric antibiotic [vancomycin, cefepime], high-flow oxygen therapy, analgesia, and pantoprazole). Later nasal swab for SARS-CoV-2 (severe acute respiratory syndrome coronavirus 2) reverse transcriptase-polymerase chain reaction tested positive. Then oral treatment with oseltamivir, hydroxychloroquine, and ribavirin was started for the patient. The patient was discharged with improved health conditions after a 15-day hospitalization. No adverse reactions happened in this period.\nConfidence: 0.6948822736740112\n\nParagraph: The ongoing pandemic of the novel coronavirus disease 2019 (COVID-19) originated in Wuhan, China, in December 2019. So far, more than 4 million people worldwide have been infected with the virus. Many different manifestations of this virus have been reported. In this study, we report the case of a patient with COVID-19 with gastrointestinal manifestations and mild respiratory involvement. This study introduces the involvement of necrotizing pancreatitis in a patient with COVID-19. 1\nConfidence: 0.6454804539680481\n\n---------------------------------------------------------------------------------------------\nDocument: Novel coronavirus disease 2019 (COVID‐19) and neurodegenerative disorders\nParagraph: During the last few months, the whole humanity is experiencing largest and most severe sudden influx of COVID‐19 outbreak caused by the novel coronavirus (CoV) originated from Wuhan, China. According to the WHO reports, total 3 862 676 positive cases and 265 961 deaths have been recorded worldwide due to COVID‐19 infection as of May 9, 2020. CoVs are a large family of viruses (enveloped, single‐stranded RNA viruses), which includes severe acute respiratory syndrome coronavirus (SARS‐CoV) and Middle East respiratory syndrome‐related coronavirus (MERS‐CoV). New SARS CoV2 is the members of Betacoronavirus genus. These viruses cause infections in bats, camels and humans, and a few other associated species. Despite many neurologic complications associated with SARS‐CoV‐2 infection, it is still unclear whether these symptoms results from direct neural injury or due to some other reason. Currently, it appears that most of the neurological symptoms of COVID‐19 are nonspecific and secondary to the systemic illness. A single case of acute hemorrhagic necrotizing encephalopathy has been reported. SARS‐CoV‐2 associated Guillain‐Barré syndrome is an atypical case. Till today, no convincing evidence is available to confirm that the SARS‐CoV‐2 virus directly affects nerves system in humans. However, postinfection surveillance will be necessary to identify the possible post‐COVID‐19 neurologic syndromes.\nConfidence: 0.6372342109680176\n\nParagraph: During the last few months, the whole humanity is experiencing the largest and most severe sudden influx of COVID-19 outbreak caused by the novel coronavirus (CoV) originated from Wuhan city.\nConfidence: 0.6110575199127197\n\nParagraph: Coronavirus was officially announced as the causative pathogen of COVID-19 by the Chinese Center for Disease Control and Prevention on January 8, 2020. The World Health Organization announced coronavirus outbreak as a public health emergency of international concern. It started to expect disturbing extents over the globe as forewarned by the World Health Organization (WHO) reports associated Guillain-Barré syndrome is an atypical case. Till today, no convincing evidence is available to confirm that the SARS-CoV-2 virus directly affects the CNS or PNS in humans. However, postinfection surveillance will be necessary to identify the possible post-COVID-19 neurologic syndromes.\nConfidence: 0.541512668132782\n\n---------------------------------------------------------------------------------------------\nDocument: Coronavirus Infections in Companion Animals: Virology, Epidemiology, Clinical and Pathologic Features\nParagraph: A genetically distinct coronavirus called ferret systemic coronavirus infection (FRSCV) was subsequently identified, which causes a systemic, progressive, and fatal pyogranulomatous inflammatory disease resembling the dry form of feline infectious peritonitis (FIP) in cats. The average age at the time of FRSCV diagnosis has been reported to be 11 months, and clinical signs include chronic weight loss, anorexia, diarrhea, palpable abdominal masses, and neurologic disease [97] .\nConfidence: 0.7171095013618469\n\nParagraph: Coronaviruses are enveloped RNA viruses capable of causing respiratory, enteric, or systemic diseases in a variety of mammalian hosts that vary in clinical severity from subclinical to fatal. The host range and tissue tropism are largely determined by the coronaviral spike protein, which initiates cellular infection by promoting fusion of the viral and host cell membranes. Companion animal coronaviruses responsible for causing enteric infection include feline enteric coronavirus, ferret enteric coronavirus, canine enteric coronavirus, equine coronavirus, and alpaca enteric coronavirus, while canine respiratory coronavirus and alpaca respiratory coronavirus result in respiratory infection. Ferret systemic coronavirus and feline infectious peritonitis virus, a mutated feline enteric coronavirus, can lead to lethal immuno-inflammatory systemic disease. Recent human viral pandemics, including severe acute respiratory syndrome (SARS), Middle East respiratory syndrome (MERS), and most recently, COVID-19, all thought to originate from bat coronaviruses, demonstrate the zoonotic potential of coronaviruses and their potential to have devastating impacts. A better understanding of the coronaviruses of companion animals, their capacity for cross-species transmission, and the sharing of genetic information may facilitate improved prevention and control strategies for future emerging zoonotic coronaviruses. This article reviews the clinical, epidemiologic, virologic, and pathologic characteristics of nine important coronaviruses of companion animals.\nConfidence: 0.682610273361206\n\nParagraph: A strong epidemiologic association has been made between alpaca respiratory coronavirus and an outbreak of alpaca respiratory syndrome (ARS) in alpacas in 2007. ARS is characterized by acute respiratory signs ranging in severity from mild upper respiratory disease to severe respiratory distress, high fever, and death [144] . Though all signalments can be affected, ARS is primarily reported in pregnant alpacas; severe fetal hypoxia in alpacas with ARS can result in abortion [145] .\nConfidence: 0.6749551892280579\n\n---------------------------------------------------------------------------------------------\nDocument: A Case Report of Varicella Zoster Meningitis as Co-Infection With Breakthrough COVID-19 in an Immunocompetent Patient\nParagraph: Emerging evidence suggests that varicella zoster virus (VZV) infection can occur after coronavirus disease 2019 (COVID-19), which is caused by severe acute respiratory syndrome coronavirus-2 (SARS-CoV-2). 1,2 VZV co-infection with SARS-CoV-2 3,4 or after BNT162b2 mRNA SARS-CoV-2 vaccination has been reported recently. 5-7 Though the mechanism of VZV co-infection in COVID-19 is not fully understood, decreased cellular immunity and inflammation after SARS-CoV-2 infection is suggested as a possible cause of VZV infection. 8, 9 Since VZV infection is dependent on cell-mediated immunity, COVID-19 may promote the VZV infection. 1,8,9 Because VZV can cause severe neurologic complications such as stroke, myelitis, and neuropathy, 10 concomitant VZV infection should be noted in COVID-19 patients.\nConfidence: 0.6862372756004333\n\nParagraph: There are several previous reports that infection or reactivation of varicella zoster virus (VZV) can occur after coronavirus disease 2019 (COVID-19), which is caused by severe acute respiratory syndrome coronavirus-2 (SARS-CoV-2). Herein, we report a rare case of VZV meningitis in breakthrough COVID-19. An 18-years-old male visited the emergency room, presenting with a headache and fever of up to 38.4°C for 5 days. He received the second dose of BNT162b2 mRNA SARS-CoV-2 vaccine 7 weeks prior to symptom onset. The symptoms persisted with headache, fever, and nausea. His cerebrospinal fluid (CSF) showed an elevated opening pressure of 27 cm H(2)O, 6/µL red blood cells, 234/µL white blood cells (polymorphonuclear leukocytes 3%, lymphocytes 83%, and other 14%), 43.9 mg/dL protein, and 59 mg/dL glucose, and CSF polymerase chain reaction (PCR) test was positive for VZV. Also, he was diagnosed with COVID-19 by reverse transcriptase-PCR examining upper and lower respiratory tract. We administered intravenous acyclovir for 12 days, and he was discharged without any neurologic complication.\nConfidence: 0.6448171734809875\n\nParagraph: We reported a rare case of VZV meningitis in breakthrough COVID-19. Because VZV infection can occur even in breakthrough COVID-19, we should be aware of the concomitant VZV infection in COVID-19 patients. \nConfidence: 0.6108859777450562\n\n---------------------------------------------------------------------------------------------\n\n=================================================================================================\n=================================================================================================\n\n\n","output_type":"stream"}]},{"cell_type":"markdown","source":"#### 5.1 Calculate the MRR of the result","metadata":{}},{"cell_type":"markdown","source":"After obtaining the test results, we will manually identify the correctness of each return snippets and calculate the reciprocal rank for each query.","metadata":{}},{"cell_type":"code","source":"reciprocal_rank = np.zeros(len(test_set))\nreciprocal_rank[0] = 1/2 # the second document is a relevant document\nreciprocal_rank[1] = 1/5 # the fifth document is a relevant document\nreciprocal_rank[2] = 0 # there is no relevant document in the top 5 documents\nreciprocal_rank[3] = 1 # the first document is a relevant document\nreciprocal_rank[4] = 0\nreciprocal_rank[5] = 0\nreciprocal_rank[6] = 1/4\nreciprocal_rank[7] = 0\nreciprocal_rank[8] = 0\nreciprocal_rank[9] = 1\n\nprint(\"The mean reciprocal rank of the test set is: \" +\n      str(np.mean(reciprocal_rank)))\n","metadata":{"execution":{"iopub.status.busy":"2023-04-05T06:50:22.375725Z","iopub.execute_input":"2023-04-05T06:50:22.376154Z","iopub.status.idle":"2023-04-05T06:50:22.385778Z","shell.execute_reply.started":"2023-04-05T06:50:22.376117Z","shell.execute_reply":"2023-04-05T06:50:22.384200Z"},"trusted":true},"execution_count":36,"outputs":[{"name":"stdout","text":"The mean reciprocal rank of the test set is: 0.29500000000000004\n","output_type":"stream"}]},{"cell_type":"markdown","source":"#### 5.2 Result analysis","metadata":{}},{"cell_type":"markdown","source":"The result shows that only half of the queries are accurately answered, which are: 1. \"what is the origin of COVID-19?\"; 2. \"what types of rapid testing for Covid-19 have been developed?\"; 3. \"what are the transmission routes of coronavirus?\"; 4. \"What vaccine candidates are being tested for Covid-19?\"; 5. \"what are the initial symptoms of Covid-19?\". The success of these queries can be attributed to the presence of specific keywords in the knowledge base. Conversely, queries that were answered incorrectly lacked keywords that were not present in the knowledge base or dataset. Therefore, it can be concluded that a well-constructed knowledge base is critical for precise answer retrieval, which can be enhanced by incorporating additional keywords and synonyms or by utilizing relation extraction to establish the connections between entities. Despite these limitations, the outcomes highlight the potential of utilizing named entity recognition and knowledge bases for question-answering systems.","metadata":{}},{"cell_type":"markdown","source":"### 6. Simple user interface","metadata":{}},{"cell_type":"code","source":"convert_json_list = []\ndef convert_to_json(row):\n    num = 0\n    if(row[\"cord_uid\"] == \"rmvw3ko0\"):\n        print(row)\n    for i in row['other_paragraph']:\n         \n        if(pd.isnull(i) == False):\n            convert_json_list.append(\n            {\n                \"id\": row[\"cord_uid\"]+\"-\"+ str(num),\n                \"contents\": i\n            })\n            num = num+1;\n        ","metadata":{"execution":{"iopub.status.busy":"2023-04-05T07:21:03.718085Z","iopub.execute_input":"2023-04-05T07:21:03.718562Z","iopub.status.idle":"2023-04-05T07:21:03.727061Z","shell.execute_reply.started":"2023-04-05T07:21:03.718502Z","shell.execute_reply":"2023-04-05T07:21:03.725742Z"},"trusted":true},"execution_count":93,"outputs":[]},{"cell_type":"code","source":"!rm -r collection.json\nx = df_sampled.progress_apply(\n    lambda row: convert_to_json(row), axis=1)\n\njson_str = json.dumps(convert_json_list)\n\nwith open(\"collection.json\", \"w\") as outfile:\n    outfile.write(json_str)\nprint(json_str[0:500])\nprint(convert_json_list[0:100])","metadata":{"_kg_hide-output":true,"execution":{"iopub.status.busy":"2023-04-05T07:21:05.471046Z","iopub.execute_input":"2023-04-05T07:21:05.471484Z","iopub.status.idle":"2023-04-05T07:21:12.434010Z","shell.execute_reply.started":"2023-04-05T07:21:05.471443Z","shell.execute_reply":"2023-04-05T07:21:12.432156Z"},"trusted":true},"execution_count":94,"outputs":[{"name":"stdout","text":"rm: cannot remove 'collection.json': No such file or directory\n","output_type":"stream"},{"name":"stderr","text":"Getting inverted index...:   0%|          | 0/10000 [00:00<?, ?it/s]","output_type":"stream"},{"name":"stdout","text":"abstract             As the COVID-19 pandemic sweeps the globe, man...\nconclusion                                                         NaN\ncord_uid                                                      rmvw3ko0\nintroduction                                                       NaN\nother_paragraph      [Since March 2020, the SARS-CoV-2 (COVID-19) p...\npdf_json_files       document_parses/pdf_json/761ee085a77ce60863fe9...\npublish_time                                                2022-04-11\ntitle                The Impact of COVID-19 on Self-Reported Substa...\npreprocessed_text    impact covid-19 self-reported substance use, w...\nname_entity          [Functioning Among, United States, Veterans, C...\nembedding            [-0.08254339, 0.176392, 0.11337998, 0.13646367...\nName: 913530, dtype: object\n","output_type":"stream"},{"name":"stderr","text":"Getting inverted index...: 100%|██████████| 10000/10000 [00:02<00:00, 3598.66it/s]\n","output_type":"stream"},{"name":"stdout","text":"[{\"id\": \"0ex7dy6g-0\", \"contents\": \"A severe acute respiratory syndrome, coronavirus 2 (SARS-CoV-2), later known as COVID-19, was identified in the Hubei Province of China towards the end of 2019. By 30 April 2020, the virus had spread to more than 200 different countries causing 217, 769 deaths worldwide, with the per-capita mortality and infection rates varying greatly between nations (World Health Organization, 2020). Part of the variance in infection and mortality rates can be attributed the \n[{'id': '0ex7dy6g-0', 'contents': 'A severe acute respiratory syndrome, coronavirus 2 (SARS-CoV-2), later known as COVID-19, was identified in the Hubei Province of China towards the end of 2019. By 30 April 2020, the virus had spread to more than 200 different countries causing 217, 769 deaths worldwide, with the per-capita mortality and infection rates varying greatly between nations (World Health Organization, 2020). Part of the variance in infection and mortality rates can be attributed the stringency, timing and effectiveness of the varying infection prevention and control (IPC) measures implemented by different governments, aiming to attenuate the spread of the pandemic.'}, {'id': '0ex7dy6g-1', 'contents': 'Typically, national/governmental IPC initiatives involved lockdowns, social distancing and quarantines. Such measures are themselves not without health implications. Several reviews and commentaries have warned Correspondence should be addressed to Mariapaola Barbato, Department of Psychology, College of Natural and Health Sciences, Zayed University, Academic City, P.O. Box 19282, Dubai, United Arab Emirates. (E-mail: mariapaola.barbato@zu.ac.ae) .'}, {'id': '0ex7dy6g-2', 'contents': 'MB designed the study, collected, and analysed the data, and wrote the first draft of the manuscript. JT contributed to data collection, interpretation of results, and manuscript preparation.'}, {'id': '0ex7dy6g-3', 'contents': 'All procedures involved in this study were in accordance with the ethical standards of the institutional research committee and with the 1964 Helsinki Declaration and its later amendments or comparable ethical standards. Informed consent was obtained from all individual adult participants included in the study. mental health professionals on the likely adverse psychological impact of interventions such as quarantine and isolation, based on findings of past pandemics (e.g. Brooks et al., 2020) . For instance, during the SARS outbreak in 2003, a study conducted in Hong Kong reported a 30% increase in completed suicides among females aged 65 years and over (Chan et al., 2006) . Furthermore, a North American study reported that, among individuals quarantining in response to SARS ourbreak, 29% presented symptoms of post-traumatic stress disorder (PTSD) and 31% presented symptoms of depression (Hawryluck et al., 2004) ; longer quarantine was associated with more severe PTSD symptoms (Reynolds et al., 2008) . Early reports from the COVID-19 pandemic have also highlighted a higher risk of experiencing clinically significant levels of psychopathology among individuals required to quarantine or isolate (e.g. Huang & Zhao, 2020; Lai et al., 2020) . Similarly, an increasing number of studies have suggested elevated rates of depression, anxiety, and insomnia in the general population as a consequence of the Covid-19 pandemic and the associated IPC measures (e.g. Huang & Zhao, 2020; Liu et al., 2020; Tian et al., 2020; Wang et al., 2020) .'}, {'id': '0ex7dy6g-4', 'contents': 'In this context, an important public mental health goal is identifying groups at elevated risk of mental health problems associated with the pandemic. Several recent studies report that women, younger individuals and people with pre-existing medical conditions or mental health problems, report increased levels of pandemic-related psychopathology (Gentile et al., 2020; Liu et al., 2020; Mazza et al., 2020; Tian et al., 2020; Wang et al., 2020) . Additionally, Liem et al. suggested that international migrant/foreign workers could be another potentially vulnerable group, due to threatened job security, access to healthcare, sub-optimal social support and potential language barriers (Liem et al., 2020) . However, to the best of our knowledge, no study has yet specifically assessed the mental health status of foreign worker populations during the COVID-19 pandemic.'}, {'id': '0ex7dy6g-5', 'contents': 'Foreign workers and their families are an increasingly sizable demographic in our globalised world. As of 2014, there were estimated to be 50.5 million such employees and their families worldwide, with the figure projected to reach 56.8 million by 2017 (Finaccord, 2014) . Globalised markets have increased the opportunity for overseas employment and also the number of foreign workers, sometimes known as guest workers or expatriates. This is typically an internationally diverse workforce spanning all employment sectors. In general, the literature on foreign workers has tended to focus on the mental health of those individuals working in low-income, low-skill occupations, for example, construction labourers (e.g. Liem et al., 2020) . There are far fewer studies conducted on foreign workers engaged in high skill, managerial, professional occupations (e.g. Bayraktar, 2019) .'}, {'id': '0ex7dy6g-6', 'contents': \"The Covid-19 pandemic and the IPC measures are likely to present some fairly unique challenges for foreign workers regardless of occupational status. For instance, the pandemic's impact on the worker's country of origin would be viewed from the host nation. This perspective might exacerbate anxieties about the wellbeing of friends and family back home. There may also be extended travel bans, creating additional obstacles for individuals who need to return home urgently on compassionate grounds. Furthermore, job insecurity and indebtedness can, in some contexts, be complicated by foreign worker status, as can access to healthcare in the host nation.\"}, {'id': '0ex7dy6g-7', 'contents': 'The United Arab Emirates (UAE) has one of the world\\'s highest foreign worker to citizen ratio with foreign workers and their accompanying families comprising around 80% of the population (National Bureau of Statistics, 2009). Compared to many other countries, the impact of COVID-19 in the UAE has been moderate, with 12,481 reported cases, 105 deaths, and 2429 recoveries (as of 30 April). Since the beginning of March 2020, the UAE government carried out an emergency plan to contain the pandemic. The closure of educational facilities (8 March) was followed by an extensive \"work from home\" campaign (14 March), and travel bans with the temporary suspension of tourist visas (26 March). Severe fines were enforced for violations of respiratory hygiene (masks) and social distancing measures.'}, {'id': '0ex7dy6g-8', 'contents': 'The present study aimed to assess the psychological correlates of the COVID-19 pandemic on a sample of UAE-based foreign workers. In line with findings from the current as well as previous pandemics, we expected to observe (a) high rates of post-traumatic stress, depressed mood, anxiety, and insomnia (e.g. Chua et al., 2004; Forte et al., 2020; Hawryluck et al., 2004; Reynolds et al., 2008) ; and (b) more severe symptoms of post-traumatic stress, depressed mood, anxiety and insomnia among individuals with a pre-existent medical or psychological condition (Gentile et al., 2020; Mazza et al., 2020) . Additionally, we hypothesised that an increased risk of psychopathology would be observed among foreign workers with family and friends living in countries that had experienced relatively high infection and mortality rates by April 2020 (e.g. Italy). To the best of our knowledge, this is one of the first studies to explore the psychological implications of the Covid-19 pandemic among foreign workers and their families. This work has potential implications for the identification of vulnerable groups and may also inform future public mental health and pandemic preparedness plans.'}, {'id': '0ex7dy6g-9', 'contents': \"Participants were adult residents of the UAE, comprised of foreign workers and their families, from various nations (n = 319). An invitation to participate was sent through social media platforms and websites, including What-sApp, Facebook, Twitter and LinkedIn. Sample optimization was achieved through snowballing. The mean age for participants was 40.6 (SD = 9.0). For a full breakdown of participant's demographics see Table 1 . Participants were asked to complete an online survey in English.\"}, {'id': '0ex7dy6g-10', 'contents': \"The study was approved by the local Research Ethics Committee (ZU20_077_F; MOHAP/DXB-REC/MMM/No. 48/2020). The online survey included socio-demographic questions and questions related to participants' experience with the COVID-19 pandemic. Questions related to COVID-19 included whether or not participants had been diagnosed with COVID-19 in the past 2 weeks, if they were affected by medical conditions, and if they had any family members or close friends currently living in countries that, at the time of data collection, were severely affected by the pandemic (e.g. China, Italy, United States, Iran, or Spain). A further question was included to assess the perceived impact (mild, moderate, severe) of COVID-19 in their home country. Following these items, participants were asked to complete a set of standardised psychometrics: Event Scale-Revised, the Patient Health Questionnaire-8; the Generalised Anxiety Disorder-7; and the Insomnia Severity Index-7. Data were collected between 6 April and 17 April 2020.\"}, {'id': '0ex7dy6g-11', 'contents': 'The Impact of Event Scale-Revised, or IES-R (Weiss & Marmar, 1997) , is a 22-item self-report that allows for the subjective assessment of responses to traumatic events in the general population. This scale has good psychometric properties and has been extensively used to assess the psychological effects of pandemics (e.g. Lai et al., 2020; Reynolds et al., 2008) . Participants were asked to rate each item in relation to their experience with COVID-19, using a 5-point scale ranging from 0 (\"not at all\") to 4 (\"extremely\"). The IES-R yields a total score (i.e. the sum of scores in the 22 items) ranging from 0 to 88, and three sub-scores relative to the sub-scales of Intrusion (the presence of repeated thoughts about the traumatic event), Avoidance (how much the individual actively tries to avoid situations that remind of the traumatic event) and Hyperarousal (the physiological component of the symptoms). A proposed cutoff score of 33 is indicative of a probable diagnosis of PTSD (Creamer et al., 2003) . This cutoff was used in the present study. The internal reliability of this scale among the current sample was excellent (α = .93).'}, {'id': '0ex7dy6g-12', 'contents': \"The Patient Health Questionnaire-8 (PHQ-8; Kroenke et al., 2009 ) is a 8-item instrument assessing the frequency, in the past 2 weeks, of depressive symptoms in the general population. Participants' responses can range from 0 to 3 with 0 = not at all, 1 = several days, 2 = more than half the days, 3 = nearly every day. The total score, ranging from 0 to 24, is obtained by summing the responses to each item. A score of less than 5 is considered to indicate an absence of depressive symptoms; scores between 5 and 9, between 10 and 14, between 15 and 19, and between 20 and 24 are thought to indicate mild, moderate, moderately severe, and severe depressive symptoms, respectively. In the present study a cutoff score ≥10 was used to identify clinically relevant (moderate) depressive symptoms, following recommendations from previous studies (Kroenke et al., 2009) . Notably, the PHQ-8 is a shortened version of the PHQ-9 in which the last item, relative to suicidal ideation, has not been included (Kroenke et al., 2009 ). In agreement with UAE cultural norms, the use of the PHQ-8 was deemed preferable in the present study. The reliability of the PHQ-8 among the current sample was excellent (α = .91).\"}, {'id': '0ex7dy6g-13', 'contents': \"The Generalised Anxiety Disorder-7 (GAD-7; Spitzer et al., 2006) is a self-report scale that allows to measure anxiety in the general population. Participants are asked 4 BARBATO AND THOMAS to indicate how often in the past 2 weeks, they have experienced each one of the seven main symptoms of generalised anxiety disorder. The total score can range from 0 to 21 and is calculated by assigning scores of 0 (not at all), 1 (several days), 2 (more than half the days) and 3 (nearly every day), to participants' responses in each item. Total scores of 5, 10 and 15 are considered cut-off points for mild, moderate and severe anxiety, respectively. The psychometric properties and normative data of the scale have been reported by Lowe et al., who suggested a cutoff score of 10 yielding a prevalence of anxiety disorders of 5%; the authors further suggest a high generalizability of their results to other populations (Löwe et al., 2008) . The suggested cutoff of 10 was used in the present study. The reliability of this scale among the current sample was excellent (α = .93).\"}, {'id': '0ex7dy6g-14', 'contents': \"The Insomnia Severity Index-7 (ISI-7; Bastien et al., 2001 ) is a 7-item self-report questionnaire assessing the nature, severity, and impact of insomnia. The first three questions assess the current (i.e. last 2 weeks) presence or frequency of sleep problems. The remaining four questions assess individuals' satisfaction, worry, and interference with everyday life, relative to their current sleep pattern. A 5-point Likert scale is used to rate each item (e.g. 0 = no problem; 4 = very severe problem), with a total score ranging from 0 to 28. The total score is interpreted as absence of insomnia (0-7), sub-threshold insomnia (8-14), moderate insomnia (15-21) and severe insomnia (22-28). A study conducted with a community sample has identified the cutoff score of 10 to have 86.1% sensitivity and 87.7% specificity (Morin et al., 2011) . This cutoff was used in the present study. The reliability of this scale among the current sample was excellent (α = .92).\"}, {'id': '0ex7dy6g-15', 'contents': \"For each outcome measure, descriptive statistics were computed. Using the recommended cutoff scores for each measure, the prevalence of probable PTSD, depression, anxiety and insomnia was reported. Correlation analysis was used to explore the relationship of each measure with the others and with age. The Shapiro-Wilk test indicated that the outcome variables were not normally distributed and attempts to transform the data to symmetry were unsuccessful. Therefore, non-parametric tests, including Mann-Whitney U test and Kruskal-Wallis H test, were used for between-group comparisons. To explore the effect of potential vulnerability factors in predicting the prevalence of probable PTSD, depression, anxiety and insomnia, a set of logistic regressions were conducted. Potential vulnerability factors included gender, age, education, pre-existent medical condition, previous diagnosis of psychological disorders, having family or friends living in a country severely affected by COVID-19, and the perceived severity of COVID-19 in the foreign worker's home country.\"}, {'id': '0ex7dy6g-16', 'contents': 'The sample included 319 individuals (mean age = 40.6; SD = 9.0). Participants were mostly female, holding graduate or post-graduate degrees, with no past history of psychological disorders, mostly Christian and living in the Emirate of Dubai. None of the study participants had received a diagnosis of COVID-19. The majority (98%) had not quarantined in the previous 2 weeks, while six participants (2%) reported having quarantined due to contact with an infected person. In total, 79% of participants reported having a family member or close friends living in countries severely affected by COVID-19. European nationalities were most prevalent (69%). About 11% of the sample reported being affected by a pre-existing medical condition likely to increase the risk of COVID-19 related complications (e.g. asthma, cardiovascular diseases, compromised immune system). About 12% of the sample reported a current or past history of psychological disorders (see Table 1 ).'}, {'id': '0ex7dy6g-17', 'contents': 'When asked to describe the impact of COVID-19 in their home country, 65.2% of participants described their home country as \"severely affected,\" 27.3% described it as \"moderately affected,\" 5% as \"mildly affected\" and the remaining 2.5% indicated not being able to estimate or not being well-informed.'}, {'id': '0ex7dy6g-18', 'contents': 'Descriptive statistics are reported in Table 2 . Data collected using the IES-R highlighted a probable PTSD in 25% of participants. Similarly, data collected using the PHQ-8 indicated that 23% of participants experienced moderate to severe depressive symptoms (4% severe). Moderate to severe anxiety symptoms (12% severe) were observed in the 24% of participants, as highlighted by the GAD-7. Insomnia symptoms were observed in 33% of participants, with 4% of participants showing severe symptoms, as highlighted by the ISI-7.'}, {'id': '0ex7dy6g-19', 'contents': 'Significant and positive correlations were observed among outcome measures; more severe symptoms were observed among younger individuals (all p < .001; see Table 3 ).'}, {'id': '0ex7dy6g-20', 'contents': 'A significant effect of gender was also observed across the four outcome measures (IES-R: U = 5075.00, GAD-7 = Generalized Anxiety Disorder-7; IES-R = Impact of Event Scale-Revised; ISI-7 = Insomnia Severity Scale-7; PHQ-8 = Patient Health Questionnaire-8. Results of the logistic regressions are presented in Table 4 . A higher reported prevalence of depression, anxiety and insomnia was observed among younger participants. Female gender was associated with higher prevalence of probable PTSD and insomnia. A previous diagnosis of a psychological disorder was associated with a higher reported prevalence of probable PTSD, depression and anxiety but not insomnia. A more severe (perceived) impact of COVID-19 in the home country predicted elevated scores for depression, anxiety and insomnia. Education, a pre-existent medical condition, and having family or friends living in a country severely affected by COVID-19 did not significantly predict psychopathology.'}, {'id': '0ex7dy6g-21', 'contents': 'The present study aimed at assessing the psychological impact of the COVID-19 pandemic in a sample of foreign workers resident in the UAE. The study was conducted during the early stages of the national response to the pandemic, which included curfews, lockdowns, travel bans and quarantines among other measures. Although pre-pandemic data were not available for our specific sample, an examination of data available in the literature seems to suggest that the rates of psychopathology observed among the foreign workers in the present study were relatively high compared to normative samples (Ashbaugh et al., 2016; Gagnon et al., 2013; Kroenke et al., 2009; Löwe et al., 2008) , and were similar to the elevated levels of psychopathology reported during the current or previous pandemics (e.g. Casagrande et al., 2020; Chua et al., 2004; Forte et al., 2020; Hawryluck et al., 2004; Reynolds et al., 2008) . While elevated levels of depression and anxiety might be expected during times of national crisis, the longer-term effect of this remains unknown. There is also a need to identify which groups, among foreign workers and their families appear to be most vulnerable to heightened levels of psychopathology during the pandemic.'}, {'id': '0ex7dy6g-22', 'contents': 'In the present study more severe symptoms of post-traumatic stress, depression, anxiety and insomnia were observed among females and younger individuals. This is in line with recent studies also reporting such age and gender effects Mazza et al., 2020; Wang et al., 2020) . Surprisingly, however, the presence of a pre-existent medical condition was not significantly associated with elevated symptom severity for post-traumatic stress, depression, anxiety or insomnia. This result contradicted our expectations and was discordant with the findings of Mazza et al. (2020) , who reported elevated anxious and depressive symptomatology among individuals with a history of medical problems. Notably, the study of Mazza et al. was undertaken in Italy at a time when Italy had one of the highest infection and mortality rates in the world; this may explain the difference between their results and ours. Furthermore, pre-existing medical conditions in our sample were reported by a minority of participants (11%), which may not provide enough data to generate adequate statistical power. It is also worth noting that the residential status of foreign workers can be highly dependent on their overall health (i.e. being fit for work). For example, in the UAE, residency visa allocation and renewal are contingent upon passing mandatory health checks. This may explain the relative dearth of pre-existing health conditions in the present sample. These links between health and residency also highlight an obvious additional stressor for foreign workers during pandemics.'}, {'id': '0ex7dy6g-23', 'contents': \"Physical health aside, the results of the logistic regression highlighted a higher reported prevalence of probable PTSD, depression, and anxiety among individuals with a current or previous diagnosis of psychological disorders. These results were both expected and in line with the literature (Cukor et al., 2011; Gentile et al., 2020) , and confirmed the idea of a higher susceptibility for this group during the pandemic. One major concern is that this elevated symptomatology could lead to the relapse for some individuals. This group would almost certainly benefit form initatives aimed at secondary prevention. Also, as with physical health, mental health conditions (psychiatric diagnoses) could potentially impact the residency status of some foreign workers, further exacerbating concerns about job security and the implications for accompanying family members. Interestingly, the perceived impact of the pandemic on participants' home country predicted the severity of depression, anxiety, and insomnia, with greater impact being associated with more severe symptomatology. Although no previous literature is available on the psychological impact of pandemics among foreign workers, these results are generally in line with our predictions. The physical, psychological and financial threats that come with COVID-19 might have a greater impact on individuals who, worried about their status in the host country, cannot envision an early repatriation as a viable option. This state of uncertainty, combined with witnessing the tragic events in their home country from afar, could exacerbate concerns and give rise to psychopathology. It is worth noting, however, that our study did not include an objective measure of the pandemic's impact in participants' home countries, therefore these results must be interpreted with caution.\"}, {'id': '0ex7dy6g-24', 'contents': 'Contrary to our predictions, we did not observe a higher reported prevalence of probable PTSD, depression, anxiety or insomnia among those who reported having family or close friends living in countries severely affected by COVID-19. Although this result was unexpected, an interpretation can be attempted. On one hand, increased worry could be expected among individuals whose family and friends live in a country with higher infection and mortality rates. Conversely, it seems possible that greater concern could have contributed to bringing people together to (virtually) share the lockdown experience with family and friends, possibly leading to individuals experiencing the benefits of greater social support. While this interpretation remains speculative, it would be partially in line with recent findings reporting an inverse association between social support and depressive symptoms during the COVID-19 pandemic (Grey et al., 2020) . Notably, previous studies conducted on foreign workers, outside of the COVID-19 context, highlight the psychological benefits of receiving social support from relatives and friends in the home country (Bayraktar, 2019). All in all, the impact of social support on the psychological wellbeing of foreign workers during pandemics seems worthy of further investigation.'}, {'id': '0ex7dy6g-25', 'contents': 'The strengths of our study include the focus on a multicultural and international sample and the specific attention given to foreign workers, a group often neglected in the psychological research literature. Aside from its strengths, our study also has several important limitations. Firstly, the study is cross-sectional and correlational limiting our ability to infer temporality or causality. Given the absence of pre-pandemic data for our specific sample, our results were compared with data available in the literature. Such a comparison is obviously sub-optimal, methodologically speaking, due to the differing sample demography and data collection procedures. These comparisons, therefore, should be interpreted with upmost caution. Moreover, our data were collected from a convenience sample; hence, it is not representative of the broader population of foreign workers living in the UAE. Notably, our sample mainly included highly educated individuals, mostly European, while \"blue-collar\" manual workers, were notably absent in our sample. This absence may be due to this group having less access to the internet, less leisure time, and being less likely to be literate in English (the language of the current survey). Further research specifically focusing on the psychological wellbeing of the lower-skilled manual labour workforce during pandemics is recommended. Another potential limitation in the present study is the use of an online questionnaire, which, although necessitated by social distancing requirements, could have affected the characteristics of our sample (self-selection bias). It is also worth noting that none of the measures used in the present study had been previously validated among samples of foreign workers. This important point should be addressed by future research.'}, {'id': '0ex7dy6g-26', 'contents': \"Taken together, our results from an early phase of the COVID-19 pandemic confirm that high rates of probable PTSD, depression, anxiety and insomnia can be expected among foreign workers and their families. Women, younger individuals and individuals affected by a pre-existing psychological disorder appear to be most vulnerable. Separation from family and friends who live in countries with higher infection and mortality rates does not seem to impact psychological wellbeing; nevertheless, the mitigating role of additional factors (e.g. social support) that were not investigated in the present study cannot be excluded. The perceived impact of COVID-19 in the participants' home country was associated with an elevated risk of depression, anxiety and insomnia. Overall, due to the lack of previous literature addressing the psychological wellbeing of foreign workers during pandemics, our results must be considered only preliminary and the need for further research must be reiterated.\"}, {'id': '0ex7dy6g-27', 'contents': \"The integration of psychological interventions during the COVID-19 outbreak can prove useful. The UAE is currently offering, through the National Program for Happiness and Wellbeing (NPHW), online support for individuals feeling overwhelmed by the COVID-19 pandemic. The present study could contribute to the development of targeted interventions to guarantee an easy access to a culturally appropriate psychological support 8 BARBATO AND THOMAS for vulnerable populations. Furthermore, foreign workers' dependence on information technology might mediate social connection and support, therefore the strengthening of IT infrastructure at a national level would be an essential part of future pandemic preparedness plans. Similarly, current social distancing measures in the UAE limit social gatherings to a small number of first-degree family members, which might not take into account the psychological needs of foreign workers, many of which rely solely on friends and compatriots for social support. While our results provide a picture of early reactions to the pandemic among foreign workers living in the UAE, future studies could examine late reactions to the pandemic as well as the efficacy of psychological interventions in reducing levels of pandemic-related psychopathology.\"}, {'id': 'rmvw3ko0-0', 'contents': 'Since March 2020, the SARS-CoV-2 (COVID-19) pandemic has been an intense stressor across the United States. In the first year of the pandemic, sudden and dramatic behavioral changes were either mandated or encouraged to \"flatten the curve, \" which included asking people to stay at home as much as possible, shutting down entire industries (e.g., restaurants and movie theaters), and shifting clinical resources to COVID-19 specific care (Emerick et al., 2020) . While these shifts were intended to slow infection rates, there was also a negative peripheral impact for many citizens due to sudden job loss (Matthews et al., 2021) , reduced social connectedness (Talcott et al., 2021) , and loss of certain health treatments (Hochstatter et al., 2021) . Across the globe, there were also reports of growing psychological distress such as heightened suicidal ideation (Brailovskaia et al., 2021) , increased negative affective states such as nervousness and fear (Zhang et al., 2020) , an increased sense of worry and loneliness (Cohn-Schwartz et al., 2021) , and decreased coping resources (Mana et al., 2021) . Pandemic stressors and negative psychological impact may have led to societal challenges exceeding individual coping resources, likely also contributing to reports of allostatic overload (Cosci and Guidi, 2021) , or the cumulative impact of an individual\\'s accumulated burden of stress on the body and mind (Danese and McEwen, 2012) . Since allostatic overload can have downstream negative health consequences, it is unsurprising that at the beginning of the pandemic many citizens, with and without reported COVID-19 infections, reported both decreased mental health (Czeisler et al., 2020) and physical functioning (Morlock et al., 2021) .'}, {'id': 'rmvw3ko0-1', 'contents': \"At the same time, the first few months of the pandemic saw increases in alcohol sales (The Nielsen Company, 2020) and self-reported use of some drugs (Janulis et al., 2021) . In accordance with the motivational model of substance use (Cooper et al., 1995) , there was concern that heightened pandemic-related psychological distress, combined with a lack of internal and external resources for stress and mood management, was leading to increased substance misuse by individuals seeking both positive affective states and coping with negative emotional experiences (Cooper et al., 1995 (Cooper et al., , 2016 . As COVID-19 stressors have the potential to increase substance misuse and health issues in vulnerable populations (Du et al., 2020) , the Veterans Health Administration (VHA) became concerned with the impact of COVID-19 stressors on veterans' drinking and drug use (Insider, 2020) . Substance use is a significant problem among United States military veterans (Oliva et al., 2017; Teeters et al., 2017) and is associated with numerous harmful effects, including adverse physical and mental health functioning (Boden and Hoggatt, 2018) . Recent COVID-19 research also supports this trend, as veterans' increases in substance use have been associated with corresponding decreases in mental and physical health (Roberts et al., 2021) .\"}, {'id': 'rmvw3ko0-2', 'contents': 'Factors predictive of veteran functioning include more than merely problematic substance use. An important framework to consider when assessing the functional impact of substance misuse for veterans is the unique social conditions of their lives, referred to by the World Health Organization (WHO) as social determinants of health (SDH; World Health Organization, 2021). Social determinants include both structural determinants (e.g., income, finances, and age), and intermediary determinants (e.g., psychosocial circumstances and social support; Hosseini Shokouh et al., 2017) , which can strongly influence health outcomes. The pandemic has had well-recorded and massive impacts on SDH areas, including changes in employment patterns, social networks, and reported quality of life (Guerin et al., 2021; Niles et al., 2021; Rogers et al., 2021) . Conditions reflecting disadvantaged SDH contexts have been widely associated with psychosocial stressors and mental health issues (Allen et al., 2014) . These SDH and environmental conditions can result in allostatic load issues and worse health outcomes (Thisted, 2003; Denny and Brownell, 2010) . Attempts to reconcile the unbalanced number of stressors without adequate access to societal and personal coping resources may have an interactive effect with negative coping strategies such as problematic substance use, further negatively impacting the functioning of veterans.'}, {'id': 'rmvw3ko0-3', 'contents': 'The combined impact of these factors-pandemic impacts on quality of life (e.g., finances and meeting basic needs)-on the health and well-being of veterans is not yet clear. Though there have been studies on the impact of COVID-19 on veteran substance use Pedersen et al., 2021) , none have specifically investigated problematic substance use, COVID-19 related changes in SDH, and functional outcomes. There is also a continued and growing need to understand more precisely COVID-19 health, stressors, and infection-control behaviors, which have occurred in very particular contexts for individuals as they balance maintaining their quality of life and staying safe. For example, social distancing can increase safety from COVID-19 infection; however, loneliness is a risk factor for relapse (Volkow, 2020) because it triggers irritability, anxiety, fear, sadness, anger, or boredom (Ornell et al., 2020) . These specific COVID-19 experiences for veterans are essential to consider when investigating the impact of COVID-19, substance use concerns, and health outcomes.'}, {'id': 'rmvw3ko0-4', 'contents': \"The main aim of the present study was to investigate the impact of the COVID-19 pandemic on United States veterans who self-reported problematic drug and alcohol use. Within the framework of SDH, the study specifically assessed veterans' lived experience of COVID-19 management, SDH (COVID-19 related quality of life indicators and loneliness), and their mental and physical health functioning. This included a particular aim to examine the contribution of SDH factors in the relationship between problematic substance use and functioning. The planned stepwise hierarchical regression model originally included a priori Hypotheses 1 and 2 (i.e., hypothesized prior to data collection but not publicly pre-registered). An additional post hoc Hypothesis 3 was added following data collection to examine a potential interaction between negative COVID-19 impacts on quality of life (e.g., finances and meeting basic needs) and COVID-19related loneliness.\"}, {'id': 'rmvw3ko0-5', 'contents': 'Frontiers in Psychology | www.frontiersin.org 3 April 2022 | Volume 13 | Article 812247'}, {'id': 'rmvw3ko0-6', 'contents': 'Hypothesis 1: Greater self-reported substance use will be associated with both lower physical and mental health functioning during COVID-19 for Veterans.'}, {'id': 'rmvw3ko0-7', 'contents': 'Hypothesis 2: Greater levels of negative COVID-19 impacts on quality of life (e.g., finances and meeting basic needs) and COVID-19-related loneliness will each have a distinctive, significant negative relationship with physical and mental health functioning, even when modeled alongside problematic substance use, resulting in a significant increase in variance explained in functioning.'}, {'id': 'rmvw3ko0-8', 'contents': 'Hypothesis 3: Loneliness was hypothesized to moderate the relationship between the negative impact of COVID-19 and both physical and mental health functioning, such that individuals reporting higher levels of loneliness and higher levels of negative COVID-19 impacts on quality of life would report lower mental and physical health functioning, compared to those with higher levels of negative COVID-19 impacts on quality of life but lower levels of loneliness.'}, {'id': 'rmvw3ko0-9', 'contents': 'Procedures and methods for collecting this cross-sectional data were completed in accordance with a protocol approved by the Institutional Review Board at the VA Bedford Healthcare System. The survey was administered using the Qualtrics federal platform between November 24, 2020 and February 2, 2021, and United States veterans were identified via a Qualtrics panel. Study staff coordinated with a project coordinator from Qualtrics, who used their panel aggregator system, an internal Qualtrics system where over 20 Web-based panel providers have been identified, screened, and utilized by Qualtrics recruiters to supply diverse, quality respondents depending on survey inclusion/exclusion criteria. Research has supported the Qualtrics panel recruitment methodology as an effective recruitment strategy, unlike non-panel recruited anonymous online samples, with evidence supporting that these panels can produce higher quality samples due to internal quality control and prescreened respondents (Ibarra et al., 2018) . This data collection method is thus becoming increasingly popular, with evidence showing data quality from the Qualtrics panel is on par with data from conventional data collection methods (e.g., Walter et al., 2019) . Potential panel participants were provided with a link to a description of the study and an eligibility survey on the Qualtrics platform. Informed consent was provided and obtained prior to accessing the survey. The survey was estimated to take approximately 30 minutes to complete. The eligibility criteria for the survey and included as pre-survey screening items were (1) reporting a minimum age of 18 years old, (2) problematic drug and alcohol use indicated by Cut Down, Annoyed, Guilty, and Eye Opener (CAGE)-AID minimum score of 1, and (3) being a veteran of the United States Armed Forces. Veteran status was further assessed by asking participants to report on employment/veteran status overall (such that participants were unaware of the veteran eligibility criteria). If they reported being a veteran, respondents were required to report their DD214 (the Certificate of Release or Discharge form) date and the years they served. Participants were excluded when (1) they had previously completed the survey, (2) did not meet the minimum CAGE-AID score of one, (3) did not meet our verification check for United States Veterans (e.g., were active duty military), or (4) were identified as non-attentive or potential survey spammers after quality-review (see below for criteria). To allow for a wide surveying of veterans with possible substance use concerns, responders were not excluded for (1) not having a formal substance use disorder (SUD) diagnosis, or (2) use of prescribed medications, but did have to report problematic use of drugs or alcohol to qualify for the survey.'}, {'id': 'rmvw3ko0-10', 'contents': 'Given the high possibility that potential recruited responders might be motivated to complete the survey even without being a veteran due to monetary gains, we instituted a series of survey items and validation checks similar to those utilized by Pedersen et al. (2017 Pedersen et al. ( , 2021 when recruiting veteran community samples. To verify veteran status, an initial screening item asked respondents for their current military status (i.e., active duty, not and have never been a member of the military, a veteran, or other). If respondents answered they were a veteran, they answered a series of additional items to verify veteran status, including when they had served and in which era (e.g., Vietnam Era, August 1964 to April 1975), and branch of service (e.g., Army). They were then asked if they had received a DD214 and, if yes, were routed to indicate their date of discharge. This date needed to reasonably match their reported service era and self-reported age. If answers were inconsistent, these respondents were removed from the survey.'}, {'id': 'rmvw3ko0-11', 'contents': 'Approximately 58,500 surveys were sent out to panel participants, who were not aware of the contents of the survey prior to selecting it in their dashboard. A total of 6,500 (11.1%) potential participants entered the survey and were provided with the informational sheet and consent document for the survey. Of these, 381 (5.9%) did not consent to take the pre-survey screening items. Of those who consented to participate in the pre-survey screening, 5,366 (87.7%) were automatically screened out for not reporting being a United States veteran (n = 3,200) and for not reporting substance abuse issues on the CAGE-AID screener (n = 2,166). Following these automatic eligibility checks (n = 753), a total of 436 participants completed the survey (completion rate of 58%). Based on proposed guidance for online recruitment quality review by Heffner et al. (2021) , potential participants were removed if three of the following four previously identified quality-review criteria were met: (1) self-reported veteran status was deemed impossible when reviewed in conjunction with age, reported military service era, and/or DD214 date, (2) attention check items showed participants reporting gibberish responses to open-ended questions, (3) total time spent on the survey was under 1/3 of median completion time (< 10 min), and (4) response patterns flagged by the Qualtrics system as potential bots using their quality check software (i.e., IP-based \"Prevent Ballot Box Stuffing\" option, attempts to take the survey multiple times) were reviewed and found to be suspicious (e.g., answering all survey items as the last Likert scale option). A total of 27 (6.2%) potential participants were removed because they did not meet these criteria.'}, {'id': 'rmvw3ko0-12', 'contents': 'The present study evaluated several demographic characteristics, including age, gender, sexual orientation, race, ethnicity, income, armed service branch, VHA connection status, and relationship status.'}, {'id': 'rmvw3ko0-13', 'contents': 'Substance use concerns and behaviors were assessed in the screening survey and within the main survey. Problematic use of alcohol and other drugs was assessed using the CAGE Adapted to Include Drugs (CAGE-AID; Brown and Rounds, 1995) , a validated four-item measure. Per the CAGE-AID protocol, the questionnaire was only given to participants who reported current alcohol or drug use. A minimum score of 1 on the CAGE-AID was used as a screener to participate in the survey, as recommended by the CAGE-AID Consensus Panel of the Substance Abuse and Mental Health Services Administration (SAMHSA) to cast a wider net and identify more patients with substance abuse or addiction issues (Sullivan and Fleming, 2008) . Previous research indicates that a CAGE-AID score of one or more indicates a positive screen with a sensitivity of 0.79 and specificity of 0.77, and that scores can accurately classify 78% of substance users (Ashman et al., 2004) .'}, {'id': 'rmvw3ko0-14', 'contents': 'A modified version of the Alcohol, Smoking, and Substance Involvement Screening Test (ASSIST; Humeniuk et al., 2010) was used to descriptively measure substance use frequency, urges, and changes in use during the pandemic for alcohol, tobacco, cannabis, cocaine, stimulants, inhalants, sedatives, hallucinogens, and opioids (four items for each substance). Item #1, assessing any lifetime non-medical use of a specific substance, was retained in its original form. As a main purpose of the current study was to describe potential changes in substance use during COVID, Item #2 was modified to read, \"Since the beginning of the COVID-19 pandemic, how often have you used [specific substance]?\" Additionally, Item #3 was modified to \"Since the beginning of COVID-19, how has your desire or urge to use [specific substance] changed?\" and Item #4 was added, \"How has your frequency of using [specific substance] changed during COVID-19?, \" with the response scale ranging from More Use/Craving, No Change, and Less Use/Craving.'}, {'id': 'rmvw3ko0-15', 'contents': 'Medical diagnoses were obtained via self-report. Several general medical diagnoses were provided, and participants self-reported whether they had ever received an official medical diagnosis from a doctor or healthcare professional. As many of these diagnoses can be long-standing, we did not specify a timeframe for when the condition might have been officially diagnosed. These medical diagnoses included chronic pain, diabetes, insomnia, heart disease, apnea, and seizures.'}, {'id': 'rmvw3ko0-16', 'contents': 'Psychiatric diagnoses were self-reported using the same criteria as medical diagnosis (e.g., had been diagnosed by a doctor or other healthcare professional specifically) by participants and included SUDs, major depressive disorder, post-traumatic stress disorder (PTSD), anxiety disorder, panic disorder, psychogenic non-epileptic seizures (PNES), schizophrenia, and bipolar disorder.'}, {'id': 'rmvw3ko0-17', 'contents': \"The Short-Form Health Survey-12 (SF-12; Ware et al., 1996) is a 12-item measure that assesses mental and physical functioning. The SF-12 has been validated for predicting populations' mental and physical health without targeting specific health outcomes and has high reliability, including with United States veterans (Salyers et al., 2000) . Scoring involves using a norm-based algorithm that produces a self-reported mental health composite score (MCS) and physical health composite score (PCS) between 0 and 100 (Jones et al., 2001) , with lower scores associated with lower functioning. Test-retest reliability is 0.80 for the PCS and 0.76 for the MCS (Ware et al., 1996) .\"}, {'id': 'rmvw3ko0-18', 'contents': 'The present study evaluated the experiences, stressors, and impact of COVID-19 across several areas of interest. A subscale score was created using a modified version of the Pain Management Collaboratory Coronavirus Pandemic (COVID-19) 5-Item Measure (PMC-5), which measured the potential negative impact of COVID-19 on quality of life on a scale of 1 (Improved) to 4 (A Lot Worse). The five areas measured by the PMC-5 included finances, emotional health, ability to meet basic needs, and two additional items on physical health and concentration. For the present study, internal consistency for the PMC-5 was satisfactory (five items: α = 0.82).'}, {'id': 'rmvw3ko0-19', 'contents': 'In addition, three modified questions from the Osteoporotic Fractures in Men (MROS) Study COVID-19 Social Impact Questionnaire (Cawthon et al., 2020) with multiple check-box options were asked to describe COVID-19 experiences, including (1) \"Which of the following statements best describes your personal experience with the coronavirus, \" (2) \"Which of the following have you done in the last month to keep yourself safe from coronavirus?, \" and (3) \"Which of the following are worries of yours related to COVID-19, or are more difficult for you now because of the pandemic?\"'}, {'id': 'rmvw3ko0-20', 'contents': 'The UCLA Loneliness Scale-8 (ULS-8; Hays and DiMatteo, 1987) was used to assess feelings loneliness during the pandemic. The scale instructions were slightly modified to assess pandemic-related loneliness and read, \"Indicate how often you have felt the following ways during your COVID-19 experience. \" The full eight-item measure was utilized, and sample items include \"There is no one I can turn to\" and \"I lack companionship. \" Scores range on a scale from 1 (Never) to 4 (Often), with higher scores indicating greater levels of loneliness. Internal validity for this scale in the current sample was satisfactory (eight items: α = 0.83).'}, {'id': 'rmvw3ko0-21', 'contents': 'Means, SDs, and frequencies were calculated for the full sample. Zero-order correlations were first conducted among all the measures of interest to assess for initial relationships among model variables. To research the unique effects of each potential predictor, two hierarchical regression analyses were conducted for mental and physical health functioning. The hypothesized model first explored the relationship between problematic substance (CAGE-AID) use on mental health and physical functioning, respectively. In addition, by adding variables related to SDH factors (PMC5 and ULS-8) to the model, we anticipated greater predictive ability for both physical and mental health functioning, with problematic substance use still accounting for a significant amount of variance in functioning. Hierarchical or sequential regression allows for the analysis of multiple predictors when the order of entry is determined by a theoretical base, while also determining the contribution of each variable after controlling for those entered earlier in the model (Tabachnick and Fidell, 2007) . In the first step, demographic control variables were included, showing significant zero-order correlations with the dependent variables (gender, age, and annual income). In the second step of the hierarchical multiple regression analysis to test Hypotheses 1, problematic alcohol and drug use concerns (CAGE-AID scores) were entered to investigate their impact on mental health. In the third step, we evaluated our proposed Hypothesis 2 by adding pandemicrelated stressors Hierarchical or sequential regression allows for the analysis of multiple predictors when the order of entry is determined by a theoretical base, while also determining the contribution of each variable after controlling for those entered earlier in the model (Tabachnick and Fidell, 2007) . In the first step, demographic control variables were included, showing significant zero-order correlations with the dependent variables (gender, age, and annual income). Hypothesis 1 was tested in the second step of the hierarchical multiple regression analysis, with problematic alcohol and drug use concerns (CAGE-AID scores) entered to investigate their impact on mental health (MCS-12). Hypothesis 2 was explored in the third step, with pandemic-related stressors (PMC-5 and ULS-8) added to the model to assess the unique impact of COVID-19 associated SDH factors on mental health functioning. A second hierarchical regression was conducted with the same variables and stepwise model to predict physical health. We tested the assumptions of regression analysis and for evidence of violations. Multicollinearity among predictor variables was set at zeroorder correlations greater than 0.70, and continuous variables scores and errors were inspected for normalcy of distribution.'}, {'id': 'rmvw3ko0-22', 'contents': 'In addition, to assess post-hoc Hypothesis 3, exploring a potential interaction between the ULS-8 and PMC5 on functioning, predictors variables were mean-centered to create a cross-product to regress on both outcome variables of interest in the final step. All tests were two-tailed, with all analyses conducted using SPSS v.26.'}, {'id': 'rmvw3ko0-23', 'contents': 'The mean age of the sample (N = 409) '}, {'id': 'rmvw3ko0-24', 'contents': 'Of the 409 participants screened into the study with problematic substance use, a minority reported a formal, medically diagnosed alcohol (27.6%; 113/409) or substance use (20.5%; 84/409) disorder. The most reported substances for those with medically diagnosed SUD other than alcohol were tobacco and cannabis. The most reported psychiatric diagnoses were anxiety disorder (36.4%; 149/409), PTSD (27.4%; 112/409), and major depressive disorder (20.3%; 83/409); the most reported physical condition was chronic pain (31.5%; 129/409: see Table 2 ).'}, {'id': 'rmvw3ko0-25', 'contents': 'Regarding alcohol and non-prescription substance use during COVID-19, a substantial number of participants reported daily or mostly daily use of substances. Of the participants who reported at least some lifetime use of a particular substance, alcohol (45.8%; 182/397) was the most commonly used substance on a daily or mostly daily basis, followed by tobacco (44.9%; 150/334), sedatives (29.8%; 48/161), and opioids (26.6%; 29/109). Participants with current substance or alcohol use also reported changes in cravings during the COVID-19 pandemic. The majority reported no changes in urges/cravings of substances, apart from alcohol, with 47.6% (189/397) reporting an increased urge to drink. For all other substances, the largest increases in cravings were reported for cannabis (38.4%; 98/255) and tobacco (37.1%; 124/334). Finally, many participants reported increased substance use during COVID-19, with the highest increased use for alcohol (45.3%; 180/397). See Table 3 for specific rates by substance.'}, {'id': 'rmvw3ko0-26', 'contents': 'Participants reported on their specific COVID-19 experiences, with 10.5% (43/409) reporting being officially diagnosed with COVID-19 and 12.7% (52/409) believing they had COVID-19 but were not officially diagnosed (see Table 4 ). The most frequently reported stressors related to the pandemic were fear of getting COVID-19 (61.4%; 251/409) or someone close to the participant getting COVID-19 (58.7%; 240/409). In terms of safety behaviors, the most common behaviors were wearing a facemask (88.8%; 363/409), avoiding public spaces and crowds (69.2%; 283/409), and avoiding in-person contact with friends and family (50.1%; 205/409).'}, {'id': 'rmvw3ko0-27', 'contents': 'Prior to multiple regressions, continuous outcome variables were inspected for normality, and all continuous variables had an approximate normal distribution with no significant outliers. In addition, control variables, predictors, and outcome variables were investigated using zero-order correlations (Table 5), which revealed that being young, having greater problematic substance use, reporting more negative COVID-19 impacts on quality of life, and experiencing greater reported loneliness were correlated with lower scores in both mental and physical health functioning. These statistically significant relationships among variables did not suggest multicollinearity (absolute correlation coefficient greater than 0.70).'}, {'id': 'rmvw3ko0-28', 'contents': \"Following the investigation of these correlations, hierarchical multiple regression analyses were conducted. In the first step of the regression, age, gender, and race were entered as covariates and explained a significant amount of the variance in mental health functioning, F(3, 403) = 13.83, p < 0.001, adjusted R 2 = 0.09. In the second step, results from the CAGE-AID were entered to control for the impact of problematic drug and alcohol use on mental health outcomes, and explained a significant amount of variance, F(4, 402) = 13.41, p < 0.001 adjusted R 2 = 0.12, with CAGE-AID score explaining a significant additional amount in variance in mental health functioning (β = −0.17, p = 0.001), such that greater levels of problematic use of substances predicted decreases in mental health. In the third and final step, indicators of COVID-19-related impacts, specifically the PMC-5 (negative COVID-19 impacts on quality of life) and ULS-8 (pandemic-related loneliness), were entered. This addition resulted in a significant change in R 2 of 0.12 (p < 0.001), with PMC-5 (β = −0.12, p = 0.03) and ULS-8 scores (β = −0.34, p < 0.001) as significant predictors of decreased mental health functioning during COVID-19. With the addition of these pandemic stressor variables, CAGE-AID scores no longer predicted mental health functioning (β = −0.07, p = 0.14). There was no significant interaction between ULS-8 and PMC-5 on mental health functioning (β = −0.05, p = 0.30). This finding indicates that negative COVID-19 impacts on quality of life (PMC-5) and pandemic-related loneliness (ULS-8) had a unique and negative impact on veterans' mental health functioning (MCS-12). The entire model accounted for 23% of the variance in the MCS (see Table 6 ).\"}, {'id': 'rmvw3ko0-29', 'contents': \"To begin testing the replicability of the previously outlined hierarchical multiple regression model on physical health, age, gender, and race were entered as covariates, which explained a significant amount of the variance in physical health functioning, F(3, 403) = 6.25, p < 0.001, adjusted R 2 = 0.04. In the second step, the CAGE-AID was entered into the model and explained a significant amount of variance, F(4, 402) = 5.70, p < 0.001, adjusted R 2 = 0.04 in physical health functioning (β = −0.10, p = 0.04), such that greater problematic use of substances predicted decreases in physical health. In the third step, indicators of COVID-19 related-impacts were entered. This addition resulted in a significant change in R 2 = 0.04, p < 0.001, and although CAGE-AID scores no longer predicted physical health (β = −0.05, p = 0.38), negative COVID-19 impacts on quality of life (PMC-5; β = −0.11, p = 0.03), and pandemic-related loneliness scores (ULS-8; β = −0.14, p = 0.02) emerged as significant predictors of decreased physical health functioning during COVID-19. There was no significant interaction between ULS-8 and PMC-5 on physical health functioning (β = 0.03, p = 0.51). This finding indicates that stressors from the pandemic, both loneliness and perceived negative impact of COVID-19 on quality of 1. Age -2. CAGE-AID −0.33*** -3. PMC-5 −0.31*** 0.31*** 4. ULS-8 −0.28*** 0.28*** −0.49*** 5. PCS-12 0.10* −0.14** −0.24** −0.24*** 6. MCS-12 0.28*** −0.24*** −0.33*** −0.45*** −0.19** -- life, had a unique and negative impact on veterans' physical health functioning during COVID-19. The entire model accounted for 9% of the variance in the physical functioning composite score.\"}, {'id': 'rmvw3ko0-30', 'contents': \"The primary aim of this study was to describe the experience of the COVID-19 pandemic on United States veterans reporting problematic substance use and model the pandemic's impact on functioning. Overall, results showed that physical and mental health functioning during COVID-19 had been strongly and negatively impacted by pandemic-related decreases in important life domains and quarantine-related loneliness for veterans reporting substance use issues. As pandemic infection and death rates continue to wane and surge, such findings are important in understanding the unique issues that might arise for veterans in our community dealing with substance use concerns.\"}, {'id': 'rmvw3ko0-31', 'contents': 'To investigate our primary descriptive aim, United States veteran participants reported on a broad range of pandemic-related health experiences and safety behaviors during the COVID-19 pandemic. Though only 23% had a confirmed or assumed COVID-19 diagnosis at the time of this survey, a third had experienced a family member or close friend becoming infected. Additionally, of those without reported coronavirus, nearly all participants sampled reported stress related to themselves or someone close to them becoming infected. Given these concerns, it is unsurprising that there was a high rate of reported safety behaviors, including mask-wearing and social distancingalthough not all participants reported these behaviors. This finding suggests that similar to civilian populations (Knowles and Olatunji, 2021) , there has been a wide range of accepted and applied COVID-19 infection-control activities within the United States veteran population. In addition, only about a third of the participants had attended at least one healthcare appointment (either telehealth or in-person), and 17.8% had canceled a healthcare appointment due to COVID-19. As 35% of our sample was not receiving healthcare at the VA and given the low rates of help-seeking yet high rates of comorbid mental and physical health concerns in the veteran population, this finding was particularly concerning. This disruption of healthcare services during COVID-19 is ongoing and suggests that the VHA should make strong attempts to enroll unconnected community veterans across the United States or reconnect with veterans who may have lapsed in their VHA healthcare utilization during the pandemic.'}, {'id': 'rmvw3ko0-32', 'contents': 'Our results also emphasized that many United States veterans reporting problematic substance use over the past year had increased their alcohol and non-prescription drug use during COVID-19 and attributed this increase specifically to the pandemic. This finding was most pronounced in drinking, where most participants reported an increase in drinking during the pandemic. Although all participants self-reported at least some problematic drug or alcohol use for study inclusion, only about a third of participants of our sample reported a formal substance use diagnosis. This discrepancy coincides with research showing problematic use of substances is perhaps more ubiquitous than during non-pandemic times (Satre et al., 2020) , possibly as a means of coping with pandemic-related stressors (Wardell et al., 2020) . However, other research suggests the COVID-19 related increases in substance use vary for different veteran populations, such as those with pre-COVID mental health concerns being associated with increased drinking Pedersen et al., 2021) . From a clinical perspective, this increase in substance use may be explained by additional underlying psychological mechanisms, such as cognitive process (e.g., Desire Thinking; Mansueto et al., 2019) , psychiatric symptom management (Cosci et al., 2019) , personality vulnerabilities such as impulsive sensation-seeking or behavioral disinhibition (Sher et al., 2000; Oh et al., 2021) , and dysfunctional coping strategies (see Cavicchioli et al., 2018) . These should be assessed in the future, to both investigate what factors may maintain increased COVID-19 substance use and as potential therapeutic targets for addictive behaviors interventions. Given our finding of increased substance use for many veterans, providers should be even more committed to screening patients, especially those with diagnosed mental health disorders, for potential SUDs during our continued COVID-19 pandemic management.'}, {'id': 'rmvw3ko0-33', 'contents': 'To explore the physical and mental health functioning of these United States veterans during the pandemic, we investigated several potentially relevant variables hypothesized to be significantly associated with functioning. Consistent with our first hypotheses, higher subjective drug and alcohol concerns were related to lower mental and physical health functioning, similar to other research on substance misuse and functioning (e.g., Sheckter et al., 2020) . Our second hypothesis, that negative COVID-19 impacts on quality of life (e.g., finances and meeting basic needs), greater self-reporting of problematic substance use, and COVID-19-related loneliness will have each a distinctive, significant negative physical and mental health functioning, was only partially supported by our data. Findings revealed COVID-19 related pandemic impacts and loneliness were negatively associated with both mental and physical functioning; indicating that distress due to the COVID-19 pandemic has strongly impacted both areas of functioning. This finding aligns with ongoing research into pandemic stressors and their impact on quality of life (Yu et al., 2021) . Researchers are likely underestimating the negative impact of COVID-19 across multiple life domains and social isolation on United States veterans with SUD concerns in terms of functioning, treatment, and relapse rates (Linas et al., 2021) . Our findings show that there have been serious, negative impacts reported by United States veterans due to COVID-19 on quality of life, social relationships, and substance use. Continued research during and beyond the pandemic should account for these stressors and model their long-term impact on SUD behaviors and coping. Our final regression model assessing our second hypothesis highlighted that, when considered in conjunction with pandemicrelated quality of life and COVID-19 related loneliness, problematic substance use behaviors alone did not significantly predict either mental or physical health functioning. Our post hoc interaction analysis, exploring the possibility that the negative impact of COVID-19 on quality of life and its relation to functioning might be moderated by reported levels of loneliness, was not supported. Instead, the final model highlighted the strong, main effect of negative COVID-19 impacts on quality of life and self-reported loneliness on decreased functioning for these veterans reporting problematic substance use concerns. The impact of COVID-19 related quality of life factors, such as financial well-being and being able to meet basic needs, emphasized the importance of attending to changes in SDH factors and environmental contexts when considering the physical and psychological functioning of veterans with substance use concerns. The strong association between loneliness and both physical and mental health functioning was also particularly important. This aligns with pre-pandemic evidence showing social support being associated with decreased all-cause mortality, cardiovascular disease, depression, and anxiety (Leigh-Hunt et al., 2017) , and pandemic-specific research showing loneliness as a strong predictor of worse functioning, affective illness, and non-accidental self-injury (Greig et al., 2022) . As loneliness can lead to worse outcomes, effective interventions that are also consistent with COVID-19 infection control and social distancing measures, such as those suggested in the systematic review by Williams et al. (2021) , may be of particular importance in veteran treatment.'}, {'id': 'rmvw3ko0-34', 'contents': \"Given these findings, it is recommended that ongoing assessments of U.S. veterans' mental and physical health functioning should evaluate not only addictive behaviors but also pandemic-related experiences and SDH. Particular emphasis should be placed upon assessing veterans' basic needs, financial insecurity, and loneliness, as these may be particularly predictive of physical and mental health functioning. These data align with decades of research into the importance of SDH impacting individuals with substance use concerns (Rhodes, 2002; Collins et al., 2019) , highlighting that these factors may be even stronger determinants of functioning in the context of a global pandemic (Russell et al., 2021) .\"}, {'id': 'rmvw3ko0-35', 'contents': \"A major limitation with this anonymous survey was having only self-reported data. First, all diagnoses were collected via participant self-report, with no diagnostic evaluation or medical record review conducted to confirm veteran's reported psychiatric, medical, and SUD diagnoses. Though a clinically validated screening instrument (CAGE-AID) was used to assess for problematic drug and alcohol use concerns, a more rigorous evaluation of substance use diagnostic criteria would have strengthened the study. Second, as changes in the frequency of use and cravings related to alcohol and drugs were also self-reported, responses might be subject to recall, response, and social desirability biases. Yet, most surveys are limited to recall limitations, and given the online and anonymous nature of the survey, biases in response and social desirability are likely low. In addition, generalizability of the current sample may be limited due to a sample that was less ethnically and racially diverse than the national United States veteran sample (i.e., sample is 90% White/Caucasian, whereas nationally, White/Caucasian veterans are 80% of the veteran population). Using a panel-recruited, web-based survey methodology may have been susceptible to fraudulent and biased responses. The research team implemented standardized quality control reviews, veteran status verification procedures, and data inclusion screening procedures to minimize this concern, including analysis of click-through behavior and scrubbing methods for web-based panel quality, resulting in higher confidence related to survey response quality. A final and important limitation of our study is that it was cross-sectional. We are thus unable to assess directionality or causality of the relationship between variables, limiting our ability to predict changes in substance use or functioning outside of participant self-report of such changes. Although logistic regressions to investigate COVID-19 related factors impacting changes in substance use may have added to this understanding, this analysis is beyond the scope of the current paper and will be examined in future analyses.\"}, {'id': 'rmvw3ko0-36', 'contents': \"The level of anxiety and stress that the COVID-19 global pandemic has inflicted on the world is staggering. United States veterans struggling with alcohol and substance addiction have reported significant increases in substance use and feelings of pandemic-related anxiety. In addition to substance-use related concerns, pandemic-related stressors and loneliness have further diminished these veterans' physical and mental health. As the spread of vaccines curtails the potential impact of COVID-19, increased research on the mental health impact of the months-long lockdowns, variant impacts, and possibly years-long social distancing protocols have had on individuals suffering from substance use issues will need to be investigated. Future research using a community sample of United States veterans would benefit from both continued, longitudinal data collection and additional assessments of drug and alcohol use (e.g., the AUDIT or DAST). Additionally, future work should further investigate the underlying mechanisms of substance use changes, as well as moderating factors related to managing allostatic load and the impact of SDH during the pandemic, such as psychological flexibility. This would allow for the assessment of alcohol and drug use disorders diagnostically, while also modeling the impact of resiliency factors on health and functioning trajectories in this population. A better understanding of how veterans with self-reported problematic substance use have been both struggling and successfully navigating their unique COVID-19 societal conditions may provide further information on areas of future interventions.\"}, {'id': 'rmvw3ko0-37', 'contents': 'The raw data supporting the conclusions of this article will be made available by the authors, without undue reservation.'}, {'id': 'rmvw3ko0-38', 'contents': 'The studies involving human participants were reviewed and approved by VA Bedford Healthcare System Institutional Review Board. The patients/participants provided their informed consent to participate in this study.'}, {'id': 'rmvw3ko0-39', 'contents': 'ER, EC, BD, SS, JH, and MK conceived the study, provided conceptual guidance, and survey distribution procedures and data collection and contributed to writing the manuscript. ER organized, cleaned, and initially analyzed the data. ER, MK, and SS interpreted the data. All authors contributed to the article and approved the submitted version. '}, {'id': 'kf7lix7f-0', 'contents': 'The concept of CP infusion evolved in 1880 whereby immunity was noticed against diphtheria which had been instigated from the existing antibody titer in blood from animals purposely immunized with non-lethal doses of toxins [13] . Besides, in the early 1950s, the purified and concentrated immunoglobulins both from the healthy individuals and the recovered patients from disease imparted an opportunity for the treatment of serious infectious diseases [13] . The severe H1N1 2009 infection could be treated with CP as the viral load was found to be decreased upon transfusion as well as the cytokine storm was under control, resulting in the decline in mortality [11] . It is worth noting that meta-analysis of 1918 influenza A (H1N1) pandemic as well as the treatment of severe influenza A by H5N1 subtype infection unraveled that the CP therapy could be a potential treatment strategy [18] . The methods for the detection of the neutralizing antibodies of severe acute respiratory syndrome (SARS) convalescent sera (i.e., anti-SARS antibodies) were well endorsed in 2003, including the enzyme-linked immunosorbent assay (ELISA), neutralization assay, and Western blot; and the evaluation of the efficacy of CP therapy with severe SARS patients revealed good outcome (decrease in the viral load within a day of transfusion with simultaneous increase in the anti-SARS-CoV immunoglobulins IgM and IgG) [11, [17] [18] [19] [20] [21] .'}, {'id': 'kf7lix7f-1', 'contents': 'Indeed, the protocol for application of the CP for SARS was established in 2015 [22] . Possible sources of CP therapy against SARS-CoV-2 are human convalescent sera from individuals who have recovered from COVID-19 ( Figure 1 ). Another option is preparations generated in certain animal hosts, such as genetically engineered cows that produce human antibody [23] . The benefits of CP therapy in the COVID-19 mitigation has been observed through the direct neutralization of invading SARS-CoV-2, regulation of the cytokine dysbiosis (precisely the cytokine storm), complement activation, and immunomodulation thereby reducing the mortality rate [13] . In a study with the hospitalized severe patients, the plasma transfusion was noticed to (1) normalize body temperature of the maximum patients, (2) the sequential organ failure assessment (SOFA) score was found to be decreased, (3) the acute respiratory distress syndrome (ARDS) was found to be eliminated in 80% cases, (4) normalized levels of C-reactive protein (CRP), alanine aminotransferase, and aspartate aminotransferase, (5) the SARS-CoV-2 load declined; and surprisingly, (6) the patients were tested to be virus negative after 12 days of transfusion [13] . Besides, as stated elsewhere, in the immunological perspective, the lymphocyte counts in the peripheral blood may significantly decrease upon CP therapy; and also, the level of the cytokines in plasma including the interleukin (IL)-6, IL-10, tumor necrosis factor (TNF)-α, and the granulocyte macrophage colony stimulating factor (GMCSF) are likely to be reduced [1, 13, 16, 24] . Moreover, the CP therapy was found to shorten the duration in the intensive care unit (ICU), the rate of mechanical ventilation (MV) support, and vasopressor support for the serious COVID-19 patients [14] . Thus, the convalescent sera obtained from the recovered COVID-19 individuals who had established humoral immunity against SARS-CoV-2 consist of significant level of neutralizing antibodies against the virus which can further help eradicating the same virus from the systemic circulation as well as the pulmonary tissues in the other COVID-19 patients if CP treatment is employed.'}, {'id': 'kf7lix7f-2', 'contents': 'Although the CP therapy appears to be a potential strategy for the treatment of the severe COVID-19 patients, still some challenges rise with the side effects such as chills, fever, transfusion-related acute lung injury, and anaphylactic reactions [25] . In addition to antibodies, the convalescent whole blood (CWB) may impart the regulation of the hemorrhagic events as was noticed in case of the Ebola virus disease (EVD) Figure 1 Preparation and pooling of the convalescent blood products (CBP). The CP manufacturing scheme presented here is adapted from Focosi et al. (2020) [4] . The convalescent donor is usually selected through the neutralizing antibody titer which is assessed through the plaque reduction neutralization test (PRNT) requiring a viable isolate, replication-competent cell lines, and skilled personnel. Alternatively, an enzyme-linked immunosorbent assay (ELISA) can also be used which targets the recombinant receptor binding domains (RBDs) of the viral anti-receptor, followed by the correlation measurements between ELISA ratios/indexes with the PRNT titers. CP can be harvested by apheresis (plasma fractionation) using the double filtration plasmapheresis (DFPP) with fractionation filter 2A20 (trial NCT04346589) or, alternatively, collecting immunoglobulins from the convalescent donors by immunoadsorption (trial NCT04264858). The pathogenic load is reduced or eliminated from the plasma by employing the (1) solvent/detergent (S/D)-filtration, providing the quick inactivation of >4 logs of most enveloped viruses, or (2) by applying photoinactivation (combinations of methylene blue and visible light, theraflex, amotosalen (S-59) and UV A, and riboflavin and UV B, or (3) by using caprylic acid and octanoic acid (which also can inactivate the enveloped viruses). The next step is the pooling of CP together with the intravenous immunoglobulin (IVIg) by means of fractionation. In case of large-pool approach, the pharmaceutical-grade facilities characteristically pool 100-2500 donors to manufacture S/D-inactivated plasma; and the IVIgs are similarly prepared from pools of 2000-4000 L of plasma. In contrast, in case of minipool fractionation scale (MPFS) process, 5-10 L of plasma can be obtained. Finally, the harvested CP can be aliquoted into 200-300 mL quantities which can be preserved below −25°C prior to use [4] . cases ( Figure 1) [4, 7] . Moreover, CP infusion may have its latent threat like the aggravating hyperimmune attacks since CP therapy is completely based on the passive immunity with administering pathogen-specific immunoglobulins to the infected individuals [25, 26] . Indeed, the CP therapy can be more effective at the earlier stage of COVID-19, and hence, the peak timing of administering CP into the COVID-19 patients demands careful consideration [26] . However, with the advantage of one dose of 200 mL CP transfusion to significantly improve the clinical symptoms as well as to increase the oxyhemoglobin saturation with the concomitant neutralization of the viral load, facial red spots may be an adverse effect [27] . The other risks may include the transmission of the virus during plasma transfusion [27] . The most important risk may be the antibody-dependent entry (ADE) of the SARS-CoV-2 during CP therapy which in turn may suppress the innate antiviral state [15, 27, 28] . In addition, in case of emergency, donor screening and the conventional viral nucleic acid testing (NAT) especially for the human immunodeficiency virus (HIV), hepatitis C virus (HCV), and hepatitis B virus (HBV) are sometimes not be sufficient to guarantee the CP safety [4] . Moreover, although the convalescent plasma is habitually administered into the COVID-19 based on the experimental data with the aim of better clinical consequences, a clinical trial (PlasmAr ClinicalTrials.gov number, NCT04383535) showed no substantial differences between the COVID-19 patients treated with the convalescent plasma and those who received placebo [29] . Such a relatively neutral result may be explained by the logic that the patients who received the CP therapy; i.e., the plasma before day 14 of the onset of symptoms are usually more likely to recover with a lower mortality rate than those who receive such CP treatment after day 14 of the development of the COVID-19 symptom. Inclusive of other clinical trials, a 75% recovery rate was noticed through the CP therapy [29] . However, the role of CP also needs to be reevaluated now that highly efficacious COVID-19 vaccines are available and a cocktail of monoclonal antibodies is likely to be effective [14, 30, 31, 32] .'}, {'id': 'kf7lix7f-3', 'contents': 'The immune plasma neutralizes the invading SARS-CoV-2 as well as also delivers the passive immunomodulatory features permitting the recipient individual to regulate the elevated cytokine levels induced by the viral infection [13] . Besides the use of several repurposed antiviral drugs including favipiravir, lopinavir and ritonavir, hydroxychloroquine, azithromycin, and remdesivir, the application of CP in SARS-CoV-1 was also found to be effective in terms of decreased mortality; and at present, such strategy is also being noticed to be very much effective in reducing the viral load with the concomitant improvement of clinical conditions in the COVID-19 patients [14, 16, 17] . While the patients who have recovered successfully from COVID-19 may play valuable role as the donors for CP therapy, the disposition of the convalescent serum prophylaxis for COVID-19 patients must comply with the availability of donors, standard settings of blood bank facilities to process the sera, serological and viral assay facilities, appropriate prophylaxis, and therapeutic protocols along with the regulatory compliance. CP transfusion has been found to be effective especially when transfused within the first 20 days of the SARS-CoV-2 infection [14] . Therefore, both the dosage and transfusion timing are important for assuring the necessary safety and usefulness of such treatment [14] . Current review outlined the pros and cons of such application of passive immunity through the CP therapy.'}, {'id': 'kf7lix7f-4', 'contents': 'In such circumstance, the CP therapy is undoubtedly operative for the mitigation of the disease. '}, {'id': 'kf7lix7f-5', 'contents': 'The immune plasma neutralizes the invading SARS-CoV-2 as well as also delivers the passive immunomodulatory features permitting the recipient individual to regulate the elevated cytokine levels induced by the viral infection [13] . Besides the use of several repurposed antiviral drugs including favipiravir, lopinavir and ritonavir, hydroxychloroquine, azithromycin, and remdesivir, the application of CP in SARS-CoV-1 was also found to be effective in terms of decreased mortality; and at present, such strategy is also being noticed to be very much effective in reducing the viral load with the concomitant improvement of clinical conditions in the COVID-19 patients [14, 16, 17] . While the patients who have recovered successfully from COVID-19 may play valuable role as the donors for CP therapy, the disposition of the convalescent serum prophylaxis for COVID-19 patients must comply with the availability of donors, standard settings of blood bank facilities to process the sera, serological and viral assay facilities, appropriate prophylaxis, and therapeutic protocols along with the regulatory compliance. CP transfusion has been found to be effective especially when transfused within the first 20 days of the SARS-CoV-2 infection [14] . Therefore, both the dosage and transfusion timing are important for assuring the necessary safety and usefulness of such treatment [14] . Current review outlined the pros and cons of such application of passive immunity through the CP therapy.'}, {'id': 'kf7lix7f-6', 'contents': 'In such circumstance, the CP therapy is undoubtedly operative for the mitigation of the disease. '}, {'id': 'kf7lix7f-7', 'contents': 'The immune plasma neutralizes the invading SARS-CoV-2 as well as also delivers the passive immunomodulatory features permitting the recipient individual to regulate the elevated cytokine levels induced by the viral infection [13] . Besides the use of several repurposed antiviral drugs including favipiravir, lopinavir and ritonavir, hydroxychloroquine, azithromycin, and remdesivir, the application of CP in SARS-CoV-1 was also found to be effective in terms of decreased mortality; and at present, such strategy is also being noticed to be very much effective in reducing the viral load with the concomitant improvement of clinical conditions in the COVID-19 patients [14, 16, 17] . While the patients who have recovered successfully from COVID-19 may play valuable role as the donors for CP therapy, the disposition of the convalescent serum prophylaxis for COVID-19 patients must comply with the availability of donors, standard settings of blood bank facilities to process the sera, serological and viral assay facilities, appropriate prophylaxis, and therapeutic protocols along with the regulatory compliance. CP transfusion has been found to be effective especially when transfused within the first 20 days of the SARS-CoV-2 infection [14] . Therefore, both the dosage and transfusion timing are important for assuring the necessary safety and usefulness of such treatment [14] . Current review outlined the pros and cons of such application of passive immunity through the CP therapy.'}, {'id': 'kf7lix7f-8', 'contents': 'In such circumstance, the CP therapy is undoubtedly operative for the mitigation of the disease. '}, {'id': 'kf7lix7f-9', 'contents': 'The immune plasma neutralizes the invading SARS-CoV-2 as well as also delivers the passive immunomodulatory features permitting the recipient individual to regulate the elevated cytokine levels induced by the viral infection [13] . Besides the use of several repurposed antiviral drugs including favipiravir, lopinavir and ritonavir, hydroxychloroquine, azithromycin, and remdesivir, the application of CP in SARS-CoV-1 was also found to be effective in terms of decreased mortality; and at present, such strategy is also being noticed to be very much effective in reducing the viral load with the concomitant improvement of clinical conditions in the COVID-19 patients [14, 16, 17] . While the patients who have recovered successfully from COVID-19 may play valuable role as the donors for CP therapy, the disposition of the convalescent serum prophylaxis for COVID-19 patients must comply with the availability of donors, standard settings of blood bank facilities to process the sera, serological and viral assay facilities, appropriate prophylaxis, and therapeutic protocols along with the regulatory compliance. CP transfusion has been found to be effective especially when transfused within the first 20 days of the SARS-CoV-2 infection [14] . Therefore, both the dosage and transfusion timing are important for assuring the necessary safety and usefulness of such treatment [14] . Current review outlined the pros and cons of such application of passive immunity through the CP therapy.'}, {'id': 'kf7lix7f-10', 'contents': 'In such circumstance, the CP therapy is undoubtedly operative for the mitigation of the disease. '}, {'id': 'kf7lix7f-11', 'contents': 'The immune plasma neutralizes the invading SARS-CoV-2 as well as also delivers the passive immunomodulatory features permitting the recipient individual to regulate the elevated cytokine levels induced by the viral infection [13] . Besides the use of several repurposed antiviral drugs including favipiravir, lopinavir and ritonavir, hydroxychloroquine, azithromycin, and remdesivir, the application of CP in SARS-CoV-1 was also found to be effective in terms of decreased mortality; and at present, such strategy is also being noticed to be very much effective in reducing the viral load with the concomitant improvement of clinical conditions in the COVID-19 patients [14, 16, 17] . While the patients who have recovered successfully from COVID-19 may play valuable role as the donors for CP therapy, the disposition of the convalescent serum prophylaxis for COVID-19 patients must comply with the availability of donors, standard settings of blood bank facilities to process the sera, serological and viral assay facilities, appropriate prophylaxis, and therapeutic protocols along with the regulatory compliance. CP transfusion has been found to be effective especially when transfused within the first 20 days of the SARS-CoV-2 infection [14] . Therefore, both the dosage and transfusion timing are important for assuring the necessary safety and usefulness of such treatment [14] . Current review outlined the pros and cons of such application of passive immunity through the CP therapy.'}, {'id': 'kf7lix7f-12', 'contents': 'In such circumstance, the CP therapy is undoubtedly operative for the mitigation of the disease. '}, {'id': 'kf7lix7f-13', 'contents': 'The immune plasma neutralizes the invading SARS-CoV-2 as well as also delivers the passive immunomodulatory features permitting the recipient individual to regulate the elevated cytokine levels induced by the viral infection [13] . Besides the use of several repurposed antiviral drugs including favipiravir, lopinavir and ritonavir, hydroxychloroquine, azithromycin, and remdesivir, the application of CP in SARS-CoV-1 was also found to be effective in terms of decreased mortality; and at present, such strategy is also being noticed to be very much effective in reducing the viral load with the concomitant improvement of clinical conditions in the COVID-19 patients [14, 16, 17] . While the patients who have recovered successfully from COVID-19 may play valuable role as the donors for CP therapy, the disposition of the convalescent serum prophylaxis for COVID-19 patients must comply with the availability of donors, standard settings of blood bank facilities to process the sera, serological and viral assay facilities, appropriate prophylaxis, and therapeutic protocols along with the regulatory compliance. CP transfusion has been found to be effective especially when transfused within the first 20 days of the SARS-CoV-2 infection [14] . Therefore, both the dosage and transfusion timing are important for assuring the necessary safety and usefulness of such treatment [14] . Current review outlined the pros and cons of such application of passive immunity through the CP therapy.'}, {'id': 'kf7lix7f-14', 'contents': 'In such circumstance, the CP therapy is undoubtedly operative for the mitigation of the disease. '}, {'id': 'kf7lix7f-15', 'contents': 'The immune plasma neutralizes the invading SARS-CoV-2 as well as also delivers the passive immunomodulatory features permitting the recipient individual to regulate the elevated cytokine levels induced by the viral infection [13] . Besides the use of several repurposed antiviral drugs including favipiravir, lopinavir and ritonavir, hydroxychloroquine, azithromycin, and remdesivir, the application of CP in SARS-CoV-1 was also found to be effective in terms of decreased mortality; and at present, such strategy is also being noticed to be very much effective in reducing the viral load with the concomitant improvement of clinical conditions in the COVID-19 patients [14, 16, 17] . While the patients who have recovered successfully from COVID-19 may play valuable role as the donors for CP therapy, the disposition of the convalescent serum prophylaxis for COVID-19 patients must comply with the availability of donors, standard settings of blood bank facilities to process the sera, serological and viral assay facilities, appropriate prophylaxis, and therapeutic protocols along with the regulatory compliance. CP transfusion has been found to be effective especially when transfused within the first 20 days of the SARS-CoV-2 infection [14] . Therefore, both the dosage and transfusion timing are important for assuring the necessary safety and usefulness of such treatment [14] . Current review outlined the pros and cons of such application of passive immunity through the CP therapy.'}, {'id': 'kf7lix7f-16', 'contents': 'In such circumstance, the CP therapy is undoubtedly operative for the mitigation of the disease. '}, {'id': 'kf7lix7f-17', 'contents': 'The immune plasma neutralizes the invading SARS-CoV-2 as well as also delivers the passive immunomodulatory features permitting the recipient individual to regulate the elevated cytokine levels induced by the viral infection [13] . Besides the use of several repurposed antiviral drugs including favipiravir, lopinavir and ritonavir, hydroxychloroquine, azithromycin, and remdesivir, the application of CP in SARS-CoV-1 was also found to be effective in terms of decreased mortality; and at present, such strategy is also being noticed to be very much effective in reducing the viral load with the concomitant improvement of clinical conditions in the COVID-19 patients [14, 16, 17] . While the patients who have recovered successfully from COVID-19 may play valuable role as the donors for CP therapy, the disposition of the convalescent serum prophylaxis for COVID-19 patients must comply with the availability of donors, standard settings of blood bank facilities to process the sera, serological and viral assay facilities, appropriate prophylaxis, and therapeutic protocols along with the regulatory compliance. CP transfusion has been found to be effective especially when transfused within the first 20 days of the SARS-CoV-2 infection [14] . Therefore, both the dosage and transfusion timing are important for assuring the necessary safety and usefulness of such treatment [14] . Current review outlined the pros and cons of such application of passive immunity through the CP therapy.'}, {'id': 'kf7lix7f-18', 'contents': 'In such circumstance, the CP therapy is undoubtedly operative for the mitigation of the disease. '}, {'id': 'kf7lix7f-19', 'contents': 'The immune plasma neutralizes the invading SARS-CoV-2 as well as also delivers the passive immunomodulatory features permitting the recipient individual to regulate the elevated cytokine levels induced by the viral infection [13] . Besides the use of several repurposed antiviral drugs including favipiravir, lopinavir and ritonavir, hydroxychloroquine, azithromycin, and remdesivir, the application of CP in SARS-CoV-1 was also found to be effective in terms of decreased mortality; and at present, such strategy is also being noticed to be very much effective in reducing the viral load with the concomitant improvement of clinical conditions in the COVID-19 patients [14, 16, 17] . While the patients who have recovered successfully from COVID-19 may play valuable role as the donors for CP therapy, the disposition of the convalescent serum prophylaxis for COVID-19 patients must comply with the availability of donors, standard settings of blood bank facilities to process the sera, serological and viral assay facilities, appropriate prophylaxis, and therapeutic protocols along with the regulatory compliance. CP transfusion has been found to be effective especially when transfused within the first 20 days of the SARS-CoV-2 infection [14] . Therefore, both the dosage and transfusion timing are important for assuring the necessary safety and usefulness of such treatment [14] . Current review outlined the pros and cons of such application of passive immunity through the CP therapy.'}, {'id': 'kf7lix7f-20', 'contents': 'In such circumstance, the CP therapy is undoubtedly operative for the mitigation of the disease. '}, {'id': 'kf7lix7f-21', 'contents': 'The immune plasma neutralizes the invading SARS-CoV-2 as well as also delivers the passive immunomodulatory features permitting the recipient individual to regulate the elevated cytokine levels induced by the viral infection [13] . Besides the use of several repurposed antiviral drugs including favipiravir, lopinavir and ritonavir, hydroxychloroquine, azithromycin, and remdesivir, the application of CP in SARS-CoV-1 was also found to be effective in terms of decreased mortality; and at present, such strategy is also being noticed to be very much effective in reducing the viral load with the concomitant improvement of clinical conditions in the COVID-19 patients [14, 16, 17] . While the patients who have recovered successfully from COVID-19 may play valuable role as the donors for CP therapy, the disposition of the convalescent serum prophylaxis for COVID-19 patients must comply with the availability of donors, standard settings of blood bank facilities to process the sera, serological and viral assay facilities, appropriate prophylaxis, and therapeutic protocols along with the regulatory compliance. CP transfusion has been found to be effective especially when transfused within the first 20 days of the SARS-CoV-2 infection [14] . Therefore, both the dosage and transfusion timing are important for assuring the necessary safety and usefulness of such treatment [14] . Current review outlined the pros and cons of such application of passive immunity through the CP therapy.'}, {'id': 'kf7lix7f-22', 'contents': 'In such circumstance, the CP therapy is undoubtedly operative for the mitigation of the disease. '}, {'id': 'kf7lix7f-23', 'contents': 'The immune plasma neutralizes the invading SARS-CoV-2 as well as also delivers the passive immunomodulatory features permitting the recipient individual to regulate the elevated cytokine levels induced by the viral infection [13] . Besides the use of several repurposed antiviral drugs including favipiravir, lopinavir and ritonavir, hydroxychloroquine, azithromycin, and remdesivir, the application of CP in SARS-CoV-1 was also found to be effective in terms of decreased mortality; and at present, such strategy is also being noticed to be very much effective in reducing the viral load with the concomitant improvement of clinical conditions in the COVID-19 patients [14, 16, 17] . While the patients who have recovered successfully from COVID-19 may play valuable role as the donors for CP therapy, the disposition of the convalescent serum prophylaxis for COVID-19 patients must comply with the availability of donors, standard settings of blood bank facilities to process the sera, serological and viral assay facilities, appropriate prophylaxis, and therapeutic protocols along with the regulatory compliance. CP transfusion has been found to be effective especially when transfused within the first 20 days of the SARS-CoV-2 infection [14] . Therefore, both the dosage and transfusion timing are important for assuring the necessary safety and usefulness of such treatment [14] . Current review outlined the pros and cons of such application of passive immunity through the CP therapy.'}, {'id': 'e0eki2rk-0', 'contents': 'The datasets generated and/or analyzed during the current study are available from the corresponding author on reasonable request.'}, {'id': 'e0eki2rk-1', 'contents': 'The study follows the principles of the Declaration of Helsinki and the collection, storage and analysis of the data comply with the General Data Protection Regulation (EU GDPR).'}, {'id': 'e0eki2rk-2', 'contents': \"The study was approved by the French Committee for the Protection of Individuals (Comité de Protection des Personnes -CPP IDF X, nr. and by the National Commission on Informatics and Liberty (Commission Nationale de l'Informatique et des Libertés -CNIL, nr. MLD/MFI/AR205600).\"}, {'id': 'e0eki2rk-3', 'contents': 'We wish to confirm that there are no known conflicts of interest associated with this publication and that there has been no significant financial support for this work that could have influenced its outcome.'}, {'id': 'e0eki2rk-4', 'contents': \"Objectifs : Décrire les caractéristiques des hésitants vis-à-vis de la vaccination dans une population adulte francophone dans le contexte de la pandémie de SRAS-CoV-2, et identifier les déterminants de l'hésitation vis-à-vis des vaccins contre la Covid-19.\"}, {'id': 'e0eki2rk-5', 'contents': 'Méthodes : 1 640 adultes francophones participant à une cohorte en ligne, entre avril et mai 2020, ont été classés en fonction de leur attitude vis-à-vis de la vaccination : \"hésitant\", \"anti-vaccination\" et \"pro-vaccination\". Les données ont été analysées à l\\'aide de statistiques descriptives, de modèles de régression multinomiale univariés et d\\'analyses multivariées.'}, {'id': 'e0eki2rk-6', 'contents': \"Résultats : Au moment de l'inclusion, par rapport aux pro-vaccination, les hésitants étaient plus souvent des femmes (p=0,044), moins vaccinés contre la grippe chaque année (p=0,026), moins optimistes quant à la découverte d'un traitement contre la Covid-19 dans quelques mois (p<0,001), moins prêts à prendre ce traitement (p<0,001), présentaient une moindre confiance dans les autorités de santé publique pour contrôler la pandémie (p=0,036) et rapportaient des scores plus faibles sur les échelles de connaissances (valeurs p de <0,001 à 0,002). Les analyses univariées ont confirmé ces résultats avec des probabilités variant de 1,51 [1,05-2,17] à 2,19 [1, 07] . Dans les modèles multivariés, les variables associées à l'hésitation par rapport aux pro-vaccination étaient la découverte d'un traitement contre la Covid-19 dans quelques mois (OR=2,57 [1, 81 \"}, {'id': 'e0eki2rk-7', 'contents': 'The CONFINS cohort study (www.confins.org) was initiated on the 8 th of April 2020 to monitor French-speaking people\\'s well-being and mental health during the Covid-19 lockdown. The cohort was online-based and participation was voluntary. Participants were recruited through social media posts, press releases, podcasts, TV programs and mailing were. The sole eligibility criterion was being aged 18+. For the purposes of this study only, we excluded subjects who had declared \"other\" for the sex variable since there were only 16 of them (0.7%). Wishing to attain sufficient statistical power, we also excluded those reporting missing data on at least one variable in the analyses. '}]\n","output_type":"stream"}]},{"cell_type":"code","source":"!python -m pyserini.index.lucene \\\n  --collection JsonCollection \\\n  --input ./ \\\n  --index index \\\n  --generator DefaultLuceneDocumentGenerator \\\n  --threads 1 \\\n  --storePositions --storeDocvectors --storeRaw","metadata":{"execution":{"iopub.status.busy":"2023-04-05T07:21:20.161081Z","iopub.execute_input":"2023-04-05T07:21:20.161773Z","iopub.status.idle":"2023-04-05T07:22:30.947138Z","shell.execute_reply.started":"2023-04-05T07:21:20.161717Z","shell.execute_reply":"2023-04-05T07:22:30.945763Z"},"trusted":true},"execution_count":95,"outputs":[{"name":"stdout","text":"WARNING: sun.reflect.Reflection.getCallerClass is not supported. This will impact performance.\n2023-04-05 07:21:23,634 INFO  [main] index.IndexCollection (IndexCollection.java:250) - Setting log level to INFO\n2023-04-05 07:21:23,637 INFO  [main] index.IndexCollection (IndexCollection.java:253) - Starting indexer...\n2023-04-05 07:21:23,638 INFO  [main] index.IndexCollection (IndexCollection.java:254) - ============ Loading Parameters ============\n2023-04-05 07:21:23,639 INFO  [main] index.IndexCollection (IndexCollection.java:255) - DocumentCollection path: ./\n2023-04-05 07:21:23,639 INFO  [main] index.IndexCollection (IndexCollection.java:256) - CollectionClass: JsonCollection\n2023-04-05 07:21:23,640 INFO  [main] index.IndexCollection (IndexCollection.java:257) - Generator: DefaultLuceneDocumentGenerator\n2023-04-05 07:21:23,641 INFO  [main] index.IndexCollection (IndexCollection.java:258) - Threads: 1\n2023-04-05 07:21:23,642 INFO  [main] index.IndexCollection (IndexCollection.java:259) - Language: en\n2023-04-05 07:21:23,645 INFO  [main] index.IndexCollection (IndexCollection.java:260) - Stemmer: porter\n2023-04-05 07:21:23,646 INFO  [main] index.IndexCollection (IndexCollection.java:261) - Keep stopwords? false\n2023-04-05 07:21:23,647 INFO  [main] index.IndexCollection (IndexCollection.java:262) - Stopwords: null\n2023-04-05 07:21:23,647 INFO  [main] index.IndexCollection (IndexCollection.java:263) - Store positions? true\n2023-04-05 07:21:23,648 INFO  [main] index.IndexCollection (IndexCollection.java:264) - Store docvectors? true\n2023-04-05 07:21:23,648 INFO  [main] index.IndexCollection (IndexCollection.java:265) - Store document \"contents\" field? false\n2023-04-05 07:21:23,648 INFO  [main] index.IndexCollection (IndexCollection.java:266) - Store document \"raw\" field? true\n2023-04-05 07:21:23,649 INFO  [main] index.IndexCollection (IndexCollection.java:267) - Additional fields to index: []\n2023-04-05 07:21:23,650 INFO  [main] index.IndexCollection (IndexCollection.java:268) - Optimize (merge segments)? false\n2023-04-05 07:21:23,650 INFO  [main] index.IndexCollection (IndexCollection.java:269) - Whitelist: null\n2023-04-05 07:21:23,651 INFO  [main] index.IndexCollection (IndexCollection.java:270) - Pretokenized?: false\n2023-04-05 07:21:23,652 INFO  [main] index.IndexCollection (IndexCollection.java:271) - Index path: index\n2023-04-05 07:21:23,654 INFO  [main] index.IndexCollection (IndexCollection.java:309) - ============ Indexing Collection ============\n2023-04-05 07:21:24,087 INFO  [main] index.IndexCollection (IndexCollection.java:424) - Thread pool with 1 threads initialized.\n2023-04-05 07:21:24,088 INFO  [main] index.IndexCollection (IndexCollection.java:426) - Initializing collection in .\n2023-04-05 07:21:24,115 INFO  [main] index.IndexCollection (IndexCollection.java:435) - 3 files found\n2023-04-05 07:21:24,116 INFO  [main] index.IndexCollection (IndexCollection.java:436) - Starting to index...\n2023-04-05 07:21:24,288 ERROR [pool-2-thread-1] index.IndexCollection$LocalIndexerThread (IndexCollection.java:218) - pool-2-thread-1: Unexpected Exception:\ncom.fasterxml.jackson.core.JsonParseException: Unrecognized token 'jdk': was expecting (JSON String, Number, Array, Object or token 'null', 'true' or 'false')\n at [Source: (BufferedReader); line: 1, column: 4]\n\tat com.fasterxml.jackson.core.JsonParser._constructError(JsonParser.java:2337) ~[anserini-0.16.1-fatjar.jar:?]\n\tat com.fasterxml.jackson.core.base.ParserMinimalBase._reportError(ParserMinimalBase.java:720) ~[anserini-0.16.1-fatjar.jar:?]\n\tat com.fasterxml.jackson.core.json.ReaderBasedJsonParser._reportInvalidToken(ReaderBasedJsonParser.java:2903) ~[anserini-0.16.1-fatjar.jar:?]\n\tat com.fasterxml.jackson.core.json.ReaderBasedJsonParser._handleOddValue(ReaderBasedJsonParser.java:1949) ~[anserini-0.16.1-fatjar.jar:?]\n\tat com.fasterxml.jackson.core.json.ReaderBasedJsonParser.nextToken(ReaderBasedJsonParser.java:781) ~[anserini-0.16.1-fatjar.jar:?]\n\tat com.fasterxml.jackson.databind.ObjectReader.readValues(ObjectReader.java:1874) ~[anserini-0.16.1-fatjar.jar:?]\n\tat io.anserini.collection.JsonCollection$Segment.<init>(JsonCollection.java:107) ~[anserini-0.16.1-fatjar.jar:?]\n\tat io.anserini.collection.JsonCollection.createFileSegment(JsonCollection.java:85) ~[anserini-0.16.1-fatjar.jar:?]\n\tat io.anserini.index.IndexCollection$LocalIndexerThread.run(IndexCollection.java:153) [anserini-0.16.1-fatjar.jar:?]\n\tat java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]\n\tat java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]\n\tat java.lang.Thread.run(Thread.java:834) [?:?]\n2023-04-05 07:21:24,478 ERROR [pool-2-thread-1] index.IndexCollection$LocalIndexerThread (IndexCollection.java:218) - pool-2-thread-1: Unexpected Exception:\njava.lang.RuntimeException: Non-standard token 'NaN': enable JsonParser.Feature.ALLOW_NON_NUMERIC_NUMBERS to allow\n at [Source: (BufferedReader); line: 1, column: 28337]\n\tat com.fasterxml.jackson.databind.MappingIterator._handleIOException(MappingIterator.java:417) ~[anserini-0.16.1-fatjar.jar:?]\n\tat com.fasterxml.jackson.databind.MappingIterator.next(MappingIterator.java:203) ~[anserini-0.16.1-fatjar.jar:?]\n\tat io.anserini.collection.JsonCollection$Segment.readNext(JsonCollection.java:124) ~[anserini-0.16.1-fatjar.jar:?]\n\tat io.anserini.collection.FileSegment$1.hasNext(FileSegment.java:132) ~[anserini-0.16.1-fatjar.jar:?]\n\tat io.anserini.index.IndexCollection$LocalIndexerThread.run(IndexCollection.java:157) [anserini-0.16.1-fatjar.jar:?]\n\tat java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]\n\tat java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]\n\tat java.lang.Thread.run(Thread.java:834) [?:?]\nCaused by: com.fasterxml.jackson.core.JsonParseException: Non-standard token 'NaN': enable JsonParser.Feature.ALLOW_NON_NUMERIC_NUMBERS to allow\n at [Source: (BufferedReader); line: 1, column: 28337]\n\tat com.fasterxml.jackson.core.JsonParser._constructError(JsonParser.java:2337) ~[anserini-0.16.1-fatjar.jar:?]\n\tat com.fasterxml.jackson.core.base.ParserMinimalBase._reportError(ParserMinimalBase.java:710) ~[anserini-0.16.1-fatjar.jar:?]\n\tat com.fasterxml.jackson.core.json.ReaderBasedJsonParser._handleOddValue(ReaderBasedJsonParser.java:1930) ~[anserini-0.16.1-fatjar.jar:?]\n\tat com.fasterxml.jackson.core.json.ReaderBasedJsonParser.nextFieldName(ReaderBasedJsonParser.java:995) ~[anserini-0.16.1-fatjar.jar:?]\n\tat com.fasterxml.jackson.databind.deser.std.BaseNodeDeserializer.deserializeObject(JsonNodeDeserializer.java:269) ~[anserini-0.16.1-fatjar.jar:?]\n\tat com.fasterxml.jackson.databind.deser.std.JsonNodeDeserializer.deserialize(JsonNodeDeserializer.java:69) ~[anserini-0.16.1-fatjar.jar:?]\n\tat com.fasterxml.jackson.databind.deser.std.JsonNodeDeserializer.deserialize(JsonNodeDeserializer.java:16) ~[anserini-0.16.1-fatjar.jar:?]\n\tat com.fasterxml.jackson.databind.MappingIterator.nextValue(MappingIterator.java:280) ~[anserini-0.16.1-fatjar.jar:?]\n\tat com.fasterxml.jackson.databind.MappingIterator.next(MappingIterator.java:199) ~[anserini-0.16.1-fatjar.jar:?]\n\t... 6 more\n2023-04-05 07:22:23,846 DEBUG [pool-2-thread-1] index.IndexCollection$LocalIndexerThread (IndexCollection.java:215) - ./collection.json: 388494 docs added.\n2023-04-05 07:22:30,685 WARN  [main] index.IndexCollection (IndexCollection.java:489) - Unexpected difference between number of indexed documents and index maxDoc.\n2023-04-05 07:22:30,685 INFO  [main] index.IndexCollection (IndexCollection.java:492) - Indexing Complete! 388,521 documents indexed\n2023-04-05 07:22:30,686 INFO  [main] index.IndexCollection (IndexCollection.java:493) - ============ Final Counter Values ============\n2023-04-05 07:22:30,686 INFO  [main] index.IndexCollection (IndexCollection.java:494) - indexed:          388,494\n2023-04-05 07:22:30,686 INFO  [main] index.IndexCollection (IndexCollection.java:495) - unindexable:            0\n2023-04-05 07:22:30,687 INFO  [main] index.IndexCollection (IndexCollection.java:496) - empty:                  0\n2023-04-05 07:22:30,687 INFO  [main] index.IndexCollection (IndexCollection.java:497) - skipped:                0\n2023-04-05 07:22:30,687 INFO  [main] index.IndexCollection (IndexCollection.java:498) - errors:                 0\n2023-04-05 07:22:30,693 INFO  [main] index.IndexCollection (IndexCollection.java:501) - Total 388,521 documents indexed in 00:01:07\n","output_type":"stream"}]},{"cell_type":"code","source":"from pyserini.search.lucene import LuceneSearcher\nimport json\nsearcher = LuceneSearcher('./index')\nf = open(\"collection.json\", \"r\")\ntext = json.loads(f.read())\n\n\nhits = searcher.search('which animal did corona spread from')\n\nfor i in range(len(hits)):\n   \n    print(f'{i+1:2} {hits[i].docid} {hits[i].contents} {hits[i].score:.5f}')\n    for j in text:\n        \n        if(j['id'] == hits[i].docid):\n            print(j['contents'])","metadata":{"execution":{"iopub.status.busy":"2023-04-05T07:48:35.409796Z","iopub.execute_input":"2023-04-05T07:48:35.410254Z","iopub.status.idle":"2023-04-05T07:48:40.052584Z","shell.execute_reply.started":"2023-04-05T07:48:35.410213Z","shell.execute_reply":"2023-04-05T07:48:40.051165Z"},"trusted":true},"execution_count":131,"outputs":[{"name":"stdout","text":" 1 wihmbqam-1 None 8.52080\nCoronavirus disease 2019 (COVID-19) is a virus of the Corona virus family and the source of a respiratory illness outbreak throughout the world that originated in Wuhan, China. Studies (1) (2) (3) show that Covid-19 has clinical characteristics akin to the SARS-CoV. The dominant symptoms include fever and cough, while gastrointestinal symptoms are uncommon. In COVID-19 infected patients the absence of fever is more frequent than in patients infected by similar viruses, i.e., MERS Corona Virus (2%) and SARS Corona Virus (1%) (4); therefore, there is a possibility of non-febrile patients being missed by a surveillance mechanism with a primary focus on detecting fever (5) . The initial patients infected by COVID-19, reportedly indicated an association with a large seafood and animal market in Wuhan that demonstrated an animal-to-person spread. Per contra, a burgeoning number of patients have not displayed any association with the animal markets, revealing the fact of human-to-human transmission of COVID-19. This pandemic has been declared a global health emergency and is spreading at an alarming rate (6) . The origin of the virus in Wuhan, China has caused 175,694 deaths globally and has 2,544,792 active patients globally (7) . With the stress on medical facilities, it is essential for governments and healthcare facilities to identify and treat cases that are most likely to survive, by so doing, judiciously utilizing the limited stock of medical resources and medications.\n 2 66b7nq68-4 None 7.82790\nOrigin In December 2019, the reports of patients with viral pneumonia in China surfaced. The origin was related to the Huanan Seafood's whole market. Researchers established that the disease has most likely originated from bats, mutated to infect humans, and transmitted by droplet routes among humans. Since the origin occurred in a wet market, the media criticized the Chinese for consuming live animals like bats, snakes, and dogs. Previously known as \"Wuhan virus\" and \"China coronavirus,\" it was subsequently called 2019-nCoV, and then finally, on February 11, the World Health Organization (WHO) gave the disease an official name, \"Covid-19\". In spite of this, renowned world leaders ignorantly kept calling it \"Chinese virus\" which left concerning effects in international communities [8, 9] . Asian people were stigmatized and called by the name 'Corona' on the streets. Many reported incidents of such racial slurs which exaggerated already existing prejudices among people. This happened in SARS 2003 in Toronto which caused the xenophobic reaction, and similarly, was repeated with the coronavirus pandemic. It had a significant impact, leading the Chinese government to order the closure of all farmer's markets and put a ban on eating live animals. This was an essential step in halting the spread of the virus. However, it also created a false sense of security among the rest of the world, and no one took the necessary precautions when the cases were limited to China.\n 3 bep6ylgh-27 None 7.68860\nRegarding the human-animal interface of SARS-CoV-2 outbreaks, on the one hand, the reason behind the efficient SARS-CoV-2 spread between mink farms could be as simple as the high stocking animal densities on such farms. On the other hand, the in-depth reasons behind this efficient spread remain ambiguous, making it difficult to stop the outbreak among farms. A possible explanation for this rapid viral spread is that it is transmitted through infected humans, fomites, infectious animal droplets, or contaminated dust [13] . However, the implementation of strict preventative measures in the Netherlands did not stop the outbreak. Therefore, other reasons, such as wild animals and wind, should also be considered. Regardless of the transmission route, halting the spread between farms could be tricky because of the difficulty to control the movement of wild animals and the dissemination of asymptomatic human cases. Furthermore, farming conditions, including many different animals, constitute a risk of forming virus reservoirs and new mutations, as well as the spread of the virus into communities from farms through humans or escaped mink. Several other animals could also be SARS-CoV-2 infected, including raccoon dogs that, just like mink, are bred in farms (in a few countries) and live in the wild close to humans, even though no similar outbreaks have been detected among them yet. As it seems a daunting task to fully control the viral spread among companion animals, another hurdle is presented by limited resources, preventing the comprehensive administration of vaccines to animals, despite the importance of vaccinating susceptible animals (i.e., minks, cats, and zoo animals) [21] . We stress the necessity of regular, comprehensive animal surveillance and monitoring, specifically for minks, raccoon dogs, cats, and zoo animals, to avoid and predict SARS-CoV-2 human-animal and animal-human transmissions. Moreover, we agree with systemic genotyping recommendations and sharing the genome sequences of isolated SARS-CoV-2 strains from all infected animals to rapidly identify possible clusters and related variants. The world is now in a situation where One Health preparedness and response strategies are a must to synchronize efforts from different sectors (i.e., human health, animal health, and agriculture) in order to achieve a timely control strategy against the pandemic [12] . Additionally, we recommend further investigations on the UK and South African variants. Such further investigations would elucidate the role of mutations on several levels, including (i) the increased SARS-CoV-2 transmissibility, (ii) their effect on the potency of vaccine candidates, and (iii) the possibility of SARS-CoV-2 re-infectivity on both previously confirmed COVID-19 patients and vaccinated individuals.\n 4 7c2uzcpx-0 None 7.54350\nCorona viruses are considered to be zoonotic pathogens. Those causing epidemics were believed to be originated from bats and transmitted to humans through an intermediate animal host with a likelihood of the spill over from animals to humans [4] . Chinese researchers proposed a hypothesis of pangolins a long-snouted, ant-eating mammals (used in traditional Chinese medicine) as animal source of the COVID-19 outbreak [5] . SARS-CoV-2 having higher reproductive number (R0) and more efficient spreading capacity, is considered to be more infectious than SARS CoV1. Both interact differently with the angiotensin-converting enzyme 2 (ACE 2) receptor. Different surface protein in SARS CoV2 allows its stronger binding to the ACE 2 receptor of host cell and greater efficiency at invading host cells [6] . It has a stronger bonding with upper respiratory tract (URT) and conjunctiva infecting more efficiently the URT and airways [7] . SARS-CoV-2 has highest viral load in URT at the time of onset of symptoms and during first week of illness compared to second week in SARS CoV1, indicating its highest infectiousness potential just before or within the first five days of symptom onset [8] .\n 5 wgmerlj5-2 None 7.54080\nBy Jan'20, SARS-CoV-2 had already reached countries around the world. None of the countries was prepared for this scale of spread of the novel virus. Due to inadequate information about 3 this virus as well as the nature, transmission and management of Covid-19, most of the countries could not control it's spread. However, the epidemics in Western countries were ahead of the rest of the world by a few months and were severe too. As a result, most of the globally confirmed COVID-19 cases in 2020 were reported from United States, the UK, Spain, Italy, Germany, and France. 11 Since most of these first world countries have curbed infectious/communicable diseases to a great extent, it is likely that their public health systems failed to identify the potential threat posed by the novel disease. They did not make testing widely available and did not impose physical distancing guidelines early in pandemic. 12 In fact, UK did not impose lockdowns initially in an effort to generate herd immunity. 13 Besides, Western countries could probably not implement strict measures to control the spread of corona virus or were late in putting them into force. Being a zoonotic virus, presence of many hosts at a given time may have led to faster evolution of the virus giving rise to a deadlier strain. This may have led to higher mortality rates in the West. Taking their learnings from the western world, stricter interventions were imposed by Eastern countries which may have slowed down the evolution of the virus leading to a less deadly variant. Countries like South Korea, Singapore, Vietnam, Taiwan, Hong Kong and India recognized the threat well in advance and took strict measures to curb the spread of Covid-19. In South Korea, Covid cases peaked by Feb, 2020, but country presented the world with one of the most successful models of handling the pandemic. They implemented social distancing and lockdowns in areas with high transmission, extensive testing including drive-through screening tests, \n 6 g8919ekh-81 None 7.45070\nIn all surveyed countries, aspects regarding pandemic control were among the most frequently mentioned positive aspects of corona apps. Thus, many citizens seem to be generally open to the idea of using mobile apps to help limiting the spread of a virus. Privacy concerns were the most frequently mentioned negative aspects of corona apps in all countries. A rigorous approach towards privacyfriendly technologies, which prevent privacy violations a priori, e. g.privacy-by-design, seems vital for the success of such apps. Somewhat surprisingly, the type of payload data did not have any significant negative influence on the willingness to use a corona app in our study. This implies that users may be willing to share additional data if other conditions are met. Citizens of Germany and the US seem to be willing to use apps for contact tracing without technical malfunctions which ensure anonymity and are administered by the health authorities. Such apps can be further developed to be ready for deployment in potential future pandemics. In China, apps facilitating individual contact tracing seem to be less relevant. This might be due to the strict regional measures implemented by the authorities as soon as infections are detected and due to the fact that authorities in China already have access to data enabling the tracking of individuals. Instead, Chinese participants expressed an increased need for apps informing about the current (regional or local) regulations and the state of the pandemic as indicated by our results.\n 7 jon0u3zi-18 None 7.43310\nIn addition to various forms of influenza and the epidemics generated by the corona virus SARS-CoV-1, SARS-CoV-2 or MERS-CoV, other epidemics occurred. It is instructive to compare them with those just mentioned, applying in addition mathematical yardsticks. We shall restrict ourselves to Ebola epidemics. Their most widespread outbreak was the Western African Ebola virus epidemic from 2013 to 2016, which caused 28,646 cases and 11,323 deaths. There is a fundamental difference in the evolution of a case of influenza or SARS-CoV-1 or SARS-CoV-2 on the one hand and of an Ebola case on the other, which leads to a basic difference in their mathematical modelling (Sects. 4 and 8). A carrier of an influenza or corona virus can transmit it to other persons well before the first symptoms appear, that is well before the end of the incubation period. A subject infected by Ebola will become infective only around the end of the incubation period. He (if it is a man) could then be immediately isolated together with his latest contacts in order to avoid further extension of the infection, provided that there is a health service nearby to do it. Therefore Ebola did not spread to countries that have a sufficiently dense primary health care network but it caused much suffering in countries that do not have it. The strategy of WHO to control the epidemic was wrong. It insisted on drugs and the search for a vaccine (which became available only in December 2019) but neglected primary health care. For the present purpose it would even have been most useful to rapidly train village health workers and \"barefoot doctors\" as it had been done decades ago.\n 8 2l1nuasc-6 None 7.34590\nNotwithstanding the lack of clinical symptoms, we and others 30, 32 did find pathological 262 abnormalities in infected ferrets. While SARS-CoV-2 replicated less efficiently in young i.t. inoculated 263 animals and almost no adaptive immune response was induced, they did display some of the most 264 affected pathology. It is however important to mention that all young ferrets euthanized on 21 dpi 265 tested positive for antibodies against ferret corona and NL63 prior to SARS- against the E protein was observed. Cellular responses against multiple accessory proteins of SARS-286\n 9 pn9nwl64-22 None 7.33230\nThe origins of more than 75% of coronavirus infections are considered zoonotic, i.e., animals are the main source of the outbreaks. For example, SARS-CoV was transmitted from palm civets to humans, and MERS-CoV from dromedary camels to humans. Bats are currently considered a reservoir for all human coronaviruses, as mentioned above [5, 71] . Many coronaviruses are circulating in animals but have not yet infected humans. The type of animal that SARS-CoV-2 originated from is still unclear. At the beginning of the outbreak in Wuhan, China, many patients were linked to the Huanan Seafood Wholesale Market, suggesting animal-to-person spread. After retrospectively studying case reports, the number of patients that did not have exposure to animal markets has risen, indicating person-to-person spread was also occurring at that time [66] . SARS-CoV-2 is closely related to bat coronaviruses and SARS-CoV [8] . A group of researchers reported early in the outbreak that the novel SARS-CoV-2 has the highest similarity of codon usage bias with snakes [72, 73] ; however, this method to determine initial host origins is dubious. Interestingly, researchers also reported one amino acid difference in the receptor-binding domain of the S protein of Pangolin-CoV compared to that of SARS-CoV-2, suggesting that pangolins might play a role as an intermediate host (Xiao et al., data currently under review). Another group of researchers reported that the virus originated from bats based on the genome sequence of SARS-CoV-2, which is 96% identical to bat coronavirus RaTG13. There were speculations that SARS-CoV-2 is a laboratory-engineered CoV and leaked directly from a laboratory in Wuhan where a bat CoV (RaTG13) was recently reported. However, there is no evidence to support this allegation [74] . Recently, a group of researchers found that SARS-CoV-2 replicates poorly in dogs, pigs, chickens, and ducks but efficiently in ferrets and cats [75] . Scientists are still trying to find the main source of the disease outbreak and identify the definitive intermediate hosts.\n10 90tfzrqc-12 None 7.29570\nThe order Nidovirales, sub-family Orthocoronavirinae, family Coronaviridae is composed of four genera: a-coronavirus, bcoronavirus, c-coronavirus, and d-coronavirus. SARS-CoV-2 belongs to the subgenus Sarbecovirus of the genus b-coronavirus, to which SARS-CoV and MERS-CoV also belong. Coronaviruses (CoVs) infect humans as well as domestic and wild animal species, with infections remaining sub-clinical in most cases [324] [325] [326] . a-coronavirus and b-coronavirus usually infect mammals, with a probable origin of bats, while c-coronavirus and d-coronavirus mainly infect birds, and sometimes mammals, and might originate from swine [327] [328] [329] . A long list of animal species has been reported as intermediate hosts, such as dogs, cats, cattle, horses, camels, rodents, rabbits, pangolins, mink, snakes, frogs, marmots, hedgehogs, and ferrets [324, [330] [331] [332] [333] [334] [335] [336] . Thus, there could be multiple viral transmission pathways from different animals to humans. The three outbreak points of coronavirus in China-namely, the livestock markets in Guangdong in 2003, the Huanan Seafood Market in Wuhan at the end of 2019, and the Xinfadi Seafood Market in Beijing in June 2020-are all related to animal markets. Civet cats and camels have been demonstrated to transmit SARS-CoV or MERS-CoV to humans, which provides an important hint of virus transmission directly from animals. However, it remains unclear which animal could be the main intermediate host of SARS-CoV-2, although positive viral RNA signals were detected in seafood markets and on the chopping boards of salmon. In 1983, Lidgerding and Hetrick [337] first reported the replication of a coronavirus in a fish cell line. Furthermore, Sano et al. [338] successfully isolated a coronavirus from common carp (Cyprinus carpio) in 1988, which induced hepatic, renal, and intestinal necrosis in experimentally infected fish. Miyazaki et al. [339] found a corona-like virus in color carp (Cyprinus carpio) in 2000, which caused dermal lesion and necrosis in internal organs.\n","output_type":"stream"}]},{"cell_type":"code","source":"","metadata":{},"execution_count":null,"outputs":[]},{"cell_type":"code","source":"test_summerized_reference = \"We introduce a new language representation model called BERT, which stands for Bidirectional Encoder Representations from Transformers. Unlike recent language representation models (Peters et al., 2018a; Radford et al., 2018), BERT is designed to pretrain deep bidirectional representations from unlabeled text by jointly conditioning on both left and right context in all layers. As a result, the pre-trained BERT model can be finetuned with just one additional output layer to create state-of-the-art models for a wide range of tasks, such as question answering and language inference, without substantial taskspecific architecture modifications. BERT is conceptually simple and empirically powerful. It obtains new state-of-the-art results on eleven natural language processing tasks, including pushing the GLUE score to 80.5% (7.7% point absolute improvement), MultiNLI accuracy to 86.7% (4.6% absolute improvement), SQuAD v1.1 question answering Test F1 to 93.2 (1.5 point absolute improvement) and SQuAD v2.0 Test F1 to 83.1 (5.1 point absolute improvement).\"\ntest_query = \"What does the 'B' in BERT stand for?\"","metadata":{},"execution_count":null,"outputs":[]},{"cell_type":"code","source":"covid_text = \"Origin In December 2019, the reports of patients with viral pneumonia in China surfaced. The origin was related to the Huanan Seafood's whole market. Researchers established that the disease has most likely originated from bats, mutated to infect humans, and transmitted by droplet routes among humans. Since the origin occurred in a wet market, the media criticized the Chinese for consuming live animals like bats, snakes, and dogs. Previously known as \\\"Wuhan virus\\\" and \\\"China coronavirus,\\\" it was subsequently called 2019-nCoV, and then finally, on February 11, the World Health Organization (WHO) gave the disease an official name, \\\"Covid-19\\\". In spite of this, renowned world leaders ignorantly kept calling it \\\"Chinese virus\\\" which left concerning effects in international communities [8, 9] . Asian people were stigmatized and called by the name 'Corona' on the streets. Many reported incidents of such racial slurs which exaggerated already existing prejudices among people. This happened in SARS 2003 in Toronto which caused the xenophobic reaction, and similarly, was repeated with the coronavirus pandemic. It had a significant impact, leading the Chinese government to order the closure of all farmer's markets and put a ban on eating live animals. This was an essential step in halting the spread of the virus. However, it also created a false sense of security among the rest of the world, and no one took the necessary precautions when the cases were limited to China\"\ncovid_question = \"which animal did corona spread from\"","metadata":{"execution":{"iopub.status.busy":"2023-04-05T07:50:43.632206Z","iopub.execute_input":"2023-04-05T07:50:43.633508Z","iopub.status.idle":"2023-04-05T07:50:43.640310Z","shell.execute_reply.started":"2023-04-05T07:50:43.633443Z","shell.execute_reply":"2023-04-05T07:50:43.639343Z"},"trusted":true},"execution_count":134,"outputs":[]},{"cell_type":"code","source":"!pip install transformers\nimport torch","metadata":{"execution":{"iopub.status.busy":"2023-04-05T07:27:31.175432Z","iopub.execute_input":"2023-04-05T07:27:31.175876Z","iopub.status.idle":"2023-04-05T07:27:43.194730Z","shell.execute_reply.started":"2023-04-05T07:27:31.175835Z","shell.execute_reply":"2023-04-05T07:27:43.193131Z"},"trusted":true},"execution_count":98,"outputs":[{"name":"stdout","text":"Requirement already satisfied: transformers in /opt/conda/lib/python3.7/site-packages (4.27.3)\nRequirement already satisfied: tokenizers!=0.11.3,<0.14,>=0.11.1 in /opt/conda/lib/python3.7/site-packages (from transformers) (0.13.2)\nRequirement already satisfied: filelock in /opt/conda/lib/python3.7/site-packages (from transformers) (3.9.0)\nRequirement already satisfied: requests in /opt/conda/lib/python3.7/site-packages (from transformers) (2.28.2)\nRequirement already satisfied: numpy>=1.17 in /opt/conda/lib/python3.7/site-packages (from transformers) (1.21.6)\nRequirement already satisfied: tqdm>=4.27 in /opt/conda/lib/python3.7/site-packages (from transformers) (4.64.1)\nRequirement already satisfied: regex!=2019.12.17 in /opt/conda/lib/python3.7/site-packages (from transformers) (2021.11.10)\nRequirement already satisfied: packaging>=20.0 in /opt/conda/lib/python3.7/site-packages (from transformers) (23.0)\nRequirement already satisfied: importlib-metadata in /opt/conda/lib/python3.7/site-packages (from transformers) (4.11.4)\nRequirement already satisfied: pyyaml>=5.1 in /opt/conda/lib/python3.7/site-packages (from transformers) (6.0)\nRequirement already satisfied: huggingface-hub<1.0,>=0.11.0 in /opt/conda/lib/python3.7/site-packages (from transformers) (0.13.3)\nRequirement already satisfied: typing-extensions>=3.7.4.3 in /opt/conda/lib/python3.7/site-packages (from huggingface-hub<1.0,>=0.11.0->transformers) (4.4.0)\nRequirement already satisfied: zipp>=0.5 in /opt/conda/lib/python3.7/site-packages (from importlib-metadata->transformers) (3.11.0)\nRequirement already satisfied: certifi>=2017.4.17 in /opt/conda/lib/python3.7/site-packages (from requests->transformers) (2022.12.7)\nRequirement already satisfied: idna<4,>=2.5 in /opt/conda/lib/python3.7/site-packages (from requests->transformers) (3.4)\nRequirement already satisfied: charset-normalizer<4,>=2 in /opt/conda/lib/python3.7/site-packages (from requests->transformers) (2.1.1)\nRequirement already satisfied: urllib3<1.27,>=1.21.1 in /opt/conda/lib/python3.7/site-packages (from requests->transformers) (1.26.14)\n\u001b[33mWARNING: Error parsing requirements for pybind11: [Errno 2] No such file or directory: '/opt/conda/lib/python3.7/site-packages/pybind11-2.10.4.dist-info/METADATA'\u001b[0m\u001b[33m\n\u001b[0m\u001b[33mWARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv\u001b[0m\u001b[33m\n\u001b[0m","output_type":"stream"}]},{"cell_type":"code","source":"from transformers import BertForQuestionAnswering\n\nmodel = BertForQuestionAnswering.from_pretrained('bert-large-uncased-whole-word-masking-finetuned-squad')","metadata":{"execution":{"iopub.status.busy":"2023-04-05T07:28:14.300067Z","iopub.execute_input":"2023-04-05T07:28:14.300472Z","iopub.status.idle":"2023-04-05T07:28:18.204838Z","shell.execute_reply.started":"2023-04-05T07:28:14.300427Z","shell.execute_reply":"2023-04-05T07:28:18.203503Z"},"trusted":true},"execution_count":101,"outputs":[]},{"cell_type":"code","source":"from transformers import BertTokenizer\n\ntokenizer = BertTokenizer.from_pretrained('bert-large-uncased-whole-word-masking-finetuned-squad')","metadata":{"execution":{"iopub.status.busy":"2023-04-05T07:28:26.394384Z","iopub.execute_input":"2023-04-05T07:28:26.394878Z","iopub.status.idle":"2023-04-05T07:28:26.806255Z","shell.execute_reply.started":"2023-04-05T07:28:26.394838Z","shell.execute_reply":"2023-04-05T07:28:26.804642Z"},"trusted":true},"execution_count":103,"outputs":[]},{"cell_type":"code","source":"def answer_question(question, answer_text):\n    '''\n    Takes a `question` string and an `answer_text` string (which contains the\n    answer), and identifies the words within the `answer_text` that are the\n    answer. Prints them out.\n    '''\n    # ======== Tokenize ========\n    # Apply the tokenizer to the input text, treating them as a text-pair.\n    input_ids = tokenizer.encode(question, answer_text)\n\n    # Report how long the input sequence is.\n#     print('Query has {:,} tokens.\\n'.format(len(input_ids)))\n\n    # ======== Set Segment IDs ========\n    # Search the input_ids for the first instance of the `[SEP]` token.\n    sep_index = input_ids.index(tokenizer.sep_token_id)\n\n    # The number of segment A tokens includes the [SEP] token istelf.\n    num_seg_a = sep_index + 1\n\n    # The remainder are segment B.\n    num_seg_b = len(input_ids) - num_seg_a\n\n    # Construct the list of 0s and 1s.\n    segment_ids = [0]*num_seg_a + [1]*num_seg_b\n\n    # There should be a segment_id for every input token.\n    assert len(segment_ids) == len(input_ids)\n\n    # ======== Evaluate ========\n    # Run our example through the model.\n    outputs = model(torch.tensor([input_ids]), # The tokens representing our input text.\n                    token_type_ids=torch.tensor([segment_ids]), # The segment IDs to differentiate question from answer_text\n                    return_dict=True) \n\n    start_scores = outputs.start_logits\n    end_scores = outputs.end_logits\n\n    # ======== Reconstruct Answer ========\n    # Find the tokens with the highest `start` and `end` scores.\n    answer_start = torch.argmax(start_scores)\n    answer_end = torch.argmax(end_scores)\n\n    # Get the string versions of the input tokens.\n    tokens = tokenizer.convert_ids_to_tokens(input_ids)\n\n    # Start with the first token.\n    answer = tokens[answer_start]\n\n    # Select the remaining answer tokens and join them with whitespace.\n    for i in range(answer_start + 1, answer_end + 1):\n        \n        # If it's a subword token, then recombine it with the previous token.\n        if tokens[i][0:2] == '##':\n            answer += tokens[i][2:]\n        \n        # Otherwise, add a space then the token.\n        else:\n            answer += ' ' + tokens[i]\n\n    print('Answer: \"' + answer + '\"')\n","metadata":{"execution":{"iopub.status.busy":"2023-04-05T07:28:31.440488Z","iopub.execute_input":"2023-04-05T07:28:31.440950Z","iopub.status.idle":"2023-04-05T07:28:31.452182Z","shell.execute_reply.started":"2023-04-05T07:28:31.440911Z","shell.execute_reply":"2023-04-05T07:28:31.450954Z"},"trusted":true},"execution_count":105,"outputs":[]},{"cell_type":"code","source":"answer_question(covid_question,covid_text )","metadata":{"execution":{"iopub.status.busy":"2023-04-05T07:50:50.839315Z","iopub.execute_input":"2023-04-05T07:50:50.840123Z","iopub.status.idle":"2023-04-05T07:50:52.986179Z","shell.execute_reply.started":"2023-04-05T07:50:50.840079Z","shell.execute_reply":"2023-04-05T07:50:52.984768Z"},"trusted":true},"execution_count":135,"outputs":[{"name":"stdout","text":"Answer: \"bats\"\n","output_type":"stream"}]},{"cell_type":"code","source":"# accept the query from the user\nquery = input(\"Please enter your question: \")\nprint(query)\n# get the answer of the query\ntop_n_paragraphs_dict = get_answer(query, 5)\n# present the answer\npresent_answer(top_n_paragraphs_dict)\n","metadata":{"execution":{"iopub.status.busy":"2023-04-05T06:20:17.827790Z","iopub.status.idle":"2023-04-05T06:20:17.828361Z","shell.execute_reply.started":"2023-04-05T06:20:17.828091Z","shell.execute_reply":"2023-04-05T06:20:17.828120Z"},"trusted":true},"execution_count":null,"outputs":[]},{"cell_type":"markdown","source":"### 7. References","metadata":{}}]}